{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Counting reads in genomic regions:"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## - install `compEpiTools` and dependencies"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Bioconductor version 3.5 (BiocInstaller 1.26.1), ?biocLite for help\n",
      "BioC_mirror: https://bioconductor.org\n",
      "Using Bioconductor 3.5 (BiocInstaller 1.26.1), R 3.4.1 (2017-06-30).\n",
      "Installing package(s) 'compEpiTools'\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "package 'compEpiTools' successfully unpacked and MD5 sums checked\n",
      "\n",
      "The downloaded binary packages are in\n",
      "\tC:\\Users\\mustafa.albahrani\\AppData\\Local\\Temp\\RtmpKm3H3E\\downloaded_packages\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Old packages: 'BiocGenerics', 'curl', 'data.table', 'GenomeInfoDb', 'IRanges',\n",
      "  'limma', 'rtracklayer', 'S4Vectors', 'backports', 'BH', 'boot', 'car',\n",
      "  'caret', 'crayon', 'DBI', 'dplyr', 'evaluate', 'foreign', 'glmnet', 'haven',\n",
      "  'htmlwidgets', 'httpuv', 'httr', 'knitr', 'lme4', 'Matrix', 'mgcv', 'modelr',\n",
      "  'openssl', 'psych', 'purrr', 'quantmod', 'R6', 'Rcpp', 'repr', 'rlang',\n",
      "  'rmarkdown', 'scales', 'shiny', 'tibble', 'tidyr', 'TTR', 'xts'\n"
     ]
    }
   ],
   "source": [
    "# install compEpiTools from bioconductor\n",
    "source(\"https://bioconductor.org/biocLite.R\")\n",
    "biocLite(\"compEpiTools\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "BioC_mirror: https://bioconductor.org\n",
      "Using Bioconductor 3.5 (BiocInstaller 1.26.1), R 3.4.1 (2017-06-30).\n",
      "Installing package(s) 'GenomicRanges'\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "package 'GenomicRanges' successfully unpacked and MD5 sums checked\n",
      "\n",
      "The downloaded binary packages are in\n",
      "\tC:\\Users\\mustafa.albahrani\\AppData\\Local\\Temp\\RtmpKm3H3E\\downloaded_packages\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Old packages: 'BiocGenerics', 'curl', 'data.table', 'GenomeInfoDb', 'IRanges',\n",
      "  'limma', 'rtracklayer', 'S4Vectors', 'backports', 'BH', 'boot', 'car',\n",
      "  'caret', 'crayon', 'DBI', 'dplyr', 'evaluate', 'foreign', 'glmnet', 'haven',\n",
      "  'htmlwidgets', 'httpuv', 'httr', 'knitr', 'lme4', 'Matrix', 'mgcv', 'modelr',\n",
      "  'openssl', 'psych', 'purrr', 'quantmod', 'R6', 'Rcpp', 'repr', 'rlang',\n",
      "  'rmarkdown', 'scales', 'shiny', 'tibble', 'tidyr', 'TTR', 'xts'\n"
     ]
    }
   ],
   "source": [
    "# install 'GenomicRanges' package\n",
    "biocLite(\"GenomicRanges\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "BioC_mirror: https://bioconductor.org\n",
      "Using Bioconductor 3.5 (BiocInstaller 1.26.1), R 3.4.1 (2017-06-30).\n",
      "Installing package(s) 'TxDb.Mmusculus.UCSC.mm9.knownGene'\n",
      "installing the source package 'TxDb.Mmusculus.UCSC.mm9.knownGene'\n",
      "\n",
      "Old packages: 'BiocGenerics', 'curl', 'data.table', 'GenomeInfoDb', 'IRanges',\n",
      "  'limma', 'rtracklayer', 'S4Vectors', 'backports', 'BH', 'boot', 'car',\n",
      "  'caret', 'crayon', 'DBI', 'dplyr', 'evaluate', 'foreign', 'glmnet', 'haven',\n",
      "  'htmlwidgets', 'httpuv', 'httr', 'knitr', 'lme4', 'Matrix', 'mgcv', 'modelr',\n",
      "  'openssl', 'psych', 'purrr', 'quantmod', 'R6', 'Rcpp', 'repr', 'rlang',\n",
      "  'rmarkdown', 'scales', 'shiny', 'tibble', 'tidyr', 'TTR', 'xts'\n",
      "BioC_mirror: https://bioconductor.org\n",
      "Using Bioconductor 3.5 (BiocInstaller 1.26.1), R 3.4.1 (2017-06-30).\n",
      "Installing package(s) 'TxDb.Hsapiens.UCSC.hg19.knownGene'\n",
      "installing the source package 'TxDb.Hsapiens.UCSC.hg19.knownGene'\n",
      "\n",
      "Old packages: 'BiocGenerics', 'curl', 'data.table', 'GenomeInfoDb', 'IRanges',\n",
      "  'limma', 'rtracklayer', 'S4Vectors', 'backports', 'BH', 'boot', 'car',\n",
      "  'caret', 'crayon', 'DBI', 'dplyr', 'evaluate', 'foreign', 'glmnet', 'haven',\n",
      "  'htmlwidgets', 'httpuv', 'httr', 'knitr', 'lme4', 'Matrix', 'mgcv', 'modelr',\n",
      "  'openssl', 'psych', 'purrr', 'quantmod', 'R6', 'Rcpp', 'repr', 'rlang',\n",
      "  'rmarkdown', 'scales', 'shiny', 'tibble', 'tidyr', 'TTR', 'xts'\n"
     ]
    }
   ],
   "source": [
    "# download the Mm genes data (mm9 build) from UCSC database using the `TxDb` package\n",
    "biocLite(\"TxDb.Mmusculus.UCSC.mm9.knownGene\")\n",
    "\n",
    "# download the Hs genes data (hg19 build) from UCSC database using the `TxDb` package\n",
    "biocLite(\"TxDb.Hsapiens.UCSC.hg19.knownGene\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "BioC_mirror: https://bioconductor.org\n",
      "Using Bioconductor 3.5 (BiocInstaller 1.26.1), R 3.4.1 (2017-06-30).\n",
      "Installing package(s) 'org.Mm.eg.db'\n",
      "installing the source package 'org.Mm.eg.db'\n",
      "\n",
      "Old packages: 'BiocGenerics', 'curl', 'data.table', 'GenomeInfoDb', 'IRanges',\n",
      "  'limma', 'rtracklayer', 'S4Vectors', 'backports', 'BH', 'boot', 'car',\n",
      "  'caret', 'crayon', 'DBI', 'dplyr', 'evaluate', 'foreign', 'glmnet', 'haven',\n",
      "  'htmlwidgets', 'httpuv', 'httr', 'knitr', 'lme4', 'Matrix', 'mgcv', 'modelr',\n",
      "  'openssl', 'psych', 'purrr', 'quantmod', 'R6', 'Rcpp', 'repr', 'rlang',\n",
      "  'rmarkdown', 'scales', 'shiny', 'tibble', 'tidyr', 'TTR', 'xts'\n",
      "BioC_mirror: https://bioconductor.org\n",
      "Using Bioconductor 3.5 (BiocInstaller 1.26.1), R 3.4.1 (2017-06-30).\n",
      "Installing package(s) 'org.Hs.eg.db'\n",
      "installing the source package 'org.Hs.eg.db'\n",
      "\n",
      "Old packages: 'BiocGenerics', 'curl', 'data.table', 'GenomeInfoDb', 'IRanges',\n",
      "  'limma', 'rtracklayer', 'S4Vectors', 'backports', 'BH', 'boot', 'car',\n",
      "  'caret', 'crayon', 'DBI', 'dplyr', 'evaluate', 'foreign', 'glmnet', 'haven',\n",
      "  'htmlwidgets', 'httpuv', 'httr', 'knitr', 'lme4', 'Matrix', 'mgcv', 'modelr',\n",
      "  'openssl', 'psych', 'purrr', 'quantmod', 'R6', 'Rcpp', 'repr', 'rlang',\n",
      "  'rmarkdown', 'scales', 'shiny', 'tibble', 'tidyr', 'TTR', 'xts'\n"
     ]
    }
   ],
   "source": [
    "# install genome wide annotation for mouse and human\n",
    "\n",
    "biocLite('org.Mm.eg.db')\n",
    "biocLite('org.Hs.eg.db')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: compEpiTools\n",
      "Loading required package: topGO\n",
      "Loading required package: BiocGenerics\n",
      "Loading required package: parallel\n",
      "\n",
      "Attaching package: 'BiocGenerics'\n",
      "\n",
      "The following objects are masked from 'package:parallel':\n",
      "\n",
      "    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,\n",
      "    clusterExport, clusterMap, parApply, parCapply, parLapply,\n",
      "    parLapplyLB, parRapply, parSapply, parSapplyLB\n",
      "\n",
      "The following objects are masked from 'package:stats':\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from 'package:base':\n",
      "\n",
      "    anyDuplicated, append, as.data.frame, cbind, colMeans, colnames,\n",
      "    colSums, do.call, duplicated, eval, evalq, Filter, Find, get, grep,\n",
      "    grepl, intersect, is.unsorted, lapply, lengths, Map, mapply, match,\n",
      "    mget, order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rowMeans, rownames, rowSums, sapply, setdiff, sort,\n",
      "    table, tapply, union, unique, unsplit, which, which.max, which.min\n",
      "\n",
      "Loading required package: graph\n",
      "Loading required package: Biobase\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "Loading required package: GO.db\n",
      "Loading required package: AnnotationDbi\n",
      "Loading required package: stats4\n",
      "Loading required package: IRanges\n",
      "Loading required package: S4Vectors\n",
      "\n",
      "Attaching package: 'S4Vectors'\n",
      "\n",
      "The following object is masked from 'package:base':\n",
      "\n",
      "    expand.grid\n",
      "\n",
      "\n",
      "Loading required package: SparseM\n",
      "\n",
      "Attaching package: 'SparseM'\n",
      "\n",
      "The following object is masked from 'package:base':\n",
      "\n",
      "    backsolve\n",
      "\n",
      "\n",
      "groupGOTerms: \tGOBPTerm, GOMFTerm, GOCCTerm environments built.\n",
      "\n",
      "Attaching package: 'topGO'\n",
      "\n",
      "The following object is masked from 'package:IRanges':\n",
      "\n",
      "    members\n",
      "\n",
      "Loading required package: GenomicRanges\n",
      "Warning message:\n",
      "\"package 'GenomicRanges' was built under R version 3.4.2\"Loading required package: GenomeInfoDb\n",
      "Loading required package: TxDb.Mmusculus.UCSC.mm9.knownGene\n",
      "Loading required package: GenomicFeatures\n",
      "\n",
      "Attaching package: 'GenomicFeatures'\n",
      "\n",
      "The following object is masked from 'package:topGO':\n",
      "\n",
      "    genes\n",
      "\n",
      "Loading required package: TxDb.Hsapiens.UCSC.hg19.knownGene\n",
      "Loading required package: org.Mm.eg.db\n",
      "\n",
      "Loading required package: org.Hs.eg.db\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# check if the packages are installed \n",
    "### NOTE: gene object from TxDb mask object with the same name in either `compEpiTools` or `GenomicRanges` associated package called `topGo`\n",
    "require('compEpiTools')\n",
    "require('GenomicRanges')\n",
    "require('TxDb.Mmusculus.UCSC.mm9.knownGene')\n",
    "require('TxDb.Hsapiens.UCSC.hg19.knownGene')\n",
    "require('org.Mm.eg.db')\n",
    "require('org.Hs.eg.db')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "TxDb object:\n",
       "# Db type: TxDb\n",
       "# Supporting package: GenomicFeatures\n",
       "# Data source: UCSC\n",
       "# Genome: mm9\n",
       "# Organism: Mus musculus\n",
       "# Taxonomy ID: 10090\n",
       "# UCSC Table: knownGene\n",
       "# Resource URL: http://genome.ucsc.edu/\n",
       "# Type of Gene ID: Entrez Gene ID\n",
       "# Full dataset: yes\n",
       "# miRBase build ID: NA\n",
       "# transcript_nrow: 55419\n",
       "# exon_nrow: 246570\n",
       "# cds_nrow: 213117\n",
       "# Db created by: GenomicFeatures package from Bioconductor\n",
       "# Creation time: 2015-10-07 18:13:02 +0000 (Wed, 07 Oct 2015)\n",
       "# GenomicFeatures version at creation time: 1.21.30\n",
       "# RSQLite version at creation time: 1.0.0\n",
       "# DBSCHEMAVERSION: 1.1"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# store `knownGene`, containing all known mouse gene from UCSC to the `txdb`   \n",
    "txdb = TxDb.Mmusculus.UCSC.mm9.knownGene\n",
    "txdb"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "TxDb object:\n",
       "# Db type: TxDb\n",
       "# Supporting package: GenomicFeatures\n",
       "# Data source: UCSC\n",
       "# Genome: hg19\n",
       "# Organism: Homo sapiens\n",
       "# Taxonomy ID: 9606\n",
       "# UCSC Table: knownGene\n",
       "# Resource URL: http://genome.ucsc.edu/\n",
       "# Type of Gene ID: Entrez Gene ID\n",
       "# Full dataset: yes\n",
       "# miRBase build ID: GRCh37\n",
       "# transcript_nrow: 82960\n",
       "# exon_nrow: 289969\n",
       "# cds_nrow: 237533\n",
       "# Db created by: GenomicFeatures package from Bioconductor\n",
       "# Creation time: 2015-10-07 18:11:28 +0000 (Wed, 07 Oct 2015)\n",
       "# GenomicFeatures version at creation time: 1.21.30\n",
       "# RSQLite version at creation time: 1.0.0\n",
       "# DBSCHEMAVERSION: 1.1"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# store `knownGene`, containing all known human gene from UCSC to the `txdb.has`\n",
    "txdb.hsa = TxDb.Hsapiens.UCSC.hg19.knownGene\n",
    "txdb.hsa"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## - Read in `BAM` file from `Rsamtools` package:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "bampath = system.file('extdata', 'ex1.bam', package='Rsamtools')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "'C:/Users/mustafa.albahrani/Documents/R/win-library/3.4/Rsamtools/extdata/ex1.bam'"
      ],
      "text/latex": [
       "'C:/Users/mustafa.albahrani/Documents/R/win-library/3.4/Rsamtools/extdata/ex1.bam'"
      ],
      "text/markdown": [
       "'C:/Users/mustafa.albahrani/Documents/R/win-library/3.4/Rsamtools/extdata/ex1.bam'"
      ],
      "text/plain": [
       "[1] \"C:/Users/mustafa.albahrani/Documents/R/win-library/3.4/Rsamtools/extdata/ex1.bam\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "bampath"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## - Define `IRanges` and `GRanges` objects"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "IRanges object with 2 ranges and 0 metadata columns:\n",
       "          start       end     width\n",
       "      <integer> <integer> <integer>\n",
       "  [1]      1000      2000      1001\n",
       "  [2]       100      1000       901"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# use `IRanges` object and assign the `start` and `end` intervals for two ranges  \n",
    "ir = IRanges(start=c(1000,100), end=c(2000,1000))\n",
    "ir"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "GRanges object with 2 ranges and 0 metadata columns:\n",
       "      seqnames       ranges strand\n",
       "         <Rle>    <IRanges>  <Rle>\n",
       "  [1]     seq1 [1000, 2000]      +\n",
       "  [2]     seq2 [ 100, 1000]      +\n",
       "  -------\n",
       "  seqinfo: 2 sequences from an unspecified genome; no seqlengths"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# pass the `seqnames, ranges, and strand` parameters/objects to `GRanges` object to assign names of each interval and the direction of the strand \n",
    "gr = GRanges(seqnames = Rle(c('seq1', 'seq2')), ranges = ir, strand = '+') #NOTE we are passing Rle() object containing a vector of sequence names to `seqnames` argument \n",
    "# gr = GRanges(seqnames = c('seq1', 'seq2'), ranges = ir, strand = '+') # another way of doing the same thing; without using Rle()\n",
    "\n",
    "gr"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## - Define `GRbaseCoverage` object "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<ol>\n",
       "\t<li><ol class=list-inline>\n",
       "\t<li>35</li>\n",
       "\t<li>35</li>\n",
       "\t<li>38</li>\n",
       "\t<li>38</li>\n",
       "\t<li>39</li>\n",
       "\t<li>39</li>\n",
       "\t<li>39</li>\n",
       "\t<li>38</li>\n",
       "\t<li>39</li>\n",
       "\t<li>39</li>\n",
       "\t<li>39</li>\n",
       "\t<li>39</li>\n",
       "\t<li>33</li>\n",
       "\t<li>40</li>\n",
       "\t<li>41</li>\n",
       "\t<li>39</li>\n",
       "\t<li>42</li>\n",
       "\t<li>36</li>\n",
       "\t<li>40</li>\n",
       "\t<li>42</li>\n",
       "\t<li>41</li>\n",
       "\t<li>40</li>\n",
       "\t<li>40</li>\n",
       "\t<li>41</li>\n",
       "\t<li>39</li>\n",
       "\t<li>41</li>\n",
       "\t<li>42</li>\n",
       "\t<li>40</li>\n",
       "\t<li>37</li>\n",
       "\t<li>37</li>\n",
       "\t<li>39</li>\n",
       "\t<li>37</li>\n",
       "\t<li>37</li>\n",
       "\t<li>40</li>\n",
       "\t<li>36</li>\n",
       "\t<li>37</li>\n",
       "\t<li>38</li>\n",
       "\t<li>37</li>\n",
       "\t<li>38</li>\n",
       "\t<li>37</li>\n",
       "\t<li>37</li>\n",
       "\t<li>38</li>\n",
       "\t<li>38</li>\n",
       "\t<li>38</li>\n",
       "\t<li>39</li>\n",
       "\t<li>37</li>\n",
       "\t<li>37</li>\n",
       "\t<li>36</li>\n",
       "\t<li>38</li>\n",
       "\t<li>37</li>\n",
       "\t<li>36</li>\n",
       "\t<li>36</li>\n",
       "\t<li>39</li>\n",
       "\t<li>40</li>\n",
       "\t<li>33</li>\n",
       "\t<li>33</li>\n",
       "\t<li>35</li>\n",
       "\t<li>35</li>\n",
       "\t<li>37</li>\n",
       "\t<li>37</li>\n",
       "\t<li>36</li>\n",
       "\t<li>33</li>\n",
       "\t<li>35</li>\n",
       "\t<li>33</li>\n",
       "\t<li>36</li>\n",
       "\t<li>35</li>\n",
       "\t<li>36</li>\n",
       "\t<li>35</li>\n",
       "\t<li>36</li>\n",
       "\t<li>37</li>\n",
       "\t<li>39</li>\n",
       "\t<li>41</li>\n",
       "\t<li>41</li>\n",
       "\t<li>43</li>\n",
       "\t<li>45</li>\n",
       "\t<li>47</li>\n",
       "\t<li>47</li>\n",
       "\t<li>47</li>\n",
       "\t<li>47</li>\n",
       "\t<li>46</li>\n",
       "\t<li>48</li>\n",
       "\t<li>49</li>\n",
       "\t<li>52</li>\n",
       "\t<li>50</li>\n",
       "\t<li>50</li>\n",
       "\t<li>50</li>\n",
       "\t<li>51</li>\n",
       "\t<li>49</li>\n",
       "\t<li>48</li>\n",
       "\t<li>51</li>\n",
       "\t<li>50</li>\n",
       "\t<li>50</li>\n",
       "\t<li>49</li>\n",
       "\t<li>49</li>\n",
       "\t<li>47</li>\n",
       "\t<li>51</li>\n",
       "\t<li>49</li>\n",
       "\t<li>51</li>\n",
       "\t<li>55</li>\n",
       "\t<li>53</li>\n",
       "\t<li>54</li>\n",
       "\t<li>52</li>\n",
       "\t<li>55</li>\n",
       "\t<li>56</li>\n",
       "\t<li>49</li>\n",
       "\t<li>55</li>\n",
       "\t<li>50</li>\n",
       "\t<li>52</li>\n",
       "\t<li>47</li>\n",
       "\t<li>48</li>\n",
       "\t<li>52</li>\n",
       "\t<li>51</li>\n",
       "\t<li>53</li>\n",
       "\t<li>54</li>\n",
       "\t<li>49</li>\n",
       "\t<li>49</li>\n",
       "\t<li>50</li>\n",
       "\t<li>46</li>\n",
       "\t<li>45</li>\n",
       "\t<li>43</li>\n",
       "\t<li>46</li>\n",
       "\t<li>48</li>\n",
       "\t<li>48</li>\n",
       "\t<li>43</li>\n",
       "\t<li>44</li>\n",
       "\t<li>43</li>\n",
       "\t<li>45</li>\n",
       "\t<li>44</li>\n",
       "\t<li>41</li>\n",
       "\t<li>40</li>\n",
       "\t<li>40</li>\n",
       "\t<li>39</li>\n",
       "\t<li>40</li>\n",
       "\t<li>37</li>\n",
       "\t<li>34</li>\n",
       "\t<li>35</li>\n",
       "\t<li>35</li>\n",
       "\t<li>34</li>\n",
       "\t<li>37</li>\n",
       "\t<li>39</li>\n",
       "\t<li>35</li>\n",
       "\t<li>36</li>\n",
       "\t<li>36</li>\n",
       "\t<li>36</li>\n",
       "\t<li>34</li>\n",
       "\t<li>30</li>\n",
       "\t<li>32</li>\n",
       "\t<li>29</li>\n",
       "\t<li>28</li>\n",
       "\t<li>30</li>\n",
       "\t<li>31</li>\n",
       "\t<li>32</li>\n",
       "\t<li>30</li>\n",
       "\t<li>33</li>\n",
       "\t<li>32</li>\n",
       "\t<li>32</li>\n",
       "\t<li>32</li>\n",
       "\t<li>30</li>\n",
       "\t<li>34</li>\n",
       "\t<li>35</li>\n",
       "\t<li>38</li>\n",
       "\t<li>38</li>\n",
       "\t<li>39</li>\n",
       "\t<li>40</li>\n",
       "\t<li>37</li>\n",
       "\t<li>42</li>\n",
       "\t<li>42</li>\n",
       "\t<li>43</li>\n",
       "\t<li>44</li>\n",
       "\t<li>44</li>\n",
       "\t<li>44</li>\n",
       "\t<li>43</li>\n",
       "\t<li>44</li>\n",
       "\t<li>43</li>\n",
       "\t<li>41</li>\n",
       "\t<li>41</li>\n",
       "\t<li>43</li>\n",
       "\t<li>41</li>\n",
       "\t<li>42</li>\n",
       "\t<li>40</li>\n",
       "\t<li>42</li>\n",
       "\t<li>41</li>\n",
       "\t<li>43</li>\n",
       "\t<li>42</li>\n",
       "\t<li>42</li>\n",
       "\t<li>40</li>\n",
       "\t<li>39</li>\n",
       "\t<li>40</li>\n",
       "\t<li>40</li>\n",
       "\t<li>40</li>\n",
       "\t<li>39</li>\n",
       "\t<li>39</li>\n",
       "\t<li>38</li>\n",
       "\t<li>39</li>\n",
       "\t<li>36</li>\n",
       "\t<li>34</li>\n",
       "\t<li>32</li>\n",
       "\t<li>36</li>\n",
       "\t<li>36</li>\n",
       "\t<li>35</li>\n",
       "\t<li>32</li>\n",
       "\t<li>33</li>\n",
       "\t<li>33</li>\n",
       "\t<li>31</li>\n",
       "\t<li>31</li>\n",
       "\t<li>31</li>\n",
       "\t<li>32</li>\n",
       "\t<li>29</li>\n",
       "\t<li>30</li>\n",
       "\t<li>31</li>\n",
       "\t<li>25</li>\n",
       "\t<li>30</li>\n",
       "\t<li>32</li>\n",
       "\t<li>30</li>\n",
       "\t<li>25</li>\n",
       "\t<li>33</li>\n",
       "\t<li>31</li>\n",
       "\t<li>29</li>\n",
       "\t<li>29</li>\n",
       "\t<li>32</li>\n",
       "\t<li>32</li>\n",
       "\t<li>31</li>\n",
       "\t<li>30</li>\n",
       "\t<li>30</li>\n",
       "\t<li>32</li>\n",
       "\t<li>29</li>\n",
       "\t<li>31</li>\n",
       "\t<li>28</li>\n",
       "\t<li>28</li>\n",
       "\t<li>30</li>\n",
       "\t<li>28</li>\n",
       "\t<li>28</li>\n",
       "\t<li>28</li>\n",
       "\t<li>33</li>\n",
       "\t<li>32</li>\n",
       "\t<li>30</li>\n",
       "\t<li>31</li>\n",
       "\t<li>32</li>\n",
       "\t<li>30</li>\n",
       "\t<li>31</li>\n",
       "\t<li>31</li>\n",
       "\t<li>29</li>\n",
       "\t<li>33</li>\n",
       "\t<li>32</li>\n",
       "\t<li>35</li>\n",
       "\t<li>36</li>\n",
       "\t<li>36</li>\n",
       "\t<li>35</li>\n",
       "\t<li>34</li>\n",
       "\t<li>33</li>\n",
       "\t<li>31</li>\n",
       "\t<li>31</li>\n",
       "\t<li>35</li>\n",
       "\t<li>33</li>\n",
       "\t<li>37</li>\n",
       "\t<li>37</li>\n",
       "\t<li>37</li>\n",
       "\t<li>39</li>\n",
       "\t<li>39</li>\n",
       "\t<li>38</li>\n",
       "\t<li>39</li>\n",
       "\t<li>36</li>\n",
       "\t<li>40</li>\n",
       "\t<li>37</li>\n",
       "\t<li>39</li>\n",
       "\t<li>42</li>\n",
       "\t<li>43</li>\n",
       "\t<li>41</li>\n",
       "\t<li>40</li>\n",
       "\t<li>41</li>\n",
       "\t<li>42</li>\n",
       "\t<li>44</li>\n",
       "\t<li>43</li>\n",
       "\t<li>43</li>\n",
       "\t<li>42</li>\n",
       "\t<li>41</li>\n",
       "\t<li>39</li>\n",
       "\t<li>38</li>\n",
       "\t<li>39</li>\n",
       "\t<li>38</li>\n",
       "\t<li>38</li>\n",
       "\t<li>37</li>\n",
       "\t<li>35</li>\n",
       "\t<li>36</li>\n",
       "\t<li>34</li>\n",
       "\t<li>38</li>\n",
       "\t<li>37</li>\n",
       "\t<li>36</li>\n",
       "\t<li>39</li>\n",
       "\t<li>40</li>\n",
       "\t<li>39</li>\n",
       "\t<li>41</li>\n",
       "\t<li>40</li>\n",
       "\t<li>39</li>\n",
       "\t<li>37</li>\n",
       "\t<li>39</li>\n",
       "\t<li>42</li>\n",
       "\t<li>46</li>\n",
       "\t<li>45</li>\n",
       "\t<li>47</li>\n",
       "\t<li>43</li>\n",
       "\t<li>44</li>\n",
       "\t<li>45</li>\n",
       "\t<li>44</li>\n",
       "\t<li>44</li>\n",
       "\t<li>45</li>\n",
       "\t<li>44</li>\n",
       "\t<li>45</li>\n",
       "\t<li>46</li>\n",
       "\t<li>45</li>\n",
       "\t<li>45</li>\n",
       "\t<li>46</li>\n",
       "\t<li>46</li>\n",
       "\t<li>49</li>\n",
       "\t<li>52</li>\n",
       "\t<li>51</li>\n",
       "\t<li>48</li>\n",
       "\t<li>49</li>\n",
       "\t<li>51</li>\n",
       "\t<li>51</li>\n",
       "\t<li>51</li>\n",
       "\t<li>54</li>\n",
       "\t<li>54</li>\n",
       "\t<li>50</li>\n",
       "\t<li>52</li>\n",
       "\t<li>54</li>\n",
       "\t<li>57</li>\n",
       "\t<li>59</li>\n",
       "\t<li>60</li>\n",
       "\t<li>59</li>\n",
       "\t<li>56</li>\n",
       "\t<li>52</li>\n",
       "\t<li>59</li>\n",
       "\t<li>56</li>\n",
       "\t<li>51</li>\n",
       "\t<li>57</li>\n",
       "\t<li>57</li>\n",
       "\t<li>59</li>\n",
       "\t<li>59</li>\n",
       "\t<li>59</li>\n",
       "\t<li>62</li>\n",
       "\t<li>60</li>\n",
       "\t<li>57</li>\n",
       "\t<li>59</li>\n",
       "\t<li>58</li>\n",
       "\t<li>60</li>\n",
       "\t<li>58</li>\n",
       "\t<li>57</li>\n",
       "\t<li>58</li>\n",
       "\t<li>57</li>\n",
       "\t<li>56</li>\n",
       "\t<li>58</li>\n",
       "\t<li>57</li>\n",
       "\t<li>56</li>\n",
       "\t<li>58</li>\n",
       "\t<li>53</li>\n",
       "\t<li>54</li>\n",
       "\t<li>49</li>\n",
       "\t<li>48</li>\n",
       "\t<li>51</li>\n",
       "\t<li>50</li>\n",
       "\t<li>47</li>\n",
       "\t<li>39</li>\n",
       "\t<li>41</li>\n",
       "\t<li>44</li>\n",
       "\t<li>44</li>\n",
       "\t<li>41</li>\n",
       "\t<li>41</li>\n",
       "\t<li>40</li>\n",
       "\t<li>39</li>\n",
       "\t<li>39</li>\n",
       "\t<li>38</li>\n",
       "\t<li>34</li>\n",
       "\t<li>33</li>\n",
       "\t<li>41</li>\n",
       "\t<li>38</li>\n",
       "\t<li>36</li>\n",
       "\t<li>36</li>\n",
       "\t<li>33</li>\n",
       "\t<li>33</li>\n",
       "\t<li>28</li>\n",
       "\t<li>33</li>\n",
       "\t<li>32</li>\n",
       "\t<li>29</li>\n",
       "\t<li>27</li>\n",
       "\t<li>26</li>\n",
       "\t<li>28</li>\n",
       "\t<li>26</li>\n",
       "\t<li>24</li>\n",
       "\t<li>27</li>\n",
       "\t<li>25</li>\n",
       "\t<li>24</li>\n",
       "\t<li>20</li>\n",
       "\t<li>25</li>\n",
       "\t<li>26</li>\n",
       "\t<li>25</li>\n",
       "\t<li>24</li>\n",
       "\t<li>19</li>\n",
       "\t<li>21</li>\n",
       "\t<li>23</li>\n",
       "\t<li>19</li>\n",
       "\t<li>18</li>\n",
       "\t<li>23</li>\n",
       "\t<li>22</li>\n",
       "\t<li>22</li>\n",
       "\t<li>22</li>\n",
       "\t<li>21</li>\n",
       "\t<li>18</li>\n",
       "\t<li>16</li>\n",
       "\t<li>14</li>\n",
       "\t<li>15</li>\n",
       "\t<li>15</li>\n",
       "\t<li>16</li>\n",
       "\t<li>16</li>\n",
       "\t<li>15</li>\n",
       "\t<li>15</li>\n",
       "\t<li>14</li>\n",
       "\t<li>12</li>\n",
       "\t<li>15</li>\n",
       "\t<li>16</li>\n",
       "\t<li>16</li>\n",
       "\t<li>15</li>\n",
       "\t<li>16</li>\n",
       "\t<li>15</li>\n",
       "\t<li>15</li>\n",
       "\t<li>13</li>\n",
       "\t<li>14</li>\n",
       "\t<li>15</li>\n",
       "\t<li>17</li>\n",
       "\t<li>15</li>\n",
       "\t<li>16</li>\n",
       "\t<li>15</li>\n",
       "\t<li>16</li>\n",
       "\t<li>14</li>\n",
       "\t<li>13</li>\n",
       "\t<li>15</li>\n",
       "\t<li>16</li>\n",
       "\t<li>16</li>\n",
       "\t<li>16</li>\n",
       "\t<li>16</li>\n",
       "\t<li>15</li>\n",
       "\t<li>15</li>\n",
       "\t<li>16</li>\n",
       "\t<li>15</li>\n",
       "\t<li>16</li>\n",
       "\t<li>15</li>\n",
       "\t<li>16</li>\n",
       "\t<li>16</li>\n",
       "\t<li>17</li>\n",
       "\t<li>18</li>\n",
       "\t<li>18</li>\n",
       "\t<li>18</li>\n",
       "\t<li>18</li>\n",
       "\t<li>17</li>\n",
       "\t<li>17</li>\n",
       "\t<li>17</li>\n",
       "\t<li>16</li>\n",
       "\t<li>16</li>\n",
       "\t<li>16</li>\n",
       "\t<li>16</li>\n",
       "\t<li>16</li>\n",
       "\t<li>18</li>\n",
       "\t<li>17</li>\n",
       "\t<li>15</li>\n",
       "\t<li>15</li>\n",
       "\t<li>18</li>\n",
       "\t<li>18</li>\n",
       "\t<li>19</li>\n",
       "\t<li>19</li>\n",
       "\t<li>21</li>\n",
       "\t<li>19</li>\n",
       "\t<li>17</li>\n",
       "\t<li>19</li>\n",
       "\t<li>20</li>\n",
       "\t<li>22</li>\n",
       "\t<li>20</li>\n",
       "\t<li>22</li>\n",
       "\t<li>20</li>\n",
       "\t<li>21</li>\n",
       "\t<li>21</li>\n",
       "\t<li>25</li>\n",
       "\t<li>26</li>\n",
       "\t<li>27</li>\n",
       "\t<li>27</li>\n",
       "\t<li>26</li>\n",
       "\t<li>29</li>\n",
       "\t<li>32</li>\n",
       "\t<li>34</li>\n",
       "\t<li>35</li>\n",
       "\t<li>36</li>\n",
       "\t<li>35</li>\n",
       "\t<li>32</li>\n",
       "\t<li>35</li>\n",
       "\t<li>37</li>\n",
       "\t<li>39</li>\n",
       "\t<li>37</li>\n",
       "\t<li>37</li>\n",
       "\t<li>37</li>\n",
       "\t<li>37</li>\n",
       "\t<li>39</li>\n",
       "\t<li>39</li>\n",
       "\t<li>43</li>\n",
       "\t<li>44</li>\n",
       "\t<li>44</li>\n",
       "\t<li>42</li>\n",
       "\t<li>40</li>\n",
       "\t<li>40</li>\n",
       "\t<li>44</li>\n",
       "\t<li>40</li>\n",
       "\t<li>40</li>\n",
       "\t<li>39</li>\n",
       "\t<li>38</li>\n",
       "\t<li>46</li>\n",
       "\t<li>46</li>\n",
       "\t<li>47</li>\n",
       "\t<li>46</li>\n",
       "\t<li>46</li>\n",
       "\t<li>46</li>\n",
       "\t<li>45</li>\n",
       "\t<li>44</li>\n",
       "\t<li>40</li>\n",
       "\t<li>39</li>\n",
       "\t<li>39</li>\n",
       "\t<li>39</li>\n",
       "\t<li>39</li>\n",
       "\t<li>41</li>\n",
       "\t<li>41</li>\n",
       "\t<li>43</li>\n",
       "\t<li>43</li>\n",
       "\t<li>44</li>\n",
       "\t<li>43</li>\n",
       "\t<li>44</li>\n",
       "\t<li>44</li>\n",
       "\t<li>45</li>\n",
       "\t<li>43</li>\n",
       "\t<li>42</li>\n",
       "\t<li>39</li>\n",
       "\t<li>36</li>\n",
       "\t<li>36</li>\n",
       "\t<li>36</li>\n",
       "\t<li>35</li>\n",
       "\t<li>35</li>\n",
       "\t<li>33</li>\n",
       "\t<li>32</li>\n",
       "\t<li>33</li>\n",
       "\t<li>30</li>\n",
       "\t<li>29</li>\n",
       "\t<li>29</li>\n",
       "\t<li>26</li>\n",
       "\t<li>25</li>\n",
       "\t<li>21</li>\n",
       "\t<li>21</li>\n",
       "\t<li>20</li>\n",
       "\t<li>19</li>\n",
       "\t<li>19</li>\n",
       "\t<li>18</li>\n",
       "\t<li>16</li>\n",
       "\t<li>16</li>\n",
       "\t<li>15</li>\n",
       "\t<li>13</li>\n",
       "\t<li>12</li>\n",
       "\t<li>12</li>\n",
       "\t<li>10</li>\n",
       "\t<li>9</li>\n",
       "\t<li>7</li>\n",
       "\t<li>6</li>\n",
       "\t<li>5</li>\n",
       "\t<li>3</li>\n",
       "\t<li>2</li>\n",
       "\t<li>1</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "</ol>\n",
       "</li>\n",
       "\t<li><ol class=list-inline>\n",
       "\t<li>27</li>\n",
       "\t<li>25</li>\n",
       "\t<li>26</li>\n",
       "\t<li>23</li>\n",
       "\t<li>24</li>\n",
       "\t<li>25</li>\n",
       "\t<li>28</li>\n",
       "\t<li>28</li>\n",
       "\t<li>28</li>\n",
       "\t<li>28</li>\n",
       "\t<li>29</li>\n",
       "\t<li>29</li>\n",
       "\t<li>27</li>\n",
       "\t<li>27</li>\n",
       "\t<li>28</li>\n",
       "\t<li>30</li>\n",
       "\t<li>28</li>\n",
       "\t<li>28</li>\n",
       "\t<li>28</li>\n",
       "\t<li>26</li>\n",
       "\t<li>26</li>\n",
       "\t<li>27</li>\n",
       "\t<li>25</li>\n",
       "\t<li>28</li>\n",
       "\t<li>30</li>\n",
       "\t<li>30</li>\n",
       "\t<li>28</li>\n",
       "\t<li>28</li>\n",
       "\t<li>26</li>\n",
       "\t<li>25</li>\n",
       "\t<li>22</li>\n",
       "\t<li>22</li>\n",
       "\t<li>20</li>\n",
       "\t<li>19</li>\n",
       "\t<li>18</li>\n",
       "\t<li>19</li>\n",
       "\t<li>19</li>\n",
       "\t<li>19</li>\n",
       "\t<li>18</li>\n",
       "\t<li>18</li>\n",
       "\t<li>17</li>\n",
       "\t<li>16</li>\n",
       "\t<li>15</li>\n",
       "\t<li>14</li>\n",
       "\t<li>13</li>\n",
       "\t<li>13</li>\n",
       "\t<li>11</li>\n",
       "\t<li>11</li>\n",
       "\t<li>13</li>\n",
       "\t<li>11</li>\n",
       "\t<li>12</li>\n",
       "\t<li>12</li>\n",
       "\t<li>12</li>\n",
       "\t<li>12</li>\n",
       "\t<li>13</li>\n",
       "\t<li>12</li>\n",
       "\t<li>11</li>\n",
       "\t<li>10</li>\n",
       "\t<li>8</li>\n",
       "\t<li>7</li>\n",
       "\t<li>6</li>\n",
       "\t<li>7</li>\n",
       "\t<li>7</li>\n",
       "\t<li>9</li>\n",
       "\t<li>9</li>\n",
       "\t<li>11</li>\n",
       "\t<li>11</li>\n",
       "\t<li>12</li>\n",
       "\t<li>12</li>\n",
       "\t<li>13</li>\n",
       "\t<li>13</li>\n",
       "\t<li>14</li>\n",
       "\t<li>14</li>\n",
       "\t<li>14</li>\n",
       "\t<li>14</li>\n",
       "\t<li>15</li>\n",
       "\t<li>15</li>\n",
       "\t<li>14</li>\n",
       "\t<li>16</li>\n",
       "\t<li>16</li>\n",
       "\t<li>17</li>\n",
       "\t<li>15</li>\n",
       "\t<li>16</li>\n",
       "\t<li>16</li>\n",
       "\t<li>17</li>\n",
       "\t<li>17</li>\n",
       "\t<li>16</li>\n",
       "\t<li>18</li>\n",
       "\t<li>19</li>\n",
       "\t<li>19</li>\n",
       "\t<li>19</li>\n",
       "\t<li>19</li>\n",
       "\t<li>19</li>\n",
       "\t<li>20</li>\n",
       "\t<li>20</li>\n",
       "\t<li>20</li>\n",
       "\t<li>19</li>\n",
       "\t<li>21</li>\n",
       "\t<li>19</li>\n",
       "\t<li>19</li>\n",
       "\t<li>20</li>\n",
       "\t<li>22</li>\n",
       "\t<li>22</li>\n",
       "\t<li>24</li>\n",
       "\t<li>22</li>\n",
       "\t<li>23</li>\n",
       "\t<li>24</li>\n",
       "\t<li>25</li>\n",
       "\t<li>25</li>\n",
       "\t<li>27</li>\n",
       "\t<li>28</li>\n",
       "\t<li>28</li>\n",
       "\t<li>30</li>\n",
       "\t<li>31</li>\n",
       "\t<li>31</li>\n",
       "\t<li>32</li>\n",
       "\t<li>34</li>\n",
       "\t<li>34</li>\n",
       "\t<li>35</li>\n",
       "\t<li>33</li>\n",
       "\t<li>35</li>\n",
       "\t<li>37</li>\n",
       "\t<li>39</li>\n",
       "\t<li>39</li>\n",
       "\t<li>42</li>\n",
       "\t<li>43</li>\n",
       "\t<li>46</li>\n",
       "\t<li>47</li>\n",
       "\t<li>46</li>\n",
       "\t<li>50</li>\n",
       "\t<li>51</li>\n",
       "\t<li>49</li>\n",
       "\t<li>45</li>\n",
       "\t<li>50</li>\n",
       "\t<li>53</li>\n",
       "\t<li>51</li>\n",
       "\t<li>48</li>\n",
       "\t<li>51</li>\n",
       "\t<li>50</li>\n",
       "\t<li>50</li>\n",
       "\t<li>53</li>\n",
       "\t<li>52</li>\n",
       "\t<li>52</li>\n",
       "\t<li>52</li>\n",
       "\t<li>50</li>\n",
       "\t<li>52</li>\n",
       "\t<li>54</li>\n",
       "\t<li>54</li>\n",
       "\t<li>55</li>\n",
       "\t<li>55</li>\n",
       "\t<li>58</li>\n",
       "\t<li>55</li>\n",
       "\t<li>55</li>\n",
       "\t<li>55</li>\n",
       "\t<li>54</li>\n",
       "\t<li>54</li>\n",
       "\t<li>52</li>\n",
       "\t<li>49</li>\n",
       "\t<li>48</li>\n",
       "\t<li>48</li>\n",
       "\t<li>46</li>\n",
       "\t<li>46</li>\n",
       "\t<li>44</li>\n",
       "\t<li>45</li>\n",
       "\t<li>44</li>\n",
       "\t<li>45</li>\n",
       "\t<li>48</li>\n",
       "\t<li>48</li>\n",
       "\t<li>47</li>\n",
       "\t<li>48</li>\n",
       "\t<li>50</li>\n",
       "\t<li>53</li>\n",
       "\t<li>51</li>\n",
       "\t<li>53</li>\n",
       "\t<li>52</li>\n",
       "\t<li>50</li>\n",
       "\t<li>51</li>\n",
       "\t<li>52</li>\n",
       "\t<li>53</li>\n",
       "\t<li>56</li>\n",
       "\t<li>55</li>\n",
       "\t<li>54</li>\n",
       "\t<li>53</li>\n",
       "\t<li>54</li>\n",
       "\t<li>53</li>\n",
       "\t<li>52</li>\n",
       "\t<li>51</li>\n",
       "\t<li>50</li>\n",
       "\t<li>50</li>\n",
       "\t<li>53</li>\n",
       "\t<li>52</li>\n",
       "\t<li>57</li>\n",
       "\t<li>56</li>\n",
       "\t<li>54</li>\n",
       "\t<li>54</li>\n",
       "\t<li>56</li>\n",
       "\t<li>57</li>\n",
       "\t<li>57</li>\n",
       "\t<li>56</li>\n",
       "\t<li>60</li>\n",
       "\t<li>58</li>\n",
       "\t<li>54</li>\n",
       "\t<li>55</li>\n",
       "\t<li>55</li>\n",
       "\t<li>50</li>\n",
       "\t<li>48</li>\n",
       "\t<li>48</li>\n",
       "\t<li>50</li>\n",
       "\t<li>51</li>\n",
       "\t<li>50</li>\n",
       "\t<li>51</li>\n",
       "\t<li>51</li>\n",
       "\t<li>49</li>\n",
       "\t<li>48</li>\n",
       "\t<li>49</li>\n",
       "\t<li>46</li>\n",
       "\t<li>46</li>\n",
       "\t<li>48</li>\n",
       "\t<li>46</li>\n",
       "\t<li>46</li>\n",
       "\t<li>45</li>\n",
       "\t<li>44</li>\n",
       "\t<li>43</li>\n",
       "\t<li>44</li>\n",
       "\t<li>45</li>\n",
       "\t<li>45</li>\n",
       "\t<li>42</li>\n",
       "\t<li>43</li>\n",
       "\t<li>46</li>\n",
       "\t<li>46</li>\n",
       "\t<li>46</li>\n",
       "\t<li>48</li>\n",
       "\t<li>50</li>\n",
       "\t<li>51</li>\n",
       "\t<li>48</li>\n",
       "\t<li>49</li>\n",
       "\t<li>48</li>\n",
       "\t<li>49</li>\n",
       "\t<li>50</li>\n",
       "\t<li>49</li>\n",
       "\t<li>47</li>\n",
       "\t<li>52</li>\n",
       "\t<li>52</li>\n",
       "\t<li>49</li>\n",
       "\t<li>48</li>\n",
       "\t<li>50</li>\n",
       "\t<li>50</li>\n",
       "\t<li>50</li>\n",
       "\t<li>48</li>\n",
       "\t<li>49</li>\n",
       "\t<li>49</li>\n",
       "\t<li>50</li>\n",
       "\t<li>48</li>\n",
       "\t<li>49</li>\n",
       "\t<li>48</li>\n",
       "\t<li>48</li>\n",
       "\t<li>50</li>\n",
       "\t<li>49</li>\n",
       "\t<li>50</li>\n",
       "\t<li>47</li>\n",
       "\t<li>51</li>\n",
       "\t<li>46</li>\n",
       "\t<li>52</li>\n",
       "\t<li>52</li>\n",
       "\t<li>50</li>\n",
       "\t<li>48</li>\n",
       "\t<li>45</li>\n",
       "\t<li>45</li>\n",
       "\t<li>44</li>\n",
       "\t<li>44</li>\n",
       "\t<li>42</li>\n",
       "\t<li>44</li>\n",
       "\t<li>45</li>\n",
       "\t<li>44</li>\n",
       "\t<li>41</li>\n",
       "\t<li>42</li>\n",
       "\t<li>42</li>\n",
       "\t<li>42</li>\n",
       "\t<li>44</li>\n",
       "\t<li>42</li>\n",
       "\t<li>43</li>\n",
       "\t<li>48</li>\n",
       "\t<li>50</li>\n",
       "\t<li>50</li>\n",
       "\t<li>51</li>\n",
       "\t<li>49</li>\n",
       "\t<li>48</li>\n",
       "\t<li>48</li>\n",
       "\t<li>53</li>\n",
       "\t<li>53</li>\n",
       "\t<li>57</li>\n",
       "\t<li>57</li>\n",
       "\t<li>56</li>\n",
       "\t<li>53</li>\n",
       "\t<li>56</li>\n",
       "\t<li>50</li>\n",
       "\t<li>51</li>\n",
       "\t<li>51</li>\n",
       "\t<li>49</li>\n",
       "\t<li>52</li>\n",
       "\t<li>50</li>\n",
       "\t<li>52</li>\n",
       "\t<li>49</li>\n",
       "\t<li>50</li>\n",
       "\t<li>46</li>\n",
       "\t<li>47</li>\n",
       "\t<li>45</li>\n",
       "\t<li>46</li>\n",
       "\t<li>48</li>\n",
       "\t<li>48</li>\n",
       "\t<li>50</li>\n",
       "\t<li>49</li>\n",
       "\t<li>50</li>\n",
       "\t<li>52</li>\n",
       "\t<li>49</li>\n",
       "\t<li>51</li>\n",
       "\t<li>49</li>\n",
       "\t<li>44</li>\n",
       "\t<li>43</li>\n",
       "\t<li>40</li>\n",
       "\t<li>42</li>\n",
       "\t<li>44</li>\n",
       "\t<li>43</li>\n",
       "\t<li>39</li>\n",
       "\t<li>39</li>\n",
       "\t<li>34</li>\n",
       "\t<li>34</li>\n",
       "\t<li>34</li>\n",
       "\t<li>34</li>\n",
       "\t<li>34</li>\n",
       "\t<li>35</li>\n",
       "\t<li>34</li>\n",
       "\t<li>37</li>\n",
       "\t<li>37</li>\n",
       "\t<li>40</li>\n",
       "\t<li>41</li>\n",
       "\t<li>39</li>\n",
       "\t<li>40</li>\n",
       "\t<li>41</li>\n",
       "\t<li>44</li>\n",
       "\t<li>43</li>\n",
       "\t<li>41</li>\n",
       "\t<li>44</li>\n",
       "\t<li>45</li>\n",
       "\t<li>42</li>\n",
       "\t<li>47</li>\n",
       "\t<li>43</li>\n",
       "\t<li>49</li>\n",
       "\t<li>49</li>\n",
       "\t<li>49</li>\n",
       "\t<li>49</li>\n",
       "\t<li>50</li>\n",
       "\t<li>50</li>\n",
       "\t<li>54</li>\n",
       "\t<li>55</li>\n",
       "\t<li>54</li>\n",
       "\t<li>54</li>\n",
       "\t<li>56</li>\n",
       "\t<li>62</li>\n",
       "\t<li>59</li>\n",
       "\t<li>58</li>\n",
       "\t<li>61</li>\n",
       "\t<li>60</li>\n",
       "\t<li>58</li>\n",
       "\t<li>62</li>\n",
       "\t<li>63</li>\n",
       "\t<li>60</li>\n",
       "\t<li>64</li>\n",
       "\t<li>60</li>\n",
       "\t<li>58</li>\n",
       "\t<li>60</li>\n",
       "\t<li>61</li>\n",
       "\t<li>60</li>\n",
       "\t<li>58</li>\n",
       "\t<li>55</li>\n",
       "\t<li>54</li>\n",
       "\t<li>54</li>\n",
       "\t<li>55</li>\n",
       "\t<li>52</li>\n",
       "\t<li>54</li>\n",
       "\t<li>49</li>\n",
       "\t<li>49</li>\n",
       "\t<li>48</li>\n",
       "\t<li>46</li>\n",
       "\t<li>48</li>\n",
       "\t<li>47</li>\n",
       "\t<li>46</li>\n",
       "\t<li>46</li>\n",
       "\t<li>45</li>\n",
       "\t<li>45</li>\n",
       "\t<li>48</li>\n",
       "\t<li>45</li>\n",
       "\t<li>44</li>\n",
       "\t<li>39</li>\n",
       "\t<li>42</li>\n",
       "\t<li>42</li>\n",
       "\t<li>43</li>\n",
       "\t<li>41</li>\n",
       "\t<li>42</li>\n",
       "\t<li>41</li>\n",
       "\t<li>44</li>\n",
       "\t<li>43</li>\n",
       "\t<li>41</li>\n",
       "\t<li>43</li>\n",
       "\t<li>43</li>\n",
       "\t<li>47</li>\n",
       "\t<li>43</li>\n",
       "\t<li>45</li>\n",
       "\t<li>46</li>\n",
       "\t<li>47</li>\n",
       "\t<li>47</li>\n",
       "\t<li>48</li>\n",
       "\t<li>47</li>\n",
       "\t<li>48</li>\n",
       "\t<li>48</li>\n",
       "\t<li>45</li>\n",
       "\t<li>48</li>\n",
       "\t<li>48</li>\n",
       "\t<li>48</li>\n",
       "\t<li>49</li>\n",
       "\t<li>48</li>\n",
       "\t<li>49</li>\n",
       "\t<li>51</li>\n",
       "\t<li>49</li>\n",
       "\t<li>49</li>\n",
       "\t<li>46</li>\n",
       "\t<li>44</li>\n",
       "\t<li>45</li>\n",
       "\t<li>44</li>\n",
       "\t<li>46</li>\n",
       "\t<li>46</li>\n",
       "\t<li>44</li>\n",
       "\t<li>44</li>\n",
       "\t<li>42</li>\n",
       "\t<li>42</li>\n",
       "\t<li>41</li>\n",
       "\t<li>41</li>\n",
       "\t<li>43</li>\n",
       "\t<li>42</li>\n",
       "\t<li>41</li>\n",
       "\t<li>39</li>\n",
       "\t<li>38</li>\n",
       "\t<li>42</li>\n",
       "\t<li>41</li>\n",
       "\t<li>37</li>\n",
       "\t<li>38</li>\n",
       "\t<li>43</li>\n",
       "\t<li>38</li>\n",
       "\t<li>40</li>\n",
       "\t<li>41</li>\n",
       "\t<li>37</li>\n",
       "\t<li>41</li>\n",
       "\t<li>43</li>\n",
       "\t<li>42</li>\n",
       "\t<li>43</li>\n",
       "\t<li>42</li>\n",
       "\t<li>39</li>\n",
       "\t<li>41</li>\n",
       "\t<li>41</li>\n",
       "\t<li>43</li>\n",
       "\t<li>48</li>\n",
       "\t<li>51</li>\n",
       "\t<li>53</li>\n",
       "\t<li>53</li>\n",
       "\t<li>56</li>\n",
       "\t<li>53</li>\n",
       "\t<li>53</li>\n",
       "\t<li>56</li>\n",
       "\t<li>59</li>\n",
       "\t<li>61</li>\n",
       "\t<li>58</li>\n",
       "\t<li>60</li>\n",
       "\t<li>60</li>\n",
       "\t<li>57</li>\n",
       "\t<li>56</li>\n",
       "\t<li>56</li>\n",
       "\t<li>57</li>\n",
       "\t<li>55</li>\n",
       "\t<li>55</li>\n",
       "\t<li>59</li>\n",
       "\t<li>59</li>\n",
       "\t<li>52</li>\n",
       "\t<li>60</li>\n",
       "\t<li>59</li>\n",
       "\t<li>61</li>\n",
       "\t<li>60</li>\n",
       "\t<li>59</li>\n",
       "\t<li>58</li>\n",
       "\t<li>59</li>\n",
       "\t<li>59</li>\n",
       "\t<li>58</li>\n",
       "\t<li>60</li>\n",
       "\t<li>53</li>\n",
       "\t<li>54</li>\n",
       "\t<li>54</li>\n",
       "\t<li>48</li>\n",
       "\t<li>48</li>\n",
       "\t<li>46</li>\n",
       "\t<li>47</li>\n",
       "\t<li>44</li>\n",
       "\t<li>44</li>\n",
       "\t<li>46</li>\n",
       "\t<li>43</li>\n",
       "\t<li>41</li>\n",
       "\t<li>39</li>\n",
       "\t<li>38</li>\n",
       "\t<li>40</li>\n",
       "\t<li>43</li>\n",
       "\t<li>44</li>\n",
       "\t<li>44</li>\n",
       "\t<li>45</li>\n",
       "\t<li>46</li>\n",
       "\t<li>46</li>\n",
       "\t<li>48</li>\n",
       "\t<li>47</li>\n",
       "\t<li>47</li>\n",
       "\t<li>47</li>\n",
       "\t<li>47</li>\n",
       "\t<li>48</li>\n",
       "\t<li>49</li>\n",
       "\t<li>49</li>\n",
       "\t<li>50</li>\n",
       "\t<li>48</li>\n",
       "\t<li>51</li>\n",
       "\t<li>54</li>\n",
       "\t<li>56</li>\n",
       "\t<li>56</li>\n",
       "\t<li>54</li>\n",
       "\t<li>56</li>\n",
       "\t<li>57</li>\n",
       "\t<li>57</li>\n",
       "\t<li>57</li>\n",
       "\t<li>56</li>\n",
       "\t<li>54</li>\n",
       "\t<li>55</li>\n",
       "\t<li>53</li>\n",
       "\t<li>54</li>\n",
       "\t<li>57</li>\n",
       "\t<li>58</li>\n",
       "\t<li>59</li>\n",
       "\t<li>59</li>\n",
       "\t<li>56</li>\n",
       "\t<li>54</li>\n",
       "\t<li>55</li>\n",
       "\t<li>56</li>\n",
       "\t<li>53</li>\n",
       "\t<li>52</li>\n",
       "\t<li>49</li>\n",
       "\t<li>49</li>\n",
       "\t<li>47</li>\n",
       "\t<li>45</li>\n",
       "\t<li>47</li>\n",
       "\t<li>44</li>\n",
       "\t<li>40</li>\n",
       "\t<li>41</li>\n",
       "\t<li>42</li>\n",
       "\t<li>41</li>\n",
       "\t<li>40</li>\n",
       "\t<li>40</li>\n",
       "\t<li>37</li>\n",
       "\t<li>36</li>\n",
       "\t<li>36</li>\n",
       "\t<li>35</li>\n",
       "\t<li>36</li>\n",
       "\t<li>33</li>\n",
       "\t<li>35</li>\n",
       "\t<li>34</li>\n",
       "\t<li>35</li>\n",
       "\t<li>38</li>\n",
       "\t<li>37</li>\n",
       "\t<li>38</li>\n",
       "\t<li>37</li>\n",
       "\t<li>35</li>\n",
       "\t<li>36</li>\n",
       "\t<li>36</li>\n",
       "\t<li>36</li>\n",
       "\t<li>39</li>\n",
       "\t<li>36</li>\n",
       "\t<li>36</li>\n",
       "\t<li>34</li>\n",
       "\t<li>36</li>\n",
       "\t<li>36</li>\n",
       "\t<li>37</li>\n",
       "\t<li>35</li>\n",
       "\t<li>36</li>\n",
       "\t<li>39</li>\n",
       "\t<li>39</li>\n",
       "\t<li>40</li>\n",
       "\t<li>40</li>\n",
       "\t<li>40</li>\n",
       "\t<li>42</li>\n",
       "\t<li>44</li>\n",
       "\t<li>43</li>\n",
       "\t<li>44</li>\n",
       "\t<li>44</li>\n",
       "\t<li>39</li>\n",
       "\t<li>42</li>\n",
       "\t<li>41</li>\n",
       "\t<li>47</li>\n",
       "\t<li>47</li>\n",
       "\t<li>46</li>\n",
       "\t<li>45</li>\n",
       "\t<li>44</li>\n",
       "\t<li>46</li>\n",
       "\t<li>46</li>\n",
       "\t<li>45</li>\n",
       "\t<li>47</li>\n",
       "\t<li>48</li>\n",
       "\t<li>50</li>\n",
       "\t<li>50</li>\n",
       "\t<li>49</li>\n",
       "\t<li>47</li>\n",
       "\t<li>47</li>\n",
       "\t<li>48</li>\n",
       "\t<li>46</li>\n",
       "\t<li>44</li>\n",
       "\t<li>47</li>\n",
       "\t<li>48</li>\n",
       "\t<li>49</li>\n",
       "\t<li>48</li>\n",
       "\t<li>46</li>\n",
       "\t<li>45</li>\n",
       "\t<li>47</li>\n",
       "\t<li>48</li>\n",
       "\t<li>45</li>\n",
       "\t<li>42</li>\n",
       "\t<li>41</li>\n",
       "\t<li>39</li>\n",
       "\t<li>38</li>\n",
       "\t<li>43</li>\n",
       "\t<li>45</li>\n",
       "\t<li>43</li>\n",
       "\t<li>44</li>\n",
       "\t<li>40</li>\n",
       "\t<li>40</li>\n",
       "\t<li>39</li>\n",
       "\t<li>42</li>\n",
       "\t<li>38</li>\n",
       "\t<li>36</li>\n",
       "\t<li>41</li>\n",
       "\t<li>41</li>\n",
       "\t<li>40</li>\n",
       "\t<li>42</li>\n",
       "\t<li>40</li>\n",
       "\t<li>42</li>\n",
       "\t<li>41</li>\n",
       "\t<li>41</li>\n",
       "\t<li>40</li>\n",
       "\t<li>40</li>\n",
       "\t<li>38</li>\n",
       "\t<li>41</li>\n",
       "\t<li>40</li>\n",
       "\t<li>41</li>\n",
       "\t<li>41</li>\n",
       "\t<li>43</li>\n",
       "\t<li>44</li>\n",
       "\t<li>44</li>\n",
       "\t<li>40</li>\n",
       "\t<li>39</li>\n",
       "\t<li>40</li>\n",
       "\t<li>44</li>\n",
       "\t<li>41</li>\n",
       "\t<li>43</li>\n",
       "\t<li>39</li>\n",
       "\t<li>37</li>\n",
       "\t<li>34</li>\n",
       "\t<li>34</li>\n",
       "\t<li>34</li>\n",
       "\t<li>36</li>\n",
       "\t<li>38</li>\n",
       "\t<li>43</li>\n",
       "\t<li>44</li>\n",
       "\t<li>44</li>\n",
       "\t<li>43</li>\n",
       "\t<li>42</li>\n",
       "\t<li>41</li>\n",
       "\t<li>40</li>\n",
       "\t<li>41</li>\n",
       "\t<li>42</li>\n",
       "\t<li>45</li>\n",
       "\t<li>45</li>\n",
       "\t<li>47</li>\n",
       "\t<li>46</li>\n",
       "\t<li>45</li>\n",
       "\t<li>44</li>\n",
       "\t<li>48</li>\n",
       "\t<li>51</li>\n",
       "\t<li>53</li>\n",
       "\t<li>53</li>\n",
       "\t<li>51</li>\n",
       "\t<li>53</li>\n",
       "\t<li>50</li>\n",
       "\t<li>50</li>\n",
       "\t<li>46</li>\n",
       "\t<li>51</li>\n",
       "\t<li>51</li>\n",
       "\t<li>46</li>\n",
       "\t<li>52</li>\n",
       "\t<li>51</li>\n",
       "\t<li>50</li>\n",
       "\t<li>47</li>\n",
       "\t<li>42</li>\n",
       "\t<li>42</li>\n",
       "\t<li>43</li>\n",
       "\t<li>37</li>\n",
       "\t<li>43</li>\n",
       "\t<li>45</li>\n",
       "\t<li>47</li>\n",
       "\t<li>49</li>\n",
       "\t<li>46</li>\n",
       "\t<li>52</li>\n",
       "\t<li>53</li>\n",
       "\t<li>53</li>\n",
       "\t<li>56</li>\n",
       "\t<li>54</li>\n",
       "\t<li>55</li>\n",
       "\t<li>55</li>\n",
       "\t<li>58</li>\n",
       "\t<li>54</li>\n",
       "\t<li>55</li>\n",
       "\t<li>53</li>\n",
       "\t<li>52</li>\n",
       "\t<li>46</li>\n",
       "\t<li>47</li>\n",
       "\t<li>47</li>\n",
       "\t<li>44</li>\n",
       "\t<li>48</li>\n",
       "\t<li>48</li>\n",
       "\t<li>47</li>\n",
       "\t<li>46</li>\n",
       "\t<li>45</li>\n",
       "\t<li>44</li>\n",
       "\t<li>46</li>\n",
       "\t<li>47</li>\n",
       "\t<li>48</li>\n",
       "\t<li>48</li>\n",
       "\t<li>48</li>\n",
       "\t<li>49</li>\n",
       "\t<li>48</li>\n",
       "\t<li>45</li>\n",
       "\t<li>43</li>\n",
       "\t<li>43</li>\n",
       "\t<li>41</li>\n",
       "\t<li>40</li>\n",
       "\t<li>38</li>\n",
       "\t<li>36</li>\n",
       "\t<li>35</li>\n",
       "\t<li>38</li>\n",
       "\t<li>34</li>\n",
       "\t<li>35</li>\n",
       "\t<li>36</li>\n",
       "\t<li>33</li>\n",
       "\t<li>32</li>\n",
       "\t<li>32</li>\n",
       "\t<li>29</li>\n",
       "\t<li>30</li>\n",
       "\t<li>31</li>\n",
       "\t<li>30</li>\n",
       "\t<li>29</li>\n",
       "\t<li>30</li>\n",
       "\t<li>32</li>\n",
       "\t<li>31</li>\n",
       "\t<li>30</li>\n",
       "\t<li>29</li>\n",
       "\t<li>30</li>\n",
       "\t<li>29</li>\n",
       "\t<li>32</li>\n",
       "\t<li>34</li>\n",
       "\t<li>37</li>\n",
       "\t<li>37</li>\n",
       "\t<li>36</li>\n",
       "\t<li>37</li>\n",
       "\t<li>36</li>\n",
       "\t<li>35</li>\n",
       "\t<li>35</li>\n",
       "\t<li>36</li>\n",
       "\t<li>36</li>\n",
       "\t<li>38</li>\n",
       "\t<li>37</li>\n",
       "\t<li>36</li>\n",
       "\t<li>35</li>\n",
       "\t<li>37</li>\n",
       "\t<li>37</li>\n",
       "\t<li>38</li>\n",
       "\t<li>36</li>\n",
       "\t<li>35</li>\n",
       "\t<li>37</li>\n",
       "\t<li>38</li>\n",
       "\t<li>37</li>\n",
       "\t<li>38</li>\n",
       "\t<li>39</li>\n",
       "\t<li>44</li>\n",
       "\t<li>42</li>\n",
       "\t<li>44</li>\n",
       "\t<li>40</li>\n",
       "\t<li>40</li>\n",
       "\t<li>39</li>\n",
       "\t<li>36</li>\n",
       "\t<li>39</li>\n",
       "\t<li>38</li>\n",
       "\t<li>41</li>\n",
       "\t<li>39</li>\n",
       "\t<li>38</li>\n",
       "\t<li>36</li>\n",
       "\t<li>37</li>\n",
       "\t<li>36</li>\n",
       "\t<li>35</li>\n",
       "\t<li>34</li>\n",
       "\t<li>37</li>\n",
       "\t<li>37</li>\n",
       "\t<li>39</li>\n",
       "\t<li>38</li>\n",
       "\t<li>34</li>\n",
       "\t<li>37</li>\n",
       "\t<li>38</li>\n",
       "\t<li>40</li>\n",
       "\t<li>40</li>\n",
       "\t<li>37</li>\n",
       "\t<li>40</li>\n",
       "\t<li>42</li>\n",
       "\t<li>44</li>\n",
       "\t<li>42</li>\n",
       "\t<li>42</li>\n",
       "\t<li>44</li>\n",
       "\t<li>43</li>\n",
       "\t<li>43</li>\n",
       "\t<li>43</li>\n",
       "\t<li>43</li>\n",
       "\t<li>44</li>\n",
       "\t<li>45</li>\n",
       "\t<li>46</li>\n",
       "\t<li>45</li>\n",
       "\t<li>46</li>\n",
       "\t<li>45</li>\n",
       "\t<li>45</li>\n",
       "\t<li>43</li>\n",
       "\t<li>44</li>\n",
       "\t<li>43</li>\n",
       "\t<li>44</li>\n",
       "\t<li>45</li>\n",
       "\t<li>44</li>\n",
       "\t<li>44</li>\n",
       "\t<li>44</li>\n",
       "\t<li>42</li>\n",
       "\t<li>40</li>\n",
       "\t<li>42</li>\n",
       "\t<li>41</li>\n",
       "\t<li>43</li>\n",
       "\t<li>42</li>\n",
       "\t<li>43</li>\n",
       "\t<li>43</li>\n",
       "\t<li>43</li>\n",
       "\t<li>45</li>\n",
       "\t<li>46</li>\n",
       "\t<li>43</li>\n",
       "\t<li>39</li>\n",
       "\t<li>42</li>\n",
       "\t<li>43</li>\n",
       "\t<li>43</li>\n",
       "\t<li>46</li>\n",
       "\t<li>44</li>\n",
       "\t<li>41</li>\n",
       "\t<li>40</li>\n",
       "\t<li>41</li>\n",
       "\t<li>45</li>\n",
       "\t<li>46</li>\n",
       "\t<li>45</li>\n",
       "\t<li>47</li>\n",
       "\t<li>47</li>\n",
       "\t<li>52</li>\n",
       "\t<li>54</li>\n",
       "\t<li>55</li>\n",
       "\t<li>55</li>\n",
       "\t<li>53</li>\n",
       "\t<li>54</li>\n",
       "\t<li>53</li>\n",
       "\t<li>53</li>\n",
       "\t<li>56</li>\n",
       "\t<li>59</li>\n",
       "\t<li>60</li>\n",
       "\t<li>61</li>\n",
       "\t<li>62</li>\n",
       "\t<li>61</li>\n",
       "\t<li>60</li>\n",
       "\t<li>60</li>\n",
       "\t<li>59</li>\n",
       "\t<li>61</li>\n",
       "\t<li>60</li>\n",
       "\t<li>61</li>\n",
       "\t<li>58</li>\n",
       "\t<li>60</li>\n",
       "\t<li>58</li>\n",
       "\t<li>59</li>\n",
       "\t<li>59</li>\n",
       "\t<li>57</li>\n",
       "\t<li>58</li>\n",
       "\t<li>59</li>\n",
       "\t<li>59</li>\n",
       "\t<li>62</li>\n",
       "\t<li>63</li>\n",
       "\t<li>61</li>\n",
       "</ol>\n",
       "</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate}\n",
       "\\item \\begin{enumerate*}\n",
       "\\item 35\n",
       "\\item 35\n",
       "\\item 38\n",
       "\\item 38\n",
       "\\item 39\n",
       "\\item 39\n",
       "\\item 39\n",
       "\\item 38\n",
       "\\item 39\n",
       "\\item 39\n",
       "\\item 39\n",
       "\\item 39\n",
       "\\item 33\n",
       "\\item 40\n",
       "\\item 41\n",
       "\\item 39\n",
       "\\item 42\n",
       "\\item 36\n",
       "\\item 40\n",
       "\\item 42\n",
       "\\item 41\n",
       "\\item 40\n",
       "\\item 40\n",
       "\\item 41\n",
       "\\item 39\n",
       "\\item 41\n",
       "\\item 42\n",
       "\\item 40\n",
       "\\item 37\n",
       "\\item 37\n",
       "\\item 39\n",
       "\\item 37\n",
       "\\item 37\n",
       "\\item 40\n",
       "\\item 36\n",
       "\\item 37\n",
       "\\item 38\n",
       "\\item 37\n",
       "\\item 38\n",
       "\\item 37\n",
       "\\item 37\n",
       "\\item 38\n",
       "\\item 38\n",
       "\\item 38\n",
       "\\item 39\n",
       "\\item 37\n",
       "\\item 37\n",
       "\\item 36\n",
       "\\item 38\n",
       "\\item 37\n",
       "\\item 36\n",
       "\\item 36\n",
       "\\item 39\n",
       "\\item 40\n",
       "\\item 33\n",
       "\\item 33\n",
       "\\item 35\n",
       "\\item 35\n",
       "\\item 37\n",
       "\\item 37\n",
       "\\item 36\n",
       "\\item 33\n",
       "\\item 35\n",
       "\\item 33\n",
       "\\item 36\n",
       "\\item 35\n",
       "\\item 36\n",
       "\\item 35\n",
       "\\item 36\n",
       "\\item 37\n",
       "\\item 39\n",
       "\\item 41\n",
       "\\item 41\n",
       "\\item 43\n",
       "\\item 45\n",
       "\\item 47\n",
       "\\item 47\n",
       "\\item 47\n",
       "\\item 47\n",
       "\\item 46\n",
       "\\item 48\n",
       "\\item 49\n",
       "\\item 52\n",
       "\\item 50\n",
       "\\item 50\n",
       "\\item 50\n",
       "\\item 51\n",
       "\\item 49\n",
       "\\item 48\n",
       "\\item 51\n",
       "\\item 50\n",
       "\\item 50\n",
       "\\item 49\n",
       "\\item 49\n",
       "\\item 47\n",
       "\\item 51\n",
       "\\item 49\n",
       "\\item 51\n",
       "\\item 55\n",
       "\\item 53\n",
       "\\item 54\n",
       "\\item 52\n",
       "\\item 55\n",
       "\\item 56\n",
       "\\item 49\n",
       "\\item 55\n",
       "\\item 50\n",
       "\\item 52\n",
       "\\item 47\n",
       "\\item 48\n",
       "\\item 52\n",
       "\\item 51\n",
       "\\item 53\n",
       "\\item 54\n",
       "\\item 49\n",
       "\\item 49\n",
       "\\item 50\n",
       "\\item 46\n",
       "\\item 45\n",
       "\\item 43\n",
       "\\item 46\n",
       "\\item 48\n",
       "\\item 48\n",
       "\\item 43\n",
       "\\item 44\n",
       "\\item 43\n",
       "\\item 45\n",
       "\\item 44\n",
       "\\item 41\n",
       "\\item 40\n",
       "\\item 40\n",
       "\\item 39\n",
       "\\item 40\n",
       "\\item 37\n",
       "\\item 34\n",
       "\\item 35\n",
       "\\item 35\n",
       "\\item 34\n",
       "\\item 37\n",
       "\\item 39\n",
       "\\item 35\n",
       "\\item 36\n",
       "\\item 36\n",
       "\\item 36\n",
       "\\item 34\n",
       "\\item 30\n",
       "\\item 32\n",
       "\\item 29\n",
       "\\item 28\n",
       "\\item 30\n",
       "\\item 31\n",
       "\\item 32\n",
       "\\item 30\n",
       "\\item 33\n",
       "\\item 32\n",
       "\\item 32\n",
       "\\item 32\n",
       "\\item 30\n",
       "\\item 34\n",
       "\\item 35\n",
       "\\item 38\n",
       "\\item 38\n",
       "\\item 39\n",
       "\\item 40\n",
       "\\item 37\n",
       "\\item 42\n",
       "\\item 42\n",
       "\\item 43\n",
       "\\item 44\n",
       "\\item 44\n",
       "\\item 44\n",
       "\\item 43\n",
       "\\item 44\n",
       "\\item 43\n",
       "\\item 41\n",
       "\\item 41\n",
       "\\item 43\n",
       "\\item 41\n",
       "\\item 42\n",
       "\\item 40\n",
       "\\item 42\n",
       "\\item 41\n",
       "\\item 43\n",
       "\\item 42\n",
       "\\item 42\n",
       "\\item 40\n",
       "\\item 39\n",
       "\\item 40\n",
       "\\item 40\n",
       "\\item 40\n",
       "\\item 39\n",
       "\\item 39\n",
       "\\item 38\n",
       "\\item 39\n",
       "\\item 36\n",
       "\\item 34\n",
       "\\item 32\n",
       "\\item 36\n",
       "\\item 36\n",
       "\\item 35\n",
       "\\item 32\n",
       "\\item 33\n",
       "\\item 33\n",
       "\\item 31\n",
       "\\item 31\n",
       "\\item 31\n",
       "\\item 32\n",
       "\\item 29\n",
       "\\item 30\n",
       "\\item 31\n",
       "\\item 25\n",
       "\\item 30\n",
       "\\item 32\n",
       "\\item 30\n",
       "\\item 25\n",
       "\\item 33\n",
       "\\item 31\n",
       "\\item 29\n",
       "\\item 29\n",
       "\\item 32\n",
       "\\item 32\n",
       "\\item 31\n",
       "\\item 30\n",
       "\\item 30\n",
       "\\item 32\n",
       "\\item 29\n",
       "\\item 31\n",
       "\\item 28\n",
       "\\item 28\n",
       "\\item 30\n",
       "\\item 28\n",
       "\\item 28\n",
       "\\item 28\n",
       "\\item 33\n",
       "\\item 32\n",
       "\\item 30\n",
       "\\item 31\n",
       "\\item 32\n",
       "\\item 30\n",
       "\\item 31\n",
       "\\item 31\n",
       "\\item 29\n",
       "\\item 33\n",
       "\\item 32\n",
       "\\item 35\n",
       "\\item 36\n",
       "\\item 36\n",
       "\\item 35\n",
       "\\item 34\n",
       "\\item 33\n",
       "\\item 31\n",
       "\\item 31\n",
       "\\item 35\n",
       "\\item 33\n",
       "\\item 37\n",
       "\\item 37\n",
       "\\item 37\n",
       "\\item 39\n",
       "\\item 39\n",
       "\\item 38\n",
       "\\item 39\n",
       "\\item 36\n",
       "\\item 40\n",
       "\\item 37\n",
       "\\item 39\n",
       "\\item 42\n",
       "\\item 43\n",
       "\\item 41\n",
       "\\item 40\n",
       "\\item 41\n",
       "\\item 42\n",
       "\\item 44\n",
       "\\item 43\n",
       "\\item 43\n",
       "\\item 42\n",
       "\\item 41\n",
       "\\item 39\n",
       "\\item 38\n",
       "\\item 39\n",
       "\\item 38\n",
       "\\item 38\n",
       "\\item 37\n",
       "\\item 35\n",
       "\\item 36\n",
       "\\item 34\n",
       "\\item 38\n",
       "\\item 37\n",
       "\\item 36\n",
       "\\item 39\n",
       "\\item 40\n",
       "\\item 39\n",
       "\\item 41\n",
       "\\item 40\n",
       "\\item 39\n",
       "\\item 37\n",
       "\\item 39\n",
       "\\item 42\n",
       "\\item 46\n",
       "\\item 45\n",
       "\\item 47\n",
       "\\item 43\n",
       "\\item 44\n",
       "\\item 45\n",
       "\\item 44\n",
       "\\item 44\n",
       "\\item 45\n",
       "\\item 44\n",
       "\\item 45\n",
       "\\item 46\n",
       "\\item 45\n",
       "\\item 45\n",
       "\\item 46\n",
       "\\item 46\n",
       "\\item 49\n",
       "\\item 52\n",
       "\\item 51\n",
       "\\item 48\n",
       "\\item 49\n",
       "\\item 51\n",
       "\\item 51\n",
       "\\item 51\n",
       "\\item 54\n",
       "\\item 54\n",
       "\\item 50\n",
       "\\item 52\n",
       "\\item 54\n",
       "\\item 57\n",
       "\\item 59\n",
       "\\item 60\n",
       "\\item 59\n",
       "\\item 56\n",
       "\\item 52\n",
       "\\item 59\n",
       "\\item 56\n",
       "\\item 51\n",
       "\\item 57\n",
       "\\item 57\n",
       "\\item 59\n",
       "\\item 59\n",
       "\\item 59\n",
       "\\item 62\n",
       "\\item 60\n",
       "\\item 57\n",
       "\\item 59\n",
       "\\item 58\n",
       "\\item 60\n",
       "\\item 58\n",
       "\\item 57\n",
       "\\item 58\n",
       "\\item 57\n",
       "\\item 56\n",
       "\\item 58\n",
       "\\item 57\n",
       "\\item 56\n",
       "\\item 58\n",
       "\\item 53\n",
       "\\item 54\n",
       "\\item 49\n",
       "\\item 48\n",
       "\\item 51\n",
       "\\item 50\n",
       "\\item 47\n",
       "\\item 39\n",
       "\\item 41\n",
       "\\item 44\n",
       "\\item 44\n",
       "\\item 41\n",
       "\\item 41\n",
       "\\item 40\n",
       "\\item 39\n",
       "\\item 39\n",
       "\\item 38\n",
       "\\item 34\n",
       "\\item 33\n",
       "\\item 41\n",
       "\\item 38\n",
       "\\item 36\n",
       "\\item 36\n",
       "\\item 33\n",
       "\\item 33\n",
       "\\item 28\n",
       "\\item 33\n",
       "\\item 32\n",
       "\\item 29\n",
       "\\item 27\n",
       "\\item 26\n",
       "\\item 28\n",
       "\\item 26\n",
       "\\item 24\n",
       "\\item 27\n",
       "\\item 25\n",
       "\\item 24\n",
       "\\item 20\n",
       "\\item 25\n",
       "\\item 26\n",
       "\\item 25\n",
       "\\item 24\n",
       "\\item 19\n",
       "\\item 21\n",
       "\\item 23\n",
       "\\item 19\n",
       "\\item 18\n",
       "\\item 23\n",
       "\\item 22\n",
       "\\item 22\n",
       "\\item 22\n",
       "\\item 21\n",
       "\\item 18\n",
       "\\item 16\n",
       "\\item 14\n",
       "\\item 15\n",
       "\\item 15\n",
       "\\item 16\n",
       "\\item 16\n",
       "\\item 15\n",
       "\\item 15\n",
       "\\item 14\n",
       "\\item 12\n",
       "\\item 15\n",
       "\\item 16\n",
       "\\item 16\n",
       "\\item 15\n",
       "\\item 16\n",
       "\\item 15\n",
       "\\item 15\n",
       "\\item 13\n",
       "\\item 14\n",
       "\\item 15\n",
       "\\item 17\n",
       "\\item 15\n",
       "\\item 16\n",
       "\\item 15\n",
       "\\item 16\n",
       "\\item 14\n",
       "\\item 13\n",
       "\\item 15\n",
       "\\item 16\n",
       "\\item 16\n",
       "\\item 16\n",
       "\\item 16\n",
       "\\item 15\n",
       "\\item 15\n",
       "\\item 16\n",
       "\\item 15\n",
       "\\item 16\n",
       "\\item 15\n",
       "\\item 16\n",
       "\\item 16\n",
       "\\item 17\n",
       "\\item 18\n",
       "\\item 18\n",
       "\\item 18\n",
       "\\item 18\n",
       "\\item 17\n",
       "\\item 17\n",
       "\\item 17\n",
       "\\item 16\n",
       "\\item 16\n",
       "\\item 16\n",
       "\\item 16\n",
       "\\item 16\n",
       "\\item 18\n",
       "\\item 17\n",
       "\\item 15\n",
       "\\item 15\n",
       "\\item 18\n",
       "\\item 18\n",
       "\\item 19\n",
       "\\item 19\n",
       "\\item 21\n",
       "\\item 19\n",
       "\\item 17\n",
       "\\item 19\n",
       "\\item 20\n",
       "\\item 22\n",
       "\\item 20\n",
       "\\item 22\n",
       "\\item 20\n",
       "\\item 21\n",
       "\\item 21\n",
       "\\item 25\n",
       "\\item 26\n",
       "\\item 27\n",
       "\\item 27\n",
       "\\item 26\n",
       "\\item 29\n",
       "\\item 32\n",
       "\\item 34\n",
       "\\item 35\n",
       "\\item 36\n",
       "\\item 35\n",
       "\\item 32\n",
       "\\item 35\n",
       "\\item 37\n",
       "\\item 39\n",
       "\\item 37\n",
       "\\item 37\n",
       "\\item 37\n",
       "\\item 37\n",
       "\\item 39\n",
       "\\item 39\n",
       "\\item 43\n",
       "\\item 44\n",
       "\\item 44\n",
       "\\item 42\n",
       "\\item 40\n",
       "\\item 40\n",
       "\\item 44\n",
       "\\item 40\n",
       "\\item 40\n",
       "\\item 39\n",
       "\\item 38\n",
       "\\item 46\n",
       "\\item 46\n",
       "\\item 47\n",
       "\\item 46\n",
       "\\item 46\n",
       "\\item 46\n",
       "\\item 45\n",
       "\\item 44\n",
       "\\item 40\n",
       "\\item 39\n",
       "\\item 39\n",
       "\\item 39\n",
       "\\item 39\n",
       "\\item 41\n",
       "\\item 41\n",
       "\\item 43\n",
       "\\item 43\n",
       "\\item 44\n",
       "\\item 43\n",
       "\\item 44\n",
       "\\item 44\n",
       "\\item 45\n",
       "\\item 43\n",
       "\\item 42\n",
       "\\item 39\n",
       "\\item 36\n",
       "\\item 36\n",
       "\\item 36\n",
       "\\item 35\n",
       "\\item 35\n",
       "\\item 33\n",
       "\\item 32\n",
       "\\item 33\n",
       "\\item 30\n",
       "\\item 29\n",
       "\\item 29\n",
       "\\item 26\n",
       "\\item 25\n",
       "\\item 21\n",
       "\\item 21\n",
       "\\item 20\n",
       "\\item 19\n",
       "\\item 19\n",
       "\\item 18\n",
       "\\item 16\n",
       "\\item 16\n",
       "\\item 15\n",
       "\\item 13\n",
       "\\item 12\n",
       "\\item 12\n",
       "\\item 10\n",
       "\\item 9\n",
       "\\item 7\n",
       "\\item 6\n",
       "\\item 5\n",
       "\\item 3\n",
       "\\item 2\n",
       "\\item 1\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\end{enumerate*}\n",
       "\n",
       "\\item \\begin{enumerate*}\n",
       "\\item 27\n",
       "\\item 25\n",
       "\\item 26\n",
       "\\item 23\n",
       "\\item 24\n",
       "\\item 25\n",
       "\\item 28\n",
       "\\item 28\n",
       "\\item 28\n",
       "\\item 28\n",
       "\\item 29\n",
       "\\item 29\n",
       "\\item 27\n",
       "\\item 27\n",
       "\\item 28\n",
       "\\item 30\n",
       "\\item 28\n",
       "\\item 28\n",
       "\\item 28\n",
       "\\item 26\n",
       "\\item 26\n",
       "\\item 27\n",
       "\\item 25\n",
       "\\item 28\n",
       "\\item 30\n",
       "\\item 30\n",
       "\\item 28\n",
       "\\item 28\n",
       "\\item 26\n",
       "\\item 25\n",
       "\\item 22\n",
       "\\item 22\n",
       "\\item 20\n",
       "\\item 19\n",
       "\\item 18\n",
       "\\item 19\n",
       "\\item 19\n",
       "\\item 19\n",
       "\\item 18\n",
       "\\item 18\n",
       "\\item 17\n",
       "\\item 16\n",
       "\\item 15\n",
       "\\item 14\n",
       "\\item 13\n",
       "\\item 13\n",
       "\\item 11\n",
       "\\item 11\n",
       "\\item 13\n",
       "\\item 11\n",
       "\\item 12\n",
       "\\item 12\n",
       "\\item 12\n",
       "\\item 12\n",
       "\\item 13\n",
       "\\item 12\n",
       "\\item 11\n",
       "\\item 10\n",
       "\\item 8\n",
       "\\item 7\n",
       "\\item 6\n",
       "\\item 7\n",
       "\\item 7\n",
       "\\item 9\n",
       "\\item 9\n",
       "\\item 11\n",
       "\\item 11\n",
       "\\item 12\n",
       "\\item 12\n",
       "\\item 13\n",
       "\\item 13\n",
       "\\item 14\n",
       "\\item 14\n",
       "\\item 14\n",
       "\\item 14\n",
       "\\item 15\n",
       "\\item 15\n",
       "\\item 14\n",
       "\\item 16\n",
       "\\item 16\n",
       "\\item 17\n",
       "\\item 15\n",
       "\\item 16\n",
       "\\item 16\n",
       "\\item 17\n",
       "\\item 17\n",
       "\\item 16\n",
       "\\item 18\n",
       "\\item 19\n",
       "\\item 19\n",
       "\\item 19\n",
       "\\item 19\n",
       "\\item 19\n",
       "\\item 20\n",
       "\\item 20\n",
       "\\item 20\n",
       "\\item 19\n",
       "\\item 21\n",
       "\\item 19\n",
       "\\item 19\n",
       "\\item 20\n",
       "\\item 22\n",
       "\\item 22\n",
       "\\item 24\n",
       "\\item 22\n",
       "\\item 23\n",
       "\\item 24\n",
       "\\item 25\n",
       "\\item 25\n",
       "\\item 27\n",
       "\\item 28\n",
       "\\item 28\n",
       "\\item 30\n",
       "\\item 31\n",
       "\\item 31\n",
       "\\item 32\n",
       "\\item 34\n",
       "\\item 34\n",
       "\\item 35\n",
       "\\item 33\n",
       "\\item 35\n",
       "\\item 37\n",
       "\\item 39\n",
       "\\item 39\n",
       "\\item 42\n",
       "\\item 43\n",
       "\\item 46\n",
       "\\item 47\n",
       "\\item 46\n",
       "\\item 50\n",
       "\\item 51\n",
       "\\item 49\n",
       "\\item 45\n",
       "\\item 50\n",
       "\\item 53\n",
       "\\item 51\n",
       "\\item 48\n",
       "\\item 51\n",
       "\\item 50\n",
       "\\item 50\n",
       "\\item 53\n",
       "\\item 52\n",
       "\\item 52\n",
       "\\item 52\n",
       "\\item 50\n",
       "\\item 52\n",
       "\\item 54\n",
       "\\item 54\n",
       "\\item 55\n",
       "\\item 55\n",
       "\\item 58\n",
       "\\item 55\n",
       "\\item 55\n",
       "\\item 55\n",
       "\\item 54\n",
       "\\item 54\n",
       "\\item 52\n",
       "\\item 49\n",
       "\\item 48\n",
       "\\item 48\n",
       "\\item 46\n",
       "\\item 46\n",
       "\\item 44\n",
       "\\item 45\n",
       "\\item 44\n",
       "\\item 45\n",
       "\\item 48\n",
       "\\item 48\n",
       "\\item 47\n",
       "\\item 48\n",
       "\\item 50\n",
       "\\item 53\n",
       "\\item 51\n",
       "\\item 53\n",
       "\\item 52\n",
       "\\item 50\n",
       "\\item 51\n",
       "\\item 52\n",
       "\\item 53\n",
       "\\item 56\n",
       "\\item 55\n",
       "\\item 54\n",
       "\\item 53\n",
       "\\item 54\n",
       "\\item 53\n",
       "\\item 52\n",
       "\\item 51\n",
       "\\item 50\n",
       "\\item 50\n",
       "\\item 53\n",
       "\\item 52\n",
       "\\item 57\n",
       "\\item 56\n",
       "\\item 54\n",
       "\\item 54\n",
       "\\item 56\n",
       "\\item 57\n",
       "\\item 57\n",
       "\\item 56\n",
       "\\item 60\n",
       "\\item 58\n",
       "\\item 54\n",
       "\\item 55\n",
       "\\item 55\n",
       "\\item 50\n",
       "\\item 48\n",
       "\\item 48\n",
       "\\item 50\n",
       "\\item 51\n",
       "\\item 50\n",
       "\\item 51\n",
       "\\item 51\n",
       "\\item 49\n",
       "\\item 48\n",
       "\\item 49\n",
       "\\item 46\n",
       "\\item 46\n",
       "\\item 48\n",
       "\\item 46\n",
       "\\item 46\n",
       "\\item 45\n",
       "\\item 44\n",
       "\\item 43\n",
       "\\item 44\n",
       "\\item 45\n",
       "\\item 45\n",
       "\\item 42\n",
       "\\item 43\n",
       "\\item 46\n",
       "\\item 46\n",
       "\\item 46\n",
       "\\item 48\n",
       "\\item 50\n",
       "\\item 51\n",
       "\\item 48\n",
       "\\item 49\n",
       "\\item 48\n",
       "\\item 49\n",
       "\\item 50\n",
       "\\item 49\n",
       "\\item 47\n",
       "\\item 52\n",
       "\\item 52\n",
       "\\item 49\n",
       "\\item 48\n",
       "\\item 50\n",
       "\\item 50\n",
       "\\item 50\n",
       "\\item 48\n",
       "\\item 49\n",
       "\\item 49\n",
       "\\item 50\n",
       "\\item 48\n",
       "\\item 49\n",
       "\\item 48\n",
       "\\item 48\n",
       "\\item 50\n",
       "\\item 49\n",
       "\\item 50\n",
       "\\item 47\n",
       "\\item 51\n",
       "\\item 46\n",
       "\\item 52\n",
       "\\item 52\n",
       "\\item 50\n",
       "\\item 48\n",
       "\\item 45\n",
       "\\item 45\n",
       "\\item 44\n",
       "\\item 44\n",
       "\\item 42\n",
       "\\item 44\n",
       "\\item 45\n",
       "\\item 44\n",
       "\\item 41\n",
       "\\item 42\n",
       "\\item 42\n",
       "\\item 42\n",
       "\\item 44\n",
       "\\item 42\n",
       "\\item 43\n",
       "\\item 48\n",
       "\\item 50\n",
       "\\item 50\n",
       "\\item 51\n",
       "\\item 49\n",
       "\\item 48\n",
       "\\item 48\n",
       "\\item 53\n",
       "\\item 53\n",
       "\\item 57\n",
       "\\item 57\n",
       "\\item 56\n",
       "\\item 53\n",
       "\\item 56\n",
       "\\item 50\n",
       "\\item 51\n",
       "\\item 51\n",
       "\\item 49\n",
       "\\item 52\n",
       "\\item 50\n",
       "\\item 52\n",
       "\\item 49\n",
       "\\item 50\n",
       "\\item 46\n",
       "\\item 47\n",
       "\\item 45\n",
       "\\item 46\n",
       "\\item 48\n",
       "\\item 48\n",
       "\\item 50\n",
       "\\item 49\n",
       "\\item 50\n",
       "\\item 52\n",
       "\\item 49\n",
       "\\item 51\n",
       "\\item 49\n",
       "\\item 44\n",
       "\\item 43\n",
       "\\item 40\n",
       "\\item 42\n",
       "\\item 44\n",
       "\\item 43\n",
       "\\item 39\n",
       "\\item 39\n",
       "\\item 34\n",
       "\\item 34\n",
       "\\item 34\n",
       "\\item 34\n",
       "\\item 34\n",
       "\\item 35\n",
       "\\item 34\n",
       "\\item 37\n",
       "\\item 37\n",
       "\\item 40\n",
       "\\item 41\n",
       "\\item 39\n",
       "\\item 40\n",
       "\\item 41\n",
       "\\item 44\n",
       "\\item 43\n",
       "\\item 41\n",
       "\\item 44\n",
       "\\item 45\n",
       "\\item 42\n",
       "\\item 47\n",
       "\\item 43\n",
       "\\item 49\n",
       "\\item 49\n",
       "\\item 49\n",
       "\\item 49\n",
       "\\item 50\n",
       "\\item 50\n",
       "\\item 54\n",
       "\\item 55\n",
       "\\item 54\n",
       "\\item 54\n",
       "\\item 56\n",
       "\\item 62\n",
       "\\item 59\n",
       "\\item 58\n",
       "\\item 61\n",
       "\\item 60\n",
       "\\item 58\n",
       "\\item 62\n",
       "\\item 63\n",
       "\\item 60\n",
       "\\item 64\n",
       "\\item 60\n",
       "\\item 58\n",
       "\\item 60\n",
       "\\item 61\n",
       "\\item 60\n",
       "\\item 58\n",
       "\\item 55\n",
       "\\item 54\n",
       "\\item 54\n",
       "\\item 55\n",
       "\\item 52\n",
       "\\item 54\n",
       "\\item 49\n",
       "\\item 49\n",
       "\\item 48\n",
       "\\item 46\n",
       "\\item 48\n",
       "\\item 47\n",
       "\\item 46\n",
       "\\item 46\n",
       "\\item 45\n",
       "\\item 45\n",
       "\\item 48\n",
       "\\item 45\n",
       "\\item 44\n",
       "\\item 39\n",
       "\\item 42\n",
       "\\item 42\n",
       "\\item 43\n",
       "\\item 41\n",
       "\\item 42\n",
       "\\item 41\n",
       "\\item 44\n",
       "\\item 43\n",
       "\\item 41\n",
       "\\item 43\n",
       "\\item 43\n",
       "\\item 47\n",
       "\\item 43\n",
       "\\item 45\n",
       "\\item 46\n",
       "\\item 47\n",
       "\\item 47\n",
       "\\item 48\n",
       "\\item 47\n",
       "\\item 48\n",
       "\\item 48\n",
       "\\item 45\n",
       "\\item 48\n",
       "\\item 48\n",
       "\\item 48\n",
       "\\item 49\n",
       "\\item 48\n",
       "\\item 49\n",
       "\\item 51\n",
       "\\item 49\n",
       "\\item 49\n",
       "\\item 46\n",
       "\\item 44\n",
       "\\item 45\n",
       "\\item 44\n",
       "\\item 46\n",
       "\\item 46\n",
       "\\item 44\n",
       "\\item 44\n",
       "\\item 42\n",
       "\\item 42\n",
       "\\item 41\n",
       "\\item 41\n",
       "\\item 43\n",
       "\\item 42\n",
       "\\item 41\n",
       "\\item 39\n",
       "\\item 38\n",
       "\\item 42\n",
       "\\item 41\n",
       "\\item 37\n",
       "\\item 38\n",
       "\\item 43\n",
       "\\item 38\n",
       "\\item 40\n",
       "\\item 41\n",
       "\\item 37\n",
       "\\item 41\n",
       "\\item 43\n",
       "\\item 42\n",
       "\\item 43\n",
       "\\item 42\n",
       "\\item 39\n",
       "\\item 41\n",
       "\\item 41\n",
       "\\item 43\n",
       "\\item 48\n",
       "\\item 51\n",
       "\\item 53\n",
       "\\item 53\n",
       "\\item 56\n",
       "\\item 53\n",
       "\\item 53\n",
       "\\item 56\n",
       "\\item 59\n",
       "\\item 61\n",
       "\\item 58\n",
       "\\item 60\n",
       "\\item 60\n",
       "\\item 57\n",
       "\\item 56\n",
       "\\item 56\n",
       "\\item 57\n",
       "\\item 55\n",
       "\\item 55\n",
       "\\item 59\n",
       "\\item 59\n",
       "\\item 52\n",
       "\\item 60\n",
       "\\item 59\n",
       "\\item 61\n",
       "\\item 60\n",
       "\\item 59\n",
       "\\item 58\n",
       "\\item 59\n",
       "\\item 59\n",
       "\\item 58\n",
       "\\item 60\n",
       "\\item 53\n",
       "\\item 54\n",
       "\\item 54\n",
       "\\item 48\n",
       "\\item 48\n",
       "\\item 46\n",
       "\\item 47\n",
       "\\item 44\n",
       "\\item 44\n",
       "\\item 46\n",
       "\\item 43\n",
       "\\item 41\n",
       "\\item 39\n",
       "\\item 38\n",
       "\\item 40\n",
       "\\item 43\n",
       "\\item 44\n",
       "\\item 44\n",
       "\\item 45\n",
       "\\item 46\n",
       "\\item 46\n",
       "\\item 48\n",
       "\\item 47\n",
       "\\item 47\n",
       "\\item 47\n",
       "\\item 47\n",
       "\\item 48\n",
       "\\item 49\n",
       "\\item 49\n",
       "\\item 50\n",
       "\\item 48\n",
       "\\item 51\n",
       "\\item 54\n",
       "\\item 56\n",
       "\\item 56\n",
       "\\item 54\n",
       "\\item 56\n",
       "\\item 57\n",
       "\\item 57\n",
       "\\item 57\n",
       "\\item 56\n",
       "\\item 54\n",
       "\\item 55\n",
       "\\item 53\n",
       "\\item 54\n",
       "\\item 57\n",
       "\\item 58\n",
       "\\item 59\n",
       "\\item 59\n",
       "\\item 56\n",
       "\\item 54\n",
       "\\item 55\n",
       "\\item 56\n",
       "\\item 53\n",
       "\\item 52\n",
       "\\item 49\n",
       "\\item 49\n",
       "\\item 47\n",
       "\\item 45\n",
       "\\item 47\n",
       "\\item 44\n",
       "\\item 40\n",
       "\\item 41\n",
       "\\item 42\n",
       "\\item 41\n",
       "\\item 40\n",
       "\\item 40\n",
       "\\item 37\n",
       "\\item 36\n",
       "\\item 36\n",
       "\\item 35\n",
       "\\item 36\n",
       "\\item 33\n",
       "\\item 35\n",
       "\\item 34\n",
       "\\item 35\n",
       "\\item 38\n",
       "\\item 37\n",
       "\\item 38\n",
       "\\item 37\n",
       "\\item 35\n",
       "\\item 36\n",
       "\\item 36\n",
       "\\item 36\n",
       "\\item 39\n",
       "\\item 36\n",
       "\\item 36\n",
       "\\item 34\n",
       "\\item 36\n",
       "\\item 36\n",
       "\\item 37\n",
       "\\item 35\n",
       "\\item 36\n",
       "\\item 39\n",
       "\\item 39\n",
       "\\item 40\n",
       "\\item 40\n",
       "\\item 40\n",
       "\\item 42\n",
       "\\item 44\n",
       "\\item 43\n",
       "\\item 44\n",
       "\\item 44\n",
       "\\item 39\n",
       "\\item 42\n",
       "\\item 41\n",
       "\\item 47\n",
       "\\item 47\n",
       "\\item 46\n",
       "\\item 45\n",
       "\\item 44\n",
       "\\item 46\n",
       "\\item 46\n",
       "\\item 45\n",
       "\\item 47\n",
       "\\item 48\n",
       "\\item 50\n",
       "\\item 50\n",
       "\\item 49\n",
       "\\item 47\n",
       "\\item 47\n",
       "\\item 48\n",
       "\\item 46\n",
       "\\item 44\n",
       "\\item 47\n",
       "\\item 48\n",
       "\\item 49\n",
       "\\item 48\n",
       "\\item 46\n",
       "\\item 45\n",
       "\\item 47\n",
       "\\item 48\n",
       "\\item 45\n",
       "\\item 42\n",
       "\\item 41\n",
       "\\item 39\n",
       "\\item 38\n",
       "\\item 43\n",
       "\\item 45\n",
       "\\item 43\n",
       "\\item 44\n",
       "\\item 40\n",
       "\\item 40\n",
       "\\item 39\n",
       "\\item 42\n",
       "\\item 38\n",
       "\\item 36\n",
       "\\item 41\n",
       "\\item 41\n",
       "\\item 40\n",
       "\\item 42\n",
       "\\item 40\n",
       "\\item 42\n",
       "\\item 41\n",
       "\\item 41\n",
       "\\item 40\n",
       "\\item 40\n",
       "\\item 38\n",
       "\\item 41\n",
       "\\item 40\n",
       "\\item 41\n",
       "\\item 41\n",
       "\\item 43\n",
       "\\item 44\n",
       "\\item 44\n",
       "\\item 40\n",
       "\\item 39\n",
       "\\item 40\n",
       "\\item 44\n",
       "\\item 41\n",
       "\\item 43\n",
       "\\item 39\n",
       "\\item 37\n",
       "\\item 34\n",
       "\\item 34\n",
       "\\item 34\n",
       "\\item 36\n",
       "\\item 38\n",
       "\\item 43\n",
       "\\item 44\n",
       "\\item 44\n",
       "\\item 43\n",
       "\\item 42\n",
       "\\item 41\n",
       "\\item 40\n",
       "\\item 41\n",
       "\\item 42\n",
       "\\item 45\n",
       "\\item 45\n",
       "\\item 47\n",
       "\\item 46\n",
       "\\item 45\n",
       "\\item 44\n",
       "\\item 48\n",
       "\\item 51\n",
       "\\item 53\n",
       "\\item 53\n",
       "\\item 51\n",
       "\\item 53\n",
       "\\item 50\n",
       "\\item 50\n",
       "\\item 46\n",
       "\\item 51\n",
       "\\item 51\n",
       "\\item 46\n",
       "\\item 52\n",
       "\\item 51\n",
       "\\item 50\n",
       "\\item 47\n",
       "\\item 42\n",
       "\\item 42\n",
       "\\item 43\n",
       "\\item 37\n",
       "\\item 43\n",
       "\\item 45\n",
       "\\item 47\n",
       "\\item 49\n",
       "\\item 46\n",
       "\\item 52\n",
       "\\item 53\n",
       "\\item 53\n",
       "\\item 56\n",
       "\\item 54\n",
       "\\item 55\n",
       "\\item 55\n",
       "\\item 58\n",
       "\\item 54\n",
       "\\item 55\n",
       "\\item 53\n",
       "\\item 52\n",
       "\\item 46\n",
       "\\item 47\n",
       "\\item 47\n",
       "\\item 44\n",
       "\\item 48\n",
       "\\item 48\n",
       "\\item 47\n",
       "\\item 46\n",
       "\\item 45\n",
       "\\item 44\n",
       "\\item 46\n",
       "\\item 47\n",
       "\\item 48\n",
       "\\item 48\n",
       "\\item 48\n",
       "\\item 49\n",
       "\\item 48\n",
       "\\item 45\n",
       "\\item 43\n",
       "\\item 43\n",
       "\\item 41\n",
       "\\item 40\n",
       "\\item 38\n",
       "\\item 36\n",
       "\\item 35\n",
       "\\item 38\n",
       "\\item 34\n",
       "\\item 35\n",
       "\\item 36\n",
       "\\item 33\n",
       "\\item 32\n",
       "\\item 32\n",
       "\\item 29\n",
       "\\item 30\n",
       "\\item 31\n",
       "\\item 30\n",
       "\\item 29\n",
       "\\item 30\n",
       "\\item 32\n",
       "\\item 31\n",
       "\\item 30\n",
       "\\item 29\n",
       "\\item 30\n",
       "\\item 29\n",
       "\\item 32\n",
       "\\item 34\n",
       "\\item 37\n",
       "\\item 37\n",
       "\\item 36\n",
       "\\item 37\n",
       "\\item 36\n",
       "\\item 35\n",
       "\\item 35\n",
       "\\item 36\n",
       "\\item 36\n",
       "\\item 38\n",
       "\\item 37\n",
       "\\item 36\n",
       "\\item 35\n",
       "\\item 37\n",
       "\\item 37\n",
       "\\item 38\n",
       "\\item 36\n",
       "\\item 35\n",
       "\\item 37\n",
       "\\item 38\n",
       "\\item 37\n",
       "\\item 38\n",
       "\\item 39\n",
       "\\item 44\n",
       "\\item 42\n",
       "\\item 44\n",
       "\\item 40\n",
       "\\item 40\n",
       "\\item 39\n",
       "\\item 36\n",
       "\\item 39\n",
       "\\item 38\n",
       "\\item 41\n",
       "\\item 39\n",
       "\\item 38\n",
       "\\item 36\n",
       "\\item 37\n",
       "\\item 36\n",
       "\\item 35\n",
       "\\item 34\n",
       "\\item 37\n",
       "\\item 37\n",
       "\\item 39\n",
       "\\item 38\n",
       "\\item 34\n",
       "\\item 37\n",
       "\\item 38\n",
       "\\item 40\n",
       "\\item 40\n",
       "\\item 37\n",
       "\\item 40\n",
       "\\item 42\n",
       "\\item 44\n",
       "\\item 42\n",
       "\\item 42\n",
       "\\item 44\n",
       "\\item 43\n",
       "\\item 43\n",
       "\\item 43\n",
       "\\item 43\n",
       "\\item 44\n",
       "\\item 45\n",
       "\\item 46\n",
       "\\item 45\n",
       "\\item 46\n",
       "\\item 45\n",
       "\\item 45\n",
       "\\item 43\n",
       "\\item 44\n",
       "\\item 43\n",
       "\\item 44\n",
       "\\item 45\n",
       "\\item 44\n",
       "\\item 44\n",
       "\\item 44\n",
       "\\item 42\n",
       "\\item 40\n",
       "\\item 42\n",
       "\\item 41\n",
       "\\item 43\n",
       "\\item 42\n",
       "\\item 43\n",
       "\\item 43\n",
       "\\item 43\n",
       "\\item 45\n",
       "\\item 46\n",
       "\\item 43\n",
       "\\item 39\n",
       "\\item 42\n",
       "\\item 43\n",
       "\\item 43\n",
       "\\item 46\n",
       "\\item 44\n",
       "\\item 41\n",
       "\\item 40\n",
       "\\item 41\n",
       "\\item 45\n",
       "\\item 46\n",
       "\\item 45\n",
       "\\item 47\n",
       "\\item 47\n",
       "\\item 52\n",
       "\\item 54\n",
       "\\item 55\n",
       "\\item 55\n",
       "\\item 53\n",
       "\\item 54\n",
       "\\item 53\n",
       "\\item 53\n",
       "\\item 56\n",
       "\\item 59\n",
       "\\item 60\n",
       "\\item 61\n",
       "\\item 62\n",
       "\\item 61\n",
       "\\item 60\n",
       "\\item 60\n",
       "\\item 59\n",
       "\\item 61\n",
       "\\item 60\n",
       "\\item 61\n",
       "\\item 58\n",
       "\\item 60\n",
       "\\item 58\n",
       "\\item 59\n",
       "\\item 59\n",
       "\\item 57\n",
       "\\item 58\n",
       "\\item 59\n",
       "\\item 59\n",
       "\\item 62\n",
       "\\item 63\n",
       "\\item 61\n",
       "\\end{enumerate*}\n",
       "\n",
       "\\end{enumerate}\n"
      ],
      "text/markdown": [
       "1. 1. 35\n",
       "2. 35\n",
       "3. 38\n",
       "4. 38\n",
       "5. 39\n",
       "6. 39\n",
       "7. 39\n",
       "8. 38\n",
       "9. 39\n",
       "10. 39\n",
       "11. 39\n",
       "12. 39\n",
       "13. 33\n",
       "14. 40\n",
       "15. 41\n",
       "16. 39\n",
       "17. 42\n",
       "18. 36\n",
       "19. 40\n",
       "20. 42\n",
       "21. 41\n",
       "22. 40\n",
       "23. 40\n",
       "24. 41\n",
       "25. 39\n",
       "26. 41\n",
       "27. 42\n",
       "28. 40\n",
       "29. 37\n",
       "30. 37\n",
       "31. 39\n",
       "32. 37\n",
       "33. 37\n",
       "34. 40\n",
       "35. 36\n",
       "36. 37\n",
       "37. 38\n",
       "38. 37\n",
       "39. 38\n",
       "40. 37\n",
       "41. 37\n",
       "42. 38\n",
       "43. 38\n",
       "44. 38\n",
       "45. 39\n",
       "46. 37\n",
       "47. 37\n",
       "48. 36\n",
       "49. 38\n",
       "50. 37\n",
       "51. 36\n",
       "52. 36\n",
       "53. 39\n",
       "54. 40\n",
       "55. 33\n",
       "56. 33\n",
       "57. 35\n",
       "58. 35\n",
       "59. 37\n",
       "60. 37\n",
       "61. 36\n",
       "62. 33\n",
       "63. 35\n",
       "64. 33\n",
       "65. 36\n",
       "66. 35\n",
       "67. 36\n",
       "68. 35\n",
       "69. 36\n",
       "70. 37\n",
       "71. 39\n",
       "72. 41\n",
       "73. 41\n",
       "74. 43\n",
       "75. 45\n",
       "76. 47\n",
       "77. 47\n",
       "78. 47\n",
       "79. 47\n",
       "80. 46\n",
       "81. 48\n",
       "82. 49\n",
       "83. 52\n",
       "84. 50\n",
       "85. 50\n",
       "86. 50\n",
       "87. 51\n",
       "88. 49\n",
       "89. 48\n",
       "90. 51\n",
       "91. 50\n",
       "92. 50\n",
       "93. 49\n",
       "94. 49\n",
       "95. 47\n",
       "96. 51\n",
       "97. 49\n",
       "98. 51\n",
       "99. 55\n",
       "100. 53\n",
       "101. 54\n",
       "102. 52\n",
       "103. 55\n",
       "104. 56\n",
       "105. 49\n",
       "106. 55\n",
       "107. 50\n",
       "108. 52\n",
       "109. 47\n",
       "110. 48\n",
       "111. 52\n",
       "112. 51\n",
       "113. 53\n",
       "114. 54\n",
       "115. 49\n",
       "116. 49\n",
       "117. 50\n",
       "118. 46\n",
       "119. 45\n",
       "120. 43\n",
       "121. 46\n",
       "122. 48\n",
       "123. 48\n",
       "124. 43\n",
       "125. 44\n",
       "126. 43\n",
       "127. 45\n",
       "128. 44\n",
       "129. 41\n",
       "130. 40\n",
       "131. 40\n",
       "132. 39\n",
       "133. 40\n",
       "134. 37\n",
       "135. 34\n",
       "136. 35\n",
       "137. 35\n",
       "138. 34\n",
       "139. 37\n",
       "140. 39\n",
       "141. 35\n",
       "142. 36\n",
       "143. 36\n",
       "144. 36\n",
       "145. 34\n",
       "146. 30\n",
       "147. 32\n",
       "148. 29\n",
       "149. 28\n",
       "150. 30\n",
       "151. 31\n",
       "152. 32\n",
       "153. 30\n",
       "154. 33\n",
       "155. 32\n",
       "156. 32\n",
       "157. 32\n",
       "158. 30\n",
       "159. 34\n",
       "160. 35\n",
       "161. 38\n",
       "162. 38\n",
       "163. 39\n",
       "164. 40\n",
       "165. 37\n",
       "166. 42\n",
       "167. 42\n",
       "168. 43\n",
       "169. 44\n",
       "170. 44\n",
       "171. 44\n",
       "172. 43\n",
       "173. 44\n",
       "174. 43\n",
       "175. 41\n",
       "176. 41\n",
       "177. 43\n",
       "178. 41\n",
       "179. 42\n",
       "180. 40\n",
       "181. 42\n",
       "182. 41\n",
       "183. 43\n",
       "184. 42\n",
       "185. 42\n",
       "186. 40\n",
       "187. 39\n",
       "188. 40\n",
       "189. 40\n",
       "190. 40\n",
       "191. 39\n",
       "192. 39\n",
       "193. 38\n",
       "194. 39\n",
       "195. 36\n",
       "196. 34\n",
       "197. 32\n",
       "198. 36\n",
       "199. 36\n",
       "200. 35\n",
       "201. 32\n",
       "202. 33\n",
       "203. 33\n",
       "204. 31\n",
       "205. 31\n",
       "206. 31\n",
       "207. 32\n",
       "208. 29\n",
       "209. 30\n",
       "210. 31\n",
       "211. 25\n",
       "212. 30\n",
       "213. 32\n",
       "214. 30\n",
       "215. 25\n",
       "216. 33\n",
       "217. 31\n",
       "218. 29\n",
       "219. 29\n",
       "220. 32\n",
       "221. 32\n",
       "222. 31\n",
       "223. 30\n",
       "224. 30\n",
       "225. 32\n",
       "226. 29\n",
       "227. 31\n",
       "228. 28\n",
       "229. 28\n",
       "230. 30\n",
       "231. 28\n",
       "232. 28\n",
       "233. 28\n",
       "234. 33\n",
       "235. 32\n",
       "236. 30\n",
       "237. 31\n",
       "238. 32\n",
       "239. 30\n",
       "240. 31\n",
       "241. 31\n",
       "242. 29\n",
       "243. 33\n",
       "244. 32\n",
       "245. 35\n",
       "246. 36\n",
       "247. 36\n",
       "248. 35\n",
       "249. 34\n",
       "250. 33\n",
       "251. 31\n",
       "252. 31\n",
       "253. 35\n",
       "254. 33\n",
       "255. 37\n",
       "256. 37\n",
       "257. 37\n",
       "258. 39\n",
       "259. 39\n",
       "260. 38\n",
       "261. 39\n",
       "262. 36\n",
       "263. 40\n",
       "264. 37\n",
       "265. 39\n",
       "266. 42\n",
       "267. 43\n",
       "268. 41\n",
       "269. 40\n",
       "270. 41\n",
       "271. 42\n",
       "272. 44\n",
       "273. 43\n",
       "274. 43\n",
       "275. 42\n",
       "276. 41\n",
       "277. 39\n",
       "278. 38\n",
       "279. 39\n",
       "280. 38\n",
       "281. 38\n",
       "282. 37\n",
       "283. 35\n",
       "284. 36\n",
       "285. 34\n",
       "286. 38\n",
       "287. 37\n",
       "288. 36\n",
       "289. 39\n",
       "290. 40\n",
       "291. 39\n",
       "292. 41\n",
       "293. 40\n",
       "294. 39\n",
       "295. 37\n",
       "296. 39\n",
       "297. 42\n",
       "298. 46\n",
       "299. 45\n",
       "300. 47\n",
       "301. 43\n",
       "302. 44\n",
       "303. 45\n",
       "304. 44\n",
       "305. 44\n",
       "306. 45\n",
       "307. 44\n",
       "308. 45\n",
       "309. 46\n",
       "310. 45\n",
       "311. 45\n",
       "312. 46\n",
       "313. 46\n",
       "314. 49\n",
       "315. 52\n",
       "316. 51\n",
       "317. 48\n",
       "318. 49\n",
       "319. 51\n",
       "320. 51\n",
       "321. 51\n",
       "322. 54\n",
       "323. 54\n",
       "324. 50\n",
       "325. 52\n",
       "326. 54\n",
       "327. 57\n",
       "328. 59\n",
       "329. 60\n",
       "330. 59\n",
       "331. 56\n",
       "332. 52\n",
       "333. 59\n",
       "334. 56\n",
       "335. 51\n",
       "336. 57\n",
       "337. 57\n",
       "338. 59\n",
       "339. 59\n",
       "340. 59\n",
       "341. 62\n",
       "342. 60\n",
       "343. 57\n",
       "344. 59\n",
       "345. 58\n",
       "346. 60\n",
       "347. 58\n",
       "348. 57\n",
       "349. 58\n",
       "350. 57\n",
       "351. 56\n",
       "352. 58\n",
       "353. 57\n",
       "354. 56\n",
       "355. 58\n",
       "356. 53\n",
       "357. 54\n",
       "358. 49\n",
       "359. 48\n",
       "360. 51\n",
       "361. 50\n",
       "362. 47\n",
       "363. 39\n",
       "364. 41\n",
       "365. 44\n",
       "366. 44\n",
       "367. 41\n",
       "368. 41\n",
       "369. 40\n",
       "370. 39\n",
       "371. 39\n",
       "372. 38\n",
       "373. 34\n",
       "374. 33\n",
       "375. 41\n",
       "376. 38\n",
       "377. 36\n",
       "378. 36\n",
       "379. 33\n",
       "380. 33\n",
       "381. 28\n",
       "382. 33\n",
       "383. 32\n",
       "384. 29\n",
       "385. 27\n",
       "386. 26\n",
       "387. 28\n",
       "388. 26\n",
       "389. 24\n",
       "390. 27\n",
       "391. 25\n",
       "392. 24\n",
       "393. 20\n",
       "394. 25\n",
       "395. 26\n",
       "396. 25\n",
       "397. 24\n",
       "398. 19\n",
       "399. 21\n",
       "400. 23\n",
       "401. 19\n",
       "402. 18\n",
       "403. 23\n",
       "404. 22\n",
       "405. 22\n",
       "406. 22\n",
       "407. 21\n",
       "408. 18\n",
       "409. 16\n",
       "410. 14\n",
       "411. 15\n",
       "412. 15\n",
       "413. 16\n",
       "414. 16\n",
       "415. 15\n",
       "416. 15\n",
       "417. 14\n",
       "418. 12\n",
       "419. 15\n",
       "420. 16\n",
       "421. 16\n",
       "422. 15\n",
       "423. 16\n",
       "424. 15\n",
       "425. 15\n",
       "426. 13\n",
       "427. 14\n",
       "428. 15\n",
       "429. 17\n",
       "430. 15\n",
       "431. 16\n",
       "432. 15\n",
       "433. 16\n",
       "434. 14\n",
       "435. 13\n",
       "436. 15\n",
       "437. 16\n",
       "438. 16\n",
       "439. 16\n",
       "440. 16\n",
       "441. 15\n",
       "442. 15\n",
       "443. 16\n",
       "444. 15\n",
       "445. 16\n",
       "446. 15\n",
       "447. 16\n",
       "448. 16\n",
       "449. 17\n",
       "450. 18\n",
       "451. 18\n",
       "452. 18\n",
       "453. 18\n",
       "454. 17\n",
       "455. 17\n",
       "456. 17\n",
       "457. 16\n",
       "458. 16\n",
       "459. 16\n",
       "460. 16\n",
       "461. 16\n",
       "462. 18\n",
       "463. 17\n",
       "464. 15\n",
       "465. 15\n",
       "466. 18\n",
       "467. 18\n",
       "468. 19\n",
       "469. 19\n",
       "470. 21\n",
       "471. 19\n",
       "472. 17\n",
       "473. 19\n",
       "474. 20\n",
       "475. 22\n",
       "476. 20\n",
       "477. 22\n",
       "478. 20\n",
       "479. 21\n",
       "480. 21\n",
       "481. 25\n",
       "482. 26\n",
       "483. 27\n",
       "484. 27\n",
       "485. 26\n",
       "486. 29\n",
       "487. 32\n",
       "488. 34\n",
       "489. 35\n",
       "490. 36\n",
       "491. 35\n",
       "492. 32\n",
       "493. 35\n",
       "494. 37\n",
       "495. 39\n",
       "496. 37\n",
       "497. 37\n",
       "498. 37\n",
       "499. 37\n",
       "500. 39\n",
       "501. 39\n",
       "502. 43\n",
       "503. 44\n",
       "504. 44\n",
       "505. 42\n",
       "506. 40\n",
       "507. 40\n",
       "508. 44\n",
       "509. 40\n",
       "510. 40\n",
       "511. 39\n",
       "512. 38\n",
       "513. 46\n",
       "514. 46\n",
       "515. 47\n",
       "516. 46\n",
       "517. 46\n",
       "518. 46\n",
       "519. 45\n",
       "520. 44\n",
       "521. 40\n",
       "522. 39\n",
       "523. 39\n",
       "524. 39\n",
       "525. 39\n",
       "526. 41\n",
       "527. 41\n",
       "528. 43\n",
       "529. 43\n",
       "530. 44\n",
       "531. 43\n",
       "532. 44\n",
       "533. 44\n",
       "534. 45\n",
       "535. 43\n",
       "536. 42\n",
       "537. 39\n",
       "538. 36\n",
       "539. 36\n",
       "540. 36\n",
       "541. 35\n",
       "542. 35\n",
       "543. 33\n",
       "544. 32\n",
       "545. 33\n",
       "546. 30\n",
       "547. 29\n",
       "548. 29\n",
       "549. 26\n",
       "550. 25\n",
       "551. 21\n",
       "552. 21\n",
       "553. 20\n",
       "554. 19\n",
       "555. 19\n",
       "556. 18\n",
       "557. 16\n",
       "558. 16\n",
       "559. 15\n",
       "560. 13\n",
       "561. 12\n",
       "562. 12\n",
       "563. 10\n",
       "564. 9\n",
       "565. 7\n",
       "566. 6\n",
       "567. 5\n",
       "568. 3\n",
       "569. 2\n",
       "570. 1\n",
       "571. 0\n",
       "572. 0\n",
       "573. 0\n",
       "574. 0\n",
       "575. 0\n",
       "576. 0\n",
       "577. 0\n",
       "578. 0\n",
       "579. 0\n",
       "580. 0\n",
       "581. 0\n",
       "582. 0\n",
       "583. 0\n",
       "584. 0\n",
       "585. 0\n",
       "586. 0\n",
       "587. 0\n",
       "588. 0\n",
       "589. 0\n",
       "590. 0\n",
       "591. 0\n",
       "592. 0\n",
       "593. 0\n",
       "594. 0\n",
       "595. 0\n",
       "596. 0\n",
       "597. 0\n",
       "598. 0\n",
       "599. 0\n",
       "600. 0\n",
       "601. 0\n",
       "602. 0\n",
       "603. 0\n",
       "604. 0\n",
       "605. 0\n",
       "606. 0\n",
       "607. 0\n",
       "608. 0\n",
       "609. 0\n",
       "610. 0\n",
       "611. 0\n",
       "612. 0\n",
       "613. 0\n",
       "614. 0\n",
       "615. 0\n",
       "616. 0\n",
       "617. 0\n",
       "618. 0\n",
       "619. 0\n",
       "620. 0\n",
       "621. 0\n",
       "622. 0\n",
       "623. 0\n",
       "624. 0\n",
       "625. 0\n",
       "626. 0\n",
       "627. 0\n",
       "628. 0\n",
       "629. 0\n",
       "630. 0\n",
       "631. 0\n",
       "632. 0\n",
       "633. 0\n",
       "634. 0\n",
       "635. 0\n",
       "636. 0\n",
       "637. 0\n",
       "638. 0\n",
       "639. 0\n",
       "640. 0\n",
       "641. 0\n",
       "642. 0\n",
       "643. 0\n",
       "644. 0\n",
       "645. 0\n",
       "646. 0\n",
       "647. 0\n",
       "648. 0\n",
       "649. 0\n",
       "650. 0\n",
       "651. 0\n",
       "652. 0\n",
       "653. 0\n",
       "654. 0\n",
       "655. 0\n",
       "656. 0\n",
       "657. 0\n",
       "658. 0\n",
       "659. 0\n",
       "660. 0\n",
       "661. 0\n",
       "662. 0\n",
       "663. 0\n",
       "664. 0\n",
       "665. 0\n",
       "666. 0\n",
       "667. 0\n",
       "668. 0\n",
       "669. 0\n",
       "670. 0\n",
       "671. 0\n",
       "672. 0\n",
       "673. 0\n",
       "674. 0\n",
       "675. 0\n",
       "676. 0\n",
       "677. 0\n",
       "678. 0\n",
       "679. 0\n",
       "680. 0\n",
       "681. 0\n",
       "682. 0\n",
       "683. 0\n",
       "684. 0\n",
       "685. 0\n",
       "686. 0\n",
       "687. 0\n",
       "688. 0\n",
       "689. 0\n",
       "690. 0\n",
       "691. 0\n",
       "692. 0\n",
       "693. 0\n",
       "694. 0\n",
       "695. 0\n",
       "696. 0\n",
       "697. 0\n",
       "698. 0\n",
       "699. 0\n",
       "700. 0\n",
       "701. 0\n",
       "702. 0\n",
       "703. 0\n",
       "704. 0\n",
       "705. 0\n",
       "706. 0\n",
       "707. 0\n",
       "708. 0\n",
       "709. 0\n",
       "710. 0\n",
       "711. 0\n",
       "712. 0\n",
       "713. 0\n",
       "714. 0\n",
       "715. 0\n",
       "716. 0\n",
       "717. 0\n",
       "718. 0\n",
       "719. 0\n",
       "720. 0\n",
       "721. 0\n",
       "722. 0\n",
       "723. 0\n",
       "724. 0\n",
       "725. 0\n",
       "726. 0\n",
       "727. 0\n",
       "728. 0\n",
       "729. 0\n",
       "730. 0\n",
       "731. 0\n",
       "732. 0\n",
       "733. 0\n",
       "734. 0\n",
       "735. 0\n",
       "736. 0\n",
       "737. 0\n",
       "738. 0\n",
       "739. 0\n",
       "740. 0\n",
       "741. 0\n",
       "742. 0\n",
       "743. 0\n",
       "744. 0\n",
       "745. 0\n",
       "746. 0\n",
       "747. 0\n",
       "748. 0\n",
       "749. 0\n",
       "750. 0\n",
       "751. 0\n",
       "752. 0\n",
       "753. 0\n",
       "754. 0\n",
       "755. 0\n",
       "756. 0\n",
       "757. 0\n",
       "758. 0\n",
       "759. 0\n",
       "760. 0\n",
       "761. 0\n",
       "762. 0\n",
       "763. 0\n",
       "764. 0\n",
       "765. 0\n",
       "766. 0\n",
       "767. 0\n",
       "768. 0\n",
       "769. 0\n",
       "770. 0\n",
       "771. 0\n",
       "772. 0\n",
       "773. 0\n",
       "774. 0\n",
       "775. 0\n",
       "776. 0\n",
       "777. 0\n",
       "778. 0\n",
       "779. 0\n",
       "780. 0\n",
       "781. 0\n",
       "782. 0\n",
       "783. 0\n",
       "784. 0\n",
       "785. 0\n",
       "786. 0\n",
       "787. 0\n",
       "788. 0\n",
       "789. 0\n",
       "790. 0\n",
       "791. 0\n",
       "792. 0\n",
       "793. 0\n",
       "794. 0\n",
       "795. 0\n",
       "796. 0\n",
       "797. 0\n",
       "798. 0\n",
       "799. 0\n",
       "800. 0\n",
       "801. 0\n",
       "802. 0\n",
       "803. 0\n",
       "804. 0\n",
       "805. 0\n",
       "806. 0\n",
       "807. 0\n",
       "808. 0\n",
       "809. 0\n",
       "810. 0\n",
       "811. 0\n",
       "812. 0\n",
       "813. 0\n",
       "814. 0\n",
       "815. 0\n",
       "816. 0\n",
       "817. 0\n",
       "818. 0\n",
       "819. 0\n",
       "820. 0\n",
       "821. 0\n",
       "822. 0\n",
       "823. 0\n",
       "824. 0\n",
       "825. 0\n",
       "826. 0\n",
       "827. 0\n",
       "828. 0\n",
       "829. 0\n",
       "830. 0\n",
       "831. 0\n",
       "832. 0\n",
       "833. 0\n",
       "834. 0\n",
       "835. 0\n",
       "836. 0\n",
       "837. 0\n",
       "838. 0\n",
       "839. 0\n",
       "840. 0\n",
       "841. 0\n",
       "842. 0\n",
       "843. 0\n",
       "844. 0\n",
       "845. 0\n",
       "846. 0\n",
       "847. 0\n",
       "848. 0\n",
       "849. 0\n",
       "850. 0\n",
       "851. 0\n",
       "852. 0\n",
       "853. 0\n",
       "854. 0\n",
       "855. 0\n",
       "856. 0\n",
       "857. 0\n",
       "858. 0\n",
       "859. 0\n",
       "860. 0\n",
       "861. 0\n",
       "862. 0\n",
       "863. 0\n",
       "864. 0\n",
       "865. 0\n",
       "866. 0\n",
       "867. 0\n",
       "868. 0\n",
       "869. 0\n",
       "870. 0\n",
       "871. 0\n",
       "872. 0\n",
       "873. 0\n",
       "874. 0\n",
       "875. 0\n",
       "876. 0\n",
       "877. 0\n",
       "878. 0\n",
       "879. 0\n",
       "880. 0\n",
       "881. 0\n",
       "882. 0\n",
       "883. 0\n",
       "884. 0\n",
       "885. 0\n",
       "886. 0\n",
       "887. 0\n",
       "888. 0\n",
       "889. 0\n",
       "890. 0\n",
       "891. 0\n",
       "892. 0\n",
       "893. 0\n",
       "894. 0\n",
       "895. 0\n",
       "896. 0\n",
       "897. 0\n",
       "898. 0\n",
       "899. 0\n",
       "900. 0\n",
       "901. 0\n",
       "902. 0\n",
       "903. 0\n",
       "904. 0\n",
       "905. 0\n",
       "906. 0\n",
       "907. 0\n",
       "908. 0\n",
       "909. 0\n",
       "910. 0\n",
       "911. 0\n",
       "912. 0\n",
       "913. 0\n",
       "914. 0\n",
       "915. 0\n",
       "916. 0\n",
       "917. 0\n",
       "918. 0\n",
       "919. 0\n",
       "920. 0\n",
       "921. 0\n",
       "922. 0\n",
       "923. 0\n",
       "924. 0\n",
       "925. 0\n",
       "926. 0\n",
       "927. 0\n",
       "928. 0\n",
       "929. 0\n",
       "930. 0\n",
       "931. 0\n",
       "932. 0\n",
       "933. 0\n",
       "934. 0\n",
       "935. 0\n",
       "936. 0\n",
       "937. 0\n",
       "938. 0\n",
       "939. 0\n",
       "940. 0\n",
       "941. 0\n",
       "942. 0\n",
       "943. 0\n",
       "944. 0\n",
       "945. 0\n",
       "946. 0\n",
       "947. 0\n",
       "948. 0\n",
       "949. 0\n",
       "950. 0\n",
       "951. 0\n",
       "952. 0\n",
       "953. 0\n",
       "954. 0\n",
       "955. 0\n",
       "956. 0\n",
       "957. 0\n",
       "958. 0\n",
       "959. 0\n",
       "960. 0\n",
       "961. 0\n",
       "962. 0\n",
       "963. 0\n",
       "964. 0\n",
       "965. 0\n",
       "966. 0\n",
       "967. 0\n",
       "968. 0\n",
       "969. 0\n",
       "970. 0\n",
       "971. 0\n",
       "972. 0\n",
       "973. 0\n",
       "974. 0\n",
       "975. 0\n",
       "976. 0\n",
       "977. 0\n",
       "978. 0\n",
       "979. 0\n",
       "980. 0\n",
       "981. 0\n",
       "982. 0\n",
       "983. 0\n",
       "984. 0\n",
       "985. 0\n",
       "986. 0\n",
       "987. 0\n",
       "988. 0\n",
       "989. 0\n",
       "990. 0\n",
       "991. 0\n",
       "992. 0\n",
       "993. 0\n",
       "994. 0\n",
       "995. 0\n",
       "996. 0\n",
       "997. 0\n",
       "998. 0\n",
       "999. 0\n",
       "1000. 0\n",
       "1001. 0\n",
       "\n",
       "\n",
       "\n",
       "2. 1. 27\n",
       "2. 25\n",
       "3. 26\n",
       "4. 23\n",
       "5. 24\n",
       "6. 25\n",
       "7. 28\n",
       "8. 28\n",
       "9. 28\n",
       "10. 28\n",
       "11. 29\n",
       "12. 29\n",
       "13. 27\n",
       "14. 27\n",
       "15. 28\n",
       "16. 30\n",
       "17. 28\n",
       "18. 28\n",
       "19. 28\n",
       "20. 26\n",
       "21. 26\n",
       "22. 27\n",
       "23. 25\n",
       "24. 28\n",
       "25. 30\n",
       "26. 30\n",
       "27. 28\n",
       "28. 28\n",
       "29. 26\n",
       "30. 25\n",
       "31. 22\n",
       "32. 22\n",
       "33. 20\n",
       "34. 19\n",
       "35. 18\n",
       "36. 19\n",
       "37. 19\n",
       "38. 19\n",
       "39. 18\n",
       "40. 18\n",
       "41. 17\n",
       "42. 16\n",
       "43. 15\n",
       "44. 14\n",
       "45. 13\n",
       "46. 13\n",
       "47. 11\n",
       "48. 11\n",
       "49. 13\n",
       "50. 11\n",
       "51. 12\n",
       "52. 12\n",
       "53. 12\n",
       "54. 12\n",
       "55. 13\n",
       "56. 12\n",
       "57. 11\n",
       "58. 10\n",
       "59. 8\n",
       "60. 7\n",
       "61. 6\n",
       "62. 7\n",
       "63. 7\n",
       "64. 9\n",
       "65. 9\n",
       "66. 11\n",
       "67. 11\n",
       "68. 12\n",
       "69. 12\n",
       "70. 13\n",
       "71. 13\n",
       "72. 14\n",
       "73. 14\n",
       "74. 14\n",
       "75. 14\n",
       "76. 15\n",
       "77. 15\n",
       "78. 14\n",
       "79. 16\n",
       "80. 16\n",
       "81. 17\n",
       "82. 15\n",
       "83. 16\n",
       "84. 16\n",
       "85. 17\n",
       "86. 17\n",
       "87. 16\n",
       "88. 18\n",
       "89. 19\n",
       "90. 19\n",
       "91. 19\n",
       "92. 19\n",
       "93. 19\n",
       "94. 20\n",
       "95. 20\n",
       "96. 20\n",
       "97. 19\n",
       "98. 21\n",
       "99. 19\n",
       "100. 19\n",
       "101. 20\n",
       "102. 22\n",
       "103. 22\n",
       "104. 24\n",
       "105. 22\n",
       "106. 23\n",
       "107. 24\n",
       "108. 25\n",
       "109. 25\n",
       "110. 27\n",
       "111. 28\n",
       "112. 28\n",
       "113. 30\n",
       "114. 31\n",
       "115. 31\n",
       "116. 32\n",
       "117. 34\n",
       "118. 34\n",
       "119. 35\n",
       "120. 33\n",
       "121. 35\n",
       "122. 37\n",
       "123. 39\n",
       "124. 39\n",
       "125. 42\n",
       "126. 43\n",
       "127. 46\n",
       "128. 47\n",
       "129. 46\n",
       "130. 50\n",
       "131. 51\n",
       "132. 49\n",
       "133. 45\n",
       "134. 50\n",
       "135. 53\n",
       "136. 51\n",
       "137. 48\n",
       "138. 51\n",
       "139. 50\n",
       "140. 50\n",
       "141. 53\n",
       "142. 52\n",
       "143. 52\n",
       "144. 52\n",
       "145. 50\n",
       "146. 52\n",
       "147. 54\n",
       "148. 54\n",
       "149. 55\n",
       "150. 55\n",
       "151. 58\n",
       "152. 55\n",
       "153. 55\n",
       "154. 55\n",
       "155. 54\n",
       "156. 54\n",
       "157. 52\n",
       "158. 49\n",
       "159. 48\n",
       "160. 48\n",
       "161. 46\n",
       "162. 46\n",
       "163. 44\n",
       "164. 45\n",
       "165. 44\n",
       "166. 45\n",
       "167. 48\n",
       "168. 48\n",
       "169. 47\n",
       "170. 48\n",
       "171. 50\n",
       "172. 53\n",
       "173. 51\n",
       "174. 53\n",
       "175. 52\n",
       "176. 50\n",
       "177. 51\n",
       "178. 52\n",
       "179. 53\n",
       "180. 56\n",
       "181. 55\n",
       "182. 54\n",
       "183. 53\n",
       "184. 54\n",
       "185. 53\n",
       "186. 52\n",
       "187. 51\n",
       "188. 50\n",
       "189. 50\n",
       "190. 53\n",
       "191. 52\n",
       "192. 57\n",
       "193. 56\n",
       "194. 54\n",
       "195. 54\n",
       "196. 56\n",
       "197. 57\n",
       "198. 57\n",
       "199. 56\n",
       "200. 60\n",
       "201. 58\n",
       "202. 54\n",
       "203. 55\n",
       "204. 55\n",
       "205. 50\n",
       "206. 48\n",
       "207. 48\n",
       "208. 50\n",
       "209. 51\n",
       "210. 50\n",
       "211. 51\n",
       "212. 51\n",
       "213. 49\n",
       "214. 48\n",
       "215. 49\n",
       "216. 46\n",
       "217. 46\n",
       "218. 48\n",
       "219. 46\n",
       "220. 46\n",
       "221. 45\n",
       "222. 44\n",
       "223. 43\n",
       "224. 44\n",
       "225. 45\n",
       "226. 45\n",
       "227. 42\n",
       "228. 43\n",
       "229. 46\n",
       "230. 46\n",
       "231. 46\n",
       "232. 48\n",
       "233. 50\n",
       "234. 51\n",
       "235. 48\n",
       "236. 49\n",
       "237. 48\n",
       "238. 49\n",
       "239. 50\n",
       "240. 49\n",
       "241. 47\n",
       "242. 52\n",
       "243. 52\n",
       "244. 49\n",
       "245. 48\n",
       "246. 50\n",
       "247. 50\n",
       "248. 50\n",
       "249. 48\n",
       "250. 49\n",
       "251. 49\n",
       "252. 50\n",
       "253. 48\n",
       "254. 49\n",
       "255. 48\n",
       "256. 48\n",
       "257. 50\n",
       "258. 49\n",
       "259. 50\n",
       "260. 47\n",
       "261. 51\n",
       "262. 46\n",
       "263. 52\n",
       "264. 52\n",
       "265. 50\n",
       "266. 48\n",
       "267. 45\n",
       "268. 45\n",
       "269. 44\n",
       "270. 44\n",
       "271. 42\n",
       "272. 44\n",
       "273. 45\n",
       "274. 44\n",
       "275. 41\n",
       "276. 42\n",
       "277. 42\n",
       "278. 42\n",
       "279. 44\n",
       "280. 42\n",
       "281. 43\n",
       "282. 48\n",
       "283. 50\n",
       "284. 50\n",
       "285. 51\n",
       "286. 49\n",
       "287. 48\n",
       "288. 48\n",
       "289. 53\n",
       "290. 53\n",
       "291. 57\n",
       "292. 57\n",
       "293. 56\n",
       "294. 53\n",
       "295. 56\n",
       "296. 50\n",
       "297. 51\n",
       "298. 51\n",
       "299. 49\n",
       "300. 52\n",
       "301. 50\n",
       "302. 52\n",
       "303. 49\n",
       "304. 50\n",
       "305. 46\n",
       "306. 47\n",
       "307. 45\n",
       "308. 46\n",
       "309. 48\n",
       "310. 48\n",
       "311. 50\n",
       "312. 49\n",
       "313. 50\n",
       "314. 52\n",
       "315. 49\n",
       "316. 51\n",
       "317. 49\n",
       "318. 44\n",
       "319. 43\n",
       "320. 40\n",
       "321. 42\n",
       "322. 44\n",
       "323. 43\n",
       "324. 39\n",
       "325. 39\n",
       "326. 34\n",
       "327. 34\n",
       "328. 34\n",
       "329. 34\n",
       "330. 34\n",
       "331. 35\n",
       "332. 34\n",
       "333. 37\n",
       "334. 37\n",
       "335. 40\n",
       "336. 41\n",
       "337. 39\n",
       "338. 40\n",
       "339. 41\n",
       "340. 44\n",
       "341. 43\n",
       "342. 41\n",
       "343. 44\n",
       "344. 45\n",
       "345. 42\n",
       "346. 47\n",
       "347. 43\n",
       "348. 49\n",
       "349. 49\n",
       "350. 49\n",
       "351. 49\n",
       "352. 50\n",
       "353. 50\n",
       "354. 54\n",
       "355. 55\n",
       "356. 54\n",
       "357. 54\n",
       "358. 56\n",
       "359. 62\n",
       "360. 59\n",
       "361. 58\n",
       "362. 61\n",
       "363. 60\n",
       "364. 58\n",
       "365. 62\n",
       "366. 63\n",
       "367. 60\n",
       "368. 64\n",
       "369. 60\n",
       "370. 58\n",
       "371. 60\n",
       "372. 61\n",
       "373. 60\n",
       "374. 58\n",
       "375. 55\n",
       "376. 54\n",
       "377. 54\n",
       "378. 55\n",
       "379. 52\n",
       "380. 54\n",
       "381. 49\n",
       "382. 49\n",
       "383. 48\n",
       "384. 46\n",
       "385. 48\n",
       "386. 47\n",
       "387. 46\n",
       "388. 46\n",
       "389. 45\n",
       "390. 45\n",
       "391. 48\n",
       "392. 45\n",
       "393. 44\n",
       "394. 39\n",
       "395. 42\n",
       "396. 42\n",
       "397. 43\n",
       "398. 41\n",
       "399. 42\n",
       "400. 41\n",
       "401. 44\n",
       "402. 43\n",
       "403. 41\n",
       "404. 43\n",
       "405. 43\n",
       "406. 47\n",
       "407. 43\n",
       "408. 45\n",
       "409. 46\n",
       "410. 47\n",
       "411. 47\n",
       "412. 48\n",
       "413. 47\n",
       "414. 48\n",
       "415. 48\n",
       "416. 45\n",
       "417. 48\n",
       "418. 48\n",
       "419. 48\n",
       "420. 49\n",
       "421. 48\n",
       "422. 49\n",
       "423. 51\n",
       "424. 49\n",
       "425. 49\n",
       "426. 46\n",
       "427. 44\n",
       "428. 45\n",
       "429. 44\n",
       "430. 46\n",
       "431. 46\n",
       "432. 44\n",
       "433. 44\n",
       "434. 42\n",
       "435. 42\n",
       "436. 41\n",
       "437. 41\n",
       "438. 43\n",
       "439. 42\n",
       "440. 41\n",
       "441. 39\n",
       "442. 38\n",
       "443. 42\n",
       "444. 41\n",
       "445. 37\n",
       "446. 38\n",
       "447. 43\n",
       "448. 38\n",
       "449. 40\n",
       "450. 41\n",
       "451. 37\n",
       "452. 41\n",
       "453. 43\n",
       "454. 42\n",
       "455. 43\n",
       "456. 42\n",
       "457. 39\n",
       "458. 41\n",
       "459. 41\n",
       "460. 43\n",
       "461. 48\n",
       "462. 51\n",
       "463. 53\n",
       "464. 53\n",
       "465. 56\n",
       "466. 53\n",
       "467. 53\n",
       "468. 56\n",
       "469. 59\n",
       "470. 61\n",
       "471. 58\n",
       "472. 60\n",
       "473. 60\n",
       "474. 57\n",
       "475. 56\n",
       "476. 56\n",
       "477. 57\n",
       "478. 55\n",
       "479. 55\n",
       "480. 59\n",
       "481. 59\n",
       "482. 52\n",
       "483. 60\n",
       "484. 59\n",
       "485. 61\n",
       "486. 60\n",
       "487. 59\n",
       "488. 58\n",
       "489. 59\n",
       "490. 59\n",
       "491. 58\n",
       "492. 60\n",
       "493. 53\n",
       "494. 54\n",
       "495. 54\n",
       "496. 48\n",
       "497. 48\n",
       "498. 46\n",
       "499. 47\n",
       "500. 44\n",
       "501. 44\n",
       "502. 46\n",
       "503. 43\n",
       "504. 41\n",
       "505. 39\n",
       "506. 38\n",
       "507. 40\n",
       "508. 43\n",
       "509. 44\n",
       "510. 44\n",
       "511. 45\n",
       "512. 46\n",
       "513. 46\n",
       "514. 48\n",
       "515. 47\n",
       "516. 47\n",
       "517. 47\n",
       "518. 47\n",
       "519. 48\n",
       "520. 49\n",
       "521. 49\n",
       "522. 50\n",
       "523. 48\n",
       "524. 51\n",
       "525. 54\n",
       "526. 56\n",
       "527. 56\n",
       "528. 54\n",
       "529. 56\n",
       "530. 57\n",
       "531. 57\n",
       "532. 57\n",
       "533. 56\n",
       "534. 54\n",
       "535. 55\n",
       "536. 53\n",
       "537. 54\n",
       "538. 57\n",
       "539. 58\n",
       "540. 59\n",
       "541. 59\n",
       "542. 56\n",
       "543. 54\n",
       "544. 55\n",
       "545. 56\n",
       "546. 53\n",
       "547. 52\n",
       "548. 49\n",
       "549. 49\n",
       "550. 47\n",
       "551. 45\n",
       "552. 47\n",
       "553. 44\n",
       "554. 40\n",
       "555. 41\n",
       "556. 42\n",
       "557. 41\n",
       "558. 40\n",
       "559. 40\n",
       "560. 37\n",
       "561. 36\n",
       "562. 36\n",
       "563. 35\n",
       "564. 36\n",
       "565. 33\n",
       "566. 35\n",
       "567. 34\n",
       "568. 35\n",
       "569. 38\n",
       "570. 37\n",
       "571. 38\n",
       "572. 37\n",
       "573. 35\n",
       "574. 36\n",
       "575. 36\n",
       "576. 36\n",
       "577. 39\n",
       "578. 36\n",
       "579. 36\n",
       "580. 34\n",
       "581. 36\n",
       "582. 36\n",
       "583. 37\n",
       "584. 35\n",
       "585. 36\n",
       "586. 39\n",
       "587. 39\n",
       "588. 40\n",
       "589. 40\n",
       "590. 40\n",
       "591. 42\n",
       "592. 44\n",
       "593. 43\n",
       "594. 44\n",
       "595. 44\n",
       "596. 39\n",
       "597. 42\n",
       "598. 41\n",
       "599. 47\n",
       "600. 47\n",
       "601. 46\n",
       "602. 45\n",
       "603. 44\n",
       "604. 46\n",
       "605. 46\n",
       "606. 45\n",
       "607. 47\n",
       "608. 48\n",
       "609. 50\n",
       "610. 50\n",
       "611. 49\n",
       "612. 47\n",
       "613. 47\n",
       "614. 48\n",
       "615. 46\n",
       "616. 44\n",
       "617. 47\n",
       "618. 48\n",
       "619. 49\n",
       "620. 48\n",
       "621. 46\n",
       "622. 45\n",
       "623. 47\n",
       "624. 48\n",
       "625. 45\n",
       "626. 42\n",
       "627. 41\n",
       "628. 39\n",
       "629. 38\n",
       "630. 43\n",
       "631. 45\n",
       "632. 43\n",
       "633. 44\n",
       "634. 40\n",
       "635. 40\n",
       "636. 39\n",
       "637. 42\n",
       "638. 38\n",
       "639. 36\n",
       "640. 41\n",
       "641. 41\n",
       "642. 40\n",
       "643. 42\n",
       "644. 40\n",
       "645. 42\n",
       "646. 41\n",
       "647. 41\n",
       "648. 40\n",
       "649. 40\n",
       "650. 38\n",
       "651. 41\n",
       "652. 40\n",
       "653. 41\n",
       "654. 41\n",
       "655. 43\n",
       "656. 44\n",
       "657. 44\n",
       "658. 40\n",
       "659. 39\n",
       "660. 40\n",
       "661. 44\n",
       "662. 41\n",
       "663. 43\n",
       "664. 39\n",
       "665. 37\n",
       "666. 34\n",
       "667. 34\n",
       "668. 34\n",
       "669. 36\n",
       "670. 38\n",
       "671. 43\n",
       "672. 44\n",
       "673. 44\n",
       "674. 43\n",
       "675. 42\n",
       "676. 41\n",
       "677. 40\n",
       "678. 41\n",
       "679. 42\n",
       "680. 45\n",
       "681. 45\n",
       "682. 47\n",
       "683. 46\n",
       "684. 45\n",
       "685. 44\n",
       "686. 48\n",
       "687. 51\n",
       "688. 53\n",
       "689. 53\n",
       "690. 51\n",
       "691. 53\n",
       "692. 50\n",
       "693. 50\n",
       "694. 46\n",
       "695. 51\n",
       "696. 51\n",
       "697. 46\n",
       "698. 52\n",
       "699. 51\n",
       "700. 50\n",
       "701. 47\n",
       "702. 42\n",
       "703. 42\n",
       "704. 43\n",
       "705. 37\n",
       "706. 43\n",
       "707. 45\n",
       "708. 47\n",
       "709. 49\n",
       "710. 46\n",
       "711. 52\n",
       "712. 53\n",
       "713. 53\n",
       "714. 56\n",
       "715. 54\n",
       "716. 55\n",
       "717. 55\n",
       "718. 58\n",
       "719. 54\n",
       "720. 55\n",
       "721. 53\n",
       "722. 52\n",
       "723. 46\n",
       "724. 47\n",
       "725. 47\n",
       "726. 44\n",
       "727. 48\n",
       "728. 48\n",
       "729. 47\n",
       "730. 46\n",
       "731. 45\n",
       "732. 44\n",
       "733. 46\n",
       "734. 47\n",
       "735. 48\n",
       "736. 48\n",
       "737. 48\n",
       "738. 49\n",
       "739. 48\n",
       "740. 45\n",
       "741. 43\n",
       "742. 43\n",
       "743. 41\n",
       "744. 40\n",
       "745. 38\n",
       "746. 36\n",
       "747. 35\n",
       "748. 38\n",
       "749. 34\n",
       "750. 35\n",
       "751. 36\n",
       "752. 33\n",
       "753. 32\n",
       "754. 32\n",
       "755. 29\n",
       "756. 30\n",
       "757. 31\n",
       "758. 30\n",
       "759. 29\n",
       "760. 30\n",
       "761. 32\n",
       "762. 31\n",
       "763. 30\n",
       "764. 29\n",
       "765. 30\n",
       "766. 29\n",
       "767. 32\n",
       "768. 34\n",
       "769. 37\n",
       "770. 37\n",
       "771. 36\n",
       "772. 37\n",
       "773. 36\n",
       "774. 35\n",
       "775. 35\n",
       "776. 36\n",
       "777. 36\n",
       "778. 38\n",
       "779. 37\n",
       "780. 36\n",
       "781. 35\n",
       "782. 37\n",
       "783. 37\n",
       "784. 38\n",
       "785. 36\n",
       "786. 35\n",
       "787. 37\n",
       "788. 38\n",
       "789. 37\n",
       "790. 38\n",
       "791. 39\n",
       "792. 44\n",
       "793. 42\n",
       "794. 44\n",
       "795. 40\n",
       "796. 40\n",
       "797. 39\n",
       "798. 36\n",
       "799. 39\n",
       "800. 38\n",
       "801. 41\n",
       "802. 39\n",
       "803. 38\n",
       "804. 36\n",
       "805. 37\n",
       "806. 36\n",
       "807. 35\n",
       "808. 34\n",
       "809. 37\n",
       "810. 37\n",
       "811. 39\n",
       "812. 38\n",
       "813. 34\n",
       "814. 37\n",
       "815. 38\n",
       "816. 40\n",
       "817. 40\n",
       "818. 37\n",
       "819. 40\n",
       "820. 42\n",
       "821. 44\n",
       "822. 42\n",
       "823. 42\n",
       "824. 44\n",
       "825. 43\n",
       "826. 43\n",
       "827. 43\n",
       "828. 43\n",
       "829. 44\n",
       "830. 45\n",
       "831. 46\n",
       "832. 45\n",
       "833. 46\n",
       "834. 45\n",
       "835. 45\n",
       "836. 43\n",
       "837. 44\n",
       "838. 43\n",
       "839. 44\n",
       "840. 45\n",
       "841. 44\n",
       "842. 44\n",
       "843. 44\n",
       "844. 42\n",
       "845. 40\n",
       "846. 42\n",
       "847. 41\n",
       "848. 43\n",
       "849. 42\n",
       "850. 43\n",
       "851. 43\n",
       "852. 43\n",
       "853. 45\n",
       "854. 46\n",
       "855. 43\n",
       "856. 39\n",
       "857. 42\n",
       "858. 43\n",
       "859. 43\n",
       "860. 46\n",
       "861. 44\n",
       "862. 41\n",
       "863. 40\n",
       "864. 41\n",
       "865. 45\n",
       "866. 46\n",
       "867. 45\n",
       "868. 47\n",
       "869. 47\n",
       "870. 52\n",
       "871. 54\n",
       "872. 55\n",
       "873. 55\n",
       "874. 53\n",
       "875. 54\n",
       "876. 53\n",
       "877. 53\n",
       "878. 56\n",
       "879. 59\n",
       "880. 60\n",
       "881. 61\n",
       "882. 62\n",
       "883. 61\n",
       "884. 60\n",
       "885. 60\n",
       "886. 59\n",
       "887. 61\n",
       "888. 60\n",
       "889. 61\n",
       "890. 58\n",
       "891. 60\n",
       "892. 58\n",
       "893. 59\n",
       "894. 59\n",
       "895. 57\n",
       "896. 58\n",
       "897. 59\n",
       "898. 59\n",
       "899. 62\n",
       "900. 63\n",
       "901. 61\n",
       "\n",
       "\n",
       "\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[[1]]\n",
       "   [1] 35 35 38 38 39 39 39 38 39 39 39 39 33 40 41 39 42 36 40 42 41 40 40 41\n",
       "  [25] 39 41 42 40 37 37 39 37 37 40 36 37 38 37 38 37 37 38 38 38 39 37 37 36\n",
       "  [49] 38 37 36 36 39 40 33 33 35 35 37 37 36 33 35 33 36 35 36 35 36 37 39 41\n",
       "  [73] 41 43 45 47 47 47 47 46 48 49 52 50 50 50 51 49 48 51 50 50 49 49 47 51\n",
       "  [97] 49 51 55 53 54 52 55 56 49 55 50 52 47 48 52 51 53 54 49 49 50 46 45 43\n",
       " [121] 46 48 48 43 44 43 45 44 41 40 40 39 40 37 34 35 35 34 37 39 35 36 36 36\n",
       " [145] 34 30 32 29 28 30 31 32 30 33 32 32 32 30 34 35 38 38 39 40 37 42 42 43\n",
       " [169] 44 44 44 43 44 43 41 41 43 41 42 40 42 41 43 42 42 40 39 40 40 40 39 39\n",
       " [193] 38 39 36 34 32 36 36 35 32 33 33 31 31 31 32 29 30 31 25 30 32 30 25 33\n",
       " [217] 31 29 29 32 32 31 30 30 32 29 31 28 28 30 28 28 28 33 32 30 31 32 30 31\n",
       " [241] 31 29 33 32 35 36 36 35 34 33 31 31 35 33 37 37 37 39 39 38 39 36 40 37\n",
       " [265] 39 42 43 41 40 41 42 44 43 43 42 41 39 38 39 38 38 37 35 36 34 38 37 36\n",
       " [289] 39 40 39 41 40 39 37 39 42 46 45 47 43 44 45 44 44 45 44 45 46 45 45 46\n",
       " [313] 46 49 52 51 48 49 51 51 51 54 54 50 52 54 57 59 60 59 56 52 59 56 51 57\n",
       " [337] 57 59 59 59 62 60 57 59 58 60 58 57 58 57 56 58 57 56 58 53 54 49 48 51\n",
       " [361] 50 47 39 41 44 44 41 41 40 39 39 38 34 33 41 38 36 36 33 33 28 33 32 29\n",
       " [385] 27 26 28 26 24 27 25 24 20 25 26 25 24 19 21 23 19 18 23 22 22 22 21 18\n",
       " [409] 16 14 15 15 16 16 15 15 14 12 15 16 16 15 16 15 15 13 14 15 17 15 16 15\n",
       " [433] 16 14 13 15 16 16 16 16 15 15 16 15 16 15 16 16 17 18 18 18 18 17 17 17\n",
       " [457] 16 16 16 16 16 18 17 15 15 18 18 19 19 21 19 17 19 20 22 20 22 20 21 21\n",
       " [481] 25 26 27 27 26 29 32 34 35 36 35 32 35 37 39 37 37 37 37 39 39 43 44 44\n",
       " [505] 42 40 40 44 40 40 39 38 46 46 47 46 46 46 45 44 40 39 39 39 39 41 41 43\n",
       " [529] 43 44 43 44 44 45 43 42 39 36 36 36 35 35 33 32 33 30 29 29 26 25 21 21\n",
       " [553] 20 19 19 18 16 16 15 13 12 12 10  9  7  6  5  3  2  1  0  0  0  0  0  0\n",
       " [577]  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0\n",
       " [601]  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0\n",
       " [625]  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0\n",
       " [649]  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0\n",
       " [673]  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0\n",
       " [697]  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0\n",
       " [721]  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0\n",
       " [745]  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0\n",
       " [769]  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0\n",
       " [793]  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0\n",
       " [817]  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0\n",
       " [841]  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0\n",
       " [865]  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0\n",
       " [889]  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0\n",
       " [913]  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0\n",
       " [937]  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0\n",
       " [961]  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0\n",
       " [985]  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0\n",
       "\n",
       "[[2]]\n",
       "  [1] 27 25 26 23 24 25 28 28 28 28 29 29 27 27 28 30 28 28 28 26 26 27 25 28 30\n",
       " [26] 30 28 28 26 25 22 22 20 19 18 19 19 19 18 18 17 16 15 14 13 13 11 11 13 11\n",
       " [51] 12 12 12 12 13 12 11 10  8  7  6  7  7  9  9 11 11 12 12 13 13 14 14 14 14\n",
       " [76] 15 15 14 16 16 17 15 16 16 17 17 16 18 19 19 19 19 19 20 20 20 19 21 19 19\n",
       "[101] 20 22 22 24 22 23 24 25 25 27 28 28 30 31 31 32 34 34 35 33 35 37 39 39 42\n",
       "[126] 43 46 47 46 50 51 49 45 50 53 51 48 51 50 50 53 52 52 52 50 52 54 54 55 55\n",
       "[151] 58 55 55 55 54 54 52 49 48 48 46 46 44 45 44 45 48 48 47 48 50 53 51 53 52\n",
       "[176] 50 51 52 53 56 55 54 53 54 53 52 51 50 50 53 52 57 56 54 54 56 57 57 56 60\n",
       "[201] 58 54 55 55 50 48 48 50 51 50 51 51 49 48 49 46 46 48 46 46 45 44 43 44 45\n",
       "[226] 45 42 43 46 46 46 48 50 51 48 49 48 49 50 49 47 52 52 49 48 50 50 50 48 49\n",
       "[251] 49 50 48 49 48 48 50 49 50 47 51 46 52 52 50 48 45 45 44 44 42 44 45 44 41\n",
       "[276] 42 42 42 44 42 43 48 50 50 51 49 48 48 53 53 57 57 56 53 56 50 51 51 49 52\n",
       "[301] 50 52 49 50 46 47 45 46 48 48 50 49 50 52 49 51 49 44 43 40 42 44 43 39 39\n",
       "[326] 34 34 34 34 34 35 34 37 37 40 41 39 40 41 44 43 41 44 45 42 47 43 49 49 49\n",
       "[351] 49 50 50 54 55 54 54 56 62 59 58 61 60 58 62 63 60 64 60 58 60 61 60 58 55\n",
       "[376] 54 54 55 52 54 49 49 48 46 48 47 46 46 45 45 48 45 44 39 42 42 43 41 42 41\n",
       "[401] 44 43 41 43 43 47 43 45 46 47 47 48 47 48 48 45 48 48 48 49 48 49 51 49 49\n",
       "[426] 46 44 45 44 46 46 44 44 42 42 41 41 43 42 41 39 38 42 41 37 38 43 38 40 41\n",
       "[451] 37 41 43 42 43 42 39 41 41 43 48 51 53 53 56 53 53 56 59 61 58 60 60 57 56\n",
       "[476] 56 57 55 55 59 59 52 60 59 61 60 59 58 59 59 58 60 53 54 54 48 48 46 47 44\n",
       "[501] 44 46 43 41 39 38 40 43 44 44 45 46 46 48 47 47 47 47 48 49 49 50 48 51 54\n",
       "[526] 56 56 54 56 57 57 57 56 54 55 53 54 57 58 59 59 56 54 55 56 53 52 49 49 47\n",
       "[551] 45 47 44 40 41 42 41 40 40 37 36 36 35 36 33 35 34 35 38 37 38 37 35 36 36\n",
       "[576] 36 39 36 36 34 36 36 37 35 36 39 39 40 40 40 42 44 43 44 44 39 42 41 47 47\n",
       "[601] 46 45 44 46 46 45 47 48 50 50 49 47 47 48 46 44 47 48 49 48 46 45 47 48 45\n",
       "[626] 42 41 39 38 43 45 43 44 40 40 39 42 38 36 41 41 40 42 40 42 41 41 40 40 38\n",
       "[651] 41 40 41 41 43 44 44 40 39 40 44 41 43 39 37 34 34 34 36 38 43 44 44 43 42\n",
       "[676] 41 40 41 42 45 45 47 46 45 44 48 51 53 53 51 53 50 50 46 51 51 46 52 51 50\n",
       "[701] 47 42 42 43 37 43 45 47 49 46 52 53 53 56 54 55 55 58 54 55 53 52 46 47 47\n",
       "[726] 44 48 48 47 46 45 44 46 47 48 48 48 49 48 45 43 43 41 40 38 36 35 38 34 35\n",
       "[751] 36 33 32 32 29 30 31 30 29 30 32 31 30 29 30 29 32 34 37 37 36 37 36 35 35\n",
       "[776] 36 36 38 37 36 35 37 37 38 36 35 37 38 37 38 39 44 42 44 40 40 39 36 39 38\n",
       "[801] 41 39 38 36 37 36 35 34 37 37 39 38 34 37 38 40 40 37 40 42 44 42 42 44 43\n",
       "[826] 43 43 43 44 45 46 45 46 45 45 43 44 43 44 45 44 44 44 42 40 42 41 43 42 43\n",
       "[851] 43 43 45 46 43 39 42 43 43 46 44 41 40 41 45 46 45 47 47 52 54 55 55 53 54\n",
       "[876] 53 53 56 59 60 61 62 61 60 60 59 61 60 61 58 60 58 59 59 57 58 59 59 62 63\n",
       "[901] 61\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# use the `GRbaseCoverage` method from `compEpiTools` package \n",
    "# to calculate the coverage of each base (i.e. average number of times a base is read) for each of the sequences in `GRanges` object\n",
    "res = GRbaseCoverage(Object=gr, bam=bampath)\n",
    "\n",
    "# the output is a list of equal length to the passed GRange object, namely `Object`\n",
    "res"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## - Use `GRcoverage` method to calculate the total coverage "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>20359</li>\n",
       "\t<li>38592</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 20359\n",
       "\\item 38592\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 20359\n",
       "2. 38592\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 20359 38592"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# total coverage of each range in GRanges (seq1 and seq2)\n",
    "GRcoverage(Object = gr, bam = bampath, Nnorm=FALSE, Snorm=FALSE)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "20359"
      ],
      "text/latex": [
       "20359"
      ],
      "text/markdown": [
       "20359"
      ],
      "text/plain": [
       "[1] 20359"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# total coverage of seq1\n",
    "GRcoverage(Object = gr, bam = bampath, Nnorm=FALSE, Snorm=FALSE)[1]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## - Use `GRcoverageInbins()` method to calculate the total coverage in equally-sized bins"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<tbody>\n",
       "\t<tr><td>8148</td><td>7969</td><td>4242</td><td>   0</td><td>   0</td></tr>\n",
       "\t<tr><td>5370</td><td>8645</td><td>8884</td><td>7850</td><td>7843</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "\\begin{tabular}{lllll}\n",
       "\t 8148 & 7969 & 4242 &    0 &    0\\\\\n",
       "\t 5370 & 8645 & 8884 & 7850 & 7843\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "| 8148 | 7969 | 4242 |    0 |    0 | \n",
       "| 5370 | 8645 | 8884 | 7850 | 7843 | \n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "     [,1] [,2] [,3] [,4] [,5]\n",
       "[1,] 8148 7969 4242    0    0\n",
       "[2,] 5370 8645 8884 7850 7843"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# divide each range in `gr` (seq1 and seq2) into 5 bins in equal size\n",
    "GRcoverageInbins(Object = gr, bam = bampath, Nnorm=FALSE, Snorm = FALSE, Nbins=5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>8148</li>\n",
       "\t<li>7969</li>\n",
       "\t<li>4242</li>\n",
       "\t<li>0</li>\n",
       "\t<li>0</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 8148\n",
       "\\item 7969\n",
       "\\item 4242\n",
       "\\item 0\n",
       "\\item 0\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 8148\n",
       "2. 7969\n",
       "3. 4242\n",
       "4. 0\n",
       "5. 0\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 8148 7969 4242    0    0"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# select the first range (seq1)\n",
    "GRcoverageInbins(Object = gr, bam = bampath, Nnorm=FALSE, Snorm = FALSE, Nbins=5)[1,]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## - Use `GRcoverageSummit()` method to calculate the maximum coverage of each range"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "GRanges object with 2 ranges and 0 metadata columns:\n",
       "      seqnames       ranges strand\n",
       "         <Rle>    <IRanges>  <Rle>\n",
       "  [1]     seq1 [1340, 1340]      +\n",
       "  [2]     seq2 [ 467,  467]      +\n",
       "  -------\n",
       "  seqinfo: 2 sequences from an unspecified genome; no seqlengths"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# return the maximum coverage with each range (seq1, seq2) in the GRange object\n",
    "# the `summit` object defined will be used to highlight the position of maximum coverage\n",
    "summit = GRcoverageSummit(Object = gr, bam = bampath)\n",
    "summit"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## - Plot the coverage (number of bases overlaping) in `seq1` interval "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAA0gAAANICAMAAADKOT/pAAAAMFBMVEUAAABNTU1oaGh8fHyM\njIyampqnp6eysrK9vb3Hx8fQ0NDZ2dnh4eHp6enw8PD////QFLu4AAAACXBIWXMAABJ0AAAS\ndAHeZh94AAAgAElEQVR4nO3djXqivBZA4aBUHaty/3c7FVEREQnZSfYO6z3n+capP5uKa6pA\n1TUAgrncCwCUgJAAAYQECCAkQAAhAQIICRBASIAAQgIEEBIggJAAAYQECCAkQAAhAQIICRBA\nSIAAQgIEEBIggJAAAYQECCAkQAAhAQIICRBASIAAQgIEEBIggJAAAYQECCAkQAAhAQIICRBA\nSIAAQgIEEBIggJAAAYQECCAkQAAhAQIICRBASIAAQgIEEBIggJAAAYQECCAkQAAhAQIICRBA\nSIAAQgIEEBIggJAAAYQECCAkQAAhAQIICRBASIAAQgIEEBIggJAAAYQECCAkQAAhAQIICRBA\nSIAAQgIEEBIggJAAAYQECCAkQAAhAQIICRBASIAAQgIEEBIggJAAAYQECCAkQAAhAQIICRBA\nSIAAQgIEEBIggJAAAYQECCAkQAAhAQIICRBASIAAQgIEEBIggJAAAYQECCAkQAAhAQIICRBA\nSIAAQgIEEBIgIEFIDjBmwaNcPpwMIwBJhAQIICQdljw1gCKEpAMhGUdIOhCScYSkAyEZR0g6\nEJJxhKQDIRlHSDoQknGEpAMhGUdIOhCScYQECCAkQAAhAQIICRBASIAAQgIEEBIggJB0YD+S\ncYSkAyEZR0g6EJJxhKQDIRlHSDoQknGEpAMhGUdIOhCScYSkAyEZR0g6EJJxhKTECr/lohCS\nEiv8lotCSEqs8FsuStKQfvd1+wkY9e431gizVvgtFyVhSJdN79NktlFGGLbCb7koCUPauerf\nqT11PlZuF2OEYWy2sy1hSJU7PU6fXBVjhGGEZFvCkF4eKtOPm/U9qBZ9diL04CeSDoRkXNrX\nSMdze4rXSG8IybiUm7+3va12m0uUEWYRknFp9yPt2v1IVb1nP9IAIRnHkQ06EJJxekJyfXFG\naNR9q9fveUXfdXlShnTZXTfV7TfObf9FGmEPIZUhYUjn6u/Bcqk4ROgFIZUhYUg/rr78/efn\n/NfUD5u/O/eQGkIyLemRDZfuP3/P8tgh2yGkMqQ+RKhyvb+IjzCIkMqQ9KndqWn2t+OELtMv\nklb0kOr+RSEk4xKGdHLV7tTU1V9Jx407xhhhECGVIeXm72P13FG0jzPCHkIqQ9odsv9+2t+S\nrffnaCOsIaQy6DmyIfEILW77j1Z3QEdxCCkzQioDIWVGSGUgpMwIqQyElBkhlYGQMiOkMhBS\nZoRUBkLKjJDKQEiZEVIZCCmv7oiG9v+3X+5bz/deFELKqx9SQ0h2EVJehFQIQsqLkApBSHkR\nUiEIKS/3eAOh6/fMpjuzCCkvQioEIeV1D+kWECGZRUh5EVIhCCmv51ENt5D4CEyjCCmvLqTe\nWzes53svCiHlRUiFIKS8CKkQhJQXIRWCkPIipEIQUl6EVAhCyuse0uOvhGQTIeVFSIVYXUjK\nHqeEVAhCymsQUqNuATEPIeVFSIUgpLwIqRCElBchFYKQsmoP+X496FvXAmImQspqJCRlS4h5\nCCkrQioFIWVFSKUgpKwIqRTrCsm1D9tIN74EIZWCkLIipFIQUlaEVApCyoqQSkFIWT1CGnwN\n1hBSVoRUCkLKipBKQUhZEVIpCCkrQirFykJ6fJKXikdr94bfg4VRsWjwREgZEVI5CCmjXkjs\nRzKOkDIipHIQUkaEVI61hXT7vFYl7x5HSOUgpIwIqRyElBEhlYOQMiKkchBSRoRUjpWF1BAS\noiCkjDiyoRyElBEhlYOQMiKkcqwupFtDKkK6LwshFYCQ8iGkghBSPoRUEELKh5AKsoaQ+rto\nXPsK32UPqSuI/UilWFtIjb6QGkIqACHlQUiFIaQ8CKkwhJQHIRWGkPIgpMIQUh7XBfgQEiVZ\nREh5EFJhCCkPQipM0pB+9/X1d1RdvfuNNWL61vSG9HJehuVBoIQhXTbuaRtlxIdbezkWR01I\n7nFs0DCk3AsHfwlD2rnq36k9dT5WbhdjxIdbew8p+7/6t4rGl4KQDEoYUuVOj9MnV8UY8eHW\nCAmxJQzp5fEx/WAhJBizgp9Iz9+dICTEkvY10vHcnkr7GqndttBtJWtshJR76eAt5ebvbW+r\n3eYSZcT4jRkI6WU/EiHZk3Y/0q7dj1TV+5T7kQgJ8a3gyAZCQnx6QnJ9ojfc3XT3YO1Hlc/9\nLe2efx+eB1PSh3TYOFcfo454vzH32HTX/akjpP7fP58HA5LvR+q2OExutCOkxEuDYKlD2rnd\npWnOO3eIMeLjjRES4kodUuXa7d4Xt4kx4uONERLiSh3S/TGS8BCht5Da/2d+sL69KyQh2ZY6\npJ97SCkPETIQ0st5hGRP0pDq/eHo/v2dvOySHiJESIgtaUiPfUTOVSkPESIkxJZyP9LpdDjU\ndbvJYTfZUZyQHCEhGj1HNkQbcd/G8Xx4usd/siGkwhBSHoRUGELKg5AKQ0h5sB+pMCsJqbEU\nEr9rbhAh5TEdUuaFgz9CyoOQCkNIeRBSYVYdUsbXIoRUmLWE9H4id0iDv7vJs6EdIeVBSIUh\npDwIqTCElMeXyYRkDSFl8W0wIVlDSFkQUmkIKQtCKg0hZUFIpSGkLAipNIWH9PI+4sOQMhxA\n4AZ/frtc9AWBkNJD+vxTR1dIqfcjEZIwQkqKkEpFSEkRUqkIKSlCKlXxIX28rZWHREmyCCkp\nQioVISVFSKUipKQIqVSElFL38B0ZnDgk6Q+8RvkhfT5LU0jDC8ZeELbbCSOklAipWISUEiEV\ni5BSIqRirTek26f4pdV91Nn3F/oJQqIkUYSUEiEVi5BSIqRiEVJKn0NKvR8pwYx1WXtIaR9N\n3YerawuJpMIRUkpdSOPnvPw19oI0hCSLkFIipGIRUkqEVCxCSomQirX6kJI+iJz7dNh1hpAe\nQzgQXAAhSY2ag5CKRUhSo+YgpGIRktSoOT6HNLxg7AXpDyEkASsPaephHX7kw9vV54cU7cHt\n7v+/Lx6/LCui7JCmb8dNtyIf0vwfgElDYqudAEL6fDYhYTZC+nw2IWE2Qvp8NiFhNkKaOlc+\npLnXjBWSG4bkCEkEIU1cNzikwfWnpg32I6ULySNvfEZIE9clJMxFSBPXJSTMRUgT1yUkzEVI\nE9ftzl62NCOPUA0h3RaCkKStOaTpi9wSEg7p86XzhZRkjRaPkKbOICTMREhTZxASZiKkqTMI\nCTMR0tQZXUhLFsc9H7A+i/Ny7QhGQ6KkcIQ0dQYhYSZCmjqDkDATIU2dQUiYyW5IczYEBITk\nbgktfvM7vSE9vqPn/UdIwQhp8utubOPbLIS0LoQ0+XVCwjyENPn1ZCG97UciJFsIafLrhIR5\njIf0ZYMaIb3NIaQ4CGny64SEeQhp8uuEhHmShvS7r9v37K13v+EjCGkJQookYUiXjXvaBo5w\nt/eRmg5pziEJX0Na+Jh+C+ktq+HF368d4RE+CMkRkpCEIe1c9e/UnjofK7cLG3EPafKi836w\nffn6922D49dXGdLtm+mF9PwygiQMqXKnx+mTq8JGENIihBRLwpBeHitfnpPNWIiyQhq9NiHZ\nwU+kb18nJMyQ9jXS8dyekniNdN/WEBrSpwsFhuTerui1zcK9X18EIcWSMKRm29tqt7mEjSCk\nRT6EREnBUobU/O7a/UhVvQ/ej0RIixBSLElDEhxBSIsQUix6QnJ9sy79COllc+CH01O3NPrV\n/o14vzvWWwieB0gQkjVZQvr6sJwZ0v2f+Q/xzFzQLyF10+bd1tvVH39+++H5eqZr4oXUEFIE\nhERIEeasT8KQnJv97I2QGkKyJWFIvxUhzV0IQrIm5VO7S+227R5Z/U/t3g59IyRMSvsa6Z9z\n/xqDIfltcmtGQ5r+rkdC6j/QpbiXaY6QxCTe2HDeuvpCSF8HEpI1iUNqmr2rjoT0bSAhWZM8\npOa0+f6Kg5C6/xCSFelDapqfoJBuRzT0Qno8HNzgqj4hDcN5D2n+9ob+Rkn3fOj6bLCIEpJ7\nC+mtdyyVI6SwEe3mr+fjMjyksc1jIw/6wJB8buC5XHFDejkLQQiJkBpCCkdIhNQQUrgSQnKE\n5Hl7hCTOXEju/qB8C8ktDWnkESsW0nMx/UPy3lb49eY+bqUjpFCE1BASIYUjpEZFSONHzhKS\nGYTUEBIhhTMcUvf37pfO78dXi4Q0tvd1dkiueQ3JdUXkDsn1bo6QpFkMqRkPqREMaey680N6\nDfEe0rdtkWO3FCuk90kIQ0gNIRFSOEJqCImQwhFSQ0iEFK6MkNz9Jf3wETx73IyQ5t6ea167\neGxnIKSCEVJDSIQUjpAaFSF9uCVCsoKQGkIipHBFhfT8vdlvt/J2m3IhOcGQ/A/Pm741QorG\nfEjd37o/l4fkBl/wW6rhpdzrFxYGMfUzZMGtLTwPcxBSQ0iEFI6QGkIipHCE1BASIYWzFtKz\nkbHNc/cjwD1f2w8f6h+uGTOk8f1IhGQGITWEREjhCKkhJEIKR0gNIRFSuFJD8nqHYNe87q38\nGNL0Tfb6fQ1pxo7VjyEJ3dvuS0iUFMhuSJ8v0f0vZ0gjX/+yCKOX+fLw9/B12WXGrBchNYTU\nEFKwwJAe67+qJJZmbMToOYTkiZAiEwrpLPskm5CeC/b9unMQUmQBIR1d3ybNUs0Oyafsx++p\nT4+PGdKniYRkRMhPpE2/o980S5U5pMkb/fTODCpC+rYQlBRG6jWSLEJ6LAgh2VDsVjtCGiwD\nIUVFSA0hdaMQIDSk/eOFktQSvY14OWNGSM41ngt0PyDiy/hZIY1cwkZIlBQkMKT9c3OD2CI1\nM5ZqMqQZD5vhVV63twmHtDyGhCHxIylMYEiVO4gtyocRvhewGVL0/UiEFJe1rXbfL0BI47dP\nSFEFhlS7i9iifBjhewFCGr99QooqMKRztRXdEzsywvcCt3dt8AupKT2kGUdMEVKQ4Kd2Kjc2\nENLbDRFSXITUENJ9FpYLDCkSQnpckZBsIKRmfkhfDqEmpBUjpJEblQ5pzmxCso3XSCM3miGk\nj0tESDYQ0siNEhJ8yTy1+93W4YsyPWL2BQhp9IYIKS6h10gX9xO8KF9GzL1AtpDauWpD+npD\nhBREamMDT+0IadWEQjq4RG/H9f0ChLTshggpiNjGhr3YIjWE1LtJQrJBKKSN7K8lZQzJfb51\n1/8N9pd7oTd34ePx834kkSfNs0IipQBSr5Fk5Vml30Jq5oS0dPbHKxKSCYQ0mEpIWCI4pH/b\nvwdX/U9ocUZHpEJIWCw0pG33ImkrtUDvI5IhJCwWGNLBVce/P47Cb4KSM6SJx1z/3e8ShiRw\nb8zZ/McbcoUIDGnjTu2fp1Rvoh8TIWGx4M3fwxMiCKl/DiEZIPYTKe2RDVEQEhbjNVJvqmsf\ntp9fdT/fI/x+oLd7fD0wpKmlIiQD2GrXm3rLYcYbM7Tb74Yh3W9CfKlkQhK6EMaF70eqy9mP\nREhYKjikKAipv1SEZAAh9aYSEpYKDemyu26uq3aybwGeLSTnFZIjJHQCQzpX3caq6iy1RMMR\n6RASFgsMaet+rj+LLjsn+u4n6wsp/n4koQthXGBIZR3ZQEhYKjCkqvt8pMustf27r9udTvXu\ny2fBaA7Jdbtl7/9vblk5Qlq1wJB2rv18pN+t23293uXxuc1fd+DmC2nGMQDuHo575hP8DpmE\nZFzoVjuPIxt2rvp3OzLvfKymwyOk/jmEZEDwfqT2yIbtnCPtqu4A16svB7kSUv8cQjIg4Q7Z\nlwfEt1f0ORASFksYUmk/kR6bGLovhM7+eE74vTHvFggpQEBI7tXX6/29RjredttqfY30ctzP\nxOUeP4xShNQIvJkCIUWXMKTHhomrzeQxResLaXqxwm9C7lIYFfrUrm5/se+3mvVhFL+7dj9S\nVe917kciJCwVvB/p/qvm3/cjLRyRDiFhscCQBA8R8nyeGMGykJ7/i7dY4TchdymMCj5EyP/N\nTw7V1/fcz7VK5xXc2/LtngkR0qoFP7Wrri93jtWcj3U51a46NPsZR0IQUn8cIRkgdYjQjN+i\nOLUX3F1/8eJcT7/rECH1xxGSATKHCNXHGdf7uW6Q2N2eA16m35l1fSGxH8m45IcIdT+7NB4i\ntCSk/i8lRZqcLCRKCpA8pH+353QaDxEipNA5KyYV0u/3F0k/t19Lv7r8aDxEiJBC56xYaEi7\n+bt+LlVvr9P01nJC6p9FSAYEb/6+m7O5YXfPp/pyHES2kOZd6vlb5oWFREmLBe+Q/dds3fm8\ndV+Onls+IqXZ/3I/H3Lf3uNh7mQNIfEjaTmBQ4T2fz+NTrLvok9Ir9OCb134cngjENLxunNV\n9lkBIb1OC7514cvhTWBI9d9Tu7PbNL+EFA0hWRAY0vH6KGoPE5r1C0lLRqRESFgodPP3/vq3\nHyf760iENJgWfhOyl8ObhEc26BoRMpeQMERIC+YSEoaCQzrW1wdSLfqpLoQ0mBZ6hxBSdKEh\nbbvfuS7h85HmS/zL8ISkX2BIB7dtP4jiUMZWu7nkQ5q8QULSL/gQoUtz/w03QdpXKCFhQODI\nBkISucWpMwlJvcCQNt1PpNP0r46HjNBI/s23ooY0+9ra73fFZF4jHavpNzMJGaERIWEgdKtd\n3f06kujB3+pXKCFhQGQ/kqv/CS3O6Ah9CAkDHNmwRPqQQubNv7L2+12xwJBq2YNVx0ZolDqk\nsK2EhJSAwObvCLSv0IhvmT86jpDUE9j8HYH2FUpIGAgM6VJvRd/1ZGSERoSEgeCndlE+0kj7\nCjUVksfCar/fFSOkJRKHFHZIEiGlwObvJQgJA4S0BCFhgJCWSLwfiZD0I6QlMoTkFt8rhJQC\nIS2ROqQmLKQYF8UrQlqCkDBASEsQEgYIaQlCwgAhLZElpKVb7nyupv2O14uQliAkDBDSEoSE\nAUJaIvWRDYSkHiEtQUgYICQbCEk5QrKBkJQjJBsISTlCsoGQlCMkG9qOlt0thJQCIenw7ccN\nISlHSDoQknGEpAMhGUdIOhCScYSkw/dNcoSkGiHpQEjGEZIOhGQcIelASMYRkg6EZBwhWUFI\nqhGSFYSkGiFZQUiqEZIVbUhLYiKkFAjJiqUheV2BO34pQrKCkFQjJCsISTVC0mHmfiRC0oqQ\ndJgRUkNIihGSDoRkHCHpQEjGEZIOhGQcIelASMYRkg7zQmrf3c7rSAXP8tZ3x0shJB3mh+R3\nyA8hJUJIOhCScYRkBSGpRkhWEJJqhGTF9T4hJLUIyQpCUo2QrCAk1ZKG9Luv3VW9+401olyE\npFrCkC4b97SNMqJk/iG1e2/9Pp5s6YeZIWFIO1f9O7WnzsfK7WKMsGvWfqQFIfneleu754Uk\nDKlyp8fpk6tijLCLkIxLGNLLY2X6gbO+1UlIxvETSQdCMi7ta6TjuT3Fa6Q3c0NyHnfOog8n\nW99dLyPl5u9tb6vd5hJlhFmEZFza/Ui7dj9SVe/ZjzRASMZxZIMOhGScnpBcX5wRikUIyRFS\nQilDuvw4tz12N8Lm7wW8Q1owwv8qaNIeIlTdDrS73QghLUBIaiXd/H34q+lQtYfZEdIShKRW\n0h2y7R/nanMmpGUISa0MhwhdtltCWoaQ1Ep4V2/cfSfsZktIi9y2Z868dwgppYR39cH9dKfO\nbktISxCSWinv6t2jnuOXXUXrW5vzdp0RklpJ7+pTfT91/iGkF/IhuYUhre++F6Hz36z1rcyZ\nj1+fkHx+4+LleliAkHQgJOMISQdCMo6QdCAk4whJB0IyjpB0mLuxLH5IK7zzJRCSDoRkHCHZ\nQkhKEZIthKQUIdnyJaT7oVftb5kvvBu585cgJHumQmpc709CSoeQ7CEkhQjJHkJSiJDsISSF\nCEkHn99e+HzRxwYGQkqNkHQgJOMISQdCMo6QdCAk4whJh8UhDf9ye8cuQkqNkHTweqsE9+H0\n7S/PkBYuyuJrrhkh6UBIxhGSDoRkHCHpQEjGEZJBM0IKuAe585cgJIMISR9CMoiQ9CEkgwhJ\nH0IyyN3flsvdT7dbF7p8Qj/Omnf/XoKQDHoJyfVC6l8k5ObhjZAMIiR9CEkHz/1IhKQNIelA\nSMYRkg4LQrpvZnC9oHoXCVmYkCuvFCHpQEjGEZIOhGQcIelASMYRkg5+IXX1tP8lJBUISQfP\nkLpNdvf/Ct9f67v7wxGSDoRkHCEZREj6EJJBhKQPIRk0TIiQ8iMkg571EJIWhGQQIelDSAYR\nkj6EZBAh6UNIOvi/r10/JOmFEb69NSAkHfx/qBCSKoSkAyEZR0g6EJJxhKTDgpCeVyOk/AhJ\nh6UhPU9KLozw7a0BIelASMYRkg6EZBwh6UBIxhGSVYSkCiFZFfE+4u73R0hWEZIqhGQVIalC\nSFYRkiqEZFXM+4j73xshWUVIqhCSDgv3I0Wyvvs/GCHpQEjGEZIOykJa3woIRUg6KHvo6loa\nCwhJB0IyjpB0ICTjCEkHQjKOkHQgJOMISQdCMo6QMIIV4IuQMIIV4CtpSL/7uv0c4Xr3G2sE\nRLACfCUM6bJxT9soIyCEFeArYUg7V/07tafOx8rtYoyAEFaAr4QhVe70OH1yVYwREMIK8JUw\npJctvNObe1mPmbECfPETSQf2IxmX9jXS8dye4jXSG0IyLuXm721vq93mEmWEWYRkXNr9SLt2\nP1JV79mPNEBIxnFkgw6EZJyekFxfnBGKKfuWdS2NBXpCSjxCGUIyjpB0ICTjCEkHQjIu6ZEN\ns18GsR4zYwX4ShjSgZDsYA14SvnU7lRN//KEwAgIYQ14Svoa6TR9YJDECMhgDXhKu7Hh0Dtu\nNdIIiGANeGKrHcawBjwREsawBjwRkg7K9iOtcA0EIiQdCMk4QtKBkIwjJB0IyThC0oGQjCMk\nHQjJOELSgZCMIyQdCMk4QtKBkIwjJIxhDXgiJIxhDXgiJIxhDXgiJIxhDXgiJIxhDXgiJIxh\nDXgiJIxhDXgiJB3Yj2QcIelASMYRkg6EZBwh6UBIxhGSDoRkHCHpQEjGEZIO2kJa4SoIQ0g6\nEJJxhKQDIRlHSBjFKvBDSBjFKvBDSBjFKvBDSBjFKvBDSBjFKvBDSBjFKvBDSBjHOvBCSDqo\n24+0wnUQhJB0ICTjCEkHQjKOkHQgJOMISQdCMo6QdCAk4whJB0IyjpB0ICTjCEkHQjKOkDCO\ndeCFkDCOdeCFkDCOdeCFkDCOdeCFkDCOdeCFkDCOdeCFkDCOdeCFkHRgP5JxhKQDIRlHSDoQ\nknGEpAMhGUdIOhCScYSkAyEZR0g6EJJxhKQDIRlHSDoQknGEhHGsAy+EhHGsAy+EhA9YCT4I\nCR+wEnwQEj5gJfggJHzASvBBSPiAleCDkHTQtx9phSshBCHpQEjGEZIOhGQcIelASMYRkg4K\nQ1rhWghASDoQknGEpAMhGUdIOhCScYSkAyEZR0j4hLXggZDwCWvBAyHhE9aCB0LCJ6wFD0lD\n+t3X7qre/cYaATmsBQ8JQ7ps3NM2yghIYi14SBjSzlX/Tu2p87FyuxgjIIm14CFhSJU7PU6f\nXBVjhF3sRzIuYUgvj5X3B47rWzjCLo3fssObz3fWgvt34XrhJ9IEjSHBQ9rXSMdze4rXSG8I\nybiUm7+3vR+Rm0uUEWYRknFp9yPt2v1IVb1nP9IAIRnHkQ06EJJxhKQDIRlHSDoQknGEBAgg\nJEAAIQECCAkQQEiAAEICBBASIICQdGA/knGEpAMhGUdIOhCScYSkAyEZR0g6EJJxhKQDIRlH\nSDoQknGEpAMhGac0JMCYBY9y+XBMzGY+80XnExLzma/txgzNZj7zCYn5zNc2n5CYz3xtN2Zo\nNvOZT0jMZ762+YTEfOZruzFDs5nPfEJiPvO1zSck5jNf240Zms185hcTElAMQgIEEBIggJAA\nAYQECCAkQAAhAQIICRBASIAAQgIEEBIggJAAAYQECCAkQAAhAQIICRCQLaRd5ardJfHQw+Yx\ntDc/6aL8dvd4lvmnH+d+ztnmX8aHJpp/uD/WoyxFrpC27Zv+b9IO3bVDq8vr/KSLcqlu93iW\n+ce83/+5us0/Z5l/un/GxPjo0KXIFNKvq07NqXK/KYee3M/l+g/Tz8v8tItS31ZnnvnV36RL\n7XaZ5v9cJ//9a5bl/v+bcHusj48OXopMIe3c8e+//9w+5dD69s1e79De/KSL8q/77J0s8/+1\nD+SLqzLNdxnv/4PbduPHRwcvRaaQanf9+X5ydYbZ1zu0Nz/lopzvqzPL/B93up/MMr97VnsN\nOfn8v39DupDGRwcvRaaQev84pXZx25f5KRdl6863OVnmb1yzr9qnt3nm77undvsM80/DQYPR\nwUuxvpAO1x/ieULau39NxpCcq9sX+7nmN4fr1obqkGk+IYk6V3WT6YHUPnHIGtJ1Y8NPlp8I\nrX27ZWzfEJKYbCFdqu1gfsKnVtcNz1lDur5GOl+38WaZf7g+tfsL+UBIYqpcIW03w/nJFuWn\n3TB0m5Njfv/BkmX+xl1fnl2uIeeY300YHx28FJlCum0kOafeanfebM/D+ckWpf/h8znm9zf/\nZ5nv8s5/2Wo3HB28FJlC2rf/Oh/bzTjpHN32fX6yRemHlGN+N+l8vROyzL/9q9/ux8pz/7d/\njI8OXopMIWU5suH86CjfkQ3d6swy/+/V0eX6GuVfpvk7dz2WbZfryIoupLKObPh7vny1/X5B\nQT/Pnwj9+WkXpVudWebvR4emm7/NOv/++md8dOhS5ArpdiBw2pm9p1b9+WkXpVudeeYftyND\nE84fHZpq/j2k8dGhS5ErJKAohAQIICRAACEBAggJEEBIgABCAgQQEiCAkAABhAQIICRAACEB\nAggJEEBIgABCAgQQEiCAkAABhAQIICRAACEBAggJEEBIgABCAgQQEiCAkAABhAQIICRAACEB\nAggJEEBIgABCAgQQEiCAkAABhGTN8k+wR0SsFWsISSXWijWEpBJrxRpCUom1Ys1fSLvu87d7\nJ5EZIVnjXO3+bNuT+/tJZEZI1jhXnZpT5f69nERmhGSNc8e//x5d/XISmRGSNd3GhusfvYbd\nqXIAAADASURBVJPIjHVgDSGpxDqwhpBUYh1Y49xv83iNdDv5k3uZQEjm3DfVHV9OIjNCssa5\nn+vOo7o9ub2fRGaEZM3tcIZ9d7J2m0PuJUJDSKaxlUEPVoVhhKQHq8IwQtKDVWEYIenBqgAE\nEBIggJAAAYQECCAkQAAhAQIICRBASIAAQgIEEBIggJAAAYQECCAkQAAhAQIICRBASIAAQgIE\nEBIggJAAAYQECCAkQAAhAQIICRBASIAAQgIEEBIggJAAAf8BijyUkG4NLsIAAAAASUVORK5C\nYII=",
      "text/plain": [
       "plot without title"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# plot the result of the base coverage `res` for the first intervel range (i.e. seq1) in IRnage object (i.e. ir)\n",
    "plot(res[[1]], type='l', xlab='bp', ylab = 'readcount')\n",
    "\n",
    "#identify the maximum coverage and use a dotted line to highlight\n",
    "abline(v=start(summit[1])-start(gr[1])+1, lty=2, lwd=2)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## - Plot the coverage (number of bases overlaping) in `seq2` interval "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAA0gAAANICAMAAADKOT/pAAAAMFBMVEUAAABNTU1oaGh8fHyM\njIyampqnp6eysrK9vb3Hx8fQ0NDZ2dnh4eHp6enw8PD////QFLu4AAAACXBIWXMAABJ0AAAS\ndAHeZh94AAAgAElEQVR4nO2diWLiuhIF24FAhgD+/7+dYLxb3ltyS1S9dxmCJZ3GVoV4ASQH\ngN3I0QUApAAiASiASAAKIBKAAogEoAAiASiASAAKIBKAAogEoAAiASiASAAKIBKAAogEoAAi\nASiASAAKIBKAAogEoAAiASiASAAKIBKAAogEoAAiASiASAAKIBKAAogEoAAiASiASAAKIBKA\nAogEoAAiASiASAAKIBKAAogEoAAiASiASAAKIBKAAogEoAAiASiASAAKIBKAAogEoAAiASiA\nSAAKIBKAAogEoAAiASiASAAKIBKAAogEoAAiASiASAAKIBKAAogEoAAiASiASAAKIBKAAogE\noAAiASiASAAKIBKAAogEoAAiASiASAAKIBKAAogEoAAiASiASAAKIBKAAogEoAAiASiASAAK\nIBKAAogEoAAiASiASAAKIBKAAogEoAAiASiASAAKIBKAAogEoAAiASiASAAKIBKAAogEoAAi\nASiASAAKIBKAAogEoAAiASiASAAKIBKAAogEoAAiASiASAAKIBKAAogEoEAAkQQgMjbMcn1x\nDogA0ASRABRApOjY8lcE+AaRogORLIJI0YFIFkGk6EAkiyBSdCCSRRApOhDJIogUHYhkEUSK\nDkSyCCJFByJZBJEAFEAkAAUQCUABRAJQAJEAFEAkAAUQCUABRIoOziNZBJGiA5EsgkjRgUgW\nQaToQCSLIFJ0IJJFECk6EMkiiBQdiGQRRIoORLIIIkUHIlkEkeKDtWMQRIoP1o5BECk+WDsG\nQaT4YO0YBJHig7VjEESKDo7aHcf4qkek6ECk40CkhECk40CkhECk40CkhECk40CkdNj0Bdqg\nwsSqR6TYQKTjQKSEQKTjQKSEQKTjQKSEQKTjQKREeK0XwaSgtFc2IiUCIoUHkRIEkcKDSAmC\nSOFBpARBpPAgUoIUInGNUEikPRkRKREQKTiIlCKIFBxEShH2kYJjUqTf6/k1DeR8+fUVkTZS\nvB6JKK0gcd6FNsW6rgSyIdLzSxpOXiJSB5GC0xVpqt2Gobdxkezfvbj3uGVy8RGROogUHIMi\nZXKv798l8xGROogUHIMidf7AnN5hZrO6QaTgGBSJV6TdIFJwDIr0t490exT32EfaSHWoRulU\nEiLN0xZpaiWFPPx9ah21+3p6iUgcRAqORZHy30txHik7XzmPtInaIkQKhUmRLEVECSIFJz6R\npI2fCG2Cl6krUrOiyzuRrPagSN7MSCsiPS+vQ3XXL5HTP08RYTlIJK3rVhFpAe/XI1MiPbK/\ncp5ZQpcIIVL6GBTpW87Pv5vvx59T30kc/kak9DEoksizvPn7Ky+JE7KIlD4mRcpflze0flCP\nCMwBIjXHGzRGQ6Q5XqvEmkjfr0uEru/rhJ7TO0mRbNHYRWq/z6aYMQqDJkYlkCWR7pJd7vk5\n+zPp9iU3HxGBQaTksShSfsuaE0VXPxFhQaTkMSlSnv/7Lt4le74+vEWEBJGSx6hIhiI0OEwk\nnfdR1PvQ7wGXXFASyYbRoxFpZqUj0g5CX8pUXvcl9a/I3cNVIuWI5KZ6wrO/vRBpB4iUPIgU\nAkRKHkQKwTEi5TkiBQORQpCUSEvevRLL+1v0aETSvxgHkSriF6m8BOY9JiINQaQQIFLyIFII\nECl5ECkEiJQ8iLSbJYewECkoB6Qj0m5MilTcSnN/93CNSEvG/FyR8rknj0hjIJKjByItaLhh\nbF8g0khgcYtIh0Qi0jYQydEDkRY03DC2LxBpJLC4PUyk4E94kH9k4nQ6Io2BSK4Oh24ZRDIY\nMQsiuTog0oKWGwb3BCKNBBa3iHRIIiJtA5FcHRBpQcsNg3vCvEjv+be7zpX9l2/WxcN1vtti\nYszy4gdEWtJyw+CeQKTZ5ohkKxGRxkCkTltE0mq5p4vBiFkQqdMWkbRa7uliMGIWROq0RSSt\nlnu6GIyYLWFOpOpNCLtCtjfXEkkWiVR/Eisi7W65p4vBiPkSpop4fzTpfpHWDeBBpHypSM3/\njwORDEbMl2BVJNXzSIiklIhIoyUgUqcpIim13NPFYMR8CYjUaYpISi33dDEYMV+Cq4jqMT2R\n1gyhKZKsEKn8KNbqEMtBHBGNSAolzIqkMJuPF6k8At4a39HWikjBsxFJoQREqtsikmLLPV0M\nRsyXgEh1W0RSbLmni8GI+RIQqW6LSIot93QxGDFfwueIlCPSaKZ+yz1dDEbMl2BVpOH9TdGI\ntChTv+WeLgYj5ktApLotIim23NPFYMR8CYhUt0UkxZZ7uhiMmC8Bkeq2iKTYck8XgxHzJbg2\nWyOStP7ZkbFqU/VE2pm9WKTqOyuOFmnJl3Oq5iGSTglTIuXpiJQvE6nx6SAk8EsSImmVgEid\nYESabL0hYH0XgxELSjAqUn0eCZF85yGSSgmI1AlGpMnWGwLWdzEYsaCCYRXSF2nvZF5jYl+k\n/RK3fmj/M9L0YJGKeR0wfOXbZBBpooJBGdJMOR2R1kyNSrr6n33ZslGkozYOIm0AkVytEQmR\n7EUsqQCRyocRaUHzDQneQSRXa0RCJHsRMwVIc9t+uGfQwSLtCXeK5B7xY0VaE4dIUwVMiJS7\nW6xNWXXUrvx3cGdT9hqROnlHiRQ4G5EUCzApUvMAInnNQyS1AhCpHYxIk803JHgHkcZraj2A\nSF7zEEmtAIdIkrJI7qcj/VaHgEgbsCtS/6NHECkUB4i0Jg6RpgpApHYwIk0235DgHUQar6n1\nACJ5zUMktQJMiuTxPBIitfMQaV9w9V+3jvITC96zV1REWnmuXlUkMS7SYEdfAosko6tjtP2G\nCN8YFilXFGndALoiuX+aFGmqjTZDkcJlb0pCpEEwIk22RSSV9tu6GIwYD0akybaIpNJ+WxeD\nEePBiDTZFpFU2m/rYjBiPLj1oRc9keqN2drKeyezfZEGj4XYOn2RpPev/wI8t9/WxWDEeDAi\nTTUdaaQNIqmASGPNc0QKAyLtDrYuUvMAInkswHP7bV0MRowHv7ahhkj9qeCekJLnsmyafKJI\ng8IQyVrEeHDr2pC2SNIWqdtjZCxdkRzLtq+msZ5WRKo+/hKR9oFI7tLmH1oIIs1X4Lf9ti4G\nI8aDEWnmMURSaL+ti8GI8WBEmnkMkRTab+tiMGI8GJFmHkMkhfbbuhiMcOdKd2Y3dRRbVppW\nrQVTg43/mG8WSePYewQivX+p9dcMIlmLcOci0oLHEEmh/bYuBiPcuYi04DFEUmi/rYvBCHcu\nIi14DJEU2m/rUvJ7PRff8X6+/PqK2MNbFnFsvkmRZOCIOJtJv81bpNa33k9+kNp6kZYMtuDx\n40RqvR8ekRqeX9Jw8hKxj1okx2Zr/3pEJO+8NwIiOblI9u9e3HvcMrn4iNgHIiFSpwKf7bd1\nKcjkXt+/S+YjYh+IhEidCny239bl3U/GflCL2AciIVKnAp/tt3UpsP6KJLMiNQ1bvVpHIbrL\nF4rUWrRMpJnHFi1FpCUV+Gy/rUvB3z7S7VHcs7mP5BCpe8e/SGvnfvoi5dI8ECh7W1BAkfJT\n66jd19NLxB4QCZG6Ffhsv61Lye+lOI+Una8WzyMhEiJ1K/DZflsXgxGOUERCpE4FPttv67Jo\n2DZ+IuYK0BSpuUyieKQvknRFkqqAqeocD848n/Flyx8fubRpIRs3ZbnSEGmCny+R881rxDY6\nvwH7hYxWJPVrS/PQEpHy9hxpRJrIWfpgp4w1o7kfP0ykHJFG+hUdyyMOkwftEMmV0xl+pqpe\nu4nRZh9HJC/tt3V593t1vMjlmeePi/z4iNgFIiFSrwKP7bd1efd7dcykOO79lC8fEbtAJETq\nVeCx/bYu736SNyt2egUfJ5L0H+vfcfRyifSeA/XBi90itR4PK5LjyA8iKbTf1uXd79Xxu1ox\nBi8Rchw22yZS/eKGSIik2uXdT87Xn5v8+7v7vFi8RAiREKlbgcf227q8+zXniEQyg5cIIRIi\ndSvw2H5blzf3+8/P+fzqnl0mPUIkR0xzF5ECYFkkUxGu0E0iSZ7n3X7voxblcO87/eOBkg9F\nal8sNAhZLZK4FJjtuEyk5ZunepKbpqVUBz1bkdJv5AtE2hGqL1IpU3tOVMNJ7+dy8oyUNlby\nKJIPnsymAV2jHChSf6V5A5F2hCISIm0dHJFaoYiESFsHR6RWKCIh0tbBEakVul0k6YtUHYIr\njTlAJJlavHhA5yirRVp97E7qzuMibTwguKYCj+23dTEY4QodilT/PC1S7hSpnAdSvjINxNkv\n0pQqpkRau0URSQdEclexfAEi7QSRdoTGIVK/33hdiLQdRNoRikiINMzx035bF4MRrlCXSHMT\nobm6pzdQIJGkvzjP62saN4gkvUYKIklHiaX98nYv6YhUPYRIFiJcoa5Ns/A3alek5rZ+bChS\nP0ZVJHfKeL5rgWPoue6OhuJJJM+zBJF2hCISInUq8Nh+WxeDEa5QREKkTgUe22/rYjDCFYpI\niNSpwGP7bV0MRrhCXbHlhlzT+WCRpP6NsF+kkXZLt4/kjUirjg0gkg6I5CiiPVQ3dkykZb6M\nLnAcxJzr7minIlKTWL7GIZKVCFdoHCINYxFJH0TaEYpIiNSpwGP7bV0MRrhCEQmROhV4bL+t\ni8GIYeY+kQYXsawVyaVfq/NUbMdh2SlSa7DRiwcWSFH/WlgvUkefgUgSRKTVl00gUp05JdKa\n3qMvKh5Eyh0itTqsF6mdtajRZEL1nCIUyXsHRJrrjUh1Z0RS7mIwYpiJSL0sRPLZAZHmevsW\nqXuFNiJ5AZE2R06ItKCcI0Xqzv3PEGnhZtkMIm2P9C5SPjFH6+7OGiZE6hYunRl8qEjNE1YX\nqfrZ4zRBpO2RiNQbHZF8dkAkd3dEcjRBJOUuBiOGkYjUGx2RfHb4OJFkUTmNSCMTxp9IMinS\neDVuZp/IzKL2ONISyb12JysYEUnqn3dPk/7zqDfS+pERqRXpjF0mUt6evDMtnI2mDrRNiJR3\nXgwVRJpWekH/VlU7RWqSpLsYkQxFuCIRKUckRNofGYdIjqWItA1E8hOJSDkiIdL+yENFGl2y\nTqRq7lb/jD2niSJlts1KkaRvx2zPXpInkfpvE6mPNC4/wtj03RC/vovBCFfkPpHmJ2BCIs2u\nktaLUDk/F4vUGX1oTDPT94vUGxmRdCIRaTjEaOYMiOSji8EIVyQiDYcYzZwBkXx0MRjhikSk\n4RCjmTMgko8uBiNcke5Jt3gKyJx1TcYakfpX10yeR5LO+GOfujApkoyVOFGDK8Il0oI1KYtE\nmi9xAWVUa+1VwyOSh8gwIo3XMSdSng9Fav3ofIkbr3GZSPOrpF11M+ASkfJwIuXdLSaVRGGs\nQKTxIRCpaYBI+l0MRiyPRKSVI5QNEEm/i8GI5ZGItHKEsgEi6XcxGLE8cfGe5+yFMM0cWSWS\n69jSSCc9kWaedDCRht3URJJhnYJI/hKXH8IZm7etlNmLW3RFGnk5DSJSq6r6dplInWb+RarH\nQSSviYi0coSqASKpdzEYsTwRkVaOUDVAJPUuBiOWJ1oTaaoTIq0AkcImaor0bjDVqr8ZF72V\np55XvVrLdwKNtB+vcabJghE6078l0qCa9rytg3si9Z9U/ciSLePcfFIP2/3yDkQykTgrUh5Y\npNbCuYf6SxVEcnRZKFKv22iYkkiNl4hkIhGR3ssQyVcXgxE+EhHpvQyRfHUxGOEjEZHeyxDJ\nVxeDET4SR2Zyu8WsSP2Fa0RafGRhXqRliWPLRo8VTorkfqeFb5HyXgGIdHxiGJFGzyMhkrMN\nIoUEkbpLEWkliKSYiEjlMkTy1MVghI/EI0V6z9IRkeqJUva2JpLU/RwiTaytPSLVJ56lWjHy\nPn+NSBYS50Qqp9mUSIOrfKTfwD2sY5a2Fpb3Aog0+pvCueAQkQZrSuoVM/ZbYDYmSBeDEZ4S\nEQmRPHYxGOEpEZEQyWMXgxGeEhEJkTx2MRjhKXFmnEqkqSabRBJEGmuzSqTR4mdjgnQxGOEp\nUVuk+auO3sPK+Atde6bmS45KHSdS7qgtuEhzf1VsL2V/F4MRnhIRCZE8djEY4SkRkRDJYxeD\nEZ4SEQmRPHYxGOEpcYFIM2rsEWlkwHIfupzJswMuE2lqdrsPe1QitSdrfVVOnh8qUpmCSPEk\nzm2mnkjLJsukIF2RFgw5+5qVTz2NaZFy6YnUFO4WabLeBSvH1X2YUbmMSNEkIhIibe5iMOKw\nxN0iuc8jIdJoE0QKCCJ1OiDSWhApdOLsLLYg0vzq8CWS+8lNZM0WikghiVuk8sjc2NDSiKYn\n0njaSI1ln0qkRh6vIjlfXREpgUREQqTNXQxGHJaISIi0uYvBiMMSEQmRNncp+b2ei88VOF9+\nfUVsxZBIrRbOyxX2iFQcdZidKLtEkrEaWyJJ+fES5U0AkaT3WL9VTCI9v6Th5CViO+ETx2j/\nCnXOoRGR5kZtiTRfwm6Rxvs0UrfmbdXpKJGqh6IQ6SLZv3tx73HL5OIjYjsRieTshEjjLVIT\nKZN7ff8umY+I7SBSt/F8IiKtrESjy7ufjP2gFrEdROo2nk9EpJWVaHQpsPyKZMejtkjzhwVa\nneYaSLOHv6DxgsQNIr2naUuk6tFqrjv3/0KKtGzQTZVodCn420e6PYp79vaRPkmkRY2XJI69\n7IxGdEWqYtoiuef4RDULnktyIuWn1lG7r6eXiK0gUq/xkkREWlOISpeS30txHik7X62dR0Kk\nXuMliYi0phCVLgYjDg4cZ1Yk937E7KiINNUpIZGkjZ+IifDQgeN4Eqm5nMCvSKIv0lQxi0Tq\n1upcq3GKtOBgbWAMiZQfL9LCA8BukSbqaUQqGw1FWpyzYFndQDrPaOr39PZ5gEjHBE6CSPM5\nC5bVDRITSbr4iNgMIs1GLAxFJJ9dCn4zRFoGIs3nLFhWN0hMpPx5llNxRpY/7aZpRHIutSyS\njC8aitRI9/Ei1TVlk5f8VPwT+Zcj0hxJi5SHFymXaER6LDxi/TjJ+YlIMyDSbM6SZU0D2yLd\nOrs8Xwt7XyW7IdI00yJN9RhvgEhNFcZEytvveP2auein4f41f8IVkd7/ItL4YPPLmgbGRcrX\nbOoW3+ZECn8pxSRVOeoiVdNqaQErQjtumBNJmiyZenaHieSJ4CIFzptjcjpuG7El0qLWS1oN\nf5juV8/mwfPbLNLsM1on0nYQ6Yi8ORBpNGfVwjweka71jpJWRYOIACDSoPWSVsMfEGljl+uy\nKxX2RIQAkQatl7Qa/oBIG7tk8qNWykhECGyKNFLVpnlgSqSq1VEizXTZxE6RPB3uQqTmdrhw\nU7WrRFrYTgb3Z/odKFJ1x6hIZ5n87IWtIFJzO1yISEsX5tGI9MhOi8/EbowIASI5C1jTCpF2\ndVn4tog9ESFAJGcBa1oh0q4uiOSFqESSlgZHiORcXA4osYjkCURqbocLN1Yrov1EV4vUFOIY\nyqtIOSKFAJG2jVjeIhIiHZM3ByINY0YXJiES+0heWDErVwyKSIjkmc8QSX3I6tiB1HIYE6l1\nQYNxkUp+T+f9pUxHeAWRNo9YT05pHpnrFkKk8uUoMpHyp3zvLmUmwieItHlERKrq0OnCn3aa\nINIwZnRhUiL9TH8Dn0aETxBp84iIVNWxq0tzrOGqVlKOSJGKtHCO+hWprkR6IkleXuFgWaQv\n3bclIVJzO1xoptq6zHUiuVpNiTQz1qhIvWuROq9R6mjtI+mCSM3tcKGZahGpHRuii8GIQ/Pm\nQKQVNSQi0r/XVyyf/ymV44zwj5mpWYJIK2pIQ6Tqm8pPWgUNIwJgZmpWTB0CQ6ROx/7qEIlR\npB/Jbn//3JQ/BAWR6hvHMjvV9iaocZHEsEhfci/+vS/+EP3VESGwMzVLEGlxCWmI1JSqWh8i\n1TeOZXaqRaQmdl+X5hWJKxs0QaTFJaQhUhr7SHZmZsWUSIYo9XlfMbD8zTTaIkn/gVIih0je\nfg1x1C583AKiEyl/T9TNJSuKlA9Fai8xKlL+7xz/eSR7MxaRFndMRSQvIFJ9YxpEag0eoovB\niAPjFoBIizsmItLz8jpcl110PwIckXKLVQ0pP1WktTO/daDNz7d3iKNU2rkKfa7VnSI9svKX\nUfbQqqgfEQB7UxaRVpSQgkgn+X69Fj0vovrpJ4iUj1dl6DwSInXG3tEljSsbDM3MEkRaUUIK\nImXl9yM9IxbJ28nuHUztFdsSSRCpHntHl4sU34/0e5KLVkX9CO8g0g5a9uwqa8dGGIo0HmJW\npBSubECkHSDSgtxFXYorG07KX8mMSIi0ooQkRPICIiHSihIQ6cCIdpilmflGhm8PaJZZKrdV\n5mEi5ZGLJF0OrmpPmKWZ+WbqjTOmRMotiNTui0g7q9oTZmpmFnh8B5oyiDSfu6TLuXhj32+m\n+mUUiIRIa7omINKlfqs555E0QaQ1XRMQKYFLhJT/KtVh8OYAs2iJpPUOjEhFyuL/8JNYZqxR\nTKy8hSIFKmFLl4tkr0uEblm8X+uCSLswsfLiF6m+RCje75BFpF2YWHkJiFR++MlNqRxnhF+i\nE8nWPp2JYlIQyQuINAEiDUCk4yJaWSbmwnIQaUBCIv1G+1ZzRNqFiWISEOkS/SVCiLQLE8XE\nL1LjkerhBkSaAJEGxC9SJv/ykzweJ/lVKykPujbEyFxYDiINEMe9w0rY1OW1Ua9/r0Z33fea\nhxUpMmyJZII0RLq9vtIl2n2k+GYlIg2IX6Tz3592D/nKfxEJjiN+kW4vgYrLhFTfkIRIsIb4\nRfrbQfq7+RbdtyMhEqwiAZH8gEiwBkQ6LuKAKPBFCiLdzq/dpLPqt7ogEqwiAZFO76uD4v1+\nJERKgPhF+pFT8UUUPxy1CwbnkQa8V4nk8Yr0+lqX6gtEFUGkCRBpSPQi1V95i0jBQKQh0Yv0\nVb4i3eVLraQckSZBpCHRi1TuI90yUf1iF0SaAJGGRC9Sfo79i8bim5WINCR+kYrzSHL+p1SO\nM8Ir8c1KRBryFkliFskLiDQBIg2JXqSz7sWqrgjPxDcrEWlI9CJ52qiIBKuIXqTX4e/l/F7f\nxybOl5lPeAi3Ovj1ngRS/i9akZ7n0+JPPXl+tb7fb/ooX0CRgiWBR6IXac1XX14k+/f+EpjH\nLZt+JyAiwSo+SqTqu5RezHyfEiLBKqIXaVW/5UcpEAlW8VEiWXxFwqM0eH9V6JEfmhtQpL99\npNv77X929pFiFIkDjQ4kl48Rqf52vxdfk4fNEWkCRHLwUSLlv5fiPFJ2vlo5jxTjnEQkB58l\nkqWIwEGKIJIDRGqGlTXH0usy9lUT45xEJAefKNJPJl8zbwNEpAkQycFHiXQ/S/aTX5UuEUIk\naPgkke6FQRf5fuaP8/Rb0xFpAkRy8Ekifb/OHV3eZ2Kf0x+WgkgTIJKDTxKp/BS/c+uHPREf\nKxI4eIt04O+Y4CL9e/9Nt/8SIUSChvdldp8h0vdr7+jN83v/JUKIBA2fJNIzq5+nTL8gIRKs\n5JNEyvNLpU8285kpiATr+CyR9CKkWmf7Vh0epQIibYuoD3XuO+KJSKAEIkUH55EsgkjRgUgW\nQaToQCSLxCmSIBLYApGiA5EsgkjRgUgWQaToQCSLIFJ0IJJFYhRJyo8DlLx1iYOHHKMgkkWi\nFCnP6+/CkV31MCVBCUQCUACRABRAJAAFPlokPAItEAlAAUQCUACRooPzSBZBpOhAJItEJpJU\nN1LeIBKYIFaR3j8hEhgBkaIDkSyCSNGBSBZBpOhAJItEKFKzUESmGm+PMQ0iWQSRogORLIJI\nAAogEoAC0YkkiAQGiUyk8kBdp+HmevAI1EAkAAUQCUABRAJQAJGig/NIFklApM0FRTohEcki\niBQdiGQRRIoORLIIIkUHIlkEkaIDkSwSm0i543tcEAkOB5GiA5EsgkjRgUgWQSQABeITabhs\nY0F4BHpEJ5JjGSLB4SASgAKIBKAAIgEo8LEiOY7+AWwGkaKD80gWQaToQCSLIFJ0IJJFECk6\nEMkinytStNMx3spTJgWRNlUU73SMt/KUQaToiLfylEGk6Ii38pRBpOiIt/KU+VSRhOkImsQn\n0v7mRRdEAk0QCUABRAJQAJEAFPhckdb3ARgFkQAUQKToiLj0hEGk6Ii49IRBpOiIuPSESUOk\n1SXFfNAu4tITJgmR1pcU81xEJIsgUnQgkkUQKToQySKIFB2IZBFEig5EsggiASiASAAKIBKA\nAogEoEBQkX6vZ3lxvvzqRqxrL4gE2gQU6fklDSfVCESCgwko0kWyf/fi3uOWyUUzApHgYAKK\nlMm9vn+XTDPio0TiPJJFAorUmQDTswGRJkAki/CKFB2IZJGw+0i3R3GPfaQ9IJJFQh7+PrWO\n2n09NSMQCQ4m7HmkS3EeKTtfOY+0HUSyyCde2YBIoI4dkaTN2r7rchAJtAkp0vNb5HQrBzns\n8HfhacxzEZEsEvISoex9od17EESClAh6+Pvnz6afrLjMDpEgKYKekC3+eWRfD0SCxDjgEqHn\n6XSgSJJvOJgBMENAkb6kOgn7dVIWaUUHyaP+mFUwSkCRfuS7vPeQEyJBUoQ8/H2pJ/BtZi4j\nEkRG0BOy93N17/GNSFuJu/pUsXNlw66IhQcQ3s0QCdRJRqRFfZKYgohkEUSKDkSyCCJFByJZ\nBJGiA5Es8lEipTEDEckiiYi07M16acxARLIIIkUHIlkEkQAUQCQABRAJQIFURFrUCZHAF4gE\noAAiASiASAAKJCTSfK80ROI8kkUQKToQySKIFB2IZBFEig5EssiniBT/O8xrEnkaiYFI0ZHI\n00gMRIqORJ5GYiBSdCTyNBIDkaIjkaeRGIgEoEBKIk11K95mgUjgC0QCUACRABRAJAAFPkSk\n1+feIRL441NE4qAxeAWRoiOdZ5ISiBQd6TyTlECk6EjnmaTE54i0sRR7IJJFECk6EMkiiBQd\niGQRRIoORLIIIkUHIlnkI0QSCVJ1IBDJIh8jEoBPEAlAAUQCUACRABT4BJFSOmQHRklKpI0F\no8AAAAerSURBVJF+iATeQSQABRApOtjjswgiRQciWSR9kSS1D2tAJIt8gEh7CrEIIlkEkaID\nkSyCSNGBSBZBpOhAJIskL1J6sw6RLJKWSI6O6c06RLIIIgEogEgACiASgALpiSTd/ogEIUAk\nAAUQCUABRAJQAJGig/NIFklSpNZUS3DSIZJF0hSpGSDBSYdIFkGk6EAkiyBSdCCSRRApOhDJ\nIsmJJNL+kIYUJ12Kzyl+EhOp8Kg1QIpzDpEsgkjRgUgWQSQABRAJQIEERWoNgEcQCEQCUACR\nABRAJAAFEAlAgaAi/V7PxYUH58uvr4gPEInzSBYJKNLzSxpOXiL6AyQ55RDJIgFFukj2717c\ne9wyufiI6A+Q5JRDJIsEFCmTe33/LpmPiP4ASU45RLJIQJE6E2B6NiDSBIhkEV6RogORLBJ2\nH+n2KO7530cqR0hyyiGSRUIe/j61jtp9Pb1E9EZIcsohkkXCnke6FOeRsvPV23mk7ghJTjlE\nskhqVzZ0R2DKQSDsiCRt9o9W3wAEwI5IuhGIBEFBJAAFEAlAgaBXNizeDUIkiIyAIv0gEiRL\nyD/t7tn0mycUIrojpCkS55EsEnQf6T59YZBGRGeENGccIlkk7MGGn9Z1q54i2iOkOeMQySIc\ntYsORLJIqiK9hkh0xiX6tCInaZH2j2IRRLIIIkUHIlkEkaIDkSyCSNGBSBZBJAAFEAlAAUQC\nUACRABRAJAAFEAlAAUQCUACRooPzSBZBpOhAJIsgUnQgkkUQKToQySKIFB2IZBFEig5Esggi\nRQciWSRhkVKdcKk+r7hJWaT9g5gEkSyCSAAKIBKAAogEoEC6IuERBCRZkXhBgpAgEoACiASg\nACJFB3t/FkGk6EAki6QrUrIgkkUQKToQySKIFB2IZBFEig5EsggiRQciWQSRogORLIJI0YFI\nFkEkAAUQCUABRAJQAJEAFEAkAAUQCUABRAJQAJGig/NIFkGk6EAkiyBSdCCSRRApOhDJIogU\nHYhkEUSKDkSyCCJFByJZBJGiA5EsYlQkgMjYMMv1xYkiewjVjGOrGmvlFCBSBdWMY6saa+UU\nIFIF1Yxjqxpr5RQgUgXVjGOrGmvlFCBSBdWMY6saa+UUIFIF1Yxjqxpr5RQgUgXVjGOrGmvl\nFCBSBdWMY6saa+UUIFIF1Yxjqxpr5RQgUgXVjGOrGmvlFCBSBdWMY6saa+UUIFIF1Yxjqxpr\n5RRYrAkgOhAJQAFEAlAAkQAUQCQABRAJQAFEAlAAkQAUQCQABRAJQAFEAlAAkQAUQCQABRAJ\nQAFEAlAAkQAUOEykSybZ5XlUesHPV11Cq5oDC/stt4aBau7fIt8PK9U83TUYmEM1R4l0Kj70\n/+ug9IJLUUL27FZzYGHP7L01DFRzM7VuHtm7nIeNcpwcJNKvZPf8nsnvMfEv7vL9N09+5LtT\nzZGFnd9fJ2Khmuwv93mWi41qvl+F/P3is7Ophhwk0kVuf7f/5HpM/Ivz+6m/Jm+rmgML+1d+\nL4+Bav4VM/cpmYlqqm9js7OpHBwk0lleL9N3OR8T3+K1dVrVHFfYQ07vCWOgmm+5V3cNVJOX\nf/K+vLZQjpODRGr9jjmWp5w61RxX2Eke71QD1XxJfs2KP30tVJNfyz/trjbKcfLpIv28/j6w\nsHWu8i83I5LIudi7t1HN31Z6HW3IfqyU4+LDRXpk59zE1in+QjEk0utgw7eZl4BrcXzumhsp\nx8Vni/TMTq0yDv1j6nWo2ZBIr32kx+vQsoFq/v5s+PvT7s/rHxvlODmoiszGSji9T0K0qjmo\nsO/iCNQ79fhq2nPUQDV/u2yvvbXny2sL5Tg59Kjd49gjLo+v06NfzUGFtb+Y/vhq2qcGDFTT\n9tpCOU4OEula/Aa+FUdjjuImp2E1BxXWFun4asrcx2sFGaimfO0pTmtZKMfJ517Z8Kg9MnO6\n3MyVDX97R8/XTsk/E9XkF3ldUXexcqGFk6P+wPwqfv2e5ht647t5DWhXc2Rh5Z8wBqq5Oks4\nbN2cbJXj4iiR3tfzHhRe0Ppjql3NkYWVIlmo5nZylHDcunHWcPwcamHjkAdA5CASgAKIBKAA\nIgEogEgACiASgAKIBKAAIgEogEgACiASgAKIBKAAIgEogEgACiASgAKIBKAAIgEogEgACiAS\ngAKIBKAAIgEogEgACiASgAKIBKAAIgEogEgACiASgAKIBKAAIgEogEgACiASgAKIBKAAIgEo\ngEgACiBSbAibzCJsldhAJJOwVWIDkUzCVokNRDIJWyU2/kS6lF/m3boLB4NIsSFylj9Oxd1r\ndRcOBpFiQyS75/dM/nXuwsEgUmyI3P5ub3Lu3IWDQaTYKA82vP5p3YWDYRvEBiKZhG0QG4hk\nErZBbIj85vU+0vvu99E1ASJFR3Wo7ta5CweDSLEh8v06eXQu7p6qu3AwiBQb78sZruXds3z9\nHF0R5IgUNRxlsAObImIQyQ5siohBJDuwKSIGkezApgBQAJEAFEAkAAUQCUABRAJQAJEAFEAk\nAAUQCUABRAJQAJEAFEAkAAUQCUABRAJQAJEAFEAkAAUQCUABRAJQAJEAFEAkAAUQCUABRAJQ\nAJEAFEAkAAUQCUABRAJQAJEAFEAkAAX+AxRKp3ICc19aAAAAAElFTkSuQmCC",
      "text/plain": [
       "plot without title"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# plot the result of the base coverage `res` for the second intervel (i.e. seq2) range in IRnage object\n",
    "plot(res[[2]], type='l', xlab='bp', ylab = 'readcount')\n",
    "#identify the maximum coverage and use a dotted line to highlight\n",
    "abline(v=start(summit[2])-start(gr[2])+1, lty=2, lwd=2)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Annoatation of genomic region"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## - find the position of the transcription start sites for all transcripts in the UCSC annotation database"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "'select()' returned 1:1 mapping between keys and columns\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "GRanges object with 55419 ranges and 2 metadata columns:\n",
       "                seqnames             ranges strand |     tx_id     tx_name\n",
       "                   <Rle>          <IRanges>  <Rle> | <integer> <character>\n",
       "      18777         chr1 [4797974, 4797974]      + |         1  uc007afg.1\n",
       "      18777         chr1 [4797974, 4797974]      + |         2  uc007afh.1\n",
       "      21399         chr1 [4847775, 4847775]      + |         3  uc007afi.2\n",
       "      21399         chr1 [4847775, 4847775]      + |         4  uc011wht.1\n",
       "      21399         chr1 [4848409, 4848409]      + |         5  uc011whu.1\n",
       "        ...          ...                ...    ... .       ...         ...\n",
       "  100041253 chrUn_random [2216886, 2216886]      - |     55415  uc009sjw.2\n",
       "      20770 chrUn_random [2678407, 2678407]      - |     55416  uc009skb.2\n",
       "     333467 chrUn_random [2891056, 2891056]      - |     55417  uc009ske.1\n",
       "         NA chrUn_random [3837247, 3837247]      - |     55418  uc009skg.1\n",
       "         NA chrUn_random [4677187, 4677187]      - |     55419  uc009skh.1\n",
       "  -------\n",
       "  seqinfo: 35 sequences (1 circular) from mm9 genome"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# use the TSS() method on mouse annotatation data from UCSC to find transcription start sites(TSS).\n",
    "# this method returns a GRanges object of width 1 with (TSS) for all transcripts\n",
    "\n",
    "TSSpos = TSS(txdb)\n",
    "TSSpos"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## - select ranges and adjust the `start` and `end`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "GRanges object with 5 ranges and 2 metadata columns:\n",
       "        seqnames             ranges strand |     tx_id     tx_name\n",
       "           <Rle>          <IRanges>  <Rle> | <integer> <character>\n",
       "  18777     chr1 [4797974, 4797974]      + |         1  uc007afg.1\n",
       "  18777     chr1 [4797974, 4797974]      + |         2  uc007afh.1\n",
       "  21399     chr1 [4847775, 4847775]      + |         3  uc007afi.2\n",
       "  21399     chr1 [4847775, 4847775]      + |         4  uc011wht.1\n",
       "  21399     chr1 [4848409, 4848409]      + |         5  uc011whu.1\n",
       "  -------\n",
       "  seqinfo: 35 sequences (1 circular) from mm9 genome"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# select the first five ranges\n",
    "gr = TSSpos[1:5]\n",
    "gr"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "GRanges object with 5 ranges and 0 metadata columns:\n",
       "        seqnames             ranges strand\n",
       "           <Rle>          <IRanges>  <Rle>\n",
       "  18777     chr1 [4796974, 4797374]      +\n",
       "  18777     chr1 [4796974, 4797374]      +\n",
       "  21399     chr1 [4846775, 4847175]      +\n",
       "  21399     chr1 [4846775, 4847175]      +\n",
       "  21399     chr1 [4847409, 4847809]      +\n",
       "  -------\n",
       "  seqinfo: 35 sequences (1 circular) from mm9 genome"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# adjust the start and end of the selected ranges\n",
    "start(gr) <- start(gr) - 1000\n",
    "end(gr) <- end(gr) - 600\n",
    "\n",
    "# remove metadate by using the `mcols()` function and assigning NULL as a value\n",
    "mcols(gr) = NULL\n",
    "gr"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## - load CpG islands annotation file from `compEpiTools` package"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# read in the mouse CpG annotation file (CGIgr_mm9.Rdata) \n",
    "cgipath = system.file('extdata', 'CGIgr_mm9.Rdata', package = 'compEpiTools')\n",
    "load(cgipath)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "## - Use `GRannotate()` method to annotate genomic intervals in `GRanges` object"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "'select()' returned 1:1 mapping between keys and columns\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "GRanges object with 5 ranges and 9 metadata columns:\n",
       "        seqnames             ranges strand | nearest_tx_name distance_fromTSS\n",
       "           <Rle>          <IRanges>  <Rle> |     <character>        <integer>\n",
       "  18777     chr1 [4797174, 4797174]      + |      uc007afg.1              799\n",
       "  18777     chr1 [4797174, 4797174]      + |      uc007afg.1              799\n",
       "  21399     chr1 [4846975, 4846975]      + |      uc007afi.2              799\n",
       "  21399     chr1 [4846975, 4846975]      + |      uc007afi.2              799\n",
       "  21399     chr1 [4847609, 4847609]      + |      uc007afi.2              165\n",
       "        nearest_gene_id nearest_gene_symbol                   location\n",
       "            <character>         <character>                <character>\n",
       "  18777           18777              Lypla1          promoter;promoter\n",
       "  18777           18777              Lypla1          promoter;promoter\n",
       "  21399           21399               Tcea1 promoter;promoter;promoter\n",
       "  21399           21399               Tcea1 promoter;promoter;promoter\n",
       "  21399           21399               Tcea1 promoter;promoter;promoter\n",
       "                          location_tx_id  location_gene_id location_gene_symbol\n",
       "                             <character>       <character>          <character>\n",
       "  18777            uc007afg.1;uc007afh.1       18777;18777        Lypla1;Lypla1\n",
       "  18777            uc007afg.1;uc007afh.1       18777;18777        Lypla1;Lypla1\n",
       "  21399 uc007afi.2;uc011wht.1;uc011whu.1 21399;21399;21399    Tcea1;Tcea1;Tcea1\n",
       "  21399 uc007afi.2;uc011wht.1;uc011whu.1 21399;21399;21399    Tcea1;Tcea1;Tcea1\n",
       "  21399 uc007afi.2;uc011wht.1;uc011whu.1 21399;21399;21399    Tcea1;Tcea1;Tcea1\n",
       "              CGI\n",
       "        <numeric>\n",
       "  18777         0\n",
       "  18777         0\n",
       "  21399         0\n",
       "  21399         0\n",
       "  21399         1\n",
       "  -------\n",
       "  seqinfo: 35 sequences (1 circular) from mm9 genome"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "\n",
    "res = GRannotate(Object = GRmidpoint(gr), # points to the peaks midpoint; other methods could be used (GRcoverageSummit()) \n",
    "                 txdb = txdb, # transcript database\n",
    "                 EG2GS = org.Mm.eg.db, # R object that contain mapping btw Entrez geneIDs and GeneBank access numbers\n",
    "                 upstream = 2000, # bp upstream TSS to define promoters\n",
    "                 downstream = 1000, # bp downstream TSS to define promoters\n",
    "                 userAnn = GRangesList(CGI=CGIgr_mm9)) # user defined annotations as GRanges object [Go to the documentation for details]\n",
    "\n",
    "res"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "H3K4me1GR = system.file(\"extdata\", \"H3K4me1GR.Rda\", package=\"compEpiTools\")\n",
    "load(H3K4me1GR)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "GRanges object with 22 ranges and 1 metadata column:\n",
       "       seqnames             ranges strand |    pvalue\n",
       "          <Rle>          <IRanges>  <Rle> | <numeric>\n",
       "   [1]    chr19 [3311040, 3311889]      * |    118.45\n",
       "   [2]    chr19 [3344836, 3346936]      * |    165.14\n",
       "   [3]    chr19 [3359293, 3361264]      * |    282.57\n",
       "   [4]    chr19 [3422317, 3423989]      * |     91.13\n",
       "   [5]    chr19 [3429800, 3432136]      * |       242\n",
       "   ...      ...                ...    ... .       ...\n",
       "  [18]    chr19 [3751893, 3754604]      * |    967.18\n",
       "  [19]    chr19 [3768768, 3771999]      * |    698.17\n",
       "  [20]    chr19 [3772830, 3781909]      * |    810.31\n",
       "  [21]    chr19 [3893099, 3895850]      * |    120.28\n",
       "  [22]    chr19 [3950991, 3952444]      * |     95.12\n",
       "  -------\n",
       "  seqinfo: 26 sequences from an unspecified genome; no seqlengths"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "enhancers(H3K4me1GR, txdb)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "# Example: annotating Mum and Hs genomes: "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## - loading data after downloading from published work (https://www.iit.it/sabonature2014)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "load('/Users/mustafa.albahrani/workspace/clustering_tool/saboEtAl2014_sourceCode/data/peaksRef.rda')\n",
    "load('/Users/mustafa.albahrani/workspace/clustering_tool/saboEtAl2014_sourceCode/data/mycSummitsRef.rda')\n",
    "load('/Users/mustafa.albahrani/workspace/clustering_tool/saboEtAl2014_sourceCode/data/CGIgr.rda')\n",
    "load('/Users/mustafa.albahrani/workspace/clustering_tool/saboEtAl2014_sourceCode/data/deseqRes.rda')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "$EuMyc\n",
       "$EuMyc$Myc.C1\n",
       "GRanges object with 6978 ranges and 2 metadata columns:\n",
       "            seqnames                 ranges strand |    pvalue enrichment\n",
       "               <Rle>              <IRanges>  <Rle> | <numeric>  <numeric>\n",
       "     [1]        chr1     [4775507, 4775623]      * |     82.65       4.72\n",
       "     [2]        chr1     [4775627, 4775942]      * |    330.87       6.63\n",
       "     [3]        chr1     [4797563, 4797990]      * |    176.67       6.25\n",
       "     [4]        chr1     [5073141, 5073326]      * |     91.73       5.27\n",
       "     [5]        chr1     [6204551, 6204731]      * |    114.56       5.59\n",
       "     ...         ...                    ...    ... .       ...        ...\n",
       "  [6974]        chrX [166434796, 166435939]      * |    704.97       9.73\n",
       "  [6975]        chrX [166437187, 166437509]      * |    174.08       7.58\n",
       "  [6976]        chrX [166504872, 166504960]      * |     80.75       5.35\n",
       "  [6977] chrX_random [   105955,    106085]      * |    150.52       7.42\n",
       "  [6978]        chrY [   427373,    427440]      * |     103.3       5.89\n",
       "  -------\n",
       "  seqinfo: 26 sequences from an unspecified genome; no seqlengths\n",
       "\n",
       "$EuMyc$Myc.P\n",
       "GRanges object with 31187 ranges and 2 metadata columns:\n",
       "          seqnames                 ranges strand |    pvalue enrichment\n",
       "             <Rle>              <IRanges>  <Rle> | <numeric>  <numeric>\n",
       "      [1]     chr1     [3754988, 3755064]      * |    167.78       6.25\n",
       "      [2]     chr1     [4313180, 4313292]      * |    102.36       5.96\n",
       "      [3]     chr1     [4775474, 4775912]      * |    304.92       7.19\n",
       "      [4]     chr1     [4797544, 4798131]      * |    486.35        7.7\n",
       "      [5]     chr1     [4831502, 4831573]      * |    109.67       4.78\n",
       "      ...      ...                    ...    ... .       ...        ...\n",
       "  [31183]     chrX [163766137, 163767108]      * |     90.93       6.19\n",
       "  [31184]     chrX [163819185, 163821457]      * |    270.61        7.8\n",
       "  [31185]     chrX [165110941, 165112485]      * |      95.9       6.16\n",
       "  [31186]     chrX [165757867, 165758893]      * |    189.52       6.55\n",
       "  [31187]     chrX [166441254, 166447026]      * |     80.98       7.74\n",
       "  -------\n",
       "  seqinfo: 27 sequences from an unspecified genome; no seqlengths\n",
       "\n",
       "$EuMyc$Myc.T27\n",
       "GRanges object with 29252 ranges and 2 metadata columns:\n",
       "             seqnames             ranges strand |    pvalue enrichment\n",
       "                <Rle>          <IRanges>  <Rle> | <numeric>  <numeric>\n",
       "      [1]        chr1 [4661358, 4662195]      * |    134.17        5.8\n",
       "      [2]        chr1 [4774952, 4776476]      * |   2141.12          9\n",
       "      [3]        chr1 [4797355, 4798676]      * |    1002.3       7.84\n",
       "      [4]        chr1 [4847305, 4848865]      * |   1075.53       8.18\n",
       "      [5]        chr1 [4908660, 4909197]      * |     85.69       5.03\n",
       "      ...         ...                ...    ... .       ...        ...\n",
       "  [29248] chrX_random [  38775,   39669]      * |    150.01       5.68\n",
       "  [29249]        chrY [ 346594,  347302]      * |    187.19       5.81\n",
       "  [29250]        chrY [ 581237,  582424]      * |    270.15       6.65\n",
       "  [29251]        chrY [ 622117,  623326]      * |    417.19       6.86\n",
       "  [29252]        chrY [2612163, 2612854]      * |     89.81       5.59\n",
       "  -------\n",
       "  seqinfo: 28 sequences from an unspecified genome; no seqlengths\n",
       "\n",
       "$EuMyc$Myc.T29\n",
       "GRanges object with 27347 ranges and 2 metadata columns:\n",
       "             seqnames                 ranges strand |    pvalue enrichment\n",
       "                <Rle>              <IRanges>  <Rle> | <numeric>  <numeric>\n",
       "      [1]        chr1     [4774916, 4776578]      * |   3137.54       9.63\n",
       "      [2]        chr1     [4797165, 4798451]      * |    980.87       8.09\n",
       "      [3]        chr1     [4847108, 4848828]      * |   1368.62       8.71\n",
       "      [4]        chr1     [5072839, 5073820]      * |    772.36       8.16\n",
       "      [5]        chr1     [6203923, 6205775]      * |    791.23       7.91\n",
       "      ...         ...                    ...    ... .       ...        ...\n",
       "  [27343]        chrX [166318541, 166319831]      * |    106.43       6.84\n",
       "  [27344]        chrX [166423380, 166425928]      * |    460.36        9.4\n",
       "  [27345]        chrX [166434694, 166438089]      * |       508       9.83\n",
       "  [27346]        chrX [166444506, 166446992]      * |     90.83       7.09\n",
       "  [27347] chrX_random [    38631,     39677]      * |    186.03       6.51\n",
       "  -------\n",
       "  seqinfo: 27 sequences from an unspecified genome; no seqlengths\n",
       "\n",
       "$EuMyc$Myc.T31\n",
       "GRanges object with 34142 ranges and 2 metadata columns:\n",
       "             seqnames                 ranges strand |    pvalue enrichment\n",
       "                <Rle>              <IRanges>  <Rle> | <numeric>  <numeric>\n",
       "      [1]        chr1     [4774803, 4776536]      * |   1537.76       8.89\n",
       "      [2]        chr1     [4797227, 4798981]      * |    648.33       7.84\n",
       "      [3]        chr1     [4847324, 4849000]      * |    587.15       7.79\n",
       "      [4]        chr1     [5072607, 5073898]      * |    387.55       7.42\n",
       "      [5]        chr1     [6204015, 6205790]      * |    370.23       7.19\n",
       "      ...         ...                    ...    ... .       ...        ...\n",
       "  [34138]        chrX [165757701, 165758907]      * |    737.73       7.51\n",
       "  [34139]        chrX [166312250, 166312848]      * |     82.58       5.03\n",
       "  [34140]        chrX [166318438, 166319612]      * |    118.69       5.96\n",
       "  [34141]        chrX [166434687, 166436418]      * |    464.28       9.06\n",
       "  [34142] chrX_random [    38706,     39700]      * |    232.24       6.31\n",
       "  -------\n",
       "  seqinfo: 27 sequences from an unspecified genome; no seqlengths\n",
       "\n",
       "$EuMyc$Pol2.C1\n",
       "GRanges object with 15571 ranges and 1 metadata column:\n",
       "          seqnames                 ranges strand |    pvalue\n",
       "             <Rle>              <IRanges>  <Rle> | <numeric>\n",
       "      [1]     chr1     [4775497, 4776192]      * |    406.08\n",
       "      [2]     chr1     [4847209, 4848281]      * |    500.42\n",
       "      [3]     chr1     [5072892, 5073456]      * |    264.62\n",
       "      [4]     chr1     [6204045, 6205172]      * |   1463.26\n",
       "      [5]     chr1     [7078221, 7079682]      * |   1266.35\n",
       "      ...      ...                    ...    ... .       ...\n",
       "  [15567]     chrX [165757883, 165758573]      * |    136.71\n",
       "  [15568]     chrY [   233835,    234303]      * |     92.57\n",
       "  [15569]     chrY [   346602,    347346]      * |    266.73\n",
       "  [15570]     chrY [   581787,    582509]      * |    222.13\n",
       "  [15571]     chrY [   622462,    623374]      * |    604.29\n",
       "  -------\n",
       "  seqinfo: 21 sequences from an unspecified genome; no seqlengths\n",
       "\n",
       "$EuMyc$Pol2.P1\n",
       "GRanges object with 17173 ranges and 1 metadata column:\n",
       "          seqnames             ranges strand |    pvalue\n",
       "             <Rle>          <IRanges>  <Rle> | <numeric>\n",
       "      [1]     chr1 [4775352, 4776138]      * |    419.32\n",
       "      [2]     chr1 [4797481, 4798204]      * |    442.47\n",
       "      [3]     chr1 [4847372, 4848129]      * |     604.9\n",
       "      [4]     chr1 [5072968, 5073405]      * |    178.37\n",
       "      [5]     chr1 [6204145, 6205217]      * |   1399.08\n",
       "      ...      ...                ...    ... .       ...\n",
       "  [17169]     chrY [ 346840,  347248]      * |    183.68\n",
       "  [17170]     chrY [ 581888,  582392]      * |    517.99\n",
       "  [17171]     chrY [ 622584,  623253]      * |    510.04\n",
       "  [17172]     chrY [2894148, 2895521]      * |     88.61\n",
       "  [17173]     chrY [2897395, 2898088]      * |    104.34\n",
       "  -------\n",
       "  seqinfo: 21 sequences from an unspecified genome; no seqlengths\n",
       "\n",
       "$EuMyc$Pol2.T27\n",
       "GRanges object with 27719 ranges and 1 metadata column:\n",
       "          seqnames                 ranges strand |    pvalue\n",
       "             <Rle>              <IRanges>  <Rle> | <numeric>\n",
       "      [1]     chr1     [4766214, 4766513]      * |     82.94\n",
       "      [2]     chr1     [4775278, 4776274]      * |    791.81\n",
       "      [3]     chr1     [4797469, 4798287]      * |    866.19\n",
       "      [4]     chr1     [4847323, 4848175]      * |   1515.47\n",
       "      [5]     chr1     [4848308, 4848575]      * |     80.93\n",
       "      ...      ...                    ...    ... .       ...\n",
       "  [27715]     chrX [166436853, 166437434]      * |    166.09\n",
       "  [27716]     chrX [166444645, 166445121]      * |     210.3\n",
       "  [27717]     chrY [   346850,    347131]      * |     92.58\n",
       "  [27718]     chrY [   581930,    582358]      * |    180.33\n",
       "  [27719]     chrY [   622836,    623316]      * |    217.43\n",
       "  -------\n",
       "  seqinfo: 22 sequences from an unspecified genome; no seqlengths\n",
       "\n",
       "$EuMyc$Pol2.T29\n",
       "GRanges object with 31442 ranges and 1 metadata column:\n",
       "          seqnames                 ranges strand |    pvalue\n",
       "             <Rle>              <IRanges>  <Rle> | <numeric>\n",
       "      [1]     chr1     [4755908, 4756700]      * |    104.57\n",
       "      [2]     chr1     [4758260, 4761371]      * |    760.62\n",
       "      [3]     chr1     [4761651, 4762505]      * |    101.07\n",
       "      [4]     chr1     [4762676, 4766194]      * |    603.55\n",
       "      [5]     chr1     [4767304, 4769374]      * |    379.47\n",
       "      ...      ...                    ...    ... .       ...\n",
       "  [31438]     chrX [165087217, 165088024]      * |    125.43\n",
       "  [31439]     chrX [165109815, 165112534]      * |   2005.45\n",
       "  [31440]     chrX [165757638, 165758904]      * |    725.49\n",
       "  [31441]     chrX [166318041, 166319523]      * |    836.55\n",
       "  [31442]     chrX [166434673, 166436417]      * |    376.46\n",
       "  -------\n",
       "  seqinfo: 21 sequences from an unspecified genome; no seqlengths\n",
       "\n",
       "$EuMyc$Pol2.T31\n",
       "GRanges object with 28315 ranges and 1 metadata column:\n",
       "          seqnames                 ranges strand |    pvalue\n",
       "             <Rle>              <IRanges>  <Rle> | <numeric>\n",
       "      [1]     chr1     [4763808, 4766770]      * |    574.55\n",
       "      [2]     chr1     [4767173, 4771000]      * |    649.94\n",
       "      [3]     chr1     [4771023, 4771818]      * |     83.13\n",
       "      [4]     chr1     [4775084, 4776472]      * |   1455.26\n",
       "      [5]     chr1     [4797230, 4798676]      * |   1902.13\n",
       "      ...      ...                    ...    ... .       ...\n",
       "  [28311]     chrX [165089491, 165090888]      * |    202.76\n",
       "  [28312]     chrX [165109911, 165112419]      * |   1976.73\n",
       "  [28313]     chrX [165757691, 165759013]      * |    677.38\n",
       "  [28314]     chrX [166317201, 166319441]      * |   1014.54\n",
       "  [28315]     chrX [166434657, 166437960]      * |   1503.03\n",
       "  -------\n",
       "  seqinfo: 20 sequences from an unspecified genome; no seqlengths\n",
       "\n",
       "$EuMyc$K4me1.C\n",
       "GRanges object with 51043 ranges and 1 metadata column:\n",
       "             seqnames                 ranges strand |    pvalue\n",
       "                <Rle>              <IRanges>  <Rle> | <numeric>\n",
       "      [1]        chr1     [4486033, 4488141]      * |    216.57\n",
       "      [2]        chr1     [4561287, 4562406]      * |     97.17\n",
       "      [3]        chr1     [4612632, 4614380]      * |    125.98\n",
       "      [4]        chr1     [4660395, 4662880]      * |    618.46\n",
       "      [5]        chr1     [4773686, 4777577]      * |    801.45\n",
       "      ...         ...                    ...    ... .       ...\n",
       "  [51039]        chrX [166334681, 166335847]      * |     92.02\n",
       "  [51040]        chrX [166409351, 166421854]      * |      3100\n",
       "  [51041] chrX_random [    36448,     39885]      * |    904.02\n",
       "  [51042] chrX_random [   159028,    162659]      * |    365.82\n",
       "  [51043] chrX_random [   163441,    165685]      * |    120.67\n",
       "  -------\n",
       "  seqinfo: 26 sequences from an unspecified genome; no seqlengths\n",
       "\n",
       "$EuMyc$K4me1.P\n",
       "GRanges object with 54912 ranges and 1 metadata column:\n",
       "             seqnames                 ranges strand |    pvalue\n",
       "                <Rle>              <IRanges>  <Rle> | <numeric>\n",
       "      [1]        chr1     [4486087, 4488167]      * |    221.61\n",
       "      [2]        chr1     [4561223, 4562241]      * |        92\n",
       "      [3]        chr1     [4612708, 4613925]      * |      87.6\n",
       "      [4]        chr1     [4660365, 4662653]      * |    323.25\n",
       "      [5]        chr1     [4764655, 4766144]      * |    121.52\n",
       "      ...         ...                    ...    ... .       ...\n",
       "  [54908]        chrX [166317016, 166320412]      * |    709.07\n",
       "  [54909]        chrX [166334989, 166336095]      * |    146.24\n",
       "  [54910]        chrX [166409493, 166421961]      * |      3100\n",
       "  [54911] chrX_random [    36504,     39861]      * |    439.12\n",
       "  [54912] chrX_random [   159230,    160297]      * |    241.04\n",
       "  -------\n",
       "  seqinfo: 25 sequences from an unspecified genome; no seqlengths\n",
       "\n",
       "$EuMyc$K4me1.T27\n",
       "GRanges object with 49519 ranges and 1 metadata column:\n",
       "             seqnames                 ranges strand |    pvalue\n",
       "                <Rle>              <IRanges>  <Rle> | <numeric>\n",
       "      [1]        chr1     [3831159, 3831813]      * |     82.63\n",
       "      [2]        chr1     [4773756, 4777379]      * |    936.06\n",
       "      [3]        chr1     [4797036, 4799574]      * |    588.44\n",
       "      [4]        chr1     [4846521, 4847744]      * |    159.52\n",
       "      [5]        chr1     [4848226, 4850943]      * |   1000.49\n",
       "      ...         ...                    ...    ... .       ...\n",
       "  [49515]        chrX [166434648, 166437929]      * |    390.53\n",
       "  [49516] chrX_random [    37951,     39754]      * |    237.11\n",
       "  [49517]        chrY [   347539,    348250]      * |    102.81\n",
       "  [49518]        chrY [   580650,    581812]      * |    115.68\n",
       "  [49519]        chrY [   621111,    622019]      * |    154.31\n",
       "  -------\n",
       "  seqinfo: 26 sequences from an unspecified genome; no seqlengths\n",
       "\n",
       "$EuMyc$K4me1.T29\n",
       "GRanges object with 47607 ranges and 1 metadata column:\n",
       "             seqnames                 ranges strand |    pvalue\n",
       "                <Rle>              <IRanges>  <Rle> | <numeric>\n",
       "      [1]        chr1     [4773666, 4777416]      * |   1411.29\n",
       "      [2]        chr1     [4796600, 4797983]      * |    314.46\n",
       "      [3]        chr1     [4798050, 4801053]      * |   1560.71\n",
       "      [4]        chr1     [4846602, 4847738]      * |    201.88\n",
       "      [5]        chr1     [4848099, 4850935]      * |   1380.28\n",
       "      ...         ...                    ...    ... .       ...\n",
       "  [47603]        chrX [166409369, 166422133]      * |    1626.7\n",
       "  [47604]        chrX [166422186, 166434587]      * |    239.46\n",
       "  [47605]        chrX [166434643, 166437845]      * |   2300.28\n",
       "  [47606]        chrX [166444609, 166447024]      * |     90.29\n",
       "  [47607] chrX_random [    37282,     39833]      * |    465.11\n",
       "  -------\n",
       "  seqinfo: 26 sequences from an unspecified genome; no seqlengths\n",
       "\n",
       "$EuMyc$K4me1.T31\n",
       "GRanges object with 66090 ranges and 1 metadata column:\n",
       "             seqnames                 ranges strand |    pvalue\n",
       "                <Rle>              <IRanges>  <Rle> | <numeric>\n",
       "      [1]        chr1     [4486109, 4487216]      * |    157.85\n",
       "      [2]        chr1     [4730009, 4732099]      * |    533.75\n",
       "      [3]        chr1     [4734268, 4735969]      * |     83.77\n",
       "      [4]        chr1     [4737340, 4739426]      * |   1155.99\n",
       "      [5]        chr1     [4773752, 4777367]      * |   1048.15\n",
       "      ...         ...                    ...    ... .       ...\n",
       "  [66086]        chrX [166409469, 166415518]      * |      3100\n",
       "  [66087]        chrX [166415700, 166421740]      * |    727.05\n",
       "  [66088]        chrX [166426367, 166437935]      * |    176.08\n",
       "  [66089]        chrX [166444290, 166447014]      * |    120.02\n",
       "  [66090] chrX_random [    38223,     39104]      * |    235.06\n",
       "  -------\n",
       "  seqinfo: 25 sequences from an unspecified genome; no seqlengths\n",
       "\n",
       "$EuMyc$K4me3.C\n",
       "GRanges object with 18649 ranges and 1 metadata column:\n",
       "              seqnames             ranges strand |    pvalue\n",
       "                 <Rle>          <IRanges>  <Rle> | <numeric>\n",
       "      [1]         chr1 [4561152, 4562489]      * |     135.9\n",
       "      [2]         chr1 [4774677, 4777094]      * |      3100\n",
       "      [3]         chr1 [4796756, 4799530]      * |      3100\n",
       "      [4]         chr1 [4846621, 4850243]      * |      3100\n",
       "      [5]         chr1 [5009304, 5010763]      * |    129.29\n",
       "      ...          ...                ...    ... .       ...\n",
       "  [18645]  chrX_random [1048144, 1051023]      * |    278.33\n",
       "  [18646]         chrY [ 345754,  348170]      * |   2984.66\n",
       "  [18647]         chrY [ 580596,  582791]      * |      3100\n",
       "  [18648]         chrY [ 620915,  623702]      * |      3100\n",
       "  [18649] chr13_random [  32007,   32471]      * |    102.21\n",
       "  -------\n",
       "  seqinfo: 28 sequences from an unspecified genome; no seqlengths\n",
       "\n",
       "$EuMyc$K4me3.P\n",
       "GRanges object with 17701 ranges and 1 metadata column:\n",
       "             seqnames             ranges strand |    pvalue\n",
       "                <Rle>          <IRanges>  <Rle> | <numeric>\n",
       "      [1]        chr1 [4774603, 4776777]      * |      3100\n",
       "      [2]        chr1 [4797001, 4799208]      * |   2908.82\n",
       "      [3]        chr1 [4846784, 4849610]      * |      3100\n",
       "      [4]        chr1 [5009089, 5010554]      * |    260.28\n",
       "      [5]        chr1 [5072611, 5074233]      * |   2016.63\n",
       "      ...         ...                ...    ... .       ...\n",
       "  [17697] chrX_random   [ 38184,  39811]      * |   2360.81\n",
       "  [17698] chrX_random   [159192, 160417]      * |    340.96\n",
       "  [17699]        chrY   [346188, 348020]      * |   1131.55\n",
       "  [17700]        chrY   [580890, 582587]      * |   1173.55\n",
       "  [17701]        chrY   [621222, 623510]      * |    2093.4\n",
       "  -------\n",
       "  seqinfo: 28 sequences from an unspecified genome; no seqlengths\n",
       "\n",
       "$EuMyc$K4me3.T27\n",
       "GRanges object with 18364 ranges and 1 metadata column:\n",
       "             seqnames                 ranges strand |    pvalue\n",
       "                <Rle>              <IRanges>  <Rle> | <numeric>\n",
       "      [1]        chr1     [4774077, 4776850]      * |      3100\n",
       "      [2]        chr1     [4796995, 4799286]      * |      3100\n",
       "      [3]        chr1     [4846838, 4849769]      * |      3100\n",
       "      [4]        chr1     [5008714, 5011422]      * |    451.45\n",
       "      [5]        chr1     [5072704, 5074089]      * |   2490.82\n",
       "      ...         ...                    ...    ... .       ...\n",
       "  [18360]        chrX [166434635, 166436479]      * |    637.65\n",
       "  [18361] chrX_random [    38230,     39833]      * |   1831.97\n",
       "  [18362]        chrY [   345564,    348261]      * |   2902.94\n",
       "  [18363]        chrY [   580572,    582874]      * |      3100\n",
       "  [18364]        chrY [   621221,    623412]      * |      3100\n",
       "  -------\n",
       "  seqinfo: 28 sequences from an unspecified genome; no seqlengths\n",
       "\n",
       "$EuMyc$K4me3.T29\n",
       "GRanges object with 20966 ranges and 1 metadata column:\n",
       "             seqnames                 ranges strand |    pvalue\n",
       "                <Rle>              <IRanges>  <Rle> | <numeric>\n",
       "      [1]        chr1     [4773769, 4776971]      * |      3100\n",
       "      [2]        chr1     [4796943, 4799279]      * |      3100\n",
       "      [3]        chr1     [4846887, 4850118]      * |      3100\n",
       "      [4]        chr1     [5072592, 5074037]      * |      3100\n",
       "      [5]        chr1     [6203146, 6206781]      * |      3100\n",
       "      ...         ...                    ...    ... .       ...\n",
       "  [20962]        chrX [166423373, 166426020]      * |    410.12\n",
       "  [20963]        chrX [166426178, 166431715]      * |    324.59\n",
       "  [20964]        chrX [166434549, 166436423]      * |    2336.9\n",
       "  [20965] chrX_random [    38058,     39813]      * |      3100\n",
       "  [20966] chrX_random [   159655,    160137]      * |    198.37\n",
       "  -------\n",
       "  seqinfo: 26 sequences from an unspecified genome; no seqlengths\n",
       "\n",
       "$EuMyc$K4me3.T31\n",
       "GRanges object with 20987 ranges and 1 metadata column:\n",
       "             seqnames                 ranges strand |    pvalue\n",
       "                <Rle>              <IRanges>  <Rle> | <numeric>\n",
       "      [1]        chr1     [4774020, 4777079]      * |      3100\n",
       "      [2]        chr1     [4797064, 4799301]      * |      3100\n",
       "      [3]        chr1     [4847026, 4849735]      * |      3100\n",
       "      [4]        chr1     [5072727, 5074017]      * |      3100\n",
       "      [5]        chr1     [6203230, 6206698]      * |      3100\n",
       "      ...         ...                    ...    ... .       ...\n",
       "  [20983]        chrX [166419051, 166425902]      * |    636.02\n",
       "  [20984]        chrX [166426163, 166431703]      * |   1704.85\n",
       "  [20985]        chrX [166431750, 166437935]      * |      3100\n",
       "  [20986] chrX_random [    38266,     39799]      * |   2738.29\n",
       "  [20987] chrX_random [   159523,    160190]      * |    148.17\n",
       "  -------\n",
       "  seqinfo: 27 sequences from an unspecified genome; no seqlengths\n",
       "\n",
       "$EuMyc$K27ac.C1\n",
       "GRanges object with 40757 ranges and 1 metadata column:\n",
       "             seqnames             ranges strand |    pvalue\n",
       "                <Rle>          <IRanges>  <Rle> | <numeric>\n",
       "      [1]        chr1 [4612917, 4613850]      * |    133.29\n",
       "      [2]        chr1 [4774927, 4776797]      * |   2090.54\n",
       "      [3]        chr1 [4796779, 4799280]      * |    3221.3\n",
       "      [4]        chr1 [4846873, 4849530]      * |      3100\n",
       "      [5]        chr1 [5072541, 5073972]      * |    2901.9\n",
       "      ...         ...                ...    ... .       ...\n",
       "  [40753] chrX_random   [159073, 160539]      * |    573.78\n",
       "  [40754]        chrY   [346072, 347926]      * |   1600.39\n",
       "  [40755]        chrY   [580748, 582929]      * |      3100\n",
       "  [40756]        chrY   [621283, 623704]      * |   2793.43\n",
       "  [40757]        chrY   [629807, 631188]      * |    234.36\n",
       "  -------\n",
       "  seqinfo: 27 sequences from an unspecified genome; no seqlengths\n",
       "\n",
       "$EuMyc$K27ac.P1\n",
       "GRanges object with 30895 ranges and 1 metadata column:\n",
       "             seqnames             ranges strand |    pvalue\n",
       "                <Rle>          <IRanges>  <Rle> | <numeric>\n",
       "      [1]        chr1 [4774719, 4776705]      * |   1418.07\n",
       "      [2]        chr1 [4796888, 4799149]      * |   1354.31\n",
       "      [3]        chr1 [4846977, 4849562]      * |   3129.69\n",
       "      [4]        chr1 [5072585, 5073986]      * |   1194.75\n",
       "      [5]        chr1 [6203418, 6206189]      * |      3100\n",
       "      ...         ...                ...    ... .       ...\n",
       "  [30891] chrX_random   [ 38201,  39699]      * |    853.95\n",
       "  [30892] chrX_random   [159553, 160288]      * |     334.5\n",
       "  [30893]        chrY   [346081, 348154]      * |   1136.21\n",
       "  [30894]        chrY   [580834, 582796]      * |      3100\n",
       "  [30895]        chrY   [621393, 623640]      * |   2728.42\n",
       "  -------\n",
       "  seqinfo: 27 sequences from an unspecified genome; no seqlengths\n",
       "\n",
       "$EuMyc$K27ac.T27\n",
       "GRanges object with 31642 ranges and 1 metadata column:\n",
       "             seqnames                 ranges strand |    pvalue\n",
       "                <Rle>              <IRanges>  <Rle> | <numeric>\n",
       "      [1]        chr1     [4774810, 4776436]      * |   1654.43\n",
       "      [2]        chr1     [4797287, 4798742]      * |   1692.62\n",
       "      [3]        chr1     [4847076, 4849108]      * |   2903.57\n",
       "      [4]        chr1     [5072763, 5073908]      * |    754.72\n",
       "      [5]        chr1     [5301437, 5302109]      * |    104.12\n",
       "      ...         ...                    ...    ... .       ...\n",
       "  [31638]        chrX [166445313, 166446972]      * |       282\n",
       "  [31639] chrX_random [    38529,     39728]      * |    283.73\n",
       "  [31640]        chrY [   346120,    347585]      * |   1274.43\n",
       "  [31641]        chrY [   580990,    582615]      * |      3100\n",
       "  [31642]        chrY [   621805,    623369]      * |   2406.19\n",
       "  -------\n",
       "  seqinfo: 27 sequences from an unspecified genome; no seqlengths\n",
       "\n",
       "$EuMyc$K27ac.T29\n",
       "GRanges object with 31954 ranges and 1 metadata column:\n",
       "             seqnames                 ranges strand |    pvalue\n",
       "                <Rle>              <IRanges>  <Rle> | <numeric>\n",
       "      [1]        chr1     [4774257, 4776611]      * |      3100\n",
       "      [2]        chr1     [4796734, 4799265]      * |      3100\n",
       "      [3]        chr1     [4846968, 4849264]      * |      3100\n",
       "      [4]        chr1     [5072593, 5073993]      * |   1748.59\n",
       "      [5]        chr1     [6203253, 6206276]      * |      3100\n",
       "      ...         ...                    ...    ... .       ...\n",
       "  [31950]        chrX [166415671, 166418941]      * |    406.11\n",
       "  [31951]        chrX [166423538, 166437962]      * |      3100\n",
       "  [31952]        chrX [166444520, 166447013]      * |    781.77\n",
       "  [31953] chrX_random [    38216,     39834]      * |   1263.91\n",
       "  [31954] chrX_random [   159450,    160316]      * |    100.98\n",
       "  -------\n",
       "  seqinfo: 26 sequences from an unspecified genome; no seqlengths\n",
       "\n",
       "$EuMyc$K27ac.T31\n",
       "GRanges object with 42542 ranges and 1 metadata column:\n",
       "             seqnames                 ranges strand |    pvalue\n",
       "                <Rle>              <IRanges>  <Rle> | <numeric>\n",
       "      [1]        chr1     [4774760, 4776592]      * |      3100\n",
       "      [2]        chr1     [4797093, 4799128]      * |   2901.76\n",
       "      [3]        chr1     [4846825, 4849113]      * |      3100\n",
       "      [4]        chr1     [5072647, 5074017]      * |   2552.47\n",
       "      [5]        chr1     [6203573, 6206348]      * |      3100\n",
       "      ...         ...                    ...    ... .       ...\n",
       "  [42538]        chrX [166415709, 166418846]      * |    272.44\n",
       "  [42539]        chrX [166423652, 166426037]      * |    753.96\n",
       "  [42540]        chrX [166426406, 166437742]      * |      3100\n",
       "  [42541]        chrX [166440451, 166446995]      * |   1511.65\n",
       "  [42542] chrX_random [    38450,     39785]      * |   1269.27\n",
       "  -------\n",
       "  seqinfo: 26 sequences from an unspecified genome; no seqlengths\n",
       "\n",
       "\n",
       "$mycER\n",
       "$mycER$Myc.0h\n",
       "GRanges object with 18000 ranges and 2 metadata columns:\n",
       "          seqnames                 ranges strand |    pvalue enrichment\n",
       "             <Rle>              <IRanges>  <Rle> | <numeric>  <numeric>\n",
       "      [1]     chr1     [4774950, 4776423]      * |    385.78       7.71\n",
       "      [2]     chr1     [4847067, 4848964]      * |    359.27        7.7\n",
       "      [3]     chr1     [5072867, 5073492]      * |    132.11       6.05\n",
       "      [4]     chr1     [6269534, 6270978]      * |    189.25       7.25\n",
       "      [5]     chr1     [6277552, 6279292]      * |     93.91       6.92\n",
       "      ...      ...                    ...    ... .       ...        ...\n",
       "  [17996]     chrX [165111862, 165112612]      * |     84.26        5.8\n",
       "  [17997]     chrX [165757891, 165758699]      * |    119.52       6.33\n",
       "  [17998]     chrX [166232376, 166233636]      * |    280.77       7.79\n",
       "  [17999]     chrX [166317263, 166318096]      * |     86.02       6.05\n",
       "  [18000]     chrX [166319126, 166320258]      * |     98.95        6.7\n",
       "  -------\n",
       "  seqinfo: 25 sequences from an unspecified genome; no seqlengths\n",
       "\n",
       "$mycER$Myc.4h\n",
       "GRanges object with 27693 ranges and 2 metadata columns:\n",
       "             seqnames                 ranges strand |    pvalue enrichment\n",
       "                <Rle>              <IRanges>  <Rle> | <numeric>  <numeric>\n",
       "      [1]        chr1     [3504586, 3505614]      * |    199.88       6.86\n",
       "      [2]        chr1     [4774851, 4776489]      * |   1161.65       8.76\n",
       "      [3]        chr1     [4847003, 4849064]      * |   1836.41       9.22\n",
       "      [4]        chr1     [5005383, 5006266]      * |    124.35        6.2\n",
       "      [5]        chr1     [5012570, 5013453]      * |    139.49       6.18\n",
       "      ...         ...                    ...    ... .       ...        ...\n",
       "  [27689]        chrX [166318473, 166320386]      * |    149.51       7.57\n",
       "  [27690]        chrX [166336708, 166338206]      * |    100.01       6.55\n",
       "  [27691]        chrX [166423283, 166426230]      * |    453.25       8.87\n",
       "  [27692]        chrX [166434632, 166439568]      * |    993.56       10.1\n",
       "  [27693] chrX_random [    38654,     39730]      * |    117.11       6.14\n",
       "  -------\n",
       "  seqinfo: 27 sequences from an unspecified genome; no seqlengths\n",
       "\n",
       "$mycER$Pol2.0h\n",
       "GRanges object with 27493 ranges and 1 metadata column:\n",
       "             seqnames                 ranges strand |    pvalue\n",
       "                <Rle>              <IRanges>  <Rle> | <numeric>\n",
       "      [1]        chr1     [4774775, 4776391]      * |   1539.03\n",
       "      [2]        chr1     [4797338, 4797952]      * |     82.14\n",
       "      [3]        chr1     [4846819, 4849234]      * |   1886.09\n",
       "      [4]        chr1     [4889169, 4890930]      * |    272.71\n",
       "      [5]        chr1     [4891001, 4894875]      * |    656.77\n",
       "      ...         ...                    ...    ... .       ...\n",
       "  [27489]        chrX [166317060, 166320392]      * |    921.25\n",
       "  [27490]        chrX [166393410, 166400601]      * |    571.25\n",
       "  [27491]        chrX [166400603, 166426157]      * |   1269.53\n",
       "  [27492] chrX_random [    38614,     39781]      * |    312.88\n",
       "  [27493] chrX_random [   159101,    160108]      * |    156.52\n",
       "  -------\n",
       "  seqinfo: 25 sequences from an unspecified genome; no seqlengths\n",
       "\n",
       "$mycER$Pol2.4h\n",
       "GRanges object with 31941 ranges and 1 metadata column:\n",
       "             seqnames                 ranges strand |    pvalue\n",
       "                <Rle>              <IRanges>  <Rle> | <numeric>\n",
       "      [1]        chr1     [4758991, 4759624]      * |    130.85\n",
       "      [2]        chr1     [4769730, 4770292]      * |     92.05\n",
       "      [3]        chr1     [4774745, 4776562]      * |   1672.07\n",
       "      [4]        chr1     [4797207, 4798172]      * |    215.52\n",
       "      [5]        chr1     [4846999, 4849265]      * |    1273.6\n",
       "      ...         ...                    ...    ... .       ...\n",
       "  [31937]        chrX [166395688, 166396787]      * |    102.09\n",
       "  [31938]        chrX [166399656, 166400494]      * |    108.91\n",
       "  [31939]        chrX [166405534, 166426019]      * |      3100\n",
       "  [31940]        chrX [166434770, 166436257]      * |    626.39\n",
       "  [31941] chrX_random [   159148,    160182]      * |    205.44\n",
       "  -------\n",
       "  seqinfo: 25 sequences from an unspecified genome; no seqlengths\n",
       "\n",
       "$mycER$K4me1.0h\n",
       "GRanges object with 77144 ranges and 1 metadata column:\n",
       "             seqnames                 ranges strand |    pvalue\n",
       "                <Rle>              <IRanges>  <Rle> | <numeric>\n",
       "      [1]        chr1     [3482902, 3484575]      * |    348.52\n",
       "      [2]        chr1     [3504154, 3508011]      * |   1192.71\n",
       "      [3]        chr1     [4773806, 4775521]      * |    810.11\n",
       "      [4]        chr1     [4775857, 4777152]      * |    747.89\n",
       "      [5]        chr1     [4796995, 4797858]      * |    297.65\n",
       "      ...         ...                    ...    ... .       ...\n",
       "  [77140]        chrX [166439972, 166440809]      * |     98.53\n",
       "  [77141]        chrX [166441851, 166444187]      * |    101.24\n",
       "  [77142]        chrX [166445544, 166447000]      * |    998.65\n",
       "  [77143] chrX_random [    37693,     39828]      * |   1374.54\n",
       "  [77144] chrX_random [   158941,    160959]      * |    517.37\n",
       "  -------\n",
       "  seqinfo: 27 sequences from an unspecified genome; no seqlengths\n",
       "\n",
       "$mycER$K4me1.4h\n",
       "GRanges object with 74104 ranges and 1 metadata column:\n",
       "             seqnames                 ranges strand |    pvalue\n",
       "                <Rle>              <IRanges>  <Rle> | <numeric>\n",
       "      [1]        chr1     [3482711, 3484606]      * |    305.44\n",
       "      [2]        chr1     [3504051, 3508091]      * |   1690.69\n",
       "      [3]        chr1     [4773683, 4775546]      * |    999.03\n",
       "      [4]        chr1     [4775601, 4777130]      * |    863.91\n",
       "      [5]        chr1     [4796879, 4797872]      * |    524.92\n",
       "      ...         ...                    ...    ... .       ...\n",
       "  [74100]        chrX [166426929, 166439738]      * |   1447.64\n",
       "  [74101]        chrX [166445284, 166447015]      * |     832.2\n",
       "  [74102] chrX_random [    37595,     39845]      * |   1716.15\n",
       "  [74103] chrX_random [   110990,    111733]      * |     106.6\n",
       "  [74104] chrX_random [   158896,    161357]      * |   1339.84\n",
       "  -------\n",
       "  seqinfo: 27 sequences from an unspecified genome; no seqlengths\n",
       "\n",
       "$mycER$K4me3.0h\n",
       "GRanges object with 13855 ranges and 1 metadata column:\n",
       "             seqnames                 ranges strand |    pvalue\n",
       "                <Rle>              <IRanges>  <Rle> | <numeric>\n",
       "      [1]        chr1     [4774420, 4776914]      * |      3100\n",
       "      [2]        chr1     [4797031, 4799395]      * |      3100\n",
       "      [3]        chr1     [4846505, 4849828]      * |      3100\n",
       "      [4]        chr1     [5008643, 5010789]      * |    938.86\n",
       "      [5]        chr1     [5072568, 5074753]      * |      3100\n",
       "      ...         ...                    ...    ... .       ...\n",
       "  [13851]        chrX [166423856, 166425845]      * |    306.91\n",
       "  [13852]        chrX [166434325, 166436976]      * |   2312.25\n",
       "  [13853] chrX_random [    37853,     39822]      * |      3100\n",
       "  [13854] chrX_random [   159079,    161033]      * |      3100\n",
       "  [13855] chrY_random [ 20940730,  20941121]      * |    491.01\n",
       "  -------\n",
       "  seqinfo: 28 sequences from an unspecified genome; no seqlengths\n",
       "\n",
       "$mycER$K4me3.4h\n",
       "GRanges object with 13407 ranges and 1 metadata column:\n",
       "             seqnames                 ranges strand |    pvalue\n",
       "                <Rle>              <IRanges>  <Rle> | <numeric>\n",
       "      [1]        chr1     [4774499, 4776750]      * |      3100\n",
       "      [2]        chr1     [4797103, 4799369]      * |      3100\n",
       "      [3]        chr1     [4846623, 4849587]      * |      3100\n",
       "      [4]        chr1     [5008736, 5010800]      * |     408.8\n",
       "      [5]        chr1     [5072543, 5074533]      * |      3100\n",
       "      ...         ...                    ...    ... .       ...\n",
       "  [13403]        chrX [166423878, 166425714]      * |    298.16\n",
       "  [13404]        chrX [166434523, 166436280]      * |   2534.35\n",
       "  [13405] chrX_random [    37869,     39736]      * |      3100\n",
       "  [13406] chrX_random [   159055,    160946]      * |   3016.38\n",
       "  [13407] chrY_random [ 20940743,  20941107]      * |    470.62\n",
       "  -------\n",
       "  seqinfo: 27 sequences from an unspecified genome; no seqlengths\n",
       "\n",
       "$mycER$K27ac.0h\n",
       "GRanges object with 34576 ranges and 1 metadata column:\n",
       "             seqnames                 ranges strand |    pvalue\n",
       "                <Rle>              <IRanges>  <Rle> | <numeric>\n",
       "      [1]        chr1     [3482449, 3484860]      * |    346.84\n",
       "      [2]        chr1     [3504134, 3504996]      * |     82.39\n",
       "      [3]        chr1     [3505057, 3508274]      * |    608.91\n",
       "      [4]        chr1     [4774350, 4776786]      * |   2142.19\n",
       "      [5]        chr1     [4797143, 4799039]      * |    611.31\n",
       "      ...         ...                    ...    ... .       ...\n",
       "  [34572]        chrX [166432110, 166437964]      * |    369.87\n",
       "  [34573]        chrX [166445667, 166447056]      * |    167.57\n",
       "  [34574] chrX_random [    38155,     39872]      * |   1011.68\n",
       "  [34575] chrX_random [   158872,    160940]      * |   1441.46\n",
       "  [34576] chrY_random [ 20940690,  20941159]      * |    148.89\n",
       "  -------\n",
       "  seqinfo: 28 sequences from an unspecified genome; no seqlengths\n",
       "\n",
       "$mycER$K27ac.4h\n",
       "GRanges object with 33702 ranges and 1 metadata column:\n",
       "             seqnames                 ranges strand |    pvalue\n",
       "                <Rle>              <IRanges>  <Rle> | <numeric>\n",
       "      [1]        chr1     [3506092, 3507929]      * |    309.37\n",
       "      [2]        chr1     [4774478, 4776760]      * |   2987.57\n",
       "      [3]        chr1     [4797179, 4799176]      * |    752.36\n",
       "      [4]        chr1     [4846639, 4849702]      * |      3100\n",
       "      [5]        chr1     [5008848, 5009757]      * |     82.36\n",
       "      ...         ...                    ...    ... .       ...\n",
       "  [33698]        chrX [166434516, 166438005]      * |   1522.09\n",
       "  [33699]        chrX [166445646, 166447034]      * |    307.33\n",
       "  [33700] chrX_random [    38337,     39636]      * |    307.47\n",
       "  [33701] chrX_random [   159005,    160777]      * |   1149.59\n",
       "  [33702] chrY_random [ 20940699,  20941149]      * |    153.96\n",
       "  -------\n",
       "  seqinfo: 27 sequences from an unspecified genome; no seqlengths\n",
       "\n",
       "\n",
       "$P493\n",
       "$P493$Myc.t0h\n",
       "GRanges object with 791 ranges and 2 metadata columns:\n",
       "        seqnames               ranges strand |    pvalue enrichment\n",
       "           <Rle>            <IRanges>  <Rle> | <numeric>  <numeric>\n",
       "    [1]     chr1   [   9984,   10351]      * |    354.81       7.73\n",
       "    [2]     chr1   [ 567518,  567730]      * |    178.59       8.02\n",
       "    [3]     chr1   [ 569850,  569987]      * |    109.47       7.58\n",
       "    [4]     chr1   [ 713900,  714174]      * |      90.8       6.16\n",
       "    [5]     chr1   [1342540, 1342890]      * |    131.91       6.79\n",
       "    ...      ...                  ...    ... .       ...        ...\n",
       "  [787]     chrY [13842621, 13842856]      * |    247.68        8.4\n",
       "  [788]     chrY [13846444, 13846527]      * |    174.75       6.77\n",
       "  [789]     chrY [13860001, 13860088]      * |    233.22       6.68\n",
       "  [790]     chrY [13868358, 13868460]      * |    314.64       7.12\n",
       "  [791]     chrY [13869031, 13869189]      * |    110.05       7.07\n",
       "  -------\n",
       "  seqinfo: 25 sequences from an unspecified genome; no seqlengths\n",
       "\n",
       "$P493$Myc.t1h\n",
       "GRanges object with 8666 ranges and 2 metadata columns:\n",
       "         seqnames               ranges strand |    pvalue enrichment\n",
       "            <Rle>            <IRanges>  <Rle> | <numeric>  <numeric>\n",
       "     [1]     chr1     [ 10039,  10385]      * |    140.83       6.32\n",
       "     [2]     chr1     [567498, 567655]      * |    118.13       7.19\n",
       "     [3]     chr1     [713942, 714343]      * |    241.78       7.12\n",
       "     [4]     chr1     [842138, 842573]      * |    165.82       6.72\n",
       "     [5]     chr1     [936087, 936458]      * |     95.39       6.07\n",
       "     ...      ...                  ...    ... .       ...        ...\n",
       "  [8662]     chrY [13846424, 13846687]      * |     81.41       6.59\n",
       "  [8663]     chrY [13859818, 13860105]      * |    183.73       6.79\n",
       "  [8664]     chrY [13866920, 13867048]      * |    194.03       6.25\n",
       "  [8665]     chrY [13868233, 13868571]      * |    212.08       7.16\n",
       "  [8666]     chrY [13869030, 13869520]      * |     85.52       7.19\n",
       "  -------\n",
       "  seqinfo: 25 sequences from an unspecified genome; no seqlengths\n",
       "\n",
       "$P493$Myc.t24h\n",
       "GRanges object with 18257 ranges and 2 metadata columns:\n",
       "          seqnames               ranges strand |    pvalue enrichment\n",
       "             <Rle>            <IRanges>  <Rle> | <numeric>  <numeric>\n",
       "      [1]     chr1     [567509, 567765]      * |    151.08        8.1\n",
       "      [2]     chr1     [569852, 570001]      * |    173.95       8.28\n",
       "      [3]     chr1     [713855, 714378]      * |     616.6       8.15\n",
       "      [4]     chr1     [740258, 740440]      * |    237.72       6.68\n",
       "      [5]     chr1     [762626, 762877]      * |    100.55        6.1\n",
       "      ...      ...                  ...    ... .       ...        ...\n",
       "  [18253]     chrY [13846314, 13846541]      * |    154.71       6.65\n",
       "  [18254]     chrY [13866924, 13867128]      * |    142.29       6.51\n",
       "  [18255]     chrY [13868195, 13868573]      * |    169.52       7.21\n",
       "  [18256]     chrY [13868950, 13869189]      * |     81.59       6.43\n",
       "  [18257]     chrY [58996456, 58996649]      * |     93.15       6.65\n",
       "  -------\n",
       "  seqinfo: 25 sequences from an unspecified genome; no seqlengths\n",
       "\n",
       "$P493$Myc.growing\n",
       "GRanges object with 26146 ranges and 2 metadata columns:\n",
       "          seqnames               ranges strand |    pvalue enrichment\n",
       "             <Rle>            <IRanges>  <Rle> | <numeric>  <numeric>\n",
       "      [1]     chr1     [713783, 714673]      * |    655.72       7.91\n",
       "      [2]     chr1     [740099, 740483]      * |    482.65       7.24\n",
       "      [3]     chr1     [839929, 840461]      * |    272.05       6.84\n",
       "      [4]     chr1     [840497, 841136]      * |    563.81        7.7\n",
       "      [5]     chr1     [841988, 842546]      * |    408.37       7.34\n",
       "      ...      ...                  ...    ... .       ...        ...\n",
       "  [26142]     chrY [13843879, 13844124]      * |     81.01       5.92\n",
       "  [26143]     chrY [13859872, 13860107]      * |    113.14       6.06\n",
       "  [26144]     chrY [13866920, 13867045]      * |    136.95       5.43\n",
       "  [26145]     chrY [13868290, 13868588]      * |      82.3       5.95\n",
       "  [26146]     chrY [21154672, 21155060]      * |     100.8       5.64\n",
       "  -------\n",
       "  seqinfo: 25 sequences from an unspecified genome; no seqlengths\n",
       "\n",
       "$P493$Myc.low\n",
       "GRanges object with 4309 ranges and 2 metadata columns:\n",
       "         seqnames               ranges strand |    pvalue enrichment\n",
       "            <Rle>            <IRanges>  <Rle> | <numeric>  <numeric>\n",
       "     [1]     chr1   [ 714026,  714228]      * |     81.46       5.52\n",
       "     [2]     chr1   [ 740247,  740467]      * |    115.31       5.75\n",
       "     [3]     chr1   [ 936147,  936370]      * |     98.31       5.78\n",
       "     [4]     chr1   [ 968536,  968733]      * |     80.04       5.25\n",
       "     [5]     chr1   [1186354, 1186653]      * |    253.86       6.74\n",
       "     ...      ...                  ...    ... .       ...        ...\n",
       "  [4305]     chrY [13846438, 13846543]      * |    200.16       6.45\n",
       "  [4306]     chrY [13859863, 13860093]      * |    145.16       6.34\n",
       "  [4307]     chrY [13866934, 13867101]      * |    142.26       6.17\n",
       "  [4308]     chrY [13868351, 13868575]      * |    224.58        6.7\n",
       "  [4309]     chrY [13869031, 13869206]      * |     95.42       6.56\n",
       "  -------\n",
       "  seqinfo: 25 sequences from an unspecified genome; no seqlengths\n",
       "\n",
       "$P493$Pol2.t0h\n",
       "GRanges object with 13026 ranges and 1 metadata column:\n",
       "          seqnames               ranges strand |    pvalue\n",
       "             <Rle>            <IRanges>  <Rle> | <numeric>\n",
       "      [1]     chr1   [ 948499,  951230]      * |      3100\n",
       "      [2]     chr1   [ 951455,  951739]      * |    133.93\n",
       "      [3]     chr1   [ 952742,  952986]      * |     86.12\n",
       "      [4]     chr1   [1093134, 1093664]      * |     128.5\n",
       "      [5]     chr1   [1310721, 1311180]      * |    178.36\n",
       "      ...      ...                  ...    ... .       ...\n",
       "  [13022]     chrY [13842607, 13842851]      * |     88.36\n",
       "  [13023]     chrY [13846430, 13846541]      * |    125.25\n",
       "  [13024]     chrY [13866925, 13867036]      * |    115.22\n",
       "  [13025]     chrY [13868344, 13868581]      * |    152.57\n",
       "  [13026]     chrY [21154551, 21154690]      * |     143.4\n",
       "  -------\n",
       "  seqinfo: 24 sequences from an unspecified genome; no seqlengths\n",
       "\n",
       "$P493$Pol2.t24h\n",
       "GRanges object with 24613 ranges and 1 metadata column:\n",
       "          seqnames               ranges strand |    pvalue\n",
       "             <Rle>            <IRanges>  <Rle> | <numeric>\n",
       "      [1]     chr1     [713780, 714505]      * |       682\n",
       "      [2]     chr1     [762622, 763125]      * |     206.3\n",
       "      [3]     chr1     [840056, 840407]      * |    154.54\n",
       "      [4]     chr1     [840482, 841126]      * |    244.95\n",
       "      [5]     chr1     [842158, 842588]      * |    202.33\n",
       "      ...      ...                  ...    ... .       ...\n",
       "  [24609]     chrY [13859819, 13860229]      * |     80.16\n",
       "  [24610]     chrY [13866916, 13867045]      * |    128.51\n",
       "  [24611]     chrY [13868335, 13868594]      * |       185\n",
       "  [24612]     chrY [21154250, 21154498]      * |     81.18\n",
       "  [24613]     chrY [21154504, 21154872]      * |    269.42\n",
       "  -------\n",
       "  seqinfo: 24 sequences from an unspecified genome; no seqlengths\n",
       "\n",
       "$P493$K4me3.t0h\n",
       "GRanges object with 27096 ranges and 1 metadata column:\n",
       "          seqnames               ranges strand |    pvalue\n",
       "             <Rle>            <IRanges>  <Rle> | <numeric>\n",
       "      [1]     chr1     [712865, 715771]      * |      3100\n",
       "      [2]     chr1     [760877, 763289]      * |   2942.17\n",
       "      [3]     chr1     [840145, 842993]      * |    1218.6\n",
       "      [4]     chr1     [858543, 859230]      * |    306.68\n",
       "      [5]     chr1     [875914, 876995]      * |    143.26\n",
       "      ...      ...                  ...    ... .       ...\n",
       "  [27092]     chrY [13810595, 13811426]      * |     107.1\n",
       "  [27093]     chrY [13842152, 13844585]      * |    116.34\n",
       "  [27094]     chrY [13851454, 13852293]      * |    193.28\n",
       "  [27095]     chrY [13867619, 13870247]      * |     93.42\n",
       "  [27096]     chrY [21153721, 21155208]      * |    467.28\n",
       "  -------\n",
       "  seqinfo: 24 sequences from an unspecified genome; no seqlengths\n",
       "\n",
       "$P493$K4me3.t24h\n",
       "GRanges object with 29953 ranges and 1 metadata column:\n",
       "          seqnames               ranges strand |    pvalue\n",
       "             <Rle>            <IRanges>  <Rle> | <numeric>\n",
       "      [1]     chr1     [712926, 715723]      * |      3100\n",
       "      [2]     chr1     [761123, 763305]      * |      3100\n",
       "      [3]     chr1     [763503, 764089]      * |    119.33\n",
       "      [4]     chr1     [839765, 842973]      * |    3204.7\n",
       "      [5]     chr1     [876054, 877988]      * |    549.85\n",
       "      ...      ...                  ...    ... .       ...\n",
       "  [29949]     chrY [13842488, 13843311]      * |     110.9\n",
       "  [29950]     chrY [13843344, 13844242]      * |     114.7\n",
       "  [29951]     chrY [13851476, 13852249]      * |     89.05\n",
       "  [29952]     chrY [13866731, 13870227]      * |    108.69\n",
       "  [29953]     chrY [21154029, 21155200]      * |    680.62\n",
       "  -------\n",
       "  seqinfo: 24 sequences from an unspecified genome; no seqlengths\n",
       "\n",
       "$P493$K27ac.t0h\n",
       "GRanges object with 53577 ranges and 1 metadata column:\n",
       "          seqnames                 ranges strand |    pvalue\n",
       "             <Rle>              <IRanges>  <Rle> | <numeric>\n",
       "      [1]     chr1       [712902, 714900]      * |    376.36\n",
       "      [2]     chr1       [761812, 763284]      * |    576.22\n",
       "      [3]     chr1       [779104, 780485]      * |    259.35\n",
       "      [4]     chr1       [800284, 802090]      * |    368.29\n",
       "      [5]     chr1       [840234, 843894]      * |   2200.87\n",
       "      ...      ...                    ...    ... .       ...\n",
       "  [53573]     chrX [154299658, 154300843]      * |    819.41\n",
       "  [53574]     chrX [154443880, 154445420]      * |    382.11\n",
       "  [53575]     chrX [154459040, 154460190]      * |    115.54\n",
       "  [53576]     chrX [154840378, 154843366]      * |   2432.03\n",
       "  [53577]     chrX [154861728, 154862952]      * |    140.89\n",
       "  -------\n",
       "  seqinfo: 23 sequences from an unspecified genome; no seqlengths\n",
       "\n",
       "$P493$K27ac.t24h\n",
       "GRanges object with 59557 ranges and 1 metadata column:\n",
       "          seqnames                 ranges strand |    pvalue\n",
       "             <Rle>              <IRanges>  <Rle> | <numeric>\n",
       "      [1]     chr1       [712751, 714867]      * |    938.28\n",
       "      [2]     chr1       [761778, 763124]      * |    413.86\n",
       "      [3]     chr1       [779912, 780490]      * |     97.66\n",
       "      [4]     chr1       [800349, 802074]      * |    530.82\n",
       "      [5]     chr1       [840175, 843943]      * |   2303.98\n",
       "      ...      ...                    ...    ... .       ...\n",
       "  [59553]     chrX [154255403, 154256930]      * |    698.56\n",
       "  [59554]     chrX [154297720, 154299427]      * |    983.57\n",
       "  [59555]     chrX [154299677, 154301063]      * |    984.44\n",
       "  [59556]     chrX [154443775, 154445641]      * |    478.02\n",
       "  [59557]     chrX [154841110, 154843238]      * |   1138.91\n",
       "  -------\n",
       "  seqinfo: 23 sequences from an unspecified genome; no seqlengths\n",
       "\n",
       "\n",
       "$Serum\n",
       "$Serum$Myc.0h\n",
       "GRanges object with 5832 ranges and 2 metadata columns:\n",
       "         seqnames                 ranges strand |    pvalue enrichment\n",
       "            <Rle>              <IRanges>  <Rle> | <numeric>  <numeric>\n",
       "     [1]     chr1     [4775038, 4776293]      * |    135.08       6.79\n",
       "     [2]     chr1     [4847295, 4848488]      * |    146.81       6.98\n",
       "     [3]     chr1     [6269800, 6270826]      * |    125.49       6.58\n",
       "     [4]     chr1     [7387117, 7388448]      * |    276.63       7.63\n",
       "     [5]     chr1     [9534725, 9536241]      * |    340.82       8.02\n",
       "     ...      ...                    ...    ... .       ...        ...\n",
       "  [5828]     chrX [156575741, 156577053]      * |    133.07       7.12\n",
       "  [5829]     chrX [156939505, 156940977]      * |    158.89       7.02\n",
       "  [5830]     chrX [159266845, 159267816]      * |    153.35       6.54\n",
       "  [5831]     chrX [160810747, 160811805]      * |     83.71       6.27\n",
       "  [5832]     chrX [162947414, 162948784]      * |    161.91       6.92\n",
       "  -------\n",
       "  seqinfo: 20 sequences from an unspecified genome; no seqlengths\n",
       "\n",
       "$Serum$Myc.1h\n",
       "GRanges object with 8105 ranges and 2 metadata columns:\n",
       "         seqnames                 ranges strand |    pvalue enrichment\n",
       "            <Rle>              <IRanges>  <Rle> | <numeric>  <numeric>\n",
       "     [1]     chr1     [4774936, 4776246]      * |    156.43       6.55\n",
       "     [2]     chr1     [4846982, 4848211]      * |     136.6       6.38\n",
       "     [3]     chr1     [7387107, 7388632]      * |    477.05       7.65\n",
       "     [4]     chr1     [9534567, 9536367]      * |    1267.3        8.9\n",
       "     [5]     chr1     [9737790, 9739085]      * |    220.45       6.89\n",
       "     ...      ...                    ...    ... .       ...        ...\n",
       "  [8101]     chrX [160810573, 160811803]      * |    119.62       5.98\n",
       "  [8102]     chrX [162608021, 162609178]      * |     98.97       6.15\n",
       "  [8103]     chrX [163099167, 163100553]      * |    116.86       6.65\n",
       "  [8104]     chrX [163643293, 163644630]      * |     90.75       6.21\n",
       "  [8105]     chrX [166423675, 166426290]      * |     96.03       7.51\n",
       "  -------\n",
       "  seqinfo: 20 sequences from an unspecified genome; no seqlengths\n",
       "\n",
       "$Serum$Myc.2h\n",
       "GRanges object with 16384 ranges and 2 metadata columns:\n",
       "          seqnames                 ranges strand |    pvalue enrichment\n",
       "             <Rle>              <IRanges>  <Rle> | <numeric>  <numeric>\n",
       "      [1]     chr1     [4774901, 4776481]      * |    594.78       7.97\n",
       "      [2]     chr1     [4847093, 4849100]      * |    430.43       7.81\n",
       "      [3]     chr1     [5072555, 5073793]      * |    104.74       6.37\n",
       "      [4]     chr1     [7078176, 7079187]      * |    112.38       6.05\n",
       "      [5]     chr1     [7387163, 7388811]      * |    597.37       8.02\n",
       "      ...      ...                    ...    ... .       ...        ...\n",
       "  [16380]     chrX [165757829, 165759043]      * |     88.65       6.24\n",
       "  [16381]     chrX [166232203, 166233821]      * |    164.44       7.17\n",
       "  [16382]     chrX [166423533, 166426028]      * |    204.44          8\n",
       "  [16383]     chrX [166428708, 166439796]      * |    646.26       10.6\n",
       "  [16384]     chrX [166444425, 166447075]      * |     92.53       7.39\n",
       "  -------\n",
       "  seqinfo: 20 sequences from an unspecified genome; no seqlengths\n",
       "\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "peaksRef"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "$EuMyc\n",
       "$EuMyc$Myc.C1\n",
       "GRanges object with 6978 ranges and 2 metadata columns:\n",
       "            seqnames                 ranges strand |    pvalue enrichment\n",
       "               <Rle>              <IRanges>  <Rle> | <numeric>  <numeric>\n",
       "     [1]        chr1     [4775590, 4775590]      * |     82.65       4.72\n",
       "     [2]        chr1     [4775669, 4775669]      * |    330.87       6.63\n",
       "     [3]        chr1     [4797702, 4797702]      * |    176.67       6.25\n",
       "     [4]        chr1     [5073237, 5073237]      * |     91.73       5.27\n",
       "     [5]        chr1     [6204580, 6204580]      * |    114.56       5.59\n",
       "     ...         ...                    ...    ... .       ...        ...\n",
       "  [6974]        chrX [166435458, 166435458]      * |    704.97       9.73\n",
       "  [6975]        chrX [166437243, 166437243]      * |    174.08       7.58\n",
       "  [6976]        chrX [166504906, 166504906]      * |     80.75       5.35\n",
       "  [6977] chrX_random [   105976,    105976]      * |    150.52       7.42\n",
       "  [6978]        chrY [   427386,    427386]      * |     103.3       5.89\n",
       "  -------\n",
       "  seqinfo: 26 sequences from an unspecified genome; no seqlengths\n",
       "\n",
       "$EuMyc$Myc.P\n",
       "GRanges object with 31187 ranges and 2 metadata columns:\n",
       "          seqnames                 ranges strand |    pvalue enrichment\n",
       "             <Rle>              <IRanges>  <Rle> | <numeric>  <numeric>\n",
       "      [1]     chr1     [3755015, 3755015]      * |    167.78       6.25\n",
       "      [2]     chr1     [4313207, 4313207]      * |    102.36       5.96\n",
       "      [3]     chr1     [4775824, 4775824]      * |    304.92       7.19\n",
       "      [4]     chr1     [4797838, 4797838]      * |    486.35        7.7\n",
       "      [5]     chr1     [4831514, 4831514]      * |    109.67       4.78\n",
       "      ...      ...                    ...    ... .       ...        ...\n",
       "  [31183]     chrX [163766631, 163766631]      * |     90.93       6.19\n",
       "  [31184]     chrX [163819823, 163819823]      * |    270.61        7.8\n",
       "  [31185]     chrX [165111552, 165111552]      * |      95.9       6.16\n",
       "  [31186]     chrX [165758330, 165758330]      * |    189.52       6.55\n",
       "  [31187]     chrX [166444985, 166444985]      * |     80.98       7.74\n",
       "  -------\n",
       "  seqinfo: 27 sequences from an unspecified genome; no seqlengths\n",
       "\n",
       "$EuMyc$Myc.T27\n",
       "GRanges object with 29252 ranges and 2 metadata columns:\n",
       "             seqnames             ranges strand |    pvalue enrichment\n",
       "                <Rle>          <IRanges>  <Rle> | <numeric>  <numeric>\n",
       "      [1]        chr1 [4661807, 4661807]      * |    134.17        5.8\n",
       "      [2]        chr1 [4775563, 4775563]      * |   2141.12          9\n",
       "      [3]        chr1 [4797641, 4797641]      * |    1002.3       7.84\n",
       "      [4]        chr1 [4847794, 4847794]      * |   1075.53       8.18\n",
       "      [5]        chr1 [4909159, 4909159]      * |     85.69       5.03\n",
       "      ...         ...                ...    ... .       ...        ...\n",
       "  [29248] chrX_random [  39243,   39243]      * |    150.01       5.68\n",
       "  [29249]        chrY [ 346773,  346773]      * |    187.19       5.81\n",
       "  [29250]        chrY [ 582157,  582157]      * |    270.15       6.65\n",
       "  [29251]        chrY [ 622794,  622794]      * |    417.19       6.86\n",
       "  [29252]        chrY [2612327, 2612327]      * |     89.81       5.59\n",
       "  -------\n",
       "  seqinfo: 28 sequences from an unspecified genome; no seqlengths\n",
       "\n",
       "$EuMyc$Myc.T29\n",
       "GRanges object with 27347 ranges and 2 metadata columns:\n",
       "             seqnames                 ranges strand |    pvalue enrichment\n",
       "                <Rle>              <IRanges>  <Rle> | <numeric>  <numeric>\n",
       "      [1]        chr1     [4775552, 4775552]      * |   3137.54       9.63\n",
       "      [2]        chr1     [4797574, 4797574]      * |    980.87       8.09\n",
       "      [3]        chr1     [4847795, 4847795]      * |   1368.62       8.71\n",
       "      [4]        chr1     [5073249, 5073249]      * |    772.36       8.16\n",
       "      [5]        chr1     [6204707, 6204707]      * |    791.23       7.91\n",
       "      ...         ...                    ...    ... .       ...        ...\n",
       "  [27343]        chrX [166318908, 166318908]      * |    106.43       6.84\n",
       "  [27344]        chrX [166424871, 166424871]      * |    460.36        9.4\n",
       "  [27345]        chrX [166436125, 166436125]      * |       508       9.83\n",
       "  [27346]        chrX [166444881, 166444881]      * |     90.83       7.09\n",
       "  [27347] chrX_random [    39235,     39235]      * |    186.03       6.51\n",
       "  -------\n",
       "  seqinfo: 27 sequences from an unspecified genome; no seqlengths\n",
       "\n",
       "$EuMyc$Myc.T31\n",
       "GRanges object with 34142 ranges and 2 metadata columns:\n",
       "             seqnames                 ranges strand |    pvalue enrichment\n",
       "                <Rle>              <IRanges>  <Rle> | <numeric>  <numeric>\n",
       "      [1]        chr1     [4775619, 4775619]      * |   1537.76       8.89\n",
       "      [2]        chr1     [4797829, 4797829]      * |    648.33       7.84\n",
       "      [3]        chr1     [4847809, 4847809]      * |    587.15       7.79\n",
       "      [4]        chr1     [5073217, 5073217]      * |    387.55       7.42\n",
       "      [5]        chr1     [6204698, 6204698]      * |    370.23       7.19\n",
       "      ...         ...                    ...    ... .       ...        ...\n",
       "  [34138]        chrX [165758346, 165758346]      * |    737.73       7.51\n",
       "  [34139]        chrX [166312497, 166312497]      * |     82.58       5.03\n",
       "  [34140]        chrX [166319426, 166319426]      * |    118.69       5.96\n",
       "  [34141]        chrX [166436132, 166436132]      * |    464.28       9.06\n",
       "  [34142] chrX_random [    38911,     38911]      * |    232.24       6.31\n",
       "  -------\n",
       "  seqinfo: 27 sequences from an unspecified genome; no seqlengths\n",
       "\n",
       "\n",
       "$mycER\n",
       "$mycER$Myc.0h\n",
       "GRanges object with 18000 ranges and 2 metadata columns:\n",
       "          seqnames                 ranges strand |    pvalue enrichment\n",
       "             <Rle>              <IRanges>  <Rle> | <numeric>  <numeric>\n",
       "      [1]     chr1     [4775553, 4775553]      * |    385.78       7.71\n",
       "      [2]     chr1     [4847801, 4847801]      * |    359.27        7.7\n",
       "      [3]     chr1     [5073109, 5073109]      * |    132.11       6.05\n",
       "      [4]     chr1     [6270298, 6270298]      * |    189.25       7.25\n",
       "      [5]     chr1     [6278378, 6278378]      * |     93.91       6.92\n",
       "      ...      ...                    ...    ... .       ...        ...\n",
       "  [17996]     chrX [165112048, 165112048]      * |     84.26        5.8\n",
       "  [17997]     chrX [165758304, 165758304]      * |    119.52       6.33\n",
       "  [17998]     chrX [166232866, 166232866]      * |    280.77       7.79\n",
       "  [17999]     chrX [166317434, 166317434]      * |     86.02       6.05\n",
       "  [18000]     chrX [166319369, 166319369]      * |     98.95        6.7\n",
       "  -------\n",
       "  seqinfo: 25 sequences from an unspecified genome; no seqlengths\n",
       "\n",
       "$mycER$Myc.4h\n",
       "GRanges object with 27693 ranges and 2 metadata columns:\n",
       "             seqnames                 ranges strand |    pvalue enrichment\n",
       "                <Rle>              <IRanges>  <Rle> | <numeric>  <numeric>\n",
       "      [1]        chr1     [3505201, 3505201]      * |    199.88       6.86\n",
       "      [2]        chr1     [4775572, 4775572]      * |   1161.65       8.76\n",
       "      [3]        chr1     [4847733, 4847733]      * |   1836.41       9.22\n",
       "      [4]        chr1     [5005504, 5005504]      * |    124.35        6.2\n",
       "      [5]        chr1     [5012974, 5012974]      * |    139.49       6.18\n",
       "      ...         ...                    ...    ... .       ...        ...\n",
       "  [27689]        chrX [166319860, 166319860]      * |    149.51       7.57\n",
       "  [27690]        chrX [166337663, 166337663]      * |    100.01       6.55\n",
       "  [27691]        chrX [166424966, 166424966]      * |    453.25       8.87\n",
       "  [27692]        chrX [166438944, 166438944]      * |    993.56       10.1\n",
       "  [27693] chrX_random [    39239,     39239]      * |    117.11       6.14\n",
       "  -------\n",
       "  seqinfo: 27 sequences from an unspecified genome; no seqlengths\n",
       "\n",
       "\n",
       "$P493\n",
       "$P493$Myc.t0h\n",
       "GRanges object with 791 ranges and 2 metadata columns:\n",
       "        seqnames               ranges strand |    pvalue enrichment\n",
       "           <Rle>            <IRanges>  <Rle> | <numeric>  <numeric>\n",
       "    [1]     chr1   [  10172,   10172]      * |    354.81       7.73\n",
       "    [2]     chr1   [ 567568,  567568]      * |    178.59       8.02\n",
       "    [3]     chr1   [ 569927,  569927]      * |    109.47       7.58\n",
       "    [4]     chr1   [ 714029,  714029]      * |      90.8       6.16\n",
       "    [5]     chr1   [1342700, 1342700]      * |    131.91       6.79\n",
       "    ...      ...                  ...    ... .       ...        ...\n",
       "  [787]     chrY [13842706, 13842706]      * |    247.68        8.4\n",
       "  [788]     chrY [13846459, 13846459]      * |    174.75       6.77\n",
       "  [789]     chrY [13860036, 13860036]      * |    233.22       6.68\n",
       "  [790]     chrY [13868401, 13868401]      * |    314.64       7.12\n",
       "  [791]     chrY [13869132, 13869132]      * |    110.05       7.07\n",
       "  -------\n",
       "  seqinfo: 25 sequences from an unspecified genome; no seqlengths\n",
       "\n",
       "$P493$Myc.t1h\n",
       "GRanges object with 8666 ranges and 2 metadata columns:\n",
       "         seqnames               ranges strand |    pvalue enrichment\n",
       "            <Rle>            <IRanges>  <Rle> | <numeric>  <numeric>\n",
       "     [1]     chr1     [ 10199,  10199]      * |    140.83       6.32\n",
       "     [2]     chr1     [567548, 567548]      * |    118.13       7.19\n",
       "     [3]     chr1     [714164, 714164]      * |    241.78       7.12\n",
       "     [4]     chr1     [842376, 842376]      * |    165.82       6.72\n",
       "     [5]     chr1     [936260, 936260]      * |     95.39       6.07\n",
       "     ...      ...                  ...    ... .       ...        ...\n",
       "  [8662]     chrY [13846461, 13846461]      * |     81.41       6.59\n",
       "  [8663]     chrY [13860028, 13860028]      * |    183.73       6.79\n",
       "  [8664]     chrY [13866951, 13866951]      * |    194.03       6.25\n",
       "  [8665]     chrY [13868407, 13868407]      * |    212.08       7.16\n",
       "  [8666]     chrY [13869114, 13869114]      * |     85.52       7.19\n",
       "  -------\n",
       "  seqinfo: 25 sequences from an unspecified genome; no seqlengths\n",
       "\n",
       "$P493$Myc.t24h\n",
       "GRanges object with 18257 ranges and 2 metadata columns:\n",
       "          seqnames               ranges strand |    pvalue enrichment\n",
       "             <Rle>            <IRanges>  <Rle> | <numeric>  <numeric>\n",
       "      [1]     chr1     [567559, 567559]      * |    151.08        8.1\n",
       "      [2]     chr1     [569901, 569901]      * |    173.95       8.28\n",
       "      [3]     chr1     [713956, 713956]      * |     616.6       8.15\n",
       "      [4]     chr1     [740322, 740322]      * |    237.72       6.68\n",
       "      [5]     chr1     [762674, 762674]      * |    100.55        6.1\n",
       "      ...      ...                  ...    ... .       ...        ...\n",
       "  [18253]     chrY [13846505, 13846505]      * |    154.71       6.65\n",
       "  [18254]     chrY [13866969, 13866969]      * |    142.29       6.51\n",
       "  [18255]     chrY [13868410, 13868410]      * |    169.52       7.21\n",
       "  [18256]     chrY [13869126, 13869126]      * |     81.59       6.43\n",
       "  [18257]     chrY [58996579, 58996579]      * |     93.15       6.65\n",
       "  -------\n",
       "  seqinfo: 25 sequences from an unspecified genome; no seqlengths\n",
       "\n",
       "$P493$Myc.growing\n",
       "GRanges object with 26146 ranges and 2 metadata columns:\n",
       "          seqnames               ranges strand |    pvalue enrichment\n",
       "             <Rle>            <IRanges>  <Rle> | <numeric>  <numeric>\n",
       "      [1]     chr1     [714301, 714301]      * |    655.72       7.91\n",
       "      [2]     chr1     [740303, 740303]      * |    482.65       7.24\n",
       "      [3]     chr1     [840195, 840195]      * |    272.05       6.84\n",
       "      [4]     chr1     [840772, 840772]      * |    563.81        7.7\n",
       "      [5]     chr1     [842202, 842202]      * |    408.37       7.34\n",
       "      ...      ...                  ...    ... .       ...        ...\n",
       "  [26142]     chrY [13843926, 13843926]      * |     81.01       5.92\n",
       "  [26143]     chrY [13860032, 13860032]      * |    113.14       6.06\n",
       "  [26144]     chrY [13866956, 13866956]      * |    136.95       5.43\n",
       "  [26145]     chrY [13868376, 13868376]      * |      82.3       5.95\n",
       "  [26146]     chrY [21154803, 21154803]      * |     100.8       5.64\n",
       "  -------\n",
       "  seqinfo: 25 sequences from an unspecified genome; no seqlengths\n",
       "\n",
       "$P493$Myc.low\n",
       "GRanges object with 4309 ranges and 2 metadata columns:\n",
       "         seqnames               ranges strand |    pvalue enrichment\n",
       "            <Rle>            <IRanges>  <Rle> | <numeric>  <numeric>\n",
       "     [1]     chr1   [ 714140,  714140]      * |     81.46       5.52\n",
       "     [2]     chr1   [ 740308,  740308]      * |    115.31       5.75\n",
       "     [3]     chr1   [ 936196,  936196]      * |     98.31       5.78\n",
       "     [4]     chr1   [ 968604,  968604]      * |     80.04       5.25\n",
       "     [5]     chr1   [1186534, 1186534]      * |    253.86       6.74\n",
       "     ...      ...                  ...    ... .       ...        ...\n",
       "  [4305]     chrY [13846476, 13846476]      * |    200.16       6.45\n",
       "  [4306]     chrY [13860036, 13860036]      * |    145.16       6.34\n",
       "  [4307]     chrY [13866966, 13866966]      * |    142.26       6.17\n",
       "  [4308]     chrY [13868388, 13868388]      * |    224.58        6.7\n",
       "  [4309]     chrY [13869128, 13869128]      * |     95.42       6.56\n",
       "  -------\n",
       "  seqinfo: 25 sequences from an unspecified genome; no seqlengths\n",
       "\n",
       "\n",
       "$Serum\n",
       "$Serum$Myc.0h\n",
       "GRanges object with 5832 ranges and 2 metadata columns:\n",
       "         seqnames                 ranges strand |    pvalue enrichment\n",
       "            <Rle>              <IRanges>  <Rle> | <numeric>  <numeric>\n",
       "     [1]     chr1     [4775692, 4775692]      * |    135.08       6.79\n",
       "     [2]     chr1     [4847810, 4847810]      * |    146.81       6.98\n",
       "     [3]     chr1     [6270371, 6270371]      * |    125.49       6.58\n",
       "     [4]     chr1     [7387634, 7387634]      * |    276.63       7.63\n",
       "     [5]     chr1     [9535091, 9535091]      * |    340.82       8.02\n",
       "     ...      ...                    ...    ... .       ...        ...\n",
       "  [5828]     chrX [156576231, 156576231]      * |    133.07       7.12\n",
       "  [5829]     chrX [156939975, 156939975]      * |    158.89       7.02\n",
       "  [5830]     chrX [159266965, 159266965]      * |    153.35       6.54\n",
       "  [5831]     chrX [160811748, 160811748]      * |     83.71       6.27\n",
       "  [5832]     chrX [162947940, 162947940]      * |    161.91       6.92\n",
       "  -------\n",
       "  seqinfo: 20 sequences from an unspecified genome; no seqlengths\n",
       "\n",
       "$Serum$Myc.1h\n",
       "GRanges object with 8105 ranges and 2 metadata columns:\n",
       "         seqnames                 ranges strand |    pvalue enrichment\n",
       "            <Rle>              <IRanges>  <Rle> | <numeric>  <numeric>\n",
       "     [1]     chr1     [4775441, 4775441]      * |    156.43       6.55\n",
       "     [2]     chr1     [4847771, 4847771]      * |     136.6       6.38\n",
       "     [3]     chr1     [7387903, 7387903]      * |    477.05       7.65\n",
       "     [4]     chr1     [9535482, 9535482]      * |    1267.3        8.9\n",
       "     [5]     chr1     [9738486, 9738486]      * |    220.45       6.89\n",
       "     ...      ...                    ...    ... .       ...        ...\n",
       "  [8101]     chrX [160811272, 160811272]      * |    119.62       5.98\n",
       "  [8102]     chrX [162608469, 162608469]      * |     98.97       6.15\n",
       "  [8103]     chrX [163099875, 163099875]      * |    116.86       6.65\n",
       "  [8104]     chrX [163643906, 163643906]      * |     90.75       6.21\n",
       "  [8105]     chrX [166424959, 166424959]      * |     96.03       7.51\n",
       "  -------\n",
       "  seqinfo: 20 sequences from an unspecified genome; no seqlengths\n",
       "\n",
       "$Serum$Myc.2h\n",
       "GRanges object with 16384 ranges and 2 metadata columns:\n",
       "          seqnames                 ranges strand |    pvalue enrichment\n",
       "             <Rle>              <IRanges>  <Rle> | <numeric>  <numeric>\n",
       "      [1]     chr1     [4775884, 4775884]      * |    594.78       7.97\n",
       "      [2]     chr1     [4847780, 4847780]      * |    430.43       7.81\n",
       "      [3]     chr1     [5073060, 5073060]      * |    104.74       6.37\n",
       "      [4]     chr1     [7078859, 7078859]      * |    112.38       6.05\n",
       "      [5]     chr1     [7388048, 7388048]      * |    597.37       8.02\n",
       "      ...      ...                    ...    ... .       ...        ...\n",
       "  [16380]     chrX [165758326, 165758326]      * |     88.65       6.24\n",
       "  [16381]     chrX [166233233, 166233233]      * |    164.44       7.17\n",
       "  [16382]     chrX [166425146, 166425146]      * |    204.44          8\n",
       "  [16383]     chrX [166439015, 166439015]      * |    646.26       10.6\n",
       "  [16384]     chrX [166444774, 166444774]      * |     92.53       7.39\n",
       "  -------\n",
       "  seqinfo: 20 sequences from an unspecified genome; no seqlengths\n",
       "\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "mycSummitsRef"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "GRanges object with 16026 ranges and 0 metadata columns:\n",
       "              seqnames                 ranges strand\n",
       "                 <Rle>              <IRanges>  <Rle>\n",
       "      [1]         chr1 [ 36568608,  36569851]      *\n",
       "      [2]         chr1 [ 82313020,  82313491]      *\n",
       "      [3]         chr1 [169213895, 169215283]      *\n",
       "      [4]         chr1 [  3521705,   3521924]      *\n",
       "      [5]         chr1 [  3660700,   3661155]      *\n",
       "      ...          ...                    ...    ...\n",
       "  [16022] chrUn_random     [3233054, 3236205]      *\n",
       "  [16023] chrUn_random     [3633779, 3634031]      *\n",
       "  [16024] chrUn_random     [3759945, 3760566]      *\n",
       "  [16025] chrUn_random     [3765218, 3765460]      *\n",
       "  [16026] chrUn_random     [4340137, 4340925]      *\n",
       "  -------\n",
       "  seqinfo: 28 sequences from an unspecified genome; no seqlengths"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "CGIgr"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<thead><tr><th></th><th scope=col>C_1</th><th scope=col>C_3</th><th scope=col>C_4</th><th scope=col>C_6</th><th scope=col>P_2</th><th scope=col>P_3</th><th scope=col>P_4</th><th scope=col>P_5</th><th scope=col>T_1</th><th scope=col>T_2</th><th scope=col>...</th><th scope=col>qvalue.CP</th><th scope=col>log2FC.CT1</th><th scope=col>log2FCvsn.CT1</th><th scope=col>qvalue.CT1</th><th scope=col>log2FC.CT2</th><th scope=col>log2FCvsn.CT2</th><th scope=col>qvalue.CT2</th><th scope=col>log2FC.CT3</th><th scope=col>log2FCvsn.CT3</th><th scope=col>qvalue.CT3</th></tr></thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>0610007C21Rik</th><td>300.42      </td><td>318.17      </td><td>245.12      </td><td>327.45      </td><td>309.13      </td><td>273.16      </td><td>237.94      </td><td>317.90      </td><td>273.52      </td><td> 398.82     </td><td>...         </td><td>9.417318e-01</td><td>-0.122658393</td><td>-0.15952224 </td><td>1.000000e+00</td><td> 0.42142989 </td><td> 0.61682006 </td><td>0.393704824 </td><td> 0.72776314 </td><td> 1.08599078 </td><td>0.0849638225</td></tr>\n",
       "\t<tr><th scope=row>0610007L01Rik</th><td>207.09      </td><td>210.95      </td><td>220.58      </td><td>267.45      </td><td>217.25      </td><td>200.22      </td><td>281.38      </td><td>283.03      </td><td>231.03      </td><td> 306.11     </td><td>...         </td><td>7.878284e-01</td><td> 0.028450720</td><td> 0.06106108 </td><td>1.000000e+00</td><td> 0.43445718 </td><td> 0.77974700 </td><td>0.391081390 </td><td> 0.59935996 </td><td> 1.07715415 </td><td>0.1662041565</td></tr>\n",
       "\t<tr><th scope=row>0610007P08Rik</th><td>134.32      </td><td>112.85      </td><td>126.30      </td><td>116.87      </td><td>146.14      </td><td>130.28      </td><td>147.10      </td><td> 97.46      </td><td>162.87      </td><td>  90.20     </td><td>...         </td><td>8.535726e-01</td><td> 0.409968779</td><td> 0.71039586 </td><td>3.821038e-01</td><td>-0.44262105 </td><td>-0.71991486 </td><td>0.557661867 </td><td> 0.09529604 </td><td> 0.16598341 </td><td>0.8933977701</td></tr>\n",
       "\t<tr><th scope=row>0610007P14Rik</th><td>220.19      </td><td>203.60      </td><td>240.95      </td><td>245.36      </td><td>253.78      </td><td>261.80      </td><td>258.35      </td><td>297.25      </td><td>309.32      </td><td> 382.49     </td><td>...         </td><td>5.572482e-01</td><td> 0.443088932</td><td> 0.63473511 </td><td>4.169869e-01</td><td> 0.74940682 </td><td> 1.09910154 </td><td>0.104290139 </td><td> 0.63176944 </td><td> 0.91806754 </td><td>0.1822683985</td></tr>\n",
       "\t<tr><th scope=row>0610007P22Rik</th><td> 78.98      </td><td>114.13      </td><td> 90.50      </td><td> 95.85      </td><td> 88.88      </td><td> 96.54      </td><td>113.98      </td><td>112.99      </td><td>140.49      </td><td> 147.13     </td><td>...         </td><td>8.120777e-01</td><td> 0.566465434</td><td> 0.65768003 </td><td>3.246861e-01</td><td> 0.63312517 </td><td> 0.73848570 </td><td>0.246905230 </td><td> 0.26233311 </td><td> 0.30157112 </td><td>0.6610278272</td></tr>\n",
       "\t<tr><th scope=row>0610008F07Rik</th><td>  0.00      </td><td>  0.00      </td><td>  0.00      </td><td>  0.00      </td><td>  0.00      </td><td>  0.00      </td><td>  0.00      </td><td>  0.00      </td><td>  0.00      </td><td>   0.00     </td><td>...         </td><td>          NA</td><td>         NaN</td><td> 0.00000000 </td><td>          NA</td><td>        NaN </td><td> 0.00000000 </td><td>         NA </td><td>        NaN </td><td> 0.00000000 </td><td>          NA</td></tr>\n",
       "\t<tr><th scope=row>0610009B22Rik</th><td> 42.55      </td><td> 79.63      </td><td> 35.92      </td><td> 66.90      </td><td> 71.29      </td><td> 92.66      </td><td> 80.27      </td><td>114.18      </td><td> 56.12      </td><td>  88.03     </td><td>...         </td><td>8.899602e-02</td><td>-0.003342027</td><td> 0.04917621 </td><td>9.905365e-01</td><td> 0.64630459 </td><td> 0.64565648 </td><td>0.336251742 </td><td> 0.97285105 </td><td> 0.96878604 </td><td>0.0990259846</td></tr>\n",
       "\t<tr><th scope=row>0610009D07Rik</th><td>533.63      </td><td>549.03      </td><td>348.60      </td><td>476.57      </td><td>534.86      </td><td>697.28      </td><td>703.17      </td><td>812.13      </td><td>583.43      </td><td>1037.25     </td><td>...         </td><td>1.480926e-02</td><td> 0.290685645</td><td> 0.50412608 </td><td>4.903655e-01</td><td> 1.12082967 </td><td> 1.95063835 </td><td>0.001107939 </td><td> 0.23678699 </td><td> 0.41441353 </td><td>0.5649703839</td></tr>\n",
       "\t<tr><th scope=row>0610009O20Rik</th><td>142.71      </td><td>124.71      </td><td>173.33      </td><td>140.49      </td><td>146.39      </td><td>164.01      </td><td>177.91      </td><td>182.31      </td><td>145.45      </td><td> 173.75     </td><td>...         </td><td>6.165242e-01</td><td> 0.001430952</td><td> 0.01342905 </td><td>1.000000e+00</td><td> 0.25793077 </td><td> 0.39644744 </td><td>0.677829395 </td><td> 0.16636690 </td><td> 0.25786454 </td><td>0.8368292328</td></tr>\n",
       "\t<tr><th scope=row>0610010B08Rik</th><td>  0.53      </td><td>  1.05      </td><td>  0.59      </td><td>  0.98      </td><td>  0.20      </td><td>  0.91      </td><td>  0.33      </td><td>  0.90      </td><td>  2.39      </td><td>   0.49     </td><td>...         </td><td>7.991074e-01</td><td> 1.601659360</td><td> 1.03301217 </td><td>1.412337e-01</td><td>-0.68644558 </td><td>-0.34649473 </td><td>1.000000000 </td><td>-0.61143788 </td><td>-0.30683543 </td><td>1.0000000000</td></tr>\n",
       "\t<tr><th scope=row>0610010F05Rik</th><td> 48.73      </td><td> 46.45      </td><td> 46.12      </td><td> 54.38      </td><td> 57.41      </td><td> 74.18      </td><td>119.09      </td><td> 83.28      </td><td>143.00      </td><td> 115.38     </td><td>...         </td><td>2.467096e-03</td><td> 1.547499287</td><td> 2.25768779 </td><td>3.195463e-05</td><td> 1.23793298 </td><td> 1.76035829 </td><td>0.002148796 </td><td> 1.37210275 </td><td> 1.97323466 </td><td>0.0003177189</td></tr>\n",
       "\t<tr><th scope=row>0610010K14Rik</th><td>139.07      </td><td>183.34      </td><td>155.09      </td><td>198.09      </td><td>153.72      </td><td>178.00      </td><td>251.22      </td><td>233.18      </td><td>189.16      </td><td> 170.52     </td><td>...         </td><td>4.907610e-01</td><td> 0.163482934</td><td> 0.20563523 </td><td>8.341150e-01</td><td> 0.01382476 </td><td> 0.02786449 </td><td>1.000000000 </td><td>-0.49083574 </td><td>-0.53639181 </td><td>0.6099562171</td></tr>\n",
       "\t<tr><th scope=row>0610010O12Rik</th><td>  8.12      </td><td> 14.05      </td><td>  4.50      </td><td> 10.43      </td><td> 13.83      </td><td>  6.98      </td><td>  2.55      </td><td>  4.55      </td><td>  2.03      </td><td>   3.73     </td><td>...         </td><td>7.435859e-01</td><td>-2.194377903</td><td>-1.12633670 </td><td>8.648071e-01</td><td>-1.31255784 </td><td>-0.68852920 </td><td>0.961124861 </td><td> 0.34741235 </td><td> 0.27590322 </td><td>0.7952688613</td></tr>\n",
       "\t<tr><th scope=row>0610011F06Rik</th><td>204.06      </td><td>201.76      </td><td>192.54      </td><td>168.32      </td><td>198.37      </td><td>168.88      </td><td>238.86      </td><td>272.29      </td><td>199.22      </td><td> 252.04     </td><td>...         </td><td>5.730909e-01</td><td> 0.055749917</td><td> 0.06967514 </td><td>9.512541e-01</td><td> 0.39504274 </td><td> 0.48754054 </td><td>0.461744128 </td><td>-0.03114370 </td><td>-0.03312322 </td><td>1.0000000000</td></tr>\n",
       "\t<tr><th scope=row>0610011L14Rik</th><td>133.58      </td><td>148.38      </td><td>144.65      </td><td>177.28      </td><td>136.48      </td><td>136.41      </td><td>149.82      </td><td>163.78      </td><td>119.23      </td><td> 135.54     </td><td>...         </td><td>9.180832e-01</td><td>-0.340562836</td><td>-0.52566824 </td><td>6.834677e-01</td><td>-0.15563093 </td><td>-0.23803711 </td><td>0.960638998 </td><td>-0.33971046 </td><td>-0.52436127 </td><td>0.6672814070</td></tr>\n",
       "\t<tr><th scope=row>0610012G03Rik</th><td> 74.73      </td><td>125.85      </td><td> 61.87      </td><td> 80.75      </td><td>102.53      </td><td> 86.76      </td><td> 95.54      </td><td>118.81      </td><td>109.84      </td><td>  76.28     </td><td>...         </td><td>4.812137e-01</td><td> 0.356306204</td><td> 0.45366946 </td><td>4.818037e-01</td><td>-0.16975144 </td><td>-0.14397236 </td><td>1.000000000 </td><td> 0.67330953 </td><td> 0.84325916 </td><td>0.1551004987</td></tr>\n",
       "\t<tr><th scope=row>0610012H03Rik</th><td>  0.00      </td><td>  0.00      </td><td>  0.00      </td><td>  0.00      </td><td>  1.14      </td><td>  0.65      </td><td>  0.48      </td><td>  0.00      </td><td>  0.00      </td><td>   0.70     </td><td>...         </td><td>4.062349e-01</td><td>         NaN</td><td> 0.00000000 </td><td>          NA</td><td>        Inf </td><td> 1.59790970 </td><td>0.394873980 </td><td>        NaN </td><td> 0.00000000 </td><td>          NA</td></tr>\n",
       "\t<tr><th scope=row>0610030E20Rik</th><td>246.86      </td><td>210.58      </td><td>244.15      </td><td>231.65      </td><td>202.98      </td><td>202.63      </td><td>208.03      </td><td>157.29      </td><td>180.58      </td><td> 161.82     </td><td>...         </td><td>3.109693e-01</td><td>-0.369645846</td><td>-0.64797212 </td><td>6.379189e-01</td><td>-0.52787395 </td><td>-0.92074926 </td><td>0.420538656 </td><td>-0.68168494 </td><td>-1.18148698 </td><td>0.2502932155</td></tr>\n",
       "\t<tr><th scope=row>0610031J06Rik</th><td>235.78      </td><td>247.11      </td><td>270.53      </td><td>289.51      </td><td>224.81      </td><td>197.12      </td><td>164.59      </td><td>222.56      </td><td>214.95      </td><td> 109.86     </td><td>...         </td><td>1.909861e-01</td><td>-0.278596389</td><td>-0.40585384 </td><td>8.012731e-01</td><td>-1.24688090 </td><td>-1.68629376 </td><td>0.030847570 </td><td>-0.04476128 </td><td>-0.06205471 </td><td>1.0000000000</td></tr>\n",
       "\t<tr><th scope=row>0610037L13Rik</th><td>120.72      </td><td>150.94      </td><td>153.03      </td><td>156.60      </td><td>157.36      </td><td>118.41      </td><td>129.79      </td><td>166.83      </td><td>106.50      </td><td> 114.60     </td><td>...         </td><td>9.939000e-01</td><td>-0.448326263</td><td>-0.55404931 </td><td>6.185147e-01</td><td>-0.34261019 </td><td>-0.42593111 </td><td>0.735352367 </td><td>-0.56716278 </td><td>-0.69506108 </td><td>0.4597216657</td></tr>\n",
       "\t<tr><th scope=row>0610037P05Rik</th><td>208.97      </td><td>204.10      </td><td>197.03      </td><td>211.32      </td><td>243.63      </td><td>191.90      </td><td>188.58      </td><td>250.09      </td><td>179.76      </td><td> 139.84     </td><td>...         </td><td>8.463243e-01</td><td>-0.192066854</td><td>-0.26320221 </td><td>9.898461e-01</td><td>-0.55435147 </td><td>-0.73609432 </td><td>0.481329466 </td><td>-0.73434086 </td><td>-0.95940851 </td><td>0.2974374726</td></tr>\n",
       "\t<tr><th scope=row>0610038B21Rik</th><td>  0.79      </td><td>  3.14      </td><td>  1.17      </td><td>  2.33      </td><td>  3.00      </td><td>  4.09      </td><td>  2.99      </td><td>  3.11      </td><td>  9.50      </td><td>  10.21     </td><td>...         </td><td>4.907610e-01</td><td> 2.354623889</td><td> 1.57269091 </td><td>2.772354e-02</td><td> 2.45883637 </td><td> 1.64794881 </td><td>0.018293855 </td><td> 0.31145165 </td><td> 0.26374880 </td><td>0.7537873673</td></tr>\n",
       "\t<tr><th scope=row>0610039K10Rik</th><td>369.63      </td><td>437.06      </td><td>366.86      </td><td>239.55      </td><td>206.66      </td><td>243.30      </td><td>218.87      </td><td>167.39      </td><td>258.65      </td><td> 282.69     </td><td>...         </td><td>9.555152e-03</td><td>-0.449816317</td><td>-0.59038159 </td><td>6.738087e-01</td><td>-0.32155836 </td><td>-0.41703417 </td><td>0.858497947 </td><td>-0.79909169 </td><td>-1.04216158 </td><td>0.2730506154</td></tr>\n",
       "\t<tr><th scope=row>0610040B10Rik</th><td> 21.96      </td><td> 32.57      </td><td> 26.36      </td><td> 16.12      </td><td> 16.62      </td><td> 14.15      </td><td> 17.27      </td><td> 20.01      </td><td> 13.70      </td><td>  12.62     </td><td>...         </td><td>6.552081e-01</td><td>-0.823887644</td><td>-0.51049225 </td><td>1.000000e+00</td><td>-0.94206758 </td><td>-0.58348453 </td><td>0.893400570 </td><td> 0.13294011 </td><td> 0.11797629 </td><td>0.9044876130</td></tr>\n",
       "\t<tr><th scope=row>0610040J01Rik</th><td>  3.88      </td><td>  5.11      </td><td>  5.25      </td><td>  8.53      </td><td>  6.35      </td><td>  7.21      </td><td>  4.06      </td><td>  2.54      </td><td> 41.27      </td><td>   2.97     </td><td>...         </td><td>7.677246e-01</td><td> 2.858036704</td><td> 2.36489744 </td><td>7.262926e-06</td><td>-0.93821514 </td><td>-0.57260694 </td><td>0.776787852 </td><td> 1.20718188 </td><td> 0.91837655 </td><td>0.2086258283</td></tr>\n",
       "\t<tr><th scope=row>0910001L09Rik</th><td>265.84      </td><td>288.72      </td><td>281.83      </td><td>296.15      </td><td>213.67      </td><td>235.43      </td><td>168.04      </td><td>243.30      </td><td>191.10      </td><td> 277.41     </td><td>...         </td><td>2.203561e-01</td><td>-0.567121735</td><td>-0.71967492 </td><td>4.858289e-01</td><td>-0.02945569 </td><td>-0.03821356 </td><td>1.000000000 </td><td>-1.92188013 </td><td>-2.16617015 </td><td>0.0019413521</td></tr>\n",
       "\t<tr><th scope=row>1100001G20Rik</th><td> 88.25      </td><td>111.80      </td><td> 30.56      </td><td> 80.95      </td><td>240.00      </td><td>357.70      </td><td>169.97      </td><td>335.60      </td><td> 16.52      </td><td>  83.70     </td><td>...         </td><td>4.176561e-09</td><td>-2.237512048</td><td>-1.53721873 </td><td>1.206218e-01</td><td> 0.10373963 </td><td> 0.18408681 </td><td>0.815898469 </td><td> 0.83479293 </td><td> 0.83249702 </td><td>0.1708197164</td></tr>\n",
       "\t<tr><th scope=row>1110001A16Rik</th><td> 55.89      </td><td> 37.89      </td><td> 52.49      </td><td> 50.97      </td><td> 72.50      </td><td> 50.06      </td><td> 57.24      </td><td> 55.07      </td><td> 63.75      </td><td>  63.88     </td><td>...         </td><td>6.552081e-01</td><td> 0.370650335</td><td> 0.38524900 </td><td>5.881720e-01</td><td> 0.37348580 </td><td> 0.38820043 </td><td>0.573125809 </td><td>-0.08755941 </td><td>-0.07358787 </td><td>1.0000000000</td></tr>\n",
       "\t<tr><th scope=row>1110001J03Rik</th><td>167.09      </td><td>184.17      </td><td> 93.07      </td><td>148.74      </td><td>172.04      </td><td>216.45      </td><td>184.32      </td><td>218.11      </td><td>117.57      </td><td> 168.78     </td><td>...         </td><td>2.730091e-01</td><td>-0.334725282</td><td>-0.27668814 </td><td>1.000000e+00</td><td> 0.18691921 </td><td> 0.21315967 </td><td>0.718532586 </td><td> 0.34558484 </td><td> 0.37025581 </td><td>0.5329533971</td></tr>\n",
       "\t<tr><th scope=row>1110002B05Rik</th><td>268.54      </td><td>255.88      </td><td>194.43      </td><td>238.68      </td><td>291.73      </td><td>383.66      </td><td>443.65      </td><td>324.53      </td><td>334.39      </td><td> 326.84     </td><td>...         </td><td>1.263570e-02</td><td> 0.482220986</td><td> 0.73280727 </td><td>2.855344e-01</td><td> 0.44925705 </td><td> 0.68167993 </td><td>0.321819917 </td><td> 1.17830251 </td><td> 1.86969136 </td><td>0.0009306154</td></tr>\n",
       "\t<tr><th scope=row>...</th><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>   </td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td></tr>\n",
       "\t<tr><th scope=row>Zscan30</th><td>  8.97      </td><td> 19.52      </td><td> 15.91      </td><td> 17.12      </td><td> 16.30      </td><td> 24.67      </td><td> 24.83      </td><td> 14.09      </td><td> 23.29      </td><td> 16.51      </td><td>...         </td><td>7.129923e-01</td><td> 0.59849256 </td><td> 0.45285540 </td><td>5.446778e-01</td><td> 0.10180099 </td><td> 0.10158047 </td><td>9.132241e-01</td><td> 1.17680869 </td><td> 0.88789237 </td><td>1.891014e-01</td></tr>\n",
       "\t<tr><th scope=row>Zscan4a</th><td>  0.00      </td><td>  0.00      </td><td>  0.40      </td><td>  0.00      </td><td>  0.00      </td><td>  0.00      </td><td>  0.00      </td><td>  0.00      </td><td>  0.00      </td><td>  0.00      </td><td>...         </td><td>6.552081e-01</td><td>       -Inf </td><td>-0.36422609 </td><td>1.000000e+00</td><td>       -Inf </td><td>-0.36422609 </td><td>1.000000e+00</td><td>       -Inf </td><td>-0.36422609 </td><td>1.000000e+00</td></tr>\n",
       "\t<tr><th scope=row>Zscan4b</th><td>  0.00      </td><td>  0.00      </td><td>  0.00      </td><td>  0.00      </td><td>  0.00      </td><td>  0.00      </td><td>  0.00      </td><td>  0.00      </td><td>  0.00      </td><td>  0.00      </td><td>...         </td><td>          NA</td><td>        NaN </td><td> 0.00000000 </td><td>          NA</td><td>        NaN </td><td> 0.00000000 </td><td>          NA</td><td>        NaN </td><td> 0.00000000 </td><td>          NA</td></tr>\n",
       "\t<tr><th scope=row>Zscan4c</th><td>  0.00      </td><td>  0.00      </td><td>  0.00      </td><td>  0.00      </td><td>  0.00      </td><td>  0.00      </td><td>  0.00      </td><td>  0.00      </td><td>  0.00      </td><td>  0.00      </td><td>...         </td><td>          NA</td><td>        NaN </td><td> 0.00000000 </td><td>          NA</td><td>        NaN </td><td> 0.00000000 </td><td>          NA</td><td>        NaN </td><td> 0.00000000 </td><td>          NA</td></tr>\n",
       "\t<tr><th scope=row>Zscan4d</th><td>  0.00      </td><td>  0.00      </td><td>  0.00      </td><td>  0.00      </td><td>  0.00      </td><td>  0.00      </td><td>  0.00      </td><td>  0.00      </td><td>  0.00      </td><td>  0.00      </td><td>...         </td><td>          NA</td><td>        NaN </td><td> 0.00000000 </td><td>          NA</td><td>        NaN </td><td> 0.00000000 </td><td>          NA</td><td>        NaN </td><td> 0.00000000 </td><td>          NA</td></tr>\n",
       "\t<tr><th scope=row>Zscan4e</th><td>  0.00      </td><td>  0.00      </td><td>  0.00      </td><td>  0.00      </td><td>  0.00      </td><td>  0.00      </td><td>  0.00      </td><td>  0.00      </td><td>  0.00      </td><td>  0.00      </td><td>...         </td><td>          NA</td><td>        NaN </td><td> 0.00000000 </td><td>          NA</td><td>        NaN </td><td> 0.00000000 </td><td>          NA</td><td>        NaN </td><td> 0.00000000 </td><td>          NA</td></tr>\n",
       "\t<tr><th scope=row>Zscan4f</th><td>  0.00      </td><td>  0.00      </td><td>  0.00      </td><td>  0.00      </td><td>  0.00      </td><td>  0.00      </td><td>  0.00      </td><td>  0.00      </td><td>  0.00      </td><td>  0.00      </td><td>...         </td><td>          NA</td><td>        NaN </td><td> 0.00000000 </td><td>          NA</td><td>        NaN </td><td> 0.00000000 </td><td>          NA</td><td>        NaN </td><td> 0.00000000 </td><td>          NA</td></tr>\n",
       "\t<tr><th scope=row>Zscan5b</th><td>  0.00      </td><td>  0.00      </td><td>  0.45      </td><td>  0.00      </td><td>  0.00      </td><td>  0.00      </td><td>  0.00      </td><td>  0.00      </td><td>  0.00      </td><td>  0.00      </td><td>...         </td><td>6.552081e-01</td><td>       -Inf </td><td>-0.36422609 </td><td>1.000000e+00</td><td>       -Inf </td><td>-0.36422609 </td><td>1.000000e+00</td><td>       -Inf </td><td>-0.36422609 </td><td>1.000000e+00</td></tr>\n",
       "\t<tr><th scope=row>Zswim1</th><td> 26.39      </td><td> 37.59      </td><td> 37.00      </td><td> 43.65      </td><td> 31.14      </td><td> 30.35      </td><td> 40.78      </td><td> 38.12      </td><td> 32.14      </td><td> 19.33      </td><td>...         </td><td>9.505141e-01</td><td>-0.16978310 </td><td>-0.15658750 </td><td>1.000000e+00</td><td>-0.90329919 </td><td>-0.85824714 </td><td>3.139871e-01</td><td>-0.33848623 </td><td>-0.32523443 </td><td>8.375345e-01</td></tr>\n",
       "\t<tr><th scope=row>Zswim2</th><td>  0.00      </td><td>  0.00      </td><td>  0.87      </td><td>  0.00      </td><td>  0.00      </td><td>  0.00      </td><td>  0.00      </td><td>  0.00      </td><td>  0.00      </td><td>  0.00      </td><td>...         </td><td>4.907610e-01</td><td>       -Inf </td><td>-0.48180498 </td><td>1.000000e+00</td><td>       -Inf </td><td>-0.48180498 </td><td>1.000000e+00</td><td>       -Inf </td><td>-0.48180498 </td><td>1.000000e+00</td></tr>\n",
       "\t<tr><th scope=row>Zswim3</th><td> 30.55      </td><td> 26.65      </td><td> 35.17      </td><td> 34.94      </td><td> 23.12      </td><td> 28.94      </td><td> 40.12      </td><td> 30.98      </td><td> 21.97      </td><td> 23.96      </td><td>...         </td><td>9.480484e-01</td><td>-0.53468486 </td><td>-0.51054247 </td><td>6.530239e-01</td><td>-0.40959365 </td><td>-0.39332060 </td><td>7.900802e-01</td><td>-0.52014160 </td><td>-0.49702733 </td><td>6.520948e-01</td></tr>\n",
       "\t<tr><th scope=row>Zswim4</th><td> 79.46      </td><td> 90.65      </td><td> 91.12      </td><td> 80.32      </td><td> 54.24      </td><td> 45.82      </td><td> 61.67      </td><td> 52.97      </td><td> 70.71      </td><td> 37.56      </td><td>...         </td><td>1.620123e-02</td><td>-0.27201873 </td><td>-0.39084477 </td><td>8.557093e-01</td><td>-1.18487977 </td><td>-1.58117757 </td><td>4.977878e-02</td><td>-0.36117718 </td><td>-0.51605673 </td><td>7.010739e-01</td></tr>\n",
       "\t<tr><th scope=row>Zswim5</th><td>  2.46      </td><td>  2.66      </td><td>  2.81      </td><td>  2.79      </td><td>  3.90      </td><td>  1.73      </td><td>  2.53      </td><td>  2.01      </td><td>  2.46      </td><td>  4.73      </td><td>...         </td><td>9.368615e-01</td><td>-0.12509575 </td><td>-0.08796732 </td><td>1.000000e+00</td><td> 0.82085465 </td><td> 0.61926496 </td><td>3.763428e-01</td><td> 0.07274011 </td><td> 0.05362971 </td><td>9.564060e-01</td></tr>\n",
       "\t<tr><th scope=row>Zswim6</th><td> 78.57      </td><td> 64.56      </td><td> 78.96      </td><td> 64.64      </td><td> 42.04      </td><td> 32.51      </td><td> 28.81      </td><td> 27.80      </td><td>  9.50      </td><td> 18.11      </td><td>...         </td><td>1.328819e-05</td><td>-2.91570432 </td><td>-3.40366320 </td><td>1.570425e-06</td><td>-1.98441559 </td><td>-2.49952744 </td><td>6.652442e-04</td><td>-2.38514132 </td><td>-2.90509218 </td><td>4.000839e-05</td></tr>\n",
       "\t<tr><th scope=row>Zswim7</th><td> 53.01      </td><td> 52.41      </td><td> 26.10      </td><td> 38.90      </td><td> 35.25      </td><td> 53.81      </td><td> 24.24      </td><td> 32.43      </td><td> 41.69      </td><td> 54.66      </td><td>...         </td><td>7.954452e-01</td><td>-0.03143804 </td><td> 0.01139782 </td><td>9.785628e-01</td><td> 0.35939567 </td><td> 0.33979318 </td><td>5.815607e-01</td><td>-1.19008750 </td><td>-0.87463236 </td><td>4.917418e-01</td></tr>\n",
       "\t<tr><th scope=row>Zufsp</th><td>591.72      </td><td>505.24      </td><td>525.78      </td><td>442.46      </td><td>501.12      </td><td>387.11      </td><td>220.12      </td><td>264.48      </td><td>233.09      </td><td>257.52      </td><td>...         </td><td>1.359496e-02</td><td>-1.14732453 </td><td>-1.96733787 </td><td>4.096078e-02</td><td>-1.00352274 </td><td>-1.73234190 </td><td>8.061953e-02</td><td>-0.48932873 </td><td>-0.85796114 </td><td>5.051271e-01</td></tr>\n",
       "\t<tr><th scope=row>Zw10</th><td>199.56      </td><td>179.26      </td><td>195.08      </td><td>205.33      </td><td>185.71      </td><td>196.94      </td><td>293.72      </td><td>221.19      </td><td>233.10      </td><td>245.22      </td><td>...         </td><td>5.339468e-01</td><td> 0.25888349 </td><td> 0.42906601 </td><td>6.312469e-01</td><td> 0.33203649 </td><td> 0.55216459 </td><td>5.152426e-01</td><td>-0.04837278 </td><td>-0.07575024 </td><td>1.000000e+00</td></tr>\n",
       "\t<tr><th scope=row>Zwilch</th><td> 28.20      </td><td> 33.03      </td><td> 33.65      </td><td> 34.71      </td><td> 82.41      </td><td> 96.15      </td><td>177.09      </td><td>139.90      </td><td>213.52      </td><td>213.88      </td><td>...         </td><td>6.037326e-16</td><td> 2.72033072 </td><td> 3.59776000 </td><td>4.772898e-15</td><td> 2.72279614 </td><td> 3.60180978 </td><td>3.773384e-15</td><td> 2.19523902 </td><td> 2.76666349 </td><td>7.365522e-09</td></tr>\n",
       "\t<tr><th scope=row>Zwint</th><td>102.02      </td><td> 87.48      </td><td> 78.36      </td><td> 83.47      </td><td>108.28      </td><td>125.24      </td><td>158.12      </td><td>126.57      </td><td>151.39      </td><td>126.62      </td><td>...         </td><td>2.875541e-02</td><td> 0.78547005 </td><td> 1.11964380 </td><td>6.879996e-02</td><td> 0.52763812 </td><td> 0.73747830 </td><td>2.575054e-01</td><td> 0.67190847 </td><td> 0.94929890 </td><td>1.221488e-01</td></tr>\n",
       "\t<tr><th scope=row>Zxda</th><td> 40.87      </td><td> 35.81      </td><td> 35.67      </td><td> 31.64      </td><td> 32.23      </td><td> 29.03      </td><td> 19.41      </td><td> 16.54      </td><td> 16.52      </td><td> 17.60      </td><td>...         </td><td>5.818633e-02</td><td>-1.12318705 </td><td>-1.35674112 </td><td>1.010015e-01</td><td>-1.03191363 </td><td>-1.25634724 </td><td>1.365961e-01</td><td>-1.07889645 </td><td>-1.30823417 </td><td>1.091052e-01</td></tr>\n",
       "\t<tr><th scope=row>Zxdb</th><td> 62.09      </td><td> 49.49      </td><td> 58.55      </td><td> 58.49      </td><td> 50.56      </td><td> 32.84      </td><td> 23.91      </td><td> 28.67      </td><td> 22.94      </td><td> 27.66      </td><td>...         </td><td>4.999058e-03</td><td>-1.31673422 </td><td>-1.66991114 </td><td>3.237706e-02</td><td>-1.04709813 </td><td>-1.35950607 </td><td>9.448219e-02</td><td>-0.84398561 </td><td>-1.11510443 </td><td>1.908914e-01</td></tr>\n",
       "\t<tr><th scope=row>Zxdc</th><td>138.97      </td><td>132.79      </td><td>139.74      </td><td>117.92      </td><td>117.88      </td><td> 98.30      </td><td> 78.73      </td><td> 82.25      </td><td> 88.82      </td><td> 86.35      </td><td>...         </td><td>5.969803e-02</td><td>-0.57539944 </td><td>-0.92930963 </td><td>4.169869e-01</td><td>-0.61617487 </td><td>-0.99262563 </td><td>3.634570e-01</td><td>-0.08893106 </td><td>-0.14444404 </td><td>1.000000e+00</td></tr>\n",
       "\t<tr><th scope=row>Zyg11a</th><td>  0.53      </td><td>  0.00      </td><td>  0.39      </td><td>  0.00      </td><td>  0.00      </td><td>  0.00      </td><td>  0.00      </td><td>  0.30      </td><td>  0.00      </td><td>  0.00      </td><td>...         </td><td>8.502624e-01</td><td>       -Inf </td><td>-0.77765562 </td><td>1.000000e+00</td><td>       -Inf </td><td>-0.77765562 </td><td>1.000000e+00</td><td> 2.15618886 </td><td> 1.35228767 </td><td>3.719981e-01</td></tr>\n",
       "\t<tr><th scope=row>Zyg11b</th><td>202.95      </td><td>186.55      </td><td>184.59      </td><td>183.46      </td><td>222.04      </td><td>200.52      </td><td>172.70      </td><td>160.35      </td><td>142.94      </td><td>120.43      </td><td>...         </td><td>1.000000e+00</td><td>-0.40594668 </td><td>-0.74501934 </td><td>6.241101e-01</td><td>-0.65308841 </td><td>-1.19550153 </td><td>3.149901e-01</td><td>-0.98688626 </td><td>-1.79550102 </td><td>7.718166e-02</td></tr>\n",
       "\t<tr><th scope=row>Zyx</th><td>118.91      </td><td>128.36      </td><td>157.01      </td><td>124.35      </td><td> 75.74      </td><td> 52.35      </td><td> 36.46      </td><td> 52.79      </td><td> 22.26      </td><td> 50.58      </td><td>...         </td><td>1.179740e-06</td><td>-2.57002123 </td><td>-2.93223204 </td><td>2.700192e-05</td><td>-1.38563841 </td><td>-1.74417753 </td><td>2.036233e-02</td><td>-1.73000714 </td><td>-2.11444476 </td><td>2.974753e-03</td></tr>\n",
       "\t<tr><th scope=row>Zzef1</th><td>337.04      </td><td>322.86      </td><td>382.60      </td><td>313.31      </td><td>289.39      </td><td>255.99      </td><td>248.20      </td><td>169.56      </td><td>230.55      </td><td>282.16      </td><td>...         </td><td>4.026519e-02</td><td>-0.55599373 </td><td>-1.01468215 </td><td>4.342302e-01</td><td>-0.26455984 </td><td>-0.47689714 </td><td>8.231671e-01</td><td>-0.46301646 </td><td>-0.84286404 </td><td>5.329534e-01</td></tr>\n",
       "\t<tr><th scope=row>Zzz3</th><td>195.02      </td><td>170.51      </td><td>202.99      </td><td>181.13      </td><td>217.82      </td><td>207.24      </td><td>214.41      </td><td>187.17      </td><td>328.25      </td><td>213.70      </td><td>...         </td><td>7.028719e-01</td><td> 0.80853939 </td><td> 1.49527585 </td><td>3.673920e-02</td><td> 0.18936912 </td><td> 0.35230353 </td><td>7.391194e-01</td><td> 0.03951529 </td><td> 0.07774637 </td><td>1.000000e+00</td></tr>\n",
       "\t<tr><th scope=row>a</th><td>  0.00      </td><td> 10.36      </td><td>  1.29      </td><td>  5.12      </td><td>  1.32      </td><td>  7.50      </td><td>  0.00      </td><td>  1.96      </td><td> 10.45      </td><td>  0.00      </td><td>...         </td><td>6.552081e-01</td><td> 1.31830017 </td><td> 1.33466617 </td><td>2.393309e-01</td><td>       -Inf </td><td>-1.74763244 </td><td>6.939833e-01</td><td> 3.21372738 </td><td> 2.69047589 </td><td>2.646548e-04</td></tr>\n",
       "\t<tr><th scope=row>l7Rn6</th><td>129.47      </td><td>149.73      </td><td>128.93      </td><td>155.23      </td><td>174.95      </td><td>235.31      </td><td>243.53      </td><td>299.47      </td><td>175.38      </td><td>186.04      </td><td>...         </td><td>5.376546e-03</td><td> 0.31644528 </td><td> 0.38346666 </td><td>5.900277e-01</td><td> 0.40154894 </td><td> 0.48934666 </td><td>4.813295e-01</td><td> 0.47655663 </td><td> 0.58407798 </td><td>3.770375e-01</td></tr>\n",
       "\t<tr><th scope=row>rp9</th><td>428.77      </td><td>609.23      </td><td>412.90      </td><td>519.73      </td><td>464.65      </td><td>541.97      </td><td>505.33      </td><td>698.16      </td><td>545.28      </td><td>508.24      </td><td>...         </td><td>5.766341e-01</td><td> 0.14640266 </td><td> 0.26363276 </td><td>7.862389e-01</td><td> 0.04490146 </td><td> 0.09660478 </td><td>9.576599e-01</td><td>-0.11683395 </td><td>-0.16486401 </td><td>1.000000e+00</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "\\begin{tabular}{r|lllllllllllllllllllllll}\n",
       "  & C\\_1 & C\\_3 & C\\_4 & C\\_6 & P\\_2 & P\\_3 & P\\_4 & P\\_5 & T\\_1 & T\\_2 & ... & qvalue.CP & log2FC.CT1 & log2FCvsn.CT1 & qvalue.CT1 & log2FC.CT2 & log2FCvsn.CT2 & qvalue.CT2 & log2FC.CT3 & log2FCvsn.CT3 & qvalue.CT3\\\\\n",
       "\\hline\n",
       "\t0610007C21Rik & 300.42       & 318.17       & 245.12       & 327.45       & 309.13       & 273.16       & 237.94       & 317.90       & 273.52       &  398.82      & ...          & 9.417318e-01 & -0.122658393 & -0.15952224  & 1.000000e+00 &  0.42142989  &  0.61682006  & 0.393704824  &  0.72776314  &  1.08599078  & 0.0849638225\\\\\n",
       "\t0610007L01Rik & 207.09       & 210.95       & 220.58       & 267.45       & 217.25       & 200.22       & 281.38       & 283.03       & 231.03       &  306.11      & ...          & 7.878284e-01 &  0.028450720 &  0.06106108  & 1.000000e+00 &  0.43445718  &  0.77974700  & 0.391081390  &  0.59935996  &  1.07715415  & 0.1662041565\\\\\n",
       "\t0610007P08Rik & 134.32       & 112.85       & 126.30       & 116.87       & 146.14       & 130.28       & 147.10       &  97.46       & 162.87       &   90.20      & ...          & 8.535726e-01 &  0.409968779 &  0.71039586  & 3.821038e-01 & -0.44262105  & -0.71991486  & 0.557661867  &  0.09529604  &  0.16598341  & 0.8933977701\\\\\n",
       "\t0610007P14Rik & 220.19       & 203.60       & 240.95       & 245.36       & 253.78       & 261.80       & 258.35       & 297.25       & 309.32       &  382.49      & ...          & 5.572482e-01 &  0.443088932 &  0.63473511  & 4.169869e-01 &  0.74940682  &  1.09910154  & 0.104290139  &  0.63176944  &  0.91806754  & 0.1822683985\\\\\n",
       "\t0610007P22Rik &  78.98       & 114.13       &  90.50       &  95.85       &  88.88       &  96.54       & 113.98       & 112.99       & 140.49       &  147.13      & ...          & 8.120777e-01 &  0.566465434 &  0.65768003  & 3.246861e-01 &  0.63312517  &  0.73848570  & 0.246905230  &  0.26233311  &  0.30157112  & 0.6610278272\\\\\n",
       "\t0610008F07Rik &   0.00       &   0.00       &   0.00       &   0.00       &   0.00       &   0.00       &   0.00       &   0.00       &   0.00       &    0.00      & ...          &           NA &          NaN &  0.00000000  &           NA &         NaN  &  0.00000000  &          NA  &         NaN  &  0.00000000  &           NA\\\\\n",
       "\t0610009B22Rik &  42.55       &  79.63       &  35.92       &  66.90       &  71.29       &  92.66       &  80.27       & 114.18       &  56.12       &   88.03      & ...          & 8.899602e-02 & -0.003342027 &  0.04917621  & 9.905365e-01 &  0.64630459  &  0.64565648  & 0.336251742  &  0.97285105  &  0.96878604  & 0.0990259846\\\\\n",
       "\t0610009D07Rik & 533.63       & 549.03       & 348.60       & 476.57       & 534.86       & 697.28       & 703.17       & 812.13       & 583.43       & 1037.25      & ...          & 1.480926e-02 &  0.290685645 &  0.50412608  & 4.903655e-01 &  1.12082967  &  1.95063835  & 0.001107939  &  0.23678699  &  0.41441353  & 0.5649703839\\\\\n",
       "\t0610009O20Rik & 142.71       & 124.71       & 173.33       & 140.49       & 146.39       & 164.01       & 177.91       & 182.31       & 145.45       &  173.75      & ...          & 6.165242e-01 &  0.001430952 &  0.01342905  & 1.000000e+00 &  0.25793077  &  0.39644744  & 0.677829395  &  0.16636690  &  0.25786454  & 0.8368292328\\\\\n",
       "\t0610010B08Rik &   0.53       &   1.05       &   0.59       &   0.98       &   0.20       &   0.91       &   0.33       &   0.90       &   2.39       &    0.49      & ...          & 7.991074e-01 &  1.601659360 &  1.03301217  & 1.412337e-01 & -0.68644558  & -0.34649473  & 1.000000000  & -0.61143788  & -0.30683543  & 1.0000000000\\\\\n",
       "\t0610010F05Rik &  48.73       &  46.45       &  46.12       &  54.38       &  57.41       &  74.18       & 119.09       &  83.28       & 143.00       &  115.38      & ...          & 2.467096e-03 &  1.547499287 &  2.25768779  & 3.195463e-05 &  1.23793298  &  1.76035829  & 0.002148796  &  1.37210275  &  1.97323466  & 0.0003177189\\\\\n",
       "\t0610010K14Rik & 139.07       & 183.34       & 155.09       & 198.09       & 153.72       & 178.00       & 251.22       & 233.18       & 189.16       &  170.52      & ...          & 4.907610e-01 &  0.163482934 &  0.20563523  & 8.341150e-01 &  0.01382476  &  0.02786449  & 1.000000000  & -0.49083574  & -0.53639181  & 0.6099562171\\\\\n",
       "\t0610010O12Rik &   8.12       &  14.05       &   4.50       &  10.43       &  13.83       &   6.98       &   2.55       &   4.55       &   2.03       &    3.73      & ...          & 7.435859e-01 & -2.194377903 & -1.12633670  & 8.648071e-01 & -1.31255784  & -0.68852920  & 0.961124861  &  0.34741235  &  0.27590322  & 0.7952688613\\\\\n",
       "\t0610011F06Rik & 204.06       & 201.76       & 192.54       & 168.32       & 198.37       & 168.88       & 238.86       & 272.29       & 199.22       &  252.04      & ...          & 5.730909e-01 &  0.055749917 &  0.06967514  & 9.512541e-01 &  0.39504274  &  0.48754054  & 0.461744128  & -0.03114370  & -0.03312322  & 1.0000000000\\\\\n",
       "\t0610011L14Rik & 133.58       & 148.38       & 144.65       & 177.28       & 136.48       & 136.41       & 149.82       & 163.78       & 119.23       &  135.54      & ...          & 9.180832e-01 & -0.340562836 & -0.52566824  & 6.834677e-01 & -0.15563093  & -0.23803711  & 0.960638998  & -0.33971046  & -0.52436127  & 0.6672814070\\\\\n",
       "\t0610012G03Rik &  74.73       & 125.85       &  61.87       &  80.75       & 102.53       &  86.76       &  95.54       & 118.81       & 109.84       &   76.28      & ...          & 4.812137e-01 &  0.356306204 &  0.45366946  & 4.818037e-01 & -0.16975144  & -0.14397236  & 1.000000000  &  0.67330953  &  0.84325916  & 0.1551004987\\\\\n",
       "\t0610012H03Rik &   0.00       &   0.00       &   0.00       &   0.00       &   1.14       &   0.65       &   0.48       &   0.00       &   0.00       &    0.70      & ...          & 4.062349e-01 &          NaN &  0.00000000  &           NA &         Inf  &  1.59790970  & 0.394873980  &         NaN  &  0.00000000  &           NA\\\\\n",
       "\t0610030E20Rik & 246.86       & 210.58       & 244.15       & 231.65       & 202.98       & 202.63       & 208.03       & 157.29       & 180.58       &  161.82      & ...          & 3.109693e-01 & -0.369645846 & -0.64797212  & 6.379189e-01 & -0.52787395  & -0.92074926  & 0.420538656  & -0.68168494  & -1.18148698  & 0.2502932155\\\\\n",
       "\t0610031J06Rik & 235.78       & 247.11       & 270.53       & 289.51       & 224.81       & 197.12       & 164.59       & 222.56       & 214.95       &  109.86      & ...          & 1.909861e-01 & -0.278596389 & -0.40585384  & 8.012731e-01 & -1.24688090  & -1.68629376  & 0.030847570  & -0.04476128  & -0.06205471  & 1.0000000000\\\\\n",
       "\t0610037L13Rik & 120.72       & 150.94       & 153.03       & 156.60       & 157.36       & 118.41       & 129.79       & 166.83       & 106.50       &  114.60      & ...          & 9.939000e-01 & -0.448326263 & -0.55404931  & 6.185147e-01 & -0.34261019  & -0.42593111  & 0.735352367  & -0.56716278  & -0.69506108  & 0.4597216657\\\\\n",
       "\t0610037P05Rik & 208.97       & 204.10       & 197.03       & 211.32       & 243.63       & 191.90       & 188.58       & 250.09       & 179.76       &  139.84      & ...          & 8.463243e-01 & -0.192066854 & -0.26320221  & 9.898461e-01 & -0.55435147  & -0.73609432  & 0.481329466  & -0.73434086  & -0.95940851  & 0.2974374726\\\\\n",
       "\t0610038B21Rik &   0.79       &   3.14       &   1.17       &   2.33       &   3.00       &   4.09       &   2.99       &   3.11       &   9.50       &   10.21      & ...          & 4.907610e-01 &  2.354623889 &  1.57269091  & 2.772354e-02 &  2.45883637  &  1.64794881  & 0.018293855  &  0.31145165  &  0.26374880  & 0.7537873673\\\\\n",
       "\t0610039K10Rik & 369.63       & 437.06       & 366.86       & 239.55       & 206.66       & 243.30       & 218.87       & 167.39       & 258.65       &  282.69      & ...          & 9.555152e-03 & -0.449816317 & -0.59038159  & 6.738087e-01 & -0.32155836  & -0.41703417  & 0.858497947  & -0.79909169  & -1.04216158  & 0.2730506154\\\\\n",
       "\t0610040B10Rik &  21.96       &  32.57       &  26.36       &  16.12       &  16.62       &  14.15       &  17.27       &  20.01       &  13.70       &   12.62      & ...          & 6.552081e-01 & -0.823887644 & -0.51049225  & 1.000000e+00 & -0.94206758  & -0.58348453  & 0.893400570  &  0.13294011  &  0.11797629  & 0.9044876130\\\\\n",
       "\t0610040J01Rik &   3.88       &   5.11       &   5.25       &   8.53       &   6.35       &   7.21       &   4.06       &   2.54       &  41.27       &    2.97      & ...          & 7.677246e-01 &  2.858036704 &  2.36489744  & 7.262926e-06 & -0.93821514  & -0.57260694  & 0.776787852  &  1.20718188  &  0.91837655  & 0.2086258283\\\\\n",
       "\t0910001L09Rik & 265.84       & 288.72       & 281.83       & 296.15       & 213.67       & 235.43       & 168.04       & 243.30       & 191.10       &  277.41      & ...          & 2.203561e-01 & -0.567121735 & -0.71967492  & 4.858289e-01 & -0.02945569  & -0.03821356  & 1.000000000  & -1.92188013  & -2.16617015  & 0.0019413521\\\\\n",
       "\t1100001G20Rik &  88.25       & 111.80       &  30.56       &  80.95       & 240.00       & 357.70       & 169.97       & 335.60       &  16.52       &   83.70      & ...          & 4.176561e-09 & -2.237512048 & -1.53721873  & 1.206218e-01 &  0.10373963  &  0.18408681  & 0.815898469  &  0.83479293  &  0.83249702  & 0.1708197164\\\\\n",
       "\t1110001A16Rik &  55.89       &  37.89       &  52.49       &  50.97       &  72.50       &  50.06       &  57.24       &  55.07       &  63.75       &   63.88      & ...          & 6.552081e-01 &  0.370650335 &  0.38524900  & 5.881720e-01 &  0.37348580  &  0.38820043  & 0.573125809  & -0.08755941  & -0.07358787  & 1.0000000000\\\\\n",
       "\t1110001J03Rik & 167.09       & 184.17       &  93.07       & 148.74       & 172.04       & 216.45       & 184.32       & 218.11       & 117.57       &  168.78      & ...          & 2.730091e-01 & -0.334725282 & -0.27668814  & 1.000000e+00 &  0.18691921  &  0.21315967  & 0.718532586  &  0.34558484  &  0.37025581  & 0.5329533971\\\\\n",
       "\t1110002B05Rik & 268.54       & 255.88       & 194.43       & 238.68       & 291.73       & 383.66       & 443.65       & 324.53       & 334.39       &  326.84      & ...          & 1.263570e-02 &  0.482220986 &  0.73280727  & 2.855344e-01 &  0.44925705  &  0.68167993  & 0.321819917  &  1.17830251  &  1.86969136  & 0.0009306154\\\\\n",
       "\t... & ... & ... & ... & ... & ... & ... & ... & ... & ... & ... &     & ... & ... & ... & ... & ... & ... & ... & ... & ... & ...\\\\\n",
       "\tZscan30 &   8.97       &  19.52       &  15.91       &  17.12       &  16.30       &  24.67       &  24.83       &  14.09       &  23.29       &  16.51       & ...          & 7.129923e-01 &  0.59849256  &  0.45285540  & 5.446778e-01 &  0.10180099  &  0.10158047  & 9.132241e-01 &  1.17680869  &  0.88789237  & 1.891014e-01\\\\\n",
       "\tZscan4a &   0.00       &   0.00       &   0.40       &   0.00       &   0.00       &   0.00       &   0.00       &   0.00       &   0.00       &   0.00       & ...          & 6.552081e-01 &        -Inf  & -0.36422609  & 1.000000e+00 &        -Inf  & -0.36422609  & 1.000000e+00 &        -Inf  & -0.36422609  & 1.000000e+00\\\\\n",
       "\tZscan4b &   0.00       &   0.00       &   0.00       &   0.00       &   0.00       &   0.00       &   0.00       &   0.00       &   0.00       &   0.00       & ...          &           NA &         NaN  &  0.00000000  &           NA &         NaN  &  0.00000000  &           NA &         NaN  &  0.00000000  &           NA\\\\\n",
       "\tZscan4c &   0.00       &   0.00       &   0.00       &   0.00       &   0.00       &   0.00       &   0.00       &   0.00       &   0.00       &   0.00       & ...          &           NA &         NaN  &  0.00000000  &           NA &         NaN  &  0.00000000  &           NA &         NaN  &  0.00000000  &           NA\\\\\n",
       "\tZscan4d &   0.00       &   0.00       &   0.00       &   0.00       &   0.00       &   0.00       &   0.00       &   0.00       &   0.00       &   0.00       & ...          &           NA &         NaN  &  0.00000000  &           NA &         NaN  &  0.00000000  &           NA &         NaN  &  0.00000000  &           NA\\\\\n",
       "\tZscan4e &   0.00       &   0.00       &   0.00       &   0.00       &   0.00       &   0.00       &   0.00       &   0.00       &   0.00       &   0.00       & ...          &           NA &         NaN  &  0.00000000  &           NA &         NaN  &  0.00000000  &           NA &         NaN  &  0.00000000  &           NA\\\\\n",
       "\tZscan4f &   0.00       &   0.00       &   0.00       &   0.00       &   0.00       &   0.00       &   0.00       &   0.00       &   0.00       &   0.00       & ...          &           NA &         NaN  &  0.00000000  &           NA &         NaN  &  0.00000000  &           NA &         NaN  &  0.00000000  &           NA\\\\\n",
       "\tZscan5b &   0.00       &   0.00       &   0.45       &   0.00       &   0.00       &   0.00       &   0.00       &   0.00       &   0.00       &   0.00       & ...          & 6.552081e-01 &        -Inf  & -0.36422609  & 1.000000e+00 &        -Inf  & -0.36422609  & 1.000000e+00 &        -Inf  & -0.36422609  & 1.000000e+00\\\\\n",
       "\tZswim1 &  26.39       &  37.59       &  37.00       &  43.65       &  31.14       &  30.35       &  40.78       &  38.12       &  32.14       &  19.33       & ...          & 9.505141e-01 & -0.16978310  & -0.15658750  & 1.000000e+00 & -0.90329919  & -0.85824714  & 3.139871e-01 & -0.33848623  & -0.32523443  & 8.375345e-01\\\\\n",
       "\tZswim2 &   0.00       &   0.00       &   0.87       &   0.00       &   0.00       &   0.00       &   0.00       &   0.00       &   0.00       &   0.00       & ...          & 4.907610e-01 &        -Inf  & -0.48180498  & 1.000000e+00 &        -Inf  & -0.48180498  & 1.000000e+00 &        -Inf  & -0.48180498  & 1.000000e+00\\\\\n",
       "\tZswim3 &  30.55       &  26.65       &  35.17       &  34.94       &  23.12       &  28.94       &  40.12       &  30.98       &  21.97       &  23.96       & ...          & 9.480484e-01 & -0.53468486  & -0.51054247  & 6.530239e-01 & -0.40959365  & -0.39332060  & 7.900802e-01 & -0.52014160  & -0.49702733  & 6.520948e-01\\\\\n",
       "\tZswim4 &  79.46       &  90.65       &  91.12       &  80.32       &  54.24       &  45.82       &  61.67       &  52.97       &  70.71       &  37.56       & ...          & 1.620123e-02 & -0.27201873  & -0.39084477  & 8.557093e-01 & -1.18487977  & -1.58117757  & 4.977878e-02 & -0.36117718  & -0.51605673  & 7.010739e-01\\\\\n",
       "\tZswim5 &   2.46       &   2.66       &   2.81       &   2.79       &   3.90       &   1.73       &   2.53       &   2.01       &   2.46       &   4.73       & ...          & 9.368615e-01 & -0.12509575  & -0.08796732  & 1.000000e+00 &  0.82085465  &  0.61926496  & 3.763428e-01 &  0.07274011  &  0.05362971  & 9.564060e-01\\\\\n",
       "\tZswim6 &  78.57       &  64.56       &  78.96       &  64.64       &  42.04       &  32.51       &  28.81       &  27.80       &   9.50       &  18.11       & ...          & 1.328819e-05 & -2.91570432  & -3.40366320  & 1.570425e-06 & -1.98441559  & -2.49952744  & 6.652442e-04 & -2.38514132  & -2.90509218  & 4.000839e-05\\\\\n",
       "\tZswim7 &  53.01       &  52.41       &  26.10       &  38.90       &  35.25       &  53.81       &  24.24       &  32.43       &  41.69       &  54.66       & ...          & 7.954452e-01 & -0.03143804  &  0.01139782  & 9.785628e-01 &  0.35939567  &  0.33979318  & 5.815607e-01 & -1.19008750  & -0.87463236  & 4.917418e-01\\\\\n",
       "\tZufsp & 591.72       & 505.24       & 525.78       & 442.46       & 501.12       & 387.11       & 220.12       & 264.48       & 233.09       & 257.52       & ...          & 1.359496e-02 & -1.14732453  & -1.96733787  & 4.096078e-02 & -1.00352274  & -1.73234190  & 8.061953e-02 & -0.48932873  & -0.85796114  & 5.051271e-01\\\\\n",
       "\tZw10 & 199.56       & 179.26       & 195.08       & 205.33       & 185.71       & 196.94       & 293.72       & 221.19       & 233.10       & 245.22       & ...          & 5.339468e-01 &  0.25888349  &  0.42906601  & 6.312469e-01 &  0.33203649  &  0.55216459  & 5.152426e-01 & -0.04837278  & -0.07575024  & 1.000000e+00\\\\\n",
       "\tZwilch &  28.20       &  33.03       &  33.65       &  34.71       &  82.41       &  96.15       & 177.09       & 139.90       & 213.52       & 213.88       & ...          & 6.037326e-16 &  2.72033072  &  3.59776000  & 4.772898e-15 &  2.72279614  &  3.60180978  & 3.773384e-15 &  2.19523902  &  2.76666349  & 7.365522e-09\\\\\n",
       "\tZwint & 102.02       &  87.48       &  78.36       &  83.47       & 108.28       & 125.24       & 158.12       & 126.57       & 151.39       & 126.62       & ...          & 2.875541e-02 &  0.78547005  &  1.11964380  & 6.879996e-02 &  0.52763812  &  0.73747830  & 2.575054e-01 &  0.67190847  &  0.94929890  & 1.221488e-01\\\\\n",
       "\tZxda &  40.87       &  35.81       &  35.67       &  31.64       &  32.23       &  29.03       &  19.41       &  16.54       &  16.52       &  17.60       & ...          & 5.818633e-02 & -1.12318705  & -1.35674112  & 1.010015e-01 & -1.03191363  & -1.25634724  & 1.365961e-01 & -1.07889645  & -1.30823417  & 1.091052e-01\\\\\n",
       "\tZxdb &  62.09       &  49.49       &  58.55       &  58.49       &  50.56       &  32.84       &  23.91       &  28.67       &  22.94       &  27.66       & ...          & 4.999058e-03 & -1.31673422  & -1.66991114  & 3.237706e-02 & -1.04709813  & -1.35950607  & 9.448219e-02 & -0.84398561  & -1.11510443  & 1.908914e-01\\\\\n",
       "\tZxdc & 138.97       & 132.79       & 139.74       & 117.92       & 117.88       &  98.30       &  78.73       &  82.25       &  88.82       &  86.35       & ...          & 5.969803e-02 & -0.57539944  & -0.92930963  & 4.169869e-01 & -0.61617487  & -0.99262563  & 3.634570e-01 & -0.08893106  & -0.14444404  & 1.000000e+00\\\\\n",
       "\tZyg11a &   0.53       &   0.00       &   0.39       &   0.00       &   0.00       &   0.00       &   0.00       &   0.30       &   0.00       &   0.00       & ...          & 8.502624e-01 &        -Inf  & -0.77765562  & 1.000000e+00 &        -Inf  & -0.77765562  & 1.000000e+00 &  2.15618886  &  1.35228767  & 3.719981e-01\\\\\n",
       "\tZyg11b & 202.95       & 186.55       & 184.59       & 183.46       & 222.04       & 200.52       & 172.70       & 160.35       & 142.94       & 120.43       & ...          & 1.000000e+00 & -0.40594668  & -0.74501934  & 6.241101e-01 & -0.65308841  & -1.19550153  & 3.149901e-01 & -0.98688626  & -1.79550102  & 7.718166e-02\\\\\n",
       "\tZyx & 118.91       & 128.36       & 157.01       & 124.35       &  75.74       &  52.35       &  36.46       &  52.79       &  22.26       &  50.58       & ...          & 1.179740e-06 & -2.57002123  & -2.93223204  & 2.700192e-05 & -1.38563841  & -1.74417753  & 2.036233e-02 & -1.73000714  & -2.11444476  & 2.974753e-03\\\\\n",
       "\tZzef1 & 337.04       & 322.86       & 382.60       & 313.31       & 289.39       & 255.99       & 248.20       & 169.56       & 230.55       & 282.16       & ...          & 4.026519e-02 & -0.55599373  & -1.01468215  & 4.342302e-01 & -0.26455984  & -0.47689714  & 8.231671e-01 & -0.46301646  & -0.84286404  & 5.329534e-01\\\\\n",
       "\tZzz3 & 195.02       & 170.51       & 202.99       & 181.13       & 217.82       & 207.24       & 214.41       & 187.17       & 328.25       & 213.70       & ...          & 7.028719e-01 &  0.80853939  &  1.49527585  & 3.673920e-02 &  0.18936912  &  0.35230353  & 7.391194e-01 &  0.03951529  &  0.07774637  & 1.000000e+00\\\\\n",
       "\ta &   0.00       &  10.36       &   1.29       &   5.12       &   1.32       &   7.50       &   0.00       &   1.96       &  10.45       &   0.00       & ...          & 6.552081e-01 &  1.31830017  &  1.33466617  & 2.393309e-01 &        -Inf  & -1.74763244  & 6.939833e-01 &  3.21372738  &  2.69047589  & 2.646548e-04\\\\\n",
       "\tl7Rn6 & 129.47       & 149.73       & 128.93       & 155.23       & 174.95       & 235.31       & 243.53       & 299.47       & 175.38       & 186.04       & ...          & 5.376546e-03 &  0.31644528  &  0.38346666  & 5.900277e-01 &  0.40154894  &  0.48934666  & 4.813295e-01 &  0.47655663  &  0.58407798  & 3.770375e-01\\\\\n",
       "\trp9 & 428.77       & 609.23       & 412.90       & 519.73       & 464.65       & 541.97       & 505.33       & 698.16       & 545.28       & 508.24       & ...          & 5.766341e-01 &  0.14640266  &  0.26363276  & 7.862389e-01 &  0.04490146  &  0.09660478  & 9.576599e-01 & -0.11683395  & -0.16486401  & 1.000000e+00\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "| <!--/--> | C_1 | C_3 | C_4 | C_6 | P_2 | P_3 | P_4 | P_5 | T_1 | T_2 | ... | qvalue.CP | log2FC.CT1 | log2FCvsn.CT1 | qvalue.CT1 | log2FC.CT2 | log2FCvsn.CT2 | qvalue.CT2 | log2FC.CT3 | log2FCvsn.CT3 | qvalue.CT3 | \n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 0610007C21Rik | 300.42       | 318.17       | 245.12       | 327.45       | 309.13       | 273.16       | 237.94       | 317.90       | 273.52       |  398.82      | ...          | 9.417318e-01 | -0.122658393 | -0.15952224  | 1.000000e+00 |  0.42142989  |  0.61682006  | 0.393704824  |  0.72776314  |  1.08599078  | 0.0849638225 | \n",
       "| 0610007L01Rik | 207.09       | 210.95       | 220.58       | 267.45       | 217.25       | 200.22       | 281.38       | 283.03       | 231.03       |  306.11      | ...          | 7.878284e-01 |  0.028450720 |  0.06106108  | 1.000000e+00 |  0.43445718  |  0.77974700  | 0.391081390  |  0.59935996  |  1.07715415  | 0.1662041565 | \n",
       "| 0610007P08Rik | 134.32       | 112.85       | 126.30       | 116.87       | 146.14       | 130.28       | 147.10       |  97.46       | 162.87       |   90.20      | ...          | 8.535726e-01 |  0.409968779 |  0.71039586  | 3.821038e-01 | -0.44262105  | -0.71991486  | 0.557661867  |  0.09529604  |  0.16598341  | 0.8933977701 | \n",
       "| 0610007P14Rik | 220.19       | 203.60       | 240.95       | 245.36       | 253.78       | 261.80       | 258.35       | 297.25       | 309.32       |  382.49      | ...          | 5.572482e-01 |  0.443088932 |  0.63473511  | 4.169869e-01 |  0.74940682  |  1.09910154  | 0.104290139  |  0.63176944  |  0.91806754  | 0.1822683985 | \n",
       "| 0610007P22Rik |  78.98       | 114.13       |  90.50       |  95.85       |  88.88       |  96.54       | 113.98       | 112.99       | 140.49       |  147.13      | ...          | 8.120777e-01 |  0.566465434 |  0.65768003  | 3.246861e-01 |  0.63312517  |  0.73848570  | 0.246905230  |  0.26233311  |  0.30157112  | 0.6610278272 | \n",
       "| 0610008F07Rik |   0.00       |   0.00       |   0.00       |   0.00       |   0.00       |   0.00       |   0.00       |   0.00       |   0.00       |    0.00      | ...          |           NA |          NaN |  0.00000000  |           NA |         NaN  |  0.00000000  |          NA  |         NaN  |  0.00000000  |           NA | \n",
       "| 0610009B22Rik |  42.55       |  79.63       |  35.92       |  66.90       |  71.29       |  92.66       |  80.27       | 114.18       |  56.12       |   88.03      | ...          | 8.899602e-02 | -0.003342027 |  0.04917621  | 9.905365e-01 |  0.64630459  |  0.64565648  | 0.336251742  |  0.97285105  |  0.96878604  | 0.0990259846 | \n",
       "| 0610009D07Rik | 533.63       | 549.03       | 348.60       | 476.57       | 534.86       | 697.28       | 703.17       | 812.13       | 583.43       | 1037.25      | ...          | 1.480926e-02 |  0.290685645 |  0.50412608  | 4.903655e-01 |  1.12082967  |  1.95063835  | 0.001107939  |  0.23678699  |  0.41441353  | 0.5649703839 | \n",
       "| 0610009O20Rik | 142.71       | 124.71       | 173.33       | 140.49       | 146.39       | 164.01       | 177.91       | 182.31       | 145.45       |  173.75      | ...          | 6.165242e-01 |  0.001430952 |  0.01342905  | 1.000000e+00 |  0.25793077  |  0.39644744  | 0.677829395  |  0.16636690  |  0.25786454  | 0.8368292328 | \n",
       "| 0610010B08Rik |   0.53       |   1.05       |   0.59       |   0.98       |   0.20       |   0.91       |   0.33       |   0.90       |   2.39       |    0.49      | ...          | 7.991074e-01 |  1.601659360 |  1.03301217  | 1.412337e-01 | -0.68644558  | -0.34649473  | 1.000000000  | -0.61143788  | -0.30683543  | 1.0000000000 | \n",
       "| 0610010F05Rik |  48.73       |  46.45       |  46.12       |  54.38       |  57.41       |  74.18       | 119.09       |  83.28       | 143.00       |  115.38      | ...          | 2.467096e-03 |  1.547499287 |  2.25768779  | 3.195463e-05 |  1.23793298  |  1.76035829  | 0.002148796  |  1.37210275  |  1.97323466  | 0.0003177189 | \n",
       "| 0610010K14Rik | 139.07       | 183.34       | 155.09       | 198.09       | 153.72       | 178.00       | 251.22       | 233.18       | 189.16       |  170.52      | ...          | 4.907610e-01 |  0.163482934 |  0.20563523  | 8.341150e-01 |  0.01382476  |  0.02786449  | 1.000000000  | -0.49083574  | -0.53639181  | 0.6099562171 | \n",
       "| 0610010O12Rik |   8.12       |  14.05       |   4.50       |  10.43       |  13.83       |   6.98       |   2.55       |   4.55       |   2.03       |    3.73      | ...          | 7.435859e-01 | -2.194377903 | -1.12633670  | 8.648071e-01 | -1.31255784  | -0.68852920  | 0.961124861  |  0.34741235  |  0.27590322  | 0.7952688613 | \n",
       "| 0610011F06Rik | 204.06       | 201.76       | 192.54       | 168.32       | 198.37       | 168.88       | 238.86       | 272.29       | 199.22       |  252.04      | ...          | 5.730909e-01 |  0.055749917 |  0.06967514  | 9.512541e-01 |  0.39504274  |  0.48754054  | 0.461744128  | -0.03114370  | -0.03312322  | 1.0000000000 | \n",
       "| 0610011L14Rik | 133.58       | 148.38       | 144.65       | 177.28       | 136.48       | 136.41       | 149.82       | 163.78       | 119.23       |  135.54      | ...          | 9.180832e-01 | -0.340562836 | -0.52566824  | 6.834677e-01 | -0.15563093  | -0.23803711  | 0.960638998  | -0.33971046  | -0.52436127  | 0.6672814070 | \n",
       "| 0610012G03Rik |  74.73       | 125.85       |  61.87       |  80.75       | 102.53       |  86.76       |  95.54       | 118.81       | 109.84       |   76.28      | ...          | 4.812137e-01 |  0.356306204 |  0.45366946  | 4.818037e-01 | -0.16975144  | -0.14397236  | 1.000000000  |  0.67330953  |  0.84325916  | 0.1551004987 | \n",
       "| 0610012H03Rik |   0.00       |   0.00       |   0.00       |   0.00       |   1.14       |   0.65       |   0.48       |   0.00       |   0.00       |    0.70      | ...          | 4.062349e-01 |          NaN |  0.00000000  |           NA |         Inf  |  1.59790970  | 0.394873980  |         NaN  |  0.00000000  |           NA | \n",
       "| 0610030E20Rik | 246.86       | 210.58       | 244.15       | 231.65       | 202.98       | 202.63       | 208.03       | 157.29       | 180.58       |  161.82      | ...          | 3.109693e-01 | -0.369645846 | -0.64797212  | 6.379189e-01 | -0.52787395  | -0.92074926  | 0.420538656  | -0.68168494  | -1.18148698  | 0.2502932155 | \n",
       "| 0610031J06Rik | 235.78       | 247.11       | 270.53       | 289.51       | 224.81       | 197.12       | 164.59       | 222.56       | 214.95       |  109.86      | ...          | 1.909861e-01 | -0.278596389 | -0.40585384  | 8.012731e-01 | -1.24688090  | -1.68629376  | 0.030847570  | -0.04476128  | -0.06205471  | 1.0000000000 | \n",
       "| 0610037L13Rik | 120.72       | 150.94       | 153.03       | 156.60       | 157.36       | 118.41       | 129.79       | 166.83       | 106.50       |  114.60      | ...          | 9.939000e-01 | -0.448326263 | -0.55404931  | 6.185147e-01 | -0.34261019  | -0.42593111  | 0.735352367  | -0.56716278  | -0.69506108  | 0.4597216657 | \n",
       "| 0610037P05Rik | 208.97       | 204.10       | 197.03       | 211.32       | 243.63       | 191.90       | 188.58       | 250.09       | 179.76       |  139.84      | ...          | 8.463243e-01 | -0.192066854 | -0.26320221  | 9.898461e-01 | -0.55435147  | -0.73609432  | 0.481329466  | -0.73434086  | -0.95940851  | 0.2974374726 | \n",
       "| 0610038B21Rik |   0.79       |   3.14       |   1.17       |   2.33       |   3.00       |   4.09       |   2.99       |   3.11       |   9.50       |   10.21      | ...          | 4.907610e-01 |  2.354623889 |  1.57269091  | 2.772354e-02 |  2.45883637  |  1.64794881  | 0.018293855  |  0.31145165  |  0.26374880  | 0.7537873673 | \n",
       "| 0610039K10Rik | 369.63       | 437.06       | 366.86       | 239.55       | 206.66       | 243.30       | 218.87       | 167.39       | 258.65       |  282.69      | ...          | 9.555152e-03 | -0.449816317 | -0.59038159  | 6.738087e-01 | -0.32155836  | -0.41703417  | 0.858497947  | -0.79909169  | -1.04216158  | 0.2730506154 | \n",
       "| 0610040B10Rik |  21.96       |  32.57       |  26.36       |  16.12       |  16.62       |  14.15       |  17.27       |  20.01       |  13.70       |   12.62      | ...          | 6.552081e-01 | -0.823887644 | -0.51049225  | 1.000000e+00 | -0.94206758  | -0.58348453  | 0.893400570  |  0.13294011  |  0.11797629  | 0.9044876130 | \n",
       "| 0610040J01Rik |   3.88       |   5.11       |   5.25       |   8.53       |   6.35       |   7.21       |   4.06       |   2.54       |  41.27       |    2.97      | ...          | 7.677246e-01 |  2.858036704 |  2.36489744  | 7.262926e-06 | -0.93821514  | -0.57260694  | 0.776787852  |  1.20718188  |  0.91837655  | 0.2086258283 | \n",
       "| 0910001L09Rik | 265.84       | 288.72       | 281.83       | 296.15       | 213.67       | 235.43       | 168.04       | 243.30       | 191.10       |  277.41      | ...          | 2.203561e-01 | -0.567121735 | -0.71967492  | 4.858289e-01 | -0.02945569  | -0.03821356  | 1.000000000  | -1.92188013  | -2.16617015  | 0.0019413521 | \n",
       "| 1100001G20Rik |  88.25       | 111.80       |  30.56       |  80.95       | 240.00       | 357.70       | 169.97       | 335.60       |  16.52       |   83.70      | ...          | 4.176561e-09 | -2.237512048 | -1.53721873  | 1.206218e-01 |  0.10373963  |  0.18408681  | 0.815898469  |  0.83479293  |  0.83249702  | 0.1708197164 | \n",
       "| 1110001A16Rik |  55.89       |  37.89       |  52.49       |  50.97       |  72.50       |  50.06       |  57.24       |  55.07       |  63.75       |   63.88      | ...          | 6.552081e-01 |  0.370650335 |  0.38524900  | 5.881720e-01 |  0.37348580  |  0.38820043  | 0.573125809  | -0.08755941  | -0.07358787  | 1.0000000000 | \n",
       "| 1110001J03Rik | 167.09       | 184.17       |  93.07       | 148.74       | 172.04       | 216.45       | 184.32       | 218.11       | 117.57       |  168.78      | ...          | 2.730091e-01 | -0.334725282 | -0.27668814  | 1.000000e+00 |  0.18691921  |  0.21315967  | 0.718532586  |  0.34558484  |  0.37025581  | 0.5329533971 | \n",
       "| 1110002B05Rik | 268.54       | 255.88       | 194.43       | 238.68       | 291.73       | 383.66       | 443.65       | 324.53       | 334.39       |  326.84      | ...          | 1.263570e-02 |  0.482220986 |  0.73280727  | 2.855344e-01 |  0.44925705  |  0.68167993  | 0.321819917  |  1.17830251  |  1.86969136  | 0.0009306154 | \n",
       "| ... | ... | ... | ... | ... | ... | ... | ... | ... | ... | ... |     | ... | ... | ... | ... | ... | ... | ... | ... | ... | ... | \n",
       "| Zscan30 |   8.97       |  19.52       |  15.91       |  17.12       |  16.30       |  24.67       |  24.83       |  14.09       |  23.29       |  16.51       | ...          | 7.129923e-01 |  0.59849256  |  0.45285540  | 5.446778e-01 |  0.10180099  |  0.10158047  | 9.132241e-01 |  1.17680869  |  0.88789237  | 1.891014e-01 | \n",
       "| Zscan4a |   0.00       |   0.00       |   0.40       |   0.00       |   0.00       |   0.00       |   0.00       |   0.00       |   0.00       |   0.00       | ...          | 6.552081e-01 |        -Inf  | -0.36422609  | 1.000000e+00 |        -Inf  | -0.36422609  | 1.000000e+00 |        -Inf  | -0.36422609  | 1.000000e+00 | \n",
       "| Zscan4b |   0.00       |   0.00       |   0.00       |   0.00       |   0.00       |   0.00       |   0.00       |   0.00       |   0.00       |   0.00       | ...          |           NA |         NaN  |  0.00000000  |           NA |         NaN  |  0.00000000  |           NA |         NaN  |  0.00000000  |           NA | \n",
       "| Zscan4c |   0.00       |   0.00       |   0.00       |   0.00       |   0.00       |   0.00       |   0.00       |   0.00       |   0.00       |   0.00       | ...          |           NA |         NaN  |  0.00000000  |           NA |         NaN  |  0.00000000  |           NA |         NaN  |  0.00000000  |           NA | \n",
       "| Zscan4d |   0.00       |   0.00       |   0.00       |   0.00       |   0.00       |   0.00       |   0.00       |   0.00       |   0.00       |   0.00       | ...          |           NA |         NaN  |  0.00000000  |           NA |         NaN  |  0.00000000  |           NA |         NaN  |  0.00000000  |           NA | \n",
       "| Zscan4e |   0.00       |   0.00       |   0.00       |   0.00       |   0.00       |   0.00       |   0.00       |   0.00       |   0.00       |   0.00       | ...          |           NA |         NaN  |  0.00000000  |           NA |         NaN  |  0.00000000  |           NA |         NaN  |  0.00000000  |           NA | \n",
       "| Zscan4f |   0.00       |   0.00       |   0.00       |   0.00       |   0.00       |   0.00       |   0.00       |   0.00       |   0.00       |   0.00       | ...          |           NA |         NaN  |  0.00000000  |           NA |         NaN  |  0.00000000  |           NA |         NaN  |  0.00000000  |           NA | \n",
       "| Zscan5b |   0.00       |   0.00       |   0.45       |   0.00       |   0.00       |   0.00       |   0.00       |   0.00       |   0.00       |   0.00       | ...          | 6.552081e-01 |        -Inf  | -0.36422609  | 1.000000e+00 |        -Inf  | -0.36422609  | 1.000000e+00 |        -Inf  | -0.36422609  | 1.000000e+00 | \n",
       "| Zswim1 |  26.39       |  37.59       |  37.00       |  43.65       |  31.14       |  30.35       |  40.78       |  38.12       |  32.14       |  19.33       | ...          | 9.505141e-01 | -0.16978310  | -0.15658750  | 1.000000e+00 | -0.90329919  | -0.85824714  | 3.139871e-01 | -0.33848623  | -0.32523443  | 8.375345e-01 | \n",
       "| Zswim2 |   0.00       |   0.00       |   0.87       |   0.00       |   0.00       |   0.00       |   0.00       |   0.00       |   0.00       |   0.00       | ...          | 4.907610e-01 |        -Inf  | -0.48180498  | 1.000000e+00 |        -Inf  | -0.48180498  | 1.000000e+00 |        -Inf  | -0.48180498  | 1.000000e+00 | \n",
       "| Zswim3 |  30.55       |  26.65       |  35.17       |  34.94       |  23.12       |  28.94       |  40.12       |  30.98       |  21.97       |  23.96       | ...          | 9.480484e-01 | -0.53468486  | -0.51054247  | 6.530239e-01 | -0.40959365  | -0.39332060  | 7.900802e-01 | -0.52014160  | -0.49702733  | 6.520948e-01 | \n",
       "| Zswim4 |  79.46       |  90.65       |  91.12       |  80.32       |  54.24       |  45.82       |  61.67       |  52.97       |  70.71       |  37.56       | ...          | 1.620123e-02 | -0.27201873  | -0.39084477  | 8.557093e-01 | -1.18487977  | -1.58117757  | 4.977878e-02 | -0.36117718  | -0.51605673  | 7.010739e-01 | \n",
       "| Zswim5 |   2.46       |   2.66       |   2.81       |   2.79       |   3.90       |   1.73       |   2.53       |   2.01       |   2.46       |   4.73       | ...          | 9.368615e-01 | -0.12509575  | -0.08796732  | 1.000000e+00 |  0.82085465  |  0.61926496  | 3.763428e-01 |  0.07274011  |  0.05362971  | 9.564060e-01 | \n",
       "| Zswim6 |  78.57       |  64.56       |  78.96       |  64.64       |  42.04       |  32.51       |  28.81       |  27.80       |   9.50       |  18.11       | ...          | 1.328819e-05 | -2.91570432  | -3.40366320  | 1.570425e-06 | -1.98441559  | -2.49952744  | 6.652442e-04 | -2.38514132  | -2.90509218  | 4.000839e-05 | \n",
       "| Zswim7 |  53.01       |  52.41       |  26.10       |  38.90       |  35.25       |  53.81       |  24.24       |  32.43       |  41.69       |  54.66       | ...          | 7.954452e-01 | -0.03143804  |  0.01139782  | 9.785628e-01 |  0.35939567  |  0.33979318  | 5.815607e-01 | -1.19008750  | -0.87463236  | 4.917418e-01 | \n",
       "| Zufsp | 591.72       | 505.24       | 525.78       | 442.46       | 501.12       | 387.11       | 220.12       | 264.48       | 233.09       | 257.52       | ...          | 1.359496e-02 | -1.14732453  | -1.96733787  | 4.096078e-02 | -1.00352274  | -1.73234190  | 8.061953e-02 | -0.48932873  | -0.85796114  | 5.051271e-01 | \n",
       "| Zw10 | 199.56       | 179.26       | 195.08       | 205.33       | 185.71       | 196.94       | 293.72       | 221.19       | 233.10       | 245.22       | ...          | 5.339468e-01 |  0.25888349  |  0.42906601  | 6.312469e-01 |  0.33203649  |  0.55216459  | 5.152426e-01 | -0.04837278  | -0.07575024  | 1.000000e+00 | \n",
       "| Zwilch |  28.20       |  33.03       |  33.65       |  34.71       |  82.41       |  96.15       | 177.09       | 139.90       | 213.52       | 213.88       | ...          | 6.037326e-16 |  2.72033072  |  3.59776000  | 4.772898e-15 |  2.72279614  |  3.60180978  | 3.773384e-15 |  2.19523902  |  2.76666349  | 7.365522e-09 | \n",
       "| Zwint | 102.02       |  87.48       |  78.36       |  83.47       | 108.28       | 125.24       | 158.12       | 126.57       | 151.39       | 126.62       | ...          | 2.875541e-02 |  0.78547005  |  1.11964380  | 6.879996e-02 |  0.52763812  |  0.73747830  | 2.575054e-01 |  0.67190847  |  0.94929890  | 1.221488e-01 | \n",
       "| Zxda |  40.87       |  35.81       |  35.67       |  31.64       |  32.23       |  29.03       |  19.41       |  16.54       |  16.52       |  17.60       | ...          | 5.818633e-02 | -1.12318705  | -1.35674112  | 1.010015e-01 | -1.03191363  | -1.25634724  | 1.365961e-01 | -1.07889645  | -1.30823417  | 1.091052e-01 | \n",
       "| Zxdb |  62.09       |  49.49       |  58.55       |  58.49       |  50.56       |  32.84       |  23.91       |  28.67       |  22.94       |  27.66       | ...          | 4.999058e-03 | -1.31673422  | -1.66991114  | 3.237706e-02 | -1.04709813  | -1.35950607  | 9.448219e-02 | -0.84398561  | -1.11510443  | 1.908914e-01 | \n",
       "| Zxdc | 138.97       | 132.79       | 139.74       | 117.92       | 117.88       |  98.30       |  78.73       |  82.25       |  88.82       |  86.35       | ...          | 5.969803e-02 | -0.57539944  | -0.92930963  | 4.169869e-01 | -0.61617487  | -0.99262563  | 3.634570e-01 | -0.08893106  | -0.14444404  | 1.000000e+00 | \n",
       "| Zyg11a |   0.53       |   0.00       |   0.39       |   0.00       |   0.00       |   0.00       |   0.00       |   0.30       |   0.00       |   0.00       | ...          | 8.502624e-01 |        -Inf  | -0.77765562  | 1.000000e+00 |        -Inf  | -0.77765562  | 1.000000e+00 |  2.15618886  |  1.35228767  | 3.719981e-01 | \n",
       "| Zyg11b | 202.95       | 186.55       | 184.59       | 183.46       | 222.04       | 200.52       | 172.70       | 160.35       | 142.94       | 120.43       | ...          | 1.000000e+00 | -0.40594668  | -0.74501934  | 6.241101e-01 | -0.65308841  | -1.19550153  | 3.149901e-01 | -0.98688626  | -1.79550102  | 7.718166e-02 | \n",
       "| Zyx | 118.91       | 128.36       | 157.01       | 124.35       |  75.74       |  52.35       |  36.46       |  52.79       |  22.26       |  50.58       | ...          | 1.179740e-06 | -2.57002123  | -2.93223204  | 2.700192e-05 | -1.38563841  | -1.74417753  | 2.036233e-02 | -1.73000714  | -2.11444476  | 2.974753e-03 | \n",
       "| Zzef1 | 337.04       | 322.86       | 382.60       | 313.31       | 289.39       | 255.99       | 248.20       | 169.56       | 230.55       | 282.16       | ...          | 4.026519e-02 | -0.55599373  | -1.01468215  | 4.342302e-01 | -0.26455984  | -0.47689714  | 8.231671e-01 | -0.46301646  | -0.84286404  | 5.329534e-01 | \n",
       "| Zzz3 | 195.02       | 170.51       | 202.99       | 181.13       | 217.82       | 207.24       | 214.41       | 187.17       | 328.25       | 213.70       | ...          | 7.028719e-01 |  0.80853939  |  1.49527585  | 3.673920e-02 |  0.18936912  |  0.35230353  | 7.391194e-01 |  0.03951529  |  0.07774637  | 1.000000e+00 | \n",
       "| a |   0.00       |  10.36       |   1.29       |   5.12       |   1.32       |   7.50       |   0.00       |   1.96       |  10.45       |   0.00       | ...          | 6.552081e-01 |  1.31830017  |  1.33466617  | 2.393309e-01 |        -Inf  | -1.74763244  | 6.939833e-01 |  3.21372738  |  2.69047589  | 2.646548e-04 | \n",
       "| l7Rn6 | 129.47       | 149.73       | 128.93       | 155.23       | 174.95       | 235.31       | 243.53       | 299.47       | 175.38       | 186.04       | ...          | 5.376546e-03 |  0.31644528  |  0.38346666  | 5.900277e-01 |  0.40154894  |  0.48934666  | 4.813295e-01 |  0.47655663  |  0.58407798  | 3.770375e-01 | \n",
       "| rp9 | 428.77       | 609.23       | 412.90       | 519.73       | 464.65       | 541.97       | 505.33       | 698.16       | 545.28       | 508.24       | ...          | 5.766341e-01 |  0.14640266  |  0.26363276  | 7.862389e-01 |  0.04490146  |  0.09660478  | 9.576599e-01 | -0.11683395  | -0.16486401  | 1.000000e+00 | \n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "              C_1    C_3    C_4    C_6    P_2    P_3    P_4    P_5    T_1   \n",
       "0610007C21Rik 300.42 318.17 245.12 327.45 309.13 273.16 237.94 317.90 273.52\n",
       "0610007L01Rik 207.09 210.95 220.58 267.45 217.25 200.22 281.38 283.03 231.03\n",
       "0610007P08Rik 134.32 112.85 126.30 116.87 146.14 130.28 147.10  97.46 162.87\n",
       "0610007P14Rik 220.19 203.60 240.95 245.36 253.78 261.80 258.35 297.25 309.32\n",
       "0610007P22Rik  78.98 114.13  90.50  95.85  88.88  96.54 113.98 112.99 140.49\n",
       "0610008F07Rik   0.00   0.00   0.00   0.00   0.00   0.00   0.00   0.00   0.00\n",
       "0610009B22Rik  42.55  79.63  35.92  66.90  71.29  92.66  80.27 114.18  56.12\n",
       "0610009D07Rik 533.63 549.03 348.60 476.57 534.86 697.28 703.17 812.13 583.43\n",
       "0610009O20Rik 142.71 124.71 173.33 140.49 146.39 164.01 177.91 182.31 145.45\n",
       "0610010B08Rik   0.53   1.05   0.59   0.98   0.20   0.91   0.33   0.90   2.39\n",
       "0610010F05Rik  48.73  46.45  46.12  54.38  57.41  74.18 119.09  83.28 143.00\n",
       "0610010K14Rik 139.07 183.34 155.09 198.09 153.72 178.00 251.22 233.18 189.16\n",
       "0610010O12Rik   8.12  14.05   4.50  10.43  13.83   6.98   2.55   4.55   2.03\n",
       "0610011F06Rik 204.06 201.76 192.54 168.32 198.37 168.88 238.86 272.29 199.22\n",
       "0610011L14Rik 133.58 148.38 144.65 177.28 136.48 136.41 149.82 163.78 119.23\n",
       "0610012G03Rik  74.73 125.85  61.87  80.75 102.53  86.76  95.54 118.81 109.84\n",
       "0610012H03Rik   0.00   0.00   0.00   0.00   1.14   0.65   0.48   0.00   0.00\n",
       "0610030E20Rik 246.86 210.58 244.15 231.65 202.98 202.63 208.03 157.29 180.58\n",
       "0610031J06Rik 235.78 247.11 270.53 289.51 224.81 197.12 164.59 222.56 214.95\n",
       "0610037L13Rik 120.72 150.94 153.03 156.60 157.36 118.41 129.79 166.83 106.50\n",
       "0610037P05Rik 208.97 204.10 197.03 211.32 243.63 191.90 188.58 250.09 179.76\n",
       "0610038B21Rik   0.79   3.14   1.17   2.33   3.00   4.09   2.99   3.11   9.50\n",
       "0610039K10Rik 369.63 437.06 366.86 239.55 206.66 243.30 218.87 167.39 258.65\n",
       "0610040B10Rik  21.96  32.57  26.36  16.12  16.62  14.15  17.27  20.01  13.70\n",
       "0610040J01Rik   3.88   5.11   5.25   8.53   6.35   7.21   4.06   2.54  41.27\n",
       "0910001L09Rik 265.84 288.72 281.83 296.15 213.67 235.43 168.04 243.30 191.10\n",
       "1100001G20Rik  88.25 111.80  30.56  80.95 240.00 357.70 169.97 335.60  16.52\n",
       "1110001A16Rik  55.89  37.89  52.49  50.97  72.50  50.06  57.24  55.07  63.75\n",
       "1110001J03Rik 167.09 184.17  93.07 148.74 172.04 216.45 184.32 218.11 117.57\n",
       "1110002B05Rik 268.54 255.88 194.43 238.68 291.73 383.66 443.65 324.53 334.39\n",
       "...           ...    ...    ...    ...    ...    ...    ...    ...    ...   \n",
       "Zscan30         8.97  19.52  15.91  17.12  16.30  24.67  24.83  14.09  23.29\n",
       "Zscan4a         0.00   0.00   0.40   0.00   0.00   0.00   0.00   0.00   0.00\n",
       "Zscan4b         0.00   0.00   0.00   0.00   0.00   0.00   0.00   0.00   0.00\n",
       "Zscan4c         0.00   0.00   0.00   0.00   0.00   0.00   0.00   0.00   0.00\n",
       "Zscan4d         0.00   0.00   0.00   0.00   0.00   0.00   0.00   0.00   0.00\n",
       "Zscan4e         0.00   0.00   0.00   0.00   0.00   0.00   0.00   0.00   0.00\n",
       "Zscan4f         0.00   0.00   0.00   0.00   0.00   0.00   0.00   0.00   0.00\n",
       "Zscan5b         0.00   0.00   0.45   0.00   0.00   0.00   0.00   0.00   0.00\n",
       "Zswim1         26.39  37.59  37.00  43.65  31.14  30.35  40.78  38.12  32.14\n",
       "Zswim2          0.00   0.00   0.87   0.00   0.00   0.00   0.00   0.00   0.00\n",
       "Zswim3         30.55  26.65  35.17  34.94  23.12  28.94  40.12  30.98  21.97\n",
       "Zswim4         79.46  90.65  91.12  80.32  54.24  45.82  61.67  52.97  70.71\n",
       "Zswim5          2.46   2.66   2.81   2.79   3.90   1.73   2.53   2.01   2.46\n",
       "Zswim6         78.57  64.56  78.96  64.64  42.04  32.51  28.81  27.80   9.50\n",
       "Zswim7         53.01  52.41  26.10  38.90  35.25  53.81  24.24  32.43  41.69\n",
       "Zufsp         591.72 505.24 525.78 442.46 501.12 387.11 220.12 264.48 233.09\n",
       "Zw10          199.56 179.26 195.08 205.33 185.71 196.94 293.72 221.19 233.10\n",
       "Zwilch         28.20  33.03  33.65  34.71  82.41  96.15 177.09 139.90 213.52\n",
       "Zwint         102.02  87.48  78.36  83.47 108.28 125.24 158.12 126.57 151.39\n",
       "Zxda           40.87  35.81  35.67  31.64  32.23  29.03  19.41  16.54  16.52\n",
       "Zxdb           62.09  49.49  58.55  58.49  50.56  32.84  23.91  28.67  22.94\n",
       "Zxdc          138.97 132.79 139.74 117.92 117.88  98.30  78.73  82.25  88.82\n",
       "Zyg11a          0.53   0.00   0.39   0.00   0.00   0.00   0.00   0.30   0.00\n",
       "Zyg11b        202.95 186.55 184.59 183.46 222.04 200.52 172.70 160.35 142.94\n",
       "Zyx           118.91 128.36 157.01 124.35  75.74  52.35  36.46  52.79  22.26\n",
       "Zzef1         337.04 322.86 382.60 313.31 289.39 255.99 248.20 169.56 230.55\n",
       "Zzz3          195.02 170.51 202.99 181.13 217.82 207.24 214.41 187.17 328.25\n",
       "a               0.00  10.36   1.29   5.12   1.32   7.50   0.00   1.96  10.45\n",
       "l7Rn6         129.47 149.73 128.93 155.23 174.95 235.31 243.53 299.47 175.38\n",
       "rp9           428.77 609.23 412.90 519.73 464.65 541.97 505.33 698.16 545.28\n",
       "              T_2     ... qvalue.CP    log2FC.CT1   log2FCvsn.CT1 qvalue.CT1  \n",
       "0610007C21Rik  398.82 ... 9.417318e-01 -0.122658393 -0.15952224   1.000000e+00\n",
       "0610007L01Rik  306.11 ... 7.878284e-01  0.028450720  0.06106108   1.000000e+00\n",
       "0610007P08Rik   90.20 ... 8.535726e-01  0.409968779  0.71039586   3.821038e-01\n",
       "0610007P14Rik  382.49 ... 5.572482e-01  0.443088932  0.63473511   4.169869e-01\n",
       "0610007P22Rik  147.13 ... 8.120777e-01  0.566465434  0.65768003   3.246861e-01\n",
       "0610008F07Rik    0.00 ...           NA          NaN  0.00000000             NA\n",
       "0610009B22Rik   88.03 ... 8.899602e-02 -0.003342027  0.04917621   9.905365e-01\n",
       "0610009D07Rik 1037.25 ... 1.480926e-02  0.290685645  0.50412608   4.903655e-01\n",
       "0610009O20Rik  173.75 ... 6.165242e-01  0.001430952  0.01342905   1.000000e+00\n",
       "0610010B08Rik    0.49 ... 7.991074e-01  1.601659360  1.03301217   1.412337e-01\n",
       "0610010F05Rik  115.38 ... 2.467096e-03  1.547499287  2.25768779   3.195463e-05\n",
       "0610010K14Rik  170.52 ... 4.907610e-01  0.163482934  0.20563523   8.341150e-01\n",
       "0610010O12Rik    3.73 ... 7.435859e-01 -2.194377903 -1.12633670   8.648071e-01\n",
       "0610011F06Rik  252.04 ... 5.730909e-01  0.055749917  0.06967514   9.512541e-01\n",
       "0610011L14Rik  135.54 ... 9.180832e-01 -0.340562836 -0.52566824   6.834677e-01\n",
       "0610012G03Rik   76.28 ... 4.812137e-01  0.356306204  0.45366946   4.818037e-01\n",
       "0610012H03Rik    0.70 ... 4.062349e-01          NaN  0.00000000             NA\n",
       "0610030E20Rik  161.82 ... 3.109693e-01 -0.369645846 -0.64797212   6.379189e-01\n",
       "0610031J06Rik  109.86 ... 1.909861e-01 -0.278596389 -0.40585384   8.012731e-01\n",
       "0610037L13Rik  114.60 ... 9.939000e-01 -0.448326263 -0.55404931   6.185147e-01\n",
       "0610037P05Rik  139.84 ... 8.463243e-01 -0.192066854 -0.26320221   9.898461e-01\n",
       "0610038B21Rik   10.21 ... 4.907610e-01  2.354623889  1.57269091   2.772354e-02\n",
       "0610039K10Rik  282.69 ... 9.555152e-03 -0.449816317 -0.59038159   6.738087e-01\n",
       "0610040B10Rik   12.62 ... 6.552081e-01 -0.823887644 -0.51049225   1.000000e+00\n",
       "0610040J01Rik    2.97 ... 7.677246e-01  2.858036704  2.36489744   7.262926e-06\n",
       "0910001L09Rik  277.41 ... 2.203561e-01 -0.567121735 -0.71967492   4.858289e-01\n",
       "1100001G20Rik   83.70 ... 4.176561e-09 -2.237512048 -1.53721873   1.206218e-01\n",
       "1110001A16Rik   63.88 ... 6.552081e-01  0.370650335  0.38524900   5.881720e-01\n",
       "1110001J03Rik  168.78 ... 2.730091e-01 -0.334725282 -0.27668814   1.000000e+00\n",
       "1110002B05Rik  326.84 ... 1.263570e-02  0.482220986  0.73280727   2.855344e-01\n",
       "...           ...         ...          ...          ...           ...         \n",
       "Zscan30        16.51  ... 7.129923e-01  0.59849256   0.45285540   5.446778e-01\n",
       "Zscan4a         0.00  ... 6.552081e-01        -Inf  -0.36422609   1.000000e+00\n",
       "Zscan4b         0.00  ...           NA         NaN   0.00000000             NA\n",
       "Zscan4c         0.00  ...           NA         NaN   0.00000000             NA\n",
       "Zscan4d         0.00  ...           NA         NaN   0.00000000             NA\n",
       "Zscan4e         0.00  ...           NA         NaN   0.00000000             NA\n",
       "Zscan4f         0.00  ...           NA         NaN   0.00000000             NA\n",
       "Zscan5b         0.00  ... 6.552081e-01        -Inf  -0.36422609   1.000000e+00\n",
       "Zswim1         19.33  ... 9.505141e-01 -0.16978310  -0.15658750   1.000000e+00\n",
       "Zswim2          0.00  ... 4.907610e-01        -Inf  -0.48180498   1.000000e+00\n",
       "Zswim3         23.96  ... 9.480484e-01 -0.53468486  -0.51054247   6.530239e-01\n",
       "Zswim4         37.56  ... 1.620123e-02 -0.27201873  -0.39084477   8.557093e-01\n",
       "Zswim5          4.73  ... 9.368615e-01 -0.12509575  -0.08796732   1.000000e+00\n",
       "Zswim6         18.11  ... 1.328819e-05 -2.91570432  -3.40366320   1.570425e-06\n",
       "Zswim7         54.66  ... 7.954452e-01 -0.03143804   0.01139782   9.785628e-01\n",
       "Zufsp         257.52  ... 1.359496e-02 -1.14732453  -1.96733787   4.096078e-02\n",
       "Zw10          245.22  ... 5.339468e-01  0.25888349   0.42906601   6.312469e-01\n",
       "Zwilch        213.88  ... 6.037326e-16  2.72033072   3.59776000   4.772898e-15\n",
       "Zwint         126.62  ... 2.875541e-02  0.78547005   1.11964380   6.879996e-02\n",
       "Zxda           17.60  ... 5.818633e-02 -1.12318705  -1.35674112   1.010015e-01\n",
       "Zxdb           27.66  ... 4.999058e-03 -1.31673422  -1.66991114   3.237706e-02\n",
       "Zxdc           86.35  ... 5.969803e-02 -0.57539944  -0.92930963   4.169869e-01\n",
       "Zyg11a          0.00  ... 8.502624e-01        -Inf  -0.77765562   1.000000e+00\n",
       "Zyg11b        120.43  ... 1.000000e+00 -0.40594668  -0.74501934   6.241101e-01\n",
       "Zyx            50.58  ... 1.179740e-06 -2.57002123  -2.93223204   2.700192e-05\n",
       "Zzef1         282.16  ... 4.026519e-02 -0.55599373  -1.01468215   4.342302e-01\n",
       "Zzz3          213.70  ... 7.028719e-01  0.80853939   1.49527585   3.673920e-02\n",
       "a               0.00  ... 6.552081e-01  1.31830017   1.33466617   2.393309e-01\n",
       "l7Rn6         186.04  ... 5.376546e-03  0.31644528   0.38346666   5.900277e-01\n",
       "rp9           508.24  ... 5.766341e-01  0.14640266   0.26363276   7.862389e-01\n",
       "              log2FC.CT2  log2FCvsn.CT2 qvalue.CT2   log2FC.CT3  log2FCvsn.CT3\n",
       "0610007C21Rik  0.42142989  0.61682006   0.393704824   0.72776314  1.08599078  \n",
       "0610007L01Rik  0.43445718  0.77974700   0.391081390   0.59935996  1.07715415  \n",
       "0610007P08Rik -0.44262105 -0.71991486   0.557661867   0.09529604  0.16598341  \n",
       "0610007P14Rik  0.74940682  1.09910154   0.104290139   0.63176944  0.91806754  \n",
       "0610007P22Rik  0.63312517  0.73848570   0.246905230   0.26233311  0.30157112  \n",
       "0610008F07Rik         NaN  0.00000000            NA          NaN  0.00000000  \n",
       "0610009B22Rik  0.64630459  0.64565648   0.336251742   0.97285105  0.96878604  \n",
       "0610009D07Rik  1.12082967  1.95063835   0.001107939   0.23678699  0.41441353  \n",
       "0610009O20Rik  0.25793077  0.39644744   0.677829395   0.16636690  0.25786454  \n",
       "0610010B08Rik -0.68644558 -0.34649473   1.000000000  -0.61143788 -0.30683543  \n",
       "0610010F05Rik  1.23793298  1.76035829   0.002148796   1.37210275  1.97323466  \n",
       "0610010K14Rik  0.01382476  0.02786449   1.000000000  -0.49083574 -0.53639181  \n",
       "0610010O12Rik -1.31255784 -0.68852920   0.961124861   0.34741235  0.27590322  \n",
       "0610011F06Rik  0.39504274  0.48754054   0.461744128  -0.03114370 -0.03312322  \n",
       "0610011L14Rik -0.15563093 -0.23803711   0.960638998  -0.33971046 -0.52436127  \n",
       "0610012G03Rik -0.16975144 -0.14397236   1.000000000   0.67330953  0.84325916  \n",
       "0610012H03Rik         Inf  1.59790970   0.394873980          NaN  0.00000000  \n",
       "0610030E20Rik -0.52787395 -0.92074926   0.420538656  -0.68168494 -1.18148698  \n",
       "0610031J06Rik -1.24688090 -1.68629376   0.030847570  -0.04476128 -0.06205471  \n",
       "0610037L13Rik -0.34261019 -0.42593111   0.735352367  -0.56716278 -0.69506108  \n",
       "0610037P05Rik -0.55435147 -0.73609432   0.481329466  -0.73434086 -0.95940851  \n",
       "0610038B21Rik  2.45883637  1.64794881   0.018293855   0.31145165  0.26374880  \n",
       "0610039K10Rik -0.32155836 -0.41703417   0.858497947  -0.79909169 -1.04216158  \n",
       "0610040B10Rik -0.94206758 -0.58348453   0.893400570   0.13294011  0.11797629  \n",
       "0610040J01Rik -0.93821514 -0.57260694   0.776787852   1.20718188  0.91837655  \n",
       "0910001L09Rik -0.02945569 -0.03821356   1.000000000  -1.92188013 -2.16617015  \n",
       "1100001G20Rik  0.10373963  0.18408681   0.815898469   0.83479293  0.83249702  \n",
       "1110001A16Rik  0.37348580  0.38820043   0.573125809  -0.08755941 -0.07358787  \n",
       "1110001J03Rik  0.18691921  0.21315967   0.718532586   0.34558484  0.37025581  \n",
       "1110002B05Rik  0.44925705  0.68167993   0.321819917   1.17830251  1.86969136  \n",
       "...           ...         ...           ...          ...         ...          \n",
       "Zscan30        0.10180099  0.10158047   9.132241e-01  1.17680869  0.88789237  \n",
       "Zscan4a              -Inf -0.36422609   1.000000e+00        -Inf -0.36422609  \n",
       "Zscan4b               NaN  0.00000000             NA         NaN  0.00000000  \n",
       "Zscan4c               NaN  0.00000000             NA         NaN  0.00000000  \n",
       "Zscan4d               NaN  0.00000000             NA         NaN  0.00000000  \n",
       "Zscan4e               NaN  0.00000000             NA         NaN  0.00000000  \n",
       "Zscan4f               NaN  0.00000000             NA         NaN  0.00000000  \n",
       "Zscan5b              -Inf -0.36422609   1.000000e+00        -Inf -0.36422609  \n",
       "Zswim1        -0.90329919 -0.85824714   3.139871e-01 -0.33848623 -0.32523443  \n",
       "Zswim2               -Inf -0.48180498   1.000000e+00        -Inf -0.48180498  \n",
       "Zswim3        -0.40959365 -0.39332060   7.900802e-01 -0.52014160 -0.49702733  \n",
       "Zswim4        -1.18487977 -1.58117757   4.977878e-02 -0.36117718 -0.51605673  \n",
       "Zswim5         0.82085465  0.61926496   3.763428e-01  0.07274011  0.05362971  \n",
       "Zswim6        -1.98441559 -2.49952744   6.652442e-04 -2.38514132 -2.90509218  \n",
       "Zswim7         0.35939567  0.33979318   5.815607e-01 -1.19008750 -0.87463236  \n",
       "Zufsp         -1.00352274 -1.73234190   8.061953e-02 -0.48932873 -0.85796114  \n",
       "Zw10           0.33203649  0.55216459   5.152426e-01 -0.04837278 -0.07575024  \n",
       "Zwilch         2.72279614  3.60180978   3.773384e-15  2.19523902  2.76666349  \n",
       "Zwint          0.52763812  0.73747830   2.575054e-01  0.67190847  0.94929890  \n",
       "Zxda          -1.03191363 -1.25634724   1.365961e-01 -1.07889645 -1.30823417  \n",
       "Zxdb          -1.04709813 -1.35950607   9.448219e-02 -0.84398561 -1.11510443  \n",
       "Zxdc          -0.61617487 -0.99262563   3.634570e-01 -0.08893106 -0.14444404  \n",
       "Zyg11a               -Inf -0.77765562   1.000000e+00  2.15618886  1.35228767  \n",
       "Zyg11b        -0.65308841 -1.19550153   3.149901e-01 -0.98688626 -1.79550102  \n",
       "Zyx           -1.38563841 -1.74417753   2.036233e-02 -1.73000714 -2.11444476  \n",
       "Zzef1         -0.26455984 -0.47689714   8.231671e-01 -0.46301646 -0.84286404  \n",
       "Zzz3           0.18936912  0.35230353   7.391194e-01  0.03951529  0.07774637  \n",
       "a                    -Inf -1.74763244   6.939833e-01  3.21372738  2.69047589  \n",
       "l7Rn6          0.40154894  0.48934666   4.813295e-01  0.47655663  0.58407798  \n",
       "rp9            0.04490146  0.09660478   9.576599e-01 -0.11683395 -0.16486401  \n",
       "              qvalue.CT3  \n",
       "0610007C21Rik 0.0849638225\n",
       "0610007L01Rik 0.1662041565\n",
       "0610007P08Rik 0.8933977701\n",
       "0610007P14Rik 0.1822683985\n",
       "0610007P22Rik 0.6610278272\n",
       "0610008F07Rik           NA\n",
       "0610009B22Rik 0.0990259846\n",
       "0610009D07Rik 0.5649703839\n",
       "0610009O20Rik 0.8368292328\n",
       "0610010B08Rik 1.0000000000\n",
       "0610010F05Rik 0.0003177189\n",
       "0610010K14Rik 0.6099562171\n",
       "0610010O12Rik 0.7952688613\n",
       "0610011F06Rik 1.0000000000\n",
       "0610011L14Rik 0.6672814070\n",
       "0610012G03Rik 0.1551004987\n",
       "0610012H03Rik           NA\n",
       "0610030E20Rik 0.2502932155\n",
       "0610031J06Rik 1.0000000000\n",
       "0610037L13Rik 0.4597216657\n",
       "0610037P05Rik 0.2974374726\n",
       "0610038B21Rik 0.7537873673\n",
       "0610039K10Rik 0.2730506154\n",
       "0610040B10Rik 0.9044876130\n",
       "0610040J01Rik 0.2086258283\n",
       "0910001L09Rik 0.0019413521\n",
       "1100001G20Rik 0.1708197164\n",
       "1110001A16Rik 1.0000000000\n",
       "1110001J03Rik 0.5329533971\n",
       "1110002B05Rik 0.0009306154\n",
       "...           ...         \n",
       "Zscan30       1.891014e-01\n",
       "Zscan4a       1.000000e+00\n",
       "Zscan4b                 NA\n",
       "Zscan4c                 NA\n",
       "Zscan4d                 NA\n",
       "Zscan4e                 NA\n",
       "Zscan4f                 NA\n",
       "Zscan5b       1.000000e+00\n",
       "Zswim1        8.375345e-01\n",
       "Zswim2        1.000000e+00\n",
       "Zswim3        6.520948e-01\n",
       "Zswim4        7.010739e-01\n",
       "Zswim5        9.564060e-01\n",
       "Zswim6        4.000839e-05\n",
       "Zswim7        4.917418e-01\n",
       "Zufsp         5.051271e-01\n",
       "Zw10          1.000000e+00\n",
       "Zwilch        7.365522e-09\n",
       "Zwint         1.221488e-01\n",
       "Zxda          1.091052e-01\n",
       "Zxdb          1.908914e-01\n",
       "Zxdc          1.000000e+00\n",
       "Zyg11a        3.719981e-01\n",
       "Zyg11b        7.718166e-02\n",
       "Zyx           2.974753e-03\n",
       "Zzef1         5.329534e-01\n",
       "Zzz3          1.000000e+00\n",
       "a             2.646548e-04\n",
       "l7Rn6         3.770375e-01\n",
       "rp9           1.000000e+00"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "deseqRes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## - Pre-processing "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# store the loaded `peaksRef` .rda file into `peakRefSimple` object\n",
    "peaksRefSimple = peaksRef"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [],
   "source": [
    "# eliminating the redundancy between 3 replicates of Myc P\n",
    "peaksRefSimple$EuMyc$Myc.P= union(peaksRef$EuMyc$Myc.P, peaksRef$EuMyc$Myc.P) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [],
   "source": [
    "# store Mm and Hs gene symbols as a list\n",
    "EG2GSlist=as.list(org.Mm.egSYMBOL)\n",
    "EG2GSlist.hsa=as.list(org.Hs.egSYMBOL)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## - Annotate Mmu promoters"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "GRanges object with 55419 ranges and 2 metadata columns:\n",
       "              seqnames             ranges strand |     tx_id     tx_name\n",
       "                 <Rle>          <IRanges>  <Rle> | <integer> <character>\n",
       "      [1]         chr1 [4795974, 4798973]      + |         1  uc007afg.1\n",
       "      [2]         chr1 [4795974, 4798973]      + |         2  uc007afh.1\n",
       "      [3]         chr1 [4845775, 4848774]      + |         3  uc007afi.2\n",
       "      [4]         chr1 [4845775, 4848774]      + |         4  uc011wht.1\n",
       "      [5]         chr1 [4846409, 4849408]      + |         5  uc011whu.1\n",
       "      ...          ...                ...    ... .       ...         ...\n",
       "  [55415] chrUn_random [2215887, 2218886]      - |     55415  uc009sjw.2\n",
       "  [55416] chrUn_random [2677408, 2680407]      - |     55416  uc009skb.2\n",
       "  [55417] chrUn_random [2890057, 2893056]      - |     55417  uc009ske.1\n",
       "  [55418] chrUn_random [3836248, 3839247]      - |     55418  uc009skg.1\n",
       "  [55419] chrUn_random [4676188, 4679187]      - |     55419  uc009skh.1\n",
       "  -------\n",
       "  seqinfo: 35 sequences (1 circular) from mm9 genome"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# retrieve annotation data using `promoters` function part of ensembldb utility package\n",
    "promoterGR= promoters(txdb, upstream= 2000, downstream= 1000)\n",
    "promoterGR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "'select()' returned 1:1 mapping between keys and columns\n"
     ]
    }
   ],
   "source": [
    "# extract data from `txdb` object using `select` \n",
    "ann=select(txdb, \n",
    "           # use `transcripts()` method from `GenomicFeatures` and select `tx_name` column\n",
    "           keys=transcripts(txdb)$tx_name, \n",
    "           keytype='TXNAME', \n",
    "           # the columns from which values should be retrieved\n",
    "           columns = c('TXNAME','GENEID')\n",
    "          )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<thead><tr><th scope=col>TXNAME</th><th scope=col>GENEID</th></tr></thead>\n",
       "<tbody>\n",
       "\t<tr><td>uc007afg.1</td><td>18777     </td></tr>\n",
       "\t<tr><td>uc007afh.1</td><td>18777     </td></tr>\n",
       "\t<tr><td>uc007afi.2</td><td>21399     </td></tr>\n",
       "\t<tr><td>uc011wht.1</td><td>21399     </td></tr>\n",
       "\t<tr><td>uc011whu.1</td><td>21399     </td></tr>\n",
       "\t<tr><td>uc007afm.1</td><td>108664    </td></tr>\n",
       "\t<tr><td>uc007afn.1</td><td>108664    </td></tr>\n",
       "\t<tr><td>uc007afo.1</td><td>18387     </td></tr>\n",
       "\t<tr><td>uc007afp.1</td><td>18387     </td></tr>\n",
       "\t<tr><td>uc007afq.1</td><td>18387     </td></tr>\n",
       "\t<tr><td>uc007afr.2</td><td>12421     </td></tr>\n",
       "\t<tr><td>uc011whx.1</td><td>12421     </td></tr>\n",
       "\t<tr><td>uc007afs.1</td><td>12421     </td></tr>\n",
       "\t<tr><td>uc007aft.2</td><td>12421     </td></tr>\n",
       "\t<tr><td>uc007afu.2</td><td>12421     </td></tr>\n",
       "\t<tr><td>uc007afv.2</td><td>620393    </td></tr>\n",
       "\t<tr><td>uc007afw.1</td><td>240690    </td></tr>\n",
       "\t<tr><td>uc007afx.1</td><td>240690    </td></tr>\n",
       "\t<tr><td>uc007afy.1</td><td>240690    </td></tr>\n",
       "\t<tr><td>uc007afz.1</td><td>240690    </td></tr>\n",
       "\t<tr><td>uc007aga.1</td><td>240690    </td></tr>\n",
       "\t<tr><td>uc007agb.1</td><td>319263    </td></tr>\n",
       "\t<tr><td>uc007agc.1</td><td>319263    </td></tr>\n",
       "\t<tr><td>uc011why.1</td><td>319263    </td></tr>\n",
       "\t<tr><td>uc007agd.1</td><td>319263    </td></tr>\n",
       "\t<tr><td>uc011wia.1</td><td>NA        </td></tr>\n",
       "\t<tr><td>uc007agk.2</td><td>59014     </td></tr>\n",
       "\t<tr><td>uc007agl.2</td><td>76187     </td></tr>\n",
       "\t<tr><td>uc007agm.2</td><td>76187     </td></tr>\n",
       "\t<tr><td>uc011wic.1</td><td>76187     </td></tr>\n",
       "\t<tr><td>...</td><td>...</td></tr>\n",
       "\t<tr><td>uc009shy.1</td><td>434632    </td></tr>\n",
       "\t<tr><td>uc009shz.1</td><td>100504239 </td></tr>\n",
       "\t<tr><td>uc009sia.1</td><td>NA        </td></tr>\n",
       "\t<tr><td>uc009sig.1</td><td>NA        </td></tr>\n",
       "\t<tr><td>uc009sio.1</td><td>NA        </td></tr>\n",
       "\t<tr><td>uc009sip.1</td><td>NA        </td></tr>\n",
       "\t<tr><td>uc009siq.1</td><td>NA        </td></tr>\n",
       "\t<tr><td>uc009sir.1</td><td>NA        </td></tr>\n",
       "\t<tr><td>uc009sis.1</td><td>NA        </td></tr>\n",
       "\t<tr><td>uc009sit.1</td><td>NA        </td></tr>\n",
       "\t<tr><td>uc009siu.1</td><td>NA        </td></tr>\n",
       "\t<tr><td>uc009siv.1</td><td>NA        </td></tr>\n",
       "\t<tr><td>uc009six.1</td><td>NA        </td></tr>\n",
       "\t<tr><td>uc009siy.1</td><td>NA        </td></tr>\n",
       "\t<tr><td>uc009sja.1</td><td>545611    </td></tr>\n",
       "\t<tr><td>uc012hdo.1</td><td>621580    </td></tr>\n",
       "\t<tr><td>uc009sjb.1</td><td>100504239 </td></tr>\n",
       "\t<tr><td>uc009sjk.1</td><td>NA        </td></tr>\n",
       "\t<tr><td>uc012hdp.1</td><td>100041253 </td></tr>\n",
       "\t<tr><td>uc012hdq.1</td><td>100041354 </td></tr>\n",
       "\t<tr><td>uc012hdr.1</td><td>100041354 </td></tr>\n",
       "\t<tr><td>uc009sjs.2</td><td>665943    </td></tr>\n",
       "\t<tr><td>uc009sju.1</td><td>NA        </td></tr>\n",
       "\t<tr><td>uc009sjv.1</td><td>NA        </td></tr>\n",
       "\t<tr><td>uc012hds.1</td><td>NA        </td></tr>\n",
       "\t<tr><td>uc009sjw.2</td><td>100041253 </td></tr>\n",
       "\t<tr><td>uc009skb.2</td><td>20770     </td></tr>\n",
       "\t<tr><td>uc009ske.1</td><td>333467    </td></tr>\n",
       "\t<tr><td>uc009skg.1</td><td>NA        </td></tr>\n",
       "\t<tr><td>uc009skh.1</td><td>NA        </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "\\begin{tabular}{r|ll}\n",
       " TXNAME & GENEID\\\\\n",
       "\\hline\n",
       "\t uc007afg.1 & 18777     \\\\\n",
       "\t uc007afh.1 & 18777     \\\\\n",
       "\t uc007afi.2 & 21399     \\\\\n",
       "\t uc011wht.1 & 21399     \\\\\n",
       "\t uc011whu.1 & 21399     \\\\\n",
       "\t uc007afm.1 & 108664    \\\\\n",
       "\t uc007afn.1 & 108664    \\\\\n",
       "\t uc007afo.1 & 18387     \\\\\n",
       "\t uc007afp.1 & 18387     \\\\\n",
       "\t uc007afq.1 & 18387     \\\\\n",
       "\t uc007afr.2 & 12421     \\\\\n",
       "\t uc011whx.1 & 12421     \\\\\n",
       "\t uc007afs.1 & 12421     \\\\\n",
       "\t uc007aft.2 & 12421     \\\\\n",
       "\t uc007afu.2 & 12421     \\\\\n",
       "\t uc007afv.2 & 620393    \\\\\n",
       "\t uc007afw.1 & 240690    \\\\\n",
       "\t uc007afx.1 & 240690    \\\\\n",
       "\t uc007afy.1 & 240690    \\\\\n",
       "\t uc007afz.1 & 240690    \\\\\n",
       "\t uc007aga.1 & 240690    \\\\\n",
       "\t uc007agb.1 & 319263    \\\\\n",
       "\t uc007agc.1 & 319263    \\\\\n",
       "\t uc011why.1 & 319263    \\\\\n",
       "\t uc007agd.1 & 319263    \\\\\n",
       "\t uc011wia.1 & NA        \\\\\n",
       "\t uc007agk.2 & 59014     \\\\\n",
       "\t uc007agl.2 & 76187     \\\\\n",
       "\t uc007agm.2 & 76187     \\\\\n",
       "\t uc011wic.1 & 76187     \\\\\n",
       "\t ... & ...\\\\\n",
       "\t uc009shy.1 & 434632    \\\\\n",
       "\t uc009shz.1 & 100504239 \\\\\n",
       "\t uc009sia.1 & NA        \\\\\n",
       "\t uc009sig.1 & NA        \\\\\n",
       "\t uc009sio.1 & NA        \\\\\n",
       "\t uc009sip.1 & NA        \\\\\n",
       "\t uc009siq.1 & NA        \\\\\n",
       "\t uc009sir.1 & NA        \\\\\n",
       "\t uc009sis.1 & NA        \\\\\n",
       "\t uc009sit.1 & NA        \\\\\n",
       "\t uc009siu.1 & NA        \\\\\n",
       "\t uc009siv.1 & NA        \\\\\n",
       "\t uc009six.1 & NA        \\\\\n",
       "\t uc009siy.1 & NA        \\\\\n",
       "\t uc009sja.1 & 545611    \\\\\n",
       "\t uc012hdo.1 & 621580    \\\\\n",
       "\t uc009sjb.1 & 100504239 \\\\\n",
       "\t uc009sjk.1 & NA        \\\\\n",
       "\t uc012hdp.1 & 100041253 \\\\\n",
       "\t uc012hdq.1 & 100041354 \\\\\n",
       "\t uc012hdr.1 & 100041354 \\\\\n",
       "\t uc009sjs.2 & 665943    \\\\\n",
       "\t uc009sju.1 & NA        \\\\\n",
       "\t uc009sjv.1 & NA        \\\\\n",
       "\t uc012hds.1 & NA        \\\\\n",
       "\t uc009sjw.2 & 100041253 \\\\\n",
       "\t uc009skb.2 & 20770     \\\\\n",
       "\t uc009ske.1 & 333467    \\\\\n",
       "\t uc009skg.1 & NA        \\\\\n",
       "\t uc009skh.1 & NA        \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "TXNAME | GENEID | \n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| uc007afg.1 | 18777      | \n",
       "| uc007afh.1 | 18777      | \n",
       "| uc007afi.2 | 21399      | \n",
       "| uc011wht.1 | 21399      | \n",
       "| uc011whu.1 | 21399      | \n",
       "| uc007afm.1 | 108664     | \n",
       "| uc007afn.1 | 108664     | \n",
       "| uc007afo.1 | 18387      | \n",
       "| uc007afp.1 | 18387      | \n",
       "| uc007afq.1 | 18387      | \n",
       "| uc007afr.2 | 12421      | \n",
       "| uc011whx.1 | 12421      | \n",
       "| uc007afs.1 | 12421      | \n",
       "| uc007aft.2 | 12421      | \n",
       "| uc007afu.2 | 12421      | \n",
       "| uc007afv.2 | 620393     | \n",
       "| uc007afw.1 | 240690     | \n",
       "| uc007afx.1 | 240690     | \n",
       "| uc007afy.1 | 240690     | \n",
       "| uc007afz.1 | 240690     | \n",
       "| uc007aga.1 | 240690     | \n",
       "| uc007agb.1 | 319263     | \n",
       "| uc007agc.1 | 319263     | \n",
       "| uc011why.1 | 319263     | \n",
       "| uc007agd.1 | 319263     | \n",
       "| uc011wia.1 | NA         | \n",
       "| uc007agk.2 | 59014      | \n",
       "| uc007agl.2 | 76187      | \n",
       "| uc007agm.2 | 76187      | \n",
       "| uc011wic.1 | 76187      | \n",
       "| ... | ... | \n",
       "| uc009shy.1 | 434632     | \n",
       "| uc009shz.1 | 100504239  | \n",
       "| uc009sia.1 | NA         | \n",
       "| uc009sig.1 | NA         | \n",
       "| uc009sio.1 | NA         | \n",
       "| uc009sip.1 | NA         | \n",
       "| uc009siq.1 | NA         | \n",
       "| uc009sir.1 | NA         | \n",
       "| uc009sis.1 | NA         | \n",
       "| uc009sit.1 | NA         | \n",
       "| uc009siu.1 | NA         | \n",
       "| uc009siv.1 | NA         | \n",
       "| uc009six.1 | NA         | \n",
       "| uc009siy.1 | NA         | \n",
       "| uc009sja.1 | 545611     | \n",
       "| uc012hdo.1 | 621580     | \n",
       "| uc009sjb.1 | 100504239  | \n",
       "| uc009sjk.1 | NA         | \n",
       "| uc012hdp.1 | 100041253  | \n",
       "| uc012hdq.1 | 100041354  | \n",
       "| uc012hdr.1 | 100041354  | \n",
       "| uc009sjs.2 | 665943     | \n",
       "| uc009sju.1 | NA         | \n",
       "| uc009sjv.1 | NA         | \n",
       "| uc012hds.1 | NA         | \n",
       "| uc009sjw.2 | 100041253  | \n",
       "| uc009skb.2 | 20770      | \n",
       "| uc009ske.1 | 333467     | \n",
       "| uc009skg.1 | NA         | \n",
       "| uc009skh.1 | NA         | \n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "      TXNAME     GENEID   \n",
       "1     uc007afg.1 18777    \n",
       "2     uc007afh.1 18777    \n",
       "3     uc007afi.2 21399    \n",
       "4     uc011wht.1 21399    \n",
       "5     uc011whu.1 21399    \n",
       "6     uc007afm.1 108664   \n",
       "7     uc007afn.1 108664   \n",
       "8     uc007afo.1 18387    \n",
       "9     uc007afp.1 18387    \n",
       "10    uc007afq.1 18387    \n",
       "11    uc007afr.2 12421    \n",
       "12    uc011whx.1 12421    \n",
       "13    uc007afs.1 12421    \n",
       "14    uc007aft.2 12421    \n",
       "15    uc007afu.2 12421    \n",
       "16    uc007afv.2 620393   \n",
       "17    uc007afw.1 240690   \n",
       "18    uc007afx.1 240690   \n",
       "19    uc007afy.1 240690   \n",
       "20    uc007afz.1 240690   \n",
       "21    uc007aga.1 240690   \n",
       "22    uc007agb.1 319263   \n",
       "23    uc007agc.1 319263   \n",
       "24    uc011why.1 319263   \n",
       "25    uc007agd.1 319263   \n",
       "26    uc011wia.1 NA       \n",
       "27    uc007agk.2 59014    \n",
       "28    uc007agl.2 76187    \n",
       "29    uc007agm.2 76187    \n",
       "30    uc011wic.1 76187    \n",
       "...   ...        ...      \n",
       "55390 uc009shy.1 434632   \n",
       "55391 uc009shz.1 100504239\n",
       "55392 uc009sia.1 NA       \n",
       "55393 uc009sig.1 NA       \n",
       "55394 uc009sio.1 NA       \n",
       "55395 uc009sip.1 NA       \n",
       "55396 uc009siq.1 NA       \n",
       "55397 uc009sir.1 NA       \n",
       "55398 uc009sis.1 NA       \n",
       "55399 uc009sit.1 NA       \n",
       "55400 uc009siu.1 NA       \n",
       "55401 uc009siv.1 NA       \n",
       "55402 uc009six.1 NA       \n",
       "55403 uc009siy.1 NA       \n",
       "55404 uc009sja.1 545611   \n",
       "55405 uc012hdo.1 621580   \n",
       "55406 uc009sjb.1 100504239\n",
       "55407 uc009sjk.1 NA       \n",
       "55408 uc012hdp.1 100041253\n",
       "55409 uc012hdq.1 100041354\n",
       "55410 uc012hdr.1 100041354\n",
       "55411 uc009sjs.2 665943   \n",
       "55412 uc009sju.1 NA       \n",
       "55413 uc009sjv.1 NA       \n",
       "55414 uc012hds.1 NA       \n",
       "55415 uc009sjw.2 100041253\n",
       "55416 uc009skb.2 20770    \n",
       "55417 uc009ske.1 333467   \n",
       "55418 uc009skg.1 NA       \n",
       "55419 uc009skh.1 NA       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# selected `TXNAME` and `GENEID` columns \n",
    "ann"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<thead><tr><th></th><th scope=col>TXNAME</th><th scope=col>GENEID</th></tr></thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>uc007afg.1</th><td>uc007afg.1</td><td>18777     </td></tr>\n",
       "\t<tr><th scope=row>uc007afh.1</th><td>uc007afh.1</td><td>18777     </td></tr>\n",
       "\t<tr><th scope=row>uc007afi.2</th><td>uc007afi.2</td><td>21399     </td></tr>\n",
       "\t<tr><th scope=row>uc011wht.1</th><td>uc011wht.1</td><td>21399     </td></tr>\n",
       "\t<tr><th scope=row>uc011whu.1</th><td>uc011whu.1</td><td>21399     </td></tr>\n",
       "\t<tr><th scope=row>uc007afm.1</th><td>uc007afm.1</td><td>108664    </td></tr>\n",
       "\t<tr><th scope=row>uc007afn.1</th><td>uc007afn.1</td><td>108664    </td></tr>\n",
       "\t<tr><th scope=row>uc007afo.1</th><td>uc007afo.1</td><td>18387     </td></tr>\n",
       "\t<tr><th scope=row>uc007afp.1</th><td>uc007afp.1</td><td>18387     </td></tr>\n",
       "\t<tr><th scope=row>uc007afq.1</th><td>uc007afq.1</td><td>18387     </td></tr>\n",
       "\t<tr><th scope=row>uc007afr.2</th><td>uc007afr.2</td><td>12421     </td></tr>\n",
       "\t<tr><th scope=row>uc011whx.1</th><td>uc011whx.1</td><td>12421     </td></tr>\n",
       "\t<tr><th scope=row>uc007afs.1</th><td>uc007afs.1</td><td>12421     </td></tr>\n",
       "\t<tr><th scope=row>uc007aft.2</th><td>uc007aft.2</td><td>12421     </td></tr>\n",
       "\t<tr><th scope=row>uc007afu.2</th><td>uc007afu.2</td><td>12421     </td></tr>\n",
       "\t<tr><th scope=row>uc007afv.2</th><td>uc007afv.2</td><td>620393    </td></tr>\n",
       "\t<tr><th scope=row>uc007afw.1</th><td>uc007afw.1</td><td>240690    </td></tr>\n",
       "\t<tr><th scope=row>uc007afx.1</th><td>uc007afx.1</td><td>240690    </td></tr>\n",
       "\t<tr><th scope=row>uc007afy.1</th><td>uc007afy.1</td><td>240690    </td></tr>\n",
       "\t<tr><th scope=row>uc007afz.1</th><td>uc007afz.1</td><td>240690    </td></tr>\n",
       "\t<tr><th scope=row>uc007aga.1</th><td>uc007aga.1</td><td>240690    </td></tr>\n",
       "\t<tr><th scope=row>uc007agb.1</th><td>uc007agb.1</td><td>319263    </td></tr>\n",
       "\t<tr><th scope=row>uc007agc.1</th><td>uc007agc.1</td><td>319263    </td></tr>\n",
       "\t<tr><th scope=row>uc011why.1</th><td>uc011why.1</td><td>319263    </td></tr>\n",
       "\t<tr><th scope=row>uc007agd.1</th><td>uc007agd.1</td><td>319263    </td></tr>\n",
       "\t<tr><th scope=row>uc011wia.1</th><td>uc011wia.1</td><td>NA        </td></tr>\n",
       "\t<tr><th scope=row>uc007agk.2</th><td>uc007agk.2</td><td>59014     </td></tr>\n",
       "\t<tr><th scope=row>uc007agl.2</th><td>uc007agl.2</td><td>76187     </td></tr>\n",
       "\t<tr><th scope=row>uc007agm.2</th><td>uc007agm.2</td><td>76187     </td></tr>\n",
       "\t<tr><th scope=row>uc011wic.1</th><td>uc011wic.1</td><td>76187     </td></tr>\n",
       "\t<tr><th scope=row>...</th><td>...</td><td>...</td></tr>\n",
       "\t<tr><th scope=row>uc009shy.1</th><td>uc009shy.1</td><td>434632    </td></tr>\n",
       "\t<tr><th scope=row>uc009shz.1</th><td>uc009shz.1</td><td>100504239 </td></tr>\n",
       "\t<tr><th scope=row>uc009sia.1</th><td>uc009sia.1</td><td>NA        </td></tr>\n",
       "\t<tr><th scope=row>uc009sig.1</th><td>uc009sig.1</td><td>NA        </td></tr>\n",
       "\t<tr><th scope=row>uc009sio.1</th><td>uc009sio.1</td><td>NA        </td></tr>\n",
       "\t<tr><th scope=row>uc009sip.1</th><td>uc009sip.1</td><td>NA        </td></tr>\n",
       "\t<tr><th scope=row>uc009siq.1</th><td>uc009siq.1</td><td>NA        </td></tr>\n",
       "\t<tr><th scope=row>uc009sir.1</th><td>uc009sir.1</td><td>NA        </td></tr>\n",
       "\t<tr><th scope=row>uc009sis.1</th><td>uc009sis.1</td><td>NA        </td></tr>\n",
       "\t<tr><th scope=row>uc009sit.1</th><td>uc009sit.1</td><td>NA        </td></tr>\n",
       "\t<tr><th scope=row>uc009siu.1</th><td>uc009siu.1</td><td>NA        </td></tr>\n",
       "\t<tr><th scope=row>uc009siv.1</th><td>uc009siv.1</td><td>NA        </td></tr>\n",
       "\t<tr><th scope=row>uc009six.1</th><td>uc009six.1</td><td>NA        </td></tr>\n",
       "\t<tr><th scope=row>uc009siy.1</th><td>uc009siy.1</td><td>NA        </td></tr>\n",
       "\t<tr><th scope=row>uc009sja.1</th><td>uc009sja.1</td><td>545611    </td></tr>\n",
       "\t<tr><th scope=row>uc012hdo.1</th><td>uc012hdo.1</td><td>621580    </td></tr>\n",
       "\t<tr><th scope=row>uc009sjb.1</th><td>uc009sjb.1</td><td>100504239 </td></tr>\n",
       "\t<tr><th scope=row>uc009sjk.1</th><td>uc009sjk.1</td><td>NA        </td></tr>\n",
       "\t<tr><th scope=row>uc012hdp.1</th><td>uc012hdp.1</td><td>100041253 </td></tr>\n",
       "\t<tr><th scope=row>uc012hdq.1</th><td>uc012hdq.1</td><td>100041354 </td></tr>\n",
       "\t<tr><th scope=row>uc012hdr.1</th><td>uc012hdr.1</td><td>100041354 </td></tr>\n",
       "\t<tr><th scope=row>uc009sjs.2</th><td>uc009sjs.2</td><td>665943    </td></tr>\n",
       "\t<tr><th scope=row>uc009sju.1</th><td>uc009sju.1</td><td>NA        </td></tr>\n",
       "\t<tr><th scope=row>uc009sjv.1</th><td>uc009sjv.1</td><td>NA        </td></tr>\n",
       "\t<tr><th scope=row>uc012hds.1</th><td>uc012hds.1</td><td>NA        </td></tr>\n",
       "\t<tr><th scope=row>uc009sjw.2</th><td>uc009sjw.2</td><td>100041253 </td></tr>\n",
       "\t<tr><th scope=row>uc009skb.2</th><td>uc009skb.2</td><td>20770     </td></tr>\n",
       "\t<tr><th scope=row>uc009ske.1</th><td>uc009ske.1</td><td>333467    </td></tr>\n",
       "\t<tr><th scope=row>uc009skg.1</th><td>uc009skg.1</td><td>NA        </td></tr>\n",
       "\t<tr><th scope=row>uc009skh.1</th><td>uc009skh.1</td><td>NA        </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "\\begin{tabular}{r|ll}\n",
       "  & TXNAME & GENEID\\\\\n",
       "\\hline\n",
       "\tuc007afg.1 & uc007afg.1 & 18777     \\\\\n",
       "\tuc007afh.1 & uc007afh.1 & 18777     \\\\\n",
       "\tuc007afi.2 & uc007afi.2 & 21399     \\\\\n",
       "\tuc011wht.1 & uc011wht.1 & 21399     \\\\\n",
       "\tuc011whu.1 & uc011whu.1 & 21399     \\\\\n",
       "\tuc007afm.1 & uc007afm.1 & 108664    \\\\\n",
       "\tuc007afn.1 & uc007afn.1 & 108664    \\\\\n",
       "\tuc007afo.1 & uc007afo.1 & 18387     \\\\\n",
       "\tuc007afp.1 & uc007afp.1 & 18387     \\\\\n",
       "\tuc007afq.1 & uc007afq.1 & 18387     \\\\\n",
       "\tuc007afr.2 & uc007afr.2 & 12421     \\\\\n",
       "\tuc011whx.1 & uc011whx.1 & 12421     \\\\\n",
       "\tuc007afs.1 & uc007afs.1 & 12421     \\\\\n",
       "\tuc007aft.2 & uc007aft.2 & 12421     \\\\\n",
       "\tuc007afu.2 & uc007afu.2 & 12421     \\\\\n",
       "\tuc007afv.2 & uc007afv.2 & 620393    \\\\\n",
       "\tuc007afw.1 & uc007afw.1 & 240690    \\\\\n",
       "\tuc007afx.1 & uc007afx.1 & 240690    \\\\\n",
       "\tuc007afy.1 & uc007afy.1 & 240690    \\\\\n",
       "\tuc007afz.1 & uc007afz.1 & 240690    \\\\\n",
       "\tuc007aga.1 & uc007aga.1 & 240690    \\\\\n",
       "\tuc007agb.1 & uc007agb.1 & 319263    \\\\\n",
       "\tuc007agc.1 & uc007agc.1 & 319263    \\\\\n",
       "\tuc011why.1 & uc011why.1 & 319263    \\\\\n",
       "\tuc007agd.1 & uc007agd.1 & 319263    \\\\\n",
       "\tuc011wia.1 & uc011wia.1 & NA        \\\\\n",
       "\tuc007agk.2 & uc007agk.2 & 59014     \\\\\n",
       "\tuc007agl.2 & uc007agl.2 & 76187     \\\\\n",
       "\tuc007agm.2 & uc007agm.2 & 76187     \\\\\n",
       "\tuc011wic.1 & uc011wic.1 & 76187     \\\\\n",
       "\t... & ... & ...\\\\\n",
       "\tuc009shy.1 & uc009shy.1 & 434632    \\\\\n",
       "\tuc009shz.1 & uc009shz.1 & 100504239 \\\\\n",
       "\tuc009sia.1 & uc009sia.1 & NA        \\\\\n",
       "\tuc009sig.1 & uc009sig.1 & NA        \\\\\n",
       "\tuc009sio.1 & uc009sio.1 & NA        \\\\\n",
       "\tuc009sip.1 & uc009sip.1 & NA        \\\\\n",
       "\tuc009siq.1 & uc009siq.1 & NA        \\\\\n",
       "\tuc009sir.1 & uc009sir.1 & NA        \\\\\n",
       "\tuc009sis.1 & uc009sis.1 & NA        \\\\\n",
       "\tuc009sit.1 & uc009sit.1 & NA        \\\\\n",
       "\tuc009siu.1 & uc009siu.1 & NA        \\\\\n",
       "\tuc009siv.1 & uc009siv.1 & NA        \\\\\n",
       "\tuc009six.1 & uc009six.1 & NA        \\\\\n",
       "\tuc009siy.1 & uc009siy.1 & NA        \\\\\n",
       "\tuc009sja.1 & uc009sja.1 & 545611    \\\\\n",
       "\tuc012hdo.1 & uc012hdo.1 & 621580    \\\\\n",
       "\tuc009sjb.1 & uc009sjb.1 & 100504239 \\\\\n",
       "\tuc009sjk.1 & uc009sjk.1 & NA        \\\\\n",
       "\tuc012hdp.1 & uc012hdp.1 & 100041253 \\\\\n",
       "\tuc012hdq.1 & uc012hdq.1 & 100041354 \\\\\n",
       "\tuc012hdr.1 & uc012hdr.1 & 100041354 \\\\\n",
       "\tuc009sjs.2 & uc009sjs.2 & 665943    \\\\\n",
       "\tuc009sju.1 & uc009sju.1 & NA        \\\\\n",
       "\tuc009sjv.1 & uc009sjv.1 & NA        \\\\\n",
       "\tuc012hds.1 & uc012hds.1 & NA        \\\\\n",
       "\tuc009sjw.2 & uc009sjw.2 & 100041253 \\\\\n",
       "\tuc009skb.2 & uc009skb.2 & 20770     \\\\\n",
       "\tuc009ske.1 & uc009ske.1 & 333467    \\\\\n",
       "\tuc009skg.1 & uc009skg.1 & NA        \\\\\n",
       "\tuc009skh.1 & uc009skh.1 & NA        \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "| <!--/--> | TXNAME | GENEID | \n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| uc007afg.1 | uc007afg.1 | 18777      | \n",
       "| uc007afh.1 | uc007afh.1 | 18777      | \n",
       "| uc007afi.2 | uc007afi.2 | 21399      | \n",
       "| uc011wht.1 | uc011wht.1 | 21399      | \n",
       "| uc011whu.1 | uc011whu.1 | 21399      | \n",
       "| uc007afm.1 | uc007afm.1 | 108664     | \n",
       "| uc007afn.1 | uc007afn.1 | 108664     | \n",
       "| uc007afo.1 | uc007afo.1 | 18387      | \n",
       "| uc007afp.1 | uc007afp.1 | 18387      | \n",
       "| uc007afq.1 | uc007afq.1 | 18387      | \n",
       "| uc007afr.2 | uc007afr.2 | 12421      | \n",
       "| uc011whx.1 | uc011whx.1 | 12421      | \n",
       "| uc007afs.1 | uc007afs.1 | 12421      | \n",
       "| uc007aft.2 | uc007aft.2 | 12421      | \n",
       "| uc007afu.2 | uc007afu.2 | 12421      | \n",
       "| uc007afv.2 | uc007afv.2 | 620393     | \n",
       "| uc007afw.1 | uc007afw.1 | 240690     | \n",
       "| uc007afx.1 | uc007afx.1 | 240690     | \n",
       "| uc007afy.1 | uc007afy.1 | 240690     | \n",
       "| uc007afz.1 | uc007afz.1 | 240690     | \n",
       "| uc007aga.1 | uc007aga.1 | 240690     | \n",
       "| uc007agb.1 | uc007agb.1 | 319263     | \n",
       "| uc007agc.1 | uc007agc.1 | 319263     | \n",
       "| uc011why.1 | uc011why.1 | 319263     | \n",
       "| uc007agd.1 | uc007agd.1 | 319263     | \n",
       "| uc011wia.1 | uc011wia.1 | NA         | \n",
       "| uc007agk.2 | uc007agk.2 | 59014      | \n",
       "| uc007agl.2 | uc007agl.2 | 76187      | \n",
       "| uc007agm.2 | uc007agm.2 | 76187      | \n",
       "| uc011wic.1 | uc011wic.1 | 76187      | \n",
       "| ... | ... | ... | \n",
       "| uc009shy.1 | uc009shy.1 | 434632     | \n",
       "| uc009shz.1 | uc009shz.1 | 100504239  | \n",
       "| uc009sia.1 | uc009sia.1 | NA         | \n",
       "| uc009sig.1 | uc009sig.1 | NA         | \n",
       "| uc009sio.1 | uc009sio.1 | NA         | \n",
       "| uc009sip.1 | uc009sip.1 | NA         | \n",
       "| uc009siq.1 | uc009siq.1 | NA         | \n",
       "| uc009sir.1 | uc009sir.1 | NA         | \n",
       "| uc009sis.1 | uc009sis.1 | NA         | \n",
       "| uc009sit.1 | uc009sit.1 | NA         | \n",
       "| uc009siu.1 | uc009siu.1 | NA         | \n",
       "| uc009siv.1 | uc009siv.1 | NA         | \n",
       "| uc009six.1 | uc009six.1 | NA         | \n",
       "| uc009siy.1 | uc009siy.1 | NA         | \n",
       "| uc009sja.1 | uc009sja.1 | 545611     | \n",
       "| uc012hdo.1 | uc012hdo.1 | 621580     | \n",
       "| uc009sjb.1 | uc009sjb.1 | 100504239  | \n",
       "| uc009sjk.1 | uc009sjk.1 | NA         | \n",
       "| uc012hdp.1 | uc012hdp.1 | 100041253  | \n",
       "| uc012hdq.1 | uc012hdq.1 | 100041354  | \n",
       "| uc012hdr.1 | uc012hdr.1 | 100041354  | \n",
       "| uc009sjs.2 | uc009sjs.2 | 665943     | \n",
       "| uc009sju.1 | uc009sju.1 | NA         | \n",
       "| uc009sjv.1 | uc009sjv.1 | NA         | \n",
       "| uc012hds.1 | uc012hds.1 | NA         | \n",
       "| uc009sjw.2 | uc009sjw.2 | 100041253  | \n",
       "| uc009skb.2 | uc009skb.2 | 20770      | \n",
       "| uc009ske.1 | uc009ske.1 | 333467     | \n",
       "| uc009skg.1 | uc009skg.1 | NA         | \n",
       "| uc009skh.1 | uc009skh.1 | NA         | \n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "           TXNAME     GENEID   \n",
       "uc007afg.1 uc007afg.1 18777    \n",
       "uc007afh.1 uc007afh.1 18777    \n",
       "uc007afi.2 uc007afi.2 21399    \n",
       "uc011wht.1 uc011wht.1 21399    \n",
       "uc011whu.1 uc011whu.1 21399    \n",
       "uc007afm.1 uc007afm.1 108664   \n",
       "uc007afn.1 uc007afn.1 108664   \n",
       "uc007afo.1 uc007afo.1 18387    \n",
       "uc007afp.1 uc007afp.1 18387    \n",
       "uc007afq.1 uc007afq.1 18387    \n",
       "uc007afr.2 uc007afr.2 12421    \n",
       "uc011whx.1 uc011whx.1 12421    \n",
       "uc007afs.1 uc007afs.1 12421    \n",
       "uc007aft.2 uc007aft.2 12421    \n",
       "uc007afu.2 uc007afu.2 12421    \n",
       "uc007afv.2 uc007afv.2 620393   \n",
       "uc007afw.1 uc007afw.1 240690   \n",
       "uc007afx.1 uc007afx.1 240690   \n",
       "uc007afy.1 uc007afy.1 240690   \n",
       "uc007afz.1 uc007afz.1 240690   \n",
       "uc007aga.1 uc007aga.1 240690   \n",
       "uc007agb.1 uc007agb.1 319263   \n",
       "uc007agc.1 uc007agc.1 319263   \n",
       "uc011why.1 uc011why.1 319263   \n",
       "uc007agd.1 uc007agd.1 319263   \n",
       "uc011wia.1 uc011wia.1 NA       \n",
       "uc007agk.2 uc007agk.2 59014    \n",
       "uc007agl.2 uc007agl.2 76187    \n",
       "uc007agm.2 uc007agm.2 76187    \n",
       "uc011wic.1 uc011wic.1 76187    \n",
       "...        ...        ...      \n",
       "uc009shy.1 uc009shy.1 434632   \n",
       "uc009shz.1 uc009shz.1 100504239\n",
       "uc009sia.1 uc009sia.1 NA       \n",
       "uc009sig.1 uc009sig.1 NA       \n",
       "uc009sio.1 uc009sio.1 NA       \n",
       "uc009sip.1 uc009sip.1 NA       \n",
       "uc009siq.1 uc009siq.1 NA       \n",
       "uc009sir.1 uc009sir.1 NA       \n",
       "uc009sis.1 uc009sis.1 NA       \n",
       "uc009sit.1 uc009sit.1 NA       \n",
       "uc009siu.1 uc009siu.1 NA       \n",
       "uc009siv.1 uc009siv.1 NA       \n",
       "uc009six.1 uc009six.1 NA       \n",
       "uc009siy.1 uc009siy.1 NA       \n",
       "uc009sja.1 uc009sja.1 545611   \n",
       "uc012hdo.1 uc012hdo.1 621580   \n",
       "uc009sjb.1 uc009sjb.1 100504239\n",
       "uc009sjk.1 uc009sjk.1 NA       \n",
       "uc012hdp.1 uc012hdp.1 100041253\n",
       "uc012hdq.1 uc012hdq.1 100041354\n",
       "uc012hdr.1 uc012hdr.1 100041354\n",
       "uc009sjs.2 uc009sjs.2 665943   \n",
       "uc009sju.1 uc009sju.1 NA       \n",
       "uc009sjv.1 uc009sjv.1 NA       \n",
       "uc012hds.1 uc012hds.1 NA       \n",
       "uc009sjw.2 uc009sjw.2 100041253\n",
       "uc009skb.2 uc009skb.2 20770    \n",
       "uc009ske.1 uc009ske.1 333467   \n",
       "uc009skg.1 uc009skg.1 NA       \n",
       "uc009skh.1 uc009skh.1 NA       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# create new index that has `TXNAME` values\n",
    "rownames(ann) = ann[,1]\n",
    "ann"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "GRanges object with 55419 ranges and 3 metadata columns:\n",
       "              seqnames             ranges strand |     tx_id     tx_name\n",
       "                 <Rle>          <IRanges>  <Rle> | <integer> <character>\n",
       "      [1]         chr1 [4795974, 4798973]      + |         1  uc007afg.1\n",
       "      [2]         chr1 [4795974, 4798973]      + |         2  uc007afh.1\n",
       "      [3]         chr1 [4845775, 4848774]      + |         3  uc007afi.2\n",
       "      [4]         chr1 [4845775, 4848774]      + |         4  uc011wht.1\n",
       "      [5]         chr1 [4846409, 4849408]      + |         5  uc011whu.1\n",
       "      ...          ...                ...    ... .       ...         ...\n",
       "  [55415] chrUn_random [2215887, 2218886]      - |     55415  uc009sjw.2\n",
       "  [55416] chrUn_random [2677408, 2680407]      - |     55416  uc009skb.2\n",
       "  [55417] chrUn_random [2890057, 2893056]      - |     55417  uc009ske.1\n",
       "  [55418] chrUn_random [3836248, 3839247]      - |     55418  uc009skg.1\n",
       "  [55419] chrUn_random [4676188, 4679187]      - |     55419  uc009skh.1\n",
       "              gene_id\n",
       "          <character>\n",
       "      [1]       18777\n",
       "      [2]       18777\n",
       "      [3]       21399\n",
       "      [4]       21399\n",
       "      [5]       21399\n",
       "      ...         ...\n",
       "  [55415]   100041253\n",
       "  [55416]       20770\n",
       "  [55417]      333467\n",
       "  [55418]        <NA>\n",
       "  [55419]        <NA>\n",
       "  -------\n",
       "  seqinfo: 35 sequences (1 circular) from mm9 genome"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# create a new metadata column in `promoterGR` and populate with `GENEID` values from the annotation list object `ann` (left join on promoters$tx_name column) \n",
    "promoterGR$gene_id = ann[promoterGR$tx_name,'GENEID']\n",
    "promoterGR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [],
   "source": [
    "# select the gene symbols coresponding to promoterGR$gene_id from `EG2GSlist` object as a list (hence, `as.character()`)\n",
    "# create `gene_symbol` as new column in promoterGR\n",
    "promoterGR$gene_symbol=as.character(EG2GSlist[promoterGR$gene_id])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "GRanges object with 55419 ranges and 4 metadata columns:\n",
       "              seqnames             ranges strand |     tx_id     tx_name\n",
       "                 <Rle>          <IRanges>  <Rle> | <integer> <character>\n",
       "      [1]         chr1 [4795974, 4798973]      + |         1  uc007afg.1\n",
       "      [2]         chr1 [4795974, 4798973]      + |         2  uc007afh.1\n",
       "      [3]         chr1 [4845775, 4848774]      + |         3  uc007afi.2\n",
       "      [4]         chr1 [4845775, 4848774]      + |         4  uc011wht.1\n",
       "      [5]         chr1 [4846409, 4849408]      + |         5  uc011whu.1\n",
       "      ...          ...                ...    ... .       ...         ...\n",
       "  [55415] chrUn_random [2215887, 2218886]      - |     55415  uc009sjw.2\n",
       "  [55416] chrUn_random [2677408, 2680407]      - |     55416  uc009skb.2\n",
       "  [55417] chrUn_random [2890057, 2893056]      - |     55417  uc009ske.1\n",
       "  [55418] chrUn_random [3836248, 3839247]      - |     55418  uc009skg.1\n",
       "  [55419] chrUn_random [4676188, 4679187]      - |     55419  uc009skh.1\n",
       "              gene_id   gene_symbol\n",
       "          <character>   <character>\n",
       "      [1]       18777        Lypla1\n",
       "      [2]       18777        Lypla1\n",
       "      [3]       21399         Tcea1\n",
       "      [4]       21399         Tcea1\n",
       "      [5]       21399         Tcea1\n",
       "      ...         ...           ...\n",
       "  [55415]   100041253       Gm16513\n",
       "  [55416]       20770         Mucl2\n",
       "  [55417]      333467 B020031M17Rik\n",
       "  [55418]        <NA>          NULL\n",
       "  [55419]        <NA>          NULL\n",
       "  -------\n",
       "  seqinfo: 35 sequences (1 circular) from mm9 genome"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Finally, the annotated Mus GRanges \n",
    "promoterGR"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## - Annotate Hs promoters"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in valid.GenomicRanges.seqinfo(x, suggest.trim = TRUE):\n",
      "\"GRanges object contains 1 out-of-bound range located on sequence\n",
      "  chrUn_gl000223. Note that only ranges located on a non-circular\n",
      "  sequence whose length is not NA can be considered out-of-bound (use\n",
      "  seqlengths() and isCircular() to get the lengths and circularity flags\n",
      "  of the underlying sequences). You can use trim() to trim these ranges.\n",
      "  See ?`trim,GenomicRanges-method` for more information.\""
     ]
    },
    {
     "data": {
      "text/plain": [
       "GRanges object with 82960 ranges and 2 metadata columns:\n",
       "                seqnames           ranges strand |     tx_id     tx_name\n",
       "                   <Rle>        <IRanges>  <Rle> | <integer> <character>\n",
       "      [1]           chr1 [  9874,  12873]      + |         1  uc001aaa.3\n",
       "      [2]           chr1 [  9874,  12873]      + |         2  uc010nxq.1\n",
       "      [3]           chr1 [  9874,  12873]      + |         3  uc010nxr.1\n",
       "      [4]           chr1 [ 67091,  70090]      + |         4  uc001aal.1\n",
       "      [5]           chr1 [319084, 322083]      + |         5  uc001aaq.2\n",
       "      ...            ...              ...    ... .       ...         ...\n",
       "  [82956] chrUn_gl000237   [ 1687,  4686]      - |     82956  uc011mgu.1\n",
       "  [82957] chrUn_gl000241   [35876, 38875]      - |     82957  uc011mgv.2\n",
       "  [82958] chrUn_gl000243   [ 9501, 12500]      + |     82958  uc011mgw.1\n",
       "  [82959] chrUn_gl000243   [11608, 14607]      + |     82959  uc022brq.1\n",
       "  [82960] chrUn_gl000247   [ 4817,  7816]      - |     82960  uc022brr.1\n",
       "  -------\n",
       "  seqinfo: 93 sequences (1 circular) from hg19 genome"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "promoterGR.hsa= promoters(txdb.hsa, upstream= 2000, downstream= 1000)\n",
    "promoterGR.hsa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "'select()' returned 1:1 mapping between keys and columns\n"
     ]
    }
   ],
   "source": [
    "# extract data from `txdb.hsa` object using `select` \n",
    "\n",
    "ann=select(txdb.hsa,\n",
    "            # use `transcripts()` method from `GenomicFeatures` and select `tx_name` column\n",
    "           keys=transcripts(txdb.hsa)$tx_name, \n",
    "           keytype='TXNAME',\n",
    "           # the columns from which values should be retrieved\n",
    "           columns = c('TXNAME','GENEID'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<thead><tr><th scope=col>TXNAME</th><th scope=col>GENEID</th></tr></thead>\n",
       "<tbody>\n",
       "\t<tr><td>uc001aaa.3</td><td>100287102 </td></tr>\n",
       "\t<tr><td>uc010nxq.1</td><td>100287102 </td></tr>\n",
       "\t<tr><td>uc010nxr.1</td><td>100287102 </td></tr>\n",
       "\t<tr><td>uc001aal.1</td><td>79501     </td></tr>\n",
       "\t<tr><td>uc001aaq.2</td><td>NA        </td></tr>\n",
       "\t<tr><td>uc001aar.2</td><td>NA        </td></tr>\n",
       "\t<tr><td>uc009vjk.2</td><td>100133331 </td></tr>\n",
       "\t<tr><td>uc001aau.3</td><td>100132062 </td></tr>\n",
       "\t<tr><td>uc021oeh.1</td><td>100133331 </td></tr>\n",
       "\t<tr><td>uc021oei.1</td><td>NA        </td></tr>\n",
       "\t<tr><td>uc010nxu.2</td><td>729759    </td></tr>\n",
       "\t<tr><td>uc001aax.1</td><td>NA        </td></tr>\n",
       "\t<tr><td>uc001abb.3</td><td>NA        </td></tr>\n",
       "\t<tr><td>uc031pjj.1</td><td>643837    </td></tr>\n",
       "\t<tr><td>uc001abp.2</td><td>643837    </td></tr>\n",
       "\t<tr><td>uc009vjn.2</td><td>643837    </td></tr>\n",
       "\t<tr><td>uc009vjo.2</td><td>643837    </td></tr>\n",
       "\t<tr><td>uc021oem.2</td><td>643837    </td></tr>\n",
       "\t<tr><td>uc031pjk.1</td><td>643837    </td></tr>\n",
       "\t<tr><td>uc001abu.1</td><td>NA        </td></tr>\n",
       "\t<tr><td>uc001abv.1</td><td>148398    </td></tr>\n",
       "\t<tr><td>uc001abw.1</td><td>148398    </td></tr>\n",
       "\t<tr><td>uc031pjl.1</td><td>148398    </td></tr>\n",
       "\t<tr><td>uc031pjm.1</td><td>148398    </td></tr>\n",
       "\t<tr><td>uc031pjn.1</td><td>148398    </td></tr>\n",
       "\t<tr><td>uc001abx.2</td><td>148398    </td></tr>\n",
       "\t<tr><td>uc031pjo.1</td><td>148398    </td></tr>\n",
       "\t<tr><td>uc031pjp.1</td><td>148398    </td></tr>\n",
       "\t<tr><td>uc031pjq.1</td><td>148398    </td></tr>\n",
       "\t<tr><td>uc031pjr.1</td><td>148398    </td></tr>\n",
       "\t<tr><td>...</td><td>...</td></tr>\n",
       "\t<tr><td>uc022brm.1</td><td>NA        </td></tr>\n",
       "\t<tr><td>uc011mfz.1</td><td>7637      </td></tr>\n",
       "\t<tr><td>uc011mga.1</td><td>7637      </td></tr>\n",
       "\t<tr><td>uc011mgb.1</td><td>7637      </td></tr>\n",
       "\t<tr><td>uc011mgc.1</td><td>7637      </td></tr>\n",
       "\t<tr><td>uc011mgd.2</td><td>7574      </td></tr>\n",
       "\t<tr><td>uc031tgg.1</td><td>7574      </td></tr>\n",
       "\t<tr><td>uc031tgh.1</td><td>7574      </td></tr>\n",
       "\t<tr><td>uc011mgg.1</td><td>NA        </td></tr>\n",
       "\t<tr><td>uc031tgi.1</td><td>653545    </td></tr>\n",
       "\t<tr><td>uc031tgj.1</td><td>22947     </td></tr>\n",
       "\t<tr><td>uc031tgk.1</td><td>728410    </td></tr>\n",
       "\t<tr><td>uc031tgl.1</td><td>22947     </td></tr>\n",
       "\t<tr><td>uc031tgm.1</td><td>100653046 </td></tr>\n",
       "\t<tr><td>uc031tgn.1</td><td>NA        </td></tr>\n",
       "\t<tr><td>uc031tgo.1</td><td>728410    </td></tr>\n",
       "\t<tr><td>uc031tgp.1</td><td>728410    </td></tr>\n",
       "\t<tr><td>uc031tgq.1</td><td>22947     </td></tr>\n",
       "\t<tr><td>uc031tgr.1</td><td>22947     </td></tr>\n",
       "\t<tr><td>uc031tgs.1</td><td>728410    </td></tr>\n",
       "\t<tr><td>uc022brn.1</td><td>100288687 </td></tr>\n",
       "\t<tr><td>uc022bro.1</td><td>100288687 </td></tr>\n",
       "\t<tr><td>uc011mgh.2</td><td>100288255 </td></tr>\n",
       "\t<tr><td>uc011mgi.2</td><td>100288255 </td></tr>\n",
       "\t<tr><td>uc022brp.1</td><td>NA        </td></tr>\n",
       "\t<tr><td>uc011mgu.1</td><td>NA        </td></tr>\n",
       "\t<tr><td>uc011mgv.2</td><td>NA        </td></tr>\n",
       "\t<tr><td>uc011mgw.1</td><td>NA        </td></tr>\n",
       "\t<tr><td>uc022brq.1</td><td>NA        </td></tr>\n",
       "\t<tr><td>uc022brr.1</td><td>NA        </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "\\begin{tabular}{r|ll}\n",
       " TXNAME & GENEID\\\\\n",
       "\\hline\n",
       "\t uc001aaa.3 & 100287102 \\\\\n",
       "\t uc010nxq.1 & 100287102 \\\\\n",
       "\t uc010nxr.1 & 100287102 \\\\\n",
       "\t uc001aal.1 & 79501     \\\\\n",
       "\t uc001aaq.2 & NA        \\\\\n",
       "\t uc001aar.2 & NA        \\\\\n",
       "\t uc009vjk.2 & 100133331 \\\\\n",
       "\t uc001aau.3 & 100132062 \\\\\n",
       "\t uc021oeh.1 & 100133331 \\\\\n",
       "\t uc021oei.1 & NA        \\\\\n",
       "\t uc010nxu.2 & 729759    \\\\\n",
       "\t uc001aax.1 & NA        \\\\\n",
       "\t uc001abb.3 & NA        \\\\\n",
       "\t uc031pjj.1 & 643837    \\\\\n",
       "\t uc001abp.2 & 643837    \\\\\n",
       "\t uc009vjn.2 & 643837    \\\\\n",
       "\t uc009vjo.2 & 643837    \\\\\n",
       "\t uc021oem.2 & 643837    \\\\\n",
       "\t uc031pjk.1 & 643837    \\\\\n",
       "\t uc001abu.1 & NA        \\\\\n",
       "\t uc001abv.1 & 148398    \\\\\n",
       "\t uc001abw.1 & 148398    \\\\\n",
       "\t uc031pjl.1 & 148398    \\\\\n",
       "\t uc031pjm.1 & 148398    \\\\\n",
       "\t uc031pjn.1 & 148398    \\\\\n",
       "\t uc001abx.2 & 148398    \\\\\n",
       "\t uc031pjo.1 & 148398    \\\\\n",
       "\t uc031pjp.1 & 148398    \\\\\n",
       "\t uc031pjq.1 & 148398    \\\\\n",
       "\t uc031pjr.1 & 148398    \\\\\n",
       "\t ... & ...\\\\\n",
       "\t uc022brm.1 & NA        \\\\\n",
       "\t uc011mfz.1 & 7637      \\\\\n",
       "\t uc011mga.1 & 7637      \\\\\n",
       "\t uc011mgb.1 & 7637      \\\\\n",
       "\t uc011mgc.1 & 7637      \\\\\n",
       "\t uc011mgd.2 & 7574      \\\\\n",
       "\t uc031tgg.1 & 7574      \\\\\n",
       "\t uc031tgh.1 & 7574      \\\\\n",
       "\t uc011mgg.1 & NA        \\\\\n",
       "\t uc031tgi.1 & 653545    \\\\\n",
       "\t uc031tgj.1 & 22947     \\\\\n",
       "\t uc031tgk.1 & 728410    \\\\\n",
       "\t uc031tgl.1 & 22947     \\\\\n",
       "\t uc031tgm.1 & 100653046 \\\\\n",
       "\t uc031tgn.1 & NA        \\\\\n",
       "\t uc031tgo.1 & 728410    \\\\\n",
       "\t uc031tgp.1 & 728410    \\\\\n",
       "\t uc031tgq.1 & 22947     \\\\\n",
       "\t uc031tgr.1 & 22947     \\\\\n",
       "\t uc031tgs.1 & 728410    \\\\\n",
       "\t uc022brn.1 & 100288687 \\\\\n",
       "\t uc022bro.1 & 100288687 \\\\\n",
       "\t uc011mgh.2 & 100288255 \\\\\n",
       "\t uc011mgi.2 & 100288255 \\\\\n",
       "\t uc022brp.1 & NA        \\\\\n",
       "\t uc011mgu.1 & NA        \\\\\n",
       "\t uc011mgv.2 & NA        \\\\\n",
       "\t uc011mgw.1 & NA        \\\\\n",
       "\t uc022brq.1 & NA        \\\\\n",
       "\t uc022brr.1 & NA        \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "TXNAME | GENEID | \n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| uc001aaa.3 | 100287102  | \n",
       "| uc010nxq.1 | 100287102  | \n",
       "| uc010nxr.1 | 100287102  | \n",
       "| uc001aal.1 | 79501      | \n",
       "| uc001aaq.2 | NA         | \n",
       "| uc001aar.2 | NA         | \n",
       "| uc009vjk.2 | 100133331  | \n",
       "| uc001aau.3 | 100132062  | \n",
       "| uc021oeh.1 | 100133331  | \n",
       "| uc021oei.1 | NA         | \n",
       "| uc010nxu.2 | 729759     | \n",
       "| uc001aax.1 | NA         | \n",
       "| uc001abb.3 | NA         | \n",
       "| uc031pjj.1 | 643837     | \n",
       "| uc001abp.2 | 643837     | \n",
       "| uc009vjn.2 | 643837     | \n",
       "| uc009vjo.2 | 643837     | \n",
       "| uc021oem.2 | 643837     | \n",
       "| uc031pjk.1 | 643837     | \n",
       "| uc001abu.1 | NA         | \n",
       "| uc001abv.1 | 148398     | \n",
       "| uc001abw.1 | 148398     | \n",
       "| uc031pjl.1 | 148398     | \n",
       "| uc031pjm.1 | 148398     | \n",
       "| uc031pjn.1 | 148398     | \n",
       "| uc001abx.2 | 148398     | \n",
       "| uc031pjo.1 | 148398     | \n",
       "| uc031pjp.1 | 148398     | \n",
       "| uc031pjq.1 | 148398     | \n",
       "| uc031pjr.1 | 148398     | \n",
       "| ... | ... | \n",
       "| uc022brm.1 | NA         | \n",
       "| uc011mfz.1 | 7637       | \n",
       "| uc011mga.1 | 7637       | \n",
       "| uc011mgb.1 | 7637       | \n",
       "| uc011mgc.1 | 7637       | \n",
       "| uc011mgd.2 | 7574       | \n",
       "| uc031tgg.1 | 7574       | \n",
       "| uc031tgh.1 | 7574       | \n",
       "| uc011mgg.1 | NA         | \n",
       "| uc031tgi.1 | 653545     | \n",
       "| uc031tgj.1 | 22947      | \n",
       "| uc031tgk.1 | 728410     | \n",
       "| uc031tgl.1 | 22947      | \n",
       "| uc031tgm.1 | 100653046  | \n",
       "| uc031tgn.1 | NA         | \n",
       "| uc031tgo.1 | 728410     | \n",
       "| uc031tgp.1 | 728410     | \n",
       "| uc031tgq.1 | 22947      | \n",
       "| uc031tgr.1 | 22947      | \n",
       "| uc031tgs.1 | 728410     | \n",
       "| uc022brn.1 | 100288687  | \n",
       "| uc022bro.1 | 100288687  | \n",
       "| uc011mgh.2 | 100288255  | \n",
       "| uc011mgi.2 | 100288255  | \n",
       "| uc022brp.1 | NA         | \n",
       "| uc011mgu.1 | NA         | \n",
       "| uc011mgv.2 | NA         | \n",
       "| uc011mgw.1 | NA         | \n",
       "| uc022brq.1 | NA         | \n",
       "| uc022brr.1 | NA         | \n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "      TXNAME     GENEID   \n",
       "1     uc001aaa.3 100287102\n",
       "2     uc010nxq.1 100287102\n",
       "3     uc010nxr.1 100287102\n",
       "4     uc001aal.1 79501    \n",
       "5     uc001aaq.2 NA       \n",
       "6     uc001aar.2 NA       \n",
       "7     uc009vjk.2 100133331\n",
       "8     uc001aau.3 100132062\n",
       "9     uc021oeh.1 100133331\n",
       "10    uc021oei.1 NA       \n",
       "11    uc010nxu.2 729759   \n",
       "12    uc001aax.1 NA       \n",
       "13    uc001abb.3 NA       \n",
       "14    uc031pjj.1 643837   \n",
       "15    uc001abp.2 643837   \n",
       "16    uc009vjn.2 643837   \n",
       "17    uc009vjo.2 643837   \n",
       "18    uc021oem.2 643837   \n",
       "19    uc031pjk.1 643837   \n",
       "20    uc001abu.1 NA       \n",
       "21    uc001abv.1 148398   \n",
       "22    uc001abw.1 148398   \n",
       "23    uc031pjl.1 148398   \n",
       "24    uc031pjm.1 148398   \n",
       "25    uc031pjn.1 148398   \n",
       "26    uc001abx.2 148398   \n",
       "27    uc031pjo.1 148398   \n",
       "28    uc031pjp.1 148398   \n",
       "29    uc031pjq.1 148398   \n",
       "30    uc031pjr.1 148398   \n",
       "...   ...        ...      \n",
       "82931 uc022brm.1 NA       \n",
       "82932 uc011mfz.1 7637     \n",
       "82933 uc011mga.1 7637     \n",
       "82934 uc011mgb.1 7637     \n",
       "82935 uc011mgc.1 7637     \n",
       "82936 uc011mgd.2 7574     \n",
       "82937 uc031tgg.1 7574     \n",
       "82938 uc031tgh.1 7574     \n",
       "82939 uc011mgg.1 NA       \n",
       "82940 uc031tgi.1 653545   \n",
       "82941 uc031tgj.1 22947    \n",
       "82942 uc031tgk.1 728410   \n",
       "82943 uc031tgl.1 22947    \n",
       "82944 uc031tgm.1 100653046\n",
       "82945 uc031tgn.1 NA       \n",
       "82946 uc031tgo.1 728410   \n",
       "82947 uc031tgp.1 728410   \n",
       "82948 uc031tgq.1 22947    \n",
       "82949 uc031tgr.1 22947    \n",
       "82950 uc031tgs.1 728410   \n",
       "82951 uc022brn.1 100288687\n",
       "82952 uc022bro.1 100288687\n",
       "82953 uc011mgh.2 100288255\n",
       "82954 uc011mgi.2 100288255\n",
       "82955 uc022brp.1 NA       \n",
       "82956 uc011mgu.1 NA       \n",
       "82957 uc011mgv.2 NA       \n",
       "82958 uc011mgw.1 NA       \n",
       "82959 uc022brq.1 NA       \n",
       "82960 uc022brr.1 NA       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# selected `TXNAME` and `GENEID` columns \n",
    "ann"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# create new index that has `TXNAME` values\n",
    "rownames(ann)= ann[,1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "GRanges object with 82960 ranges and 4 metadata columns:\n",
       "                seqnames           ranges strand |     tx_id     tx_name\n",
       "                   <Rle>        <IRanges>  <Rle> | <integer> <character>\n",
       "      [1]           chr1 [  9874,  12873]      + |         1  uc001aaa.3\n",
       "      [2]           chr1 [  9874,  12873]      + |         2  uc010nxq.1\n",
       "      [3]           chr1 [  9874,  12873]      + |         3  uc010nxr.1\n",
       "      [4]           chr1 [ 67091,  70090]      + |         4  uc001aal.1\n",
       "      [5]           chr1 [319084, 322083]      + |         5  uc001aaq.2\n",
       "      ...            ...              ...    ... .       ...         ...\n",
       "  [82956] chrUn_gl000237   [ 1687,  4686]      - |     82956  uc011mgu.1\n",
       "  [82957] chrUn_gl000241   [35876, 38875]      - |     82957  uc011mgv.2\n",
       "  [82958] chrUn_gl000243   [ 9501, 12500]      + |     82958  uc011mgw.1\n",
       "  [82959] chrUn_gl000243   [11608, 14607]      + |     82959  uc022brq.1\n",
       "  [82960] chrUn_gl000247   [ 4817,  7816]      - |     82960  uc022brr.1\n",
       "              gene_id gene_symbol\n",
       "          <character> <character>\n",
       "      [1]   100287102     DDX11L1\n",
       "      [2]   100287102     DDX11L1\n",
       "      [3]   100287102     DDX11L1\n",
       "      [4]       79501       OR4F5\n",
       "      [5]        <NA>        NULL\n",
       "      ...         ...         ...\n",
       "  [82956]        <NA>        NULL\n",
       "  [82957]        <NA>        NULL\n",
       "  [82958]        <NA>        NULL\n",
       "  [82959]        <NA>        NULL\n",
       "  [82960]        <NA>        NULL\n",
       "  -------\n",
       "  seqinfo: 93 sequences (1 circular) from hg19 genome"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# create a new metadata column in `promoterGR` and populate with `GENEID` values from the annotation list object `ann` (left join on promoters$tx_name column) \n",
    "promoterGR.hsa$gene_id=ann[promoterGR.hsa$tx_name,'GENEID']\n",
    "\n",
    "# select the gene symbols coresponding to promoterGR$gene_id from `EG2GSlist` object as a list (hence, `as.character()`)\n",
    "# create `gene_symbol` as new column in promoterGR\n",
    "promoterGR.hsa$gene_symbol=as.character(EG2GSlist.hsa[promoterGR.hsa$gene_id])\n",
    "\n",
    "# Finally, the annotated Hs GRanges\n",
    "promoterGR.hsa"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "3.4.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
